Index: kdeaddons-3.5.10/kate/cppsymbolviewer/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/cppsymbolviewer/Makefile.in
+++ kdeaddons-3.5.10/kate/cppsymbolviewer/Makefile.in
@@ -367,7 +367,7 @@ INCLUDES = $(all_includes)
 # Install this plugin in the KDE modules directory
 kde_module_LTLIBRARIES = katecppsymbolviewerplugin.la
 katecppsymbolviewerplugin_la_SOURCES = cpp_parser.cpp tcl_parser.cpp plugin_katesymbolviewer.cpp
-katecppsymbolviewerplugin_la_LIBADD = -lkateinterfaces
+katecppsymbolviewerplugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI)
 katecppsymbolviewerplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 pluginsdir = $(kde_datadir)/kate/plugins/katecppsymbolviewer
 plugins_DATA = ui.rc
Index: kdeaddons-3.5.10/kate/filelistloader/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/filelistloader/Makefile.in
+++ kdeaddons-3.5.10/kate/filelistloader/Makefile.in
@@ -370,7 +370,7 @@ kde_module_LTLIBRARIES = katefll_plugin.
 #katefll_initplugin_la_LIBADD = -lkateinterfaces
 #katefll_initplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 katefll_plugin_la_SOURCES = katefll_plugin.cpp
-katefll_plugin_la_LIBADD = -lkateinterfaces
+katefll_plugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI) $(LIB_KIO)
 katefll_plugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 
 # Install the .rc file in the Part's directory (in this case, the part
Index: kdeaddons-3.5.10/kate/helloworld/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/helloworld/Makefile.in
+++ kdeaddons-3.5.10/kate/helloworld/Makefile.in
@@ -362,7 +362,7 @@ INCLUDES = $(all_includes)
 # Install this plugin in the KDE modules directory
 kde_module_LTLIBRARIES = katehelloworldplugin.la
 katehelloworldplugin_la_SOURCES = plugin_katehelloworld.cpp
-katehelloworldplugin_la_LIBADD = -lkateinterfaces
+katehelloworldplugin_la_LIBADD = -lkateinterfaces  $(LIB_QT) $(LIB_KDEUI)
 katehelloworldplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 pluginsdir = $(kde_datadir)/kate/plugins/katehelloworld
 plugins_DATA = ui.rc
Index: kdeaddons-3.5.10/kate/htmltools/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/htmltools/Makefile.in
+++ kdeaddons-3.5.10/kate/htmltools/Makefile.in
@@ -362,7 +362,7 @@ kde_module_LTLIBRARIES = katehtmltoolspl
 # This is all standard.  Remove the LIB_KHTML reference if you are not
 # using the KHTML Part
 katehtmltoolsplugin_la_SOURCES = plugin_katehtmltools.cpp
-katehtmltoolsplugin_la_LIBADD = -lkateinterfaces
+katehtmltoolsplugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI)
 katehtmltoolsplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 
 # Install the .rc file in the Part's directory (in this case, the part
Index: kdeaddons-3.5.10/kate/insertcommand/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/insertcommand/Makefile.in
+++ kdeaddons-3.5.10/kate/insertcommand/Makefile.in
@@ -365,7 +365,7 @@ kde_module_LTLIBRARIES = kateinsertcomma
 # This is all standard.  Remove the LIB_KHTML reference if you are not
 # using the KHTML Part
 kateinsertcommandplugin_la_SOURCES = plugin_kateinsertcommand.cpp
-kateinsertcommandplugin_la_LIBADD = -lkateinterfaces
+kateinsertcommandplugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI) $(LIB_KIO)
 kateinsertcommandplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 
 # Install the .rc file in the Part's directory (in this case, the part
Index: kdeaddons-3.5.10/kate/kpybrowser/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/kpybrowser/Makefile.in
+++ kdeaddons-3.5.10/kate/kpybrowser/Makefile.in
@@ -364,7 +364,7 @@ INCLUDES = $(all_includes)
 #>- METASOURCES = AUTO 
 kde_module_LTLIBRARIES = katepybrowseplugin.la
 katepybrowseplugin_la_SOURCES = kpybrowser.cpp pybrowsenode.cpp pybrowse_part.cpp
-katepybrowseplugin_la_LIBADD = -lkateinterfaces
+katepybrowseplugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI)
 katepybrowseplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 pluginsdir = $(kde_datadir)/kate/plugins/katepybrowse
 plugins_DATA = ui.rc
Index: kdeaddons-3.5.10/kate/make/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/make/Makefile.in
+++ kdeaddons-3.5.10/kate/make/Makefile.in
@@ -362,7 +362,7 @@ kde_module_LTLIBRARIES = katemakeplugin.
 # This is all standard.  Remove the LIB_KHTML reference if you are not
 # using the KHTML Part
 katemakeplugin_la_SOURCES = plugin_katemake.cpp
-katemakeplugin_la_LIBADD = -lkateinterfaces
+katemakeplugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI)
 katemakeplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 
 # Install the .rc file in the Part's directory (in this case, the part
Index: kdeaddons-3.5.10/kate/modeline/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/modeline/Makefile.in
+++ kdeaddons-3.5.10/kate/modeline/Makefile.in
@@ -362,7 +362,7 @@ kde_module_LTLIBRARIES = katemodelineplu
 # This is all standard.  Remove the LIB_KHTML reference if you are not
 # using the KHTML Part
 katemodelineplugin_la_SOURCES = ModelinePlugin.cpp
-katemodelineplugin_la_LIBADD = -lkateinterfaces
+katemodelineplugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI)
 katemodelineplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 
 # Install the .rc file in the Part's directory (in this case, the part
Index: kdeaddons-3.5.10/kate/openheader/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/openheader/Makefile.in
+++ kdeaddons-3.5.10/kate/openheader/Makefile.in
@@ -365,7 +365,7 @@ kde_module_LTLIBRARIES = kateopenheaderp
 # This is all standard.  Remove the LIB_KHTML reference if you are not
 # using the KHTML Part
 kateopenheaderplugin_la_SOURCES = plugin_kateopenheader.cpp
-kateopenheaderplugin_la_LIBADD = -lkateinterfaces
+kateopenheaderplugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI) $(LIB_KIO)
 kateopenheaderplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 
 # Install the .rc file in the Part's directory (in this case, the part
Index: kdeaddons-3.5.10/kate/snippets/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/snippets/Makefile.in
+++ kdeaddons-3.5.10/kate/snippets/Makefile.in
@@ -375,7 +375,7 @@ kde_module_LTLIBRARIES = katesnippetsplu
 #>- katesnippetsplugin_la_SOURCES = plugin_katesnippets.cpp CWidgetSnippetsBase.ui cwidgetsnippets.h cwidgetsnippets.cpp csnippet.h csnippet.cpp CWidgetSnippetsBase.ui.h
 #>+ 1
 katesnippetsplugin_la_SOURCES=plugin_katesnippets.cpp  cwidgetsnippets.h cwidgetsnippets.cpp csnippet.h csnippet.cpp CWidgetSnippetsBase.ui.h CWidgetSnippetsBase.cpp
-katesnippetsplugin_la_LIBADD = -lkateinterfaces
+katesnippetsplugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI)
 katesnippetsplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 pluginsdir = $(kde_datadir)/kate/plugins/katesnippets
 plugins_DATA = plugin_katesnippets.rc
Index: kdeaddons-3.5.10/kate/tabbarextension/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/tabbarextension/Makefile.in
+++ kdeaddons-3.5.10/kate/tabbarextension/Makefile.in
@@ -363,7 +363,7 @@ INCLUDES = $(all_includes)
 # Install this plugin in the KDE modules directory
 kde_module_LTLIBRARIES = libkatetabbarextensionplugin.la
 libkatetabbarextensionplugin_la_SOURCES = plugin_katetabbarextension.cpp
-libkatetabbarextensionplugin_la_LIBADD = -lkateinterfaces
+libkatetabbarextensionplugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI) -lktexteditor
 libkatetabbarextensionplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 pluginsdir = $(kde_datadir)/kate/plugins/katetabbarextension
 plugins_DATA = ui.rc
Index: kdeaddons-3.5.10/kate/textfilter/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/textfilter/Makefile.in
+++ kdeaddons-3.5.10/kate/textfilter/Makefile.in
@@ -365,7 +365,7 @@ kde_module_LTLIBRARIES = katetextfilterp
 # This is all standard.  Remove the LIB_KHTML reference if you are not
 # using the KHTML Part
 katetextfilterplugin_la_SOURCES = plugin_katetextfilter.cpp
-katetextfilterplugin_la_LIBADD = -lkateinterfaces
+katetextfilterplugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI) -lktexteditor -lkatepartinterfaces
 katetextfilterplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 
 # Install the .rc file in the Part's directory (in this case, the part
Index: kdeaddons-3.5.10/kate/xmlcheck/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/xmlcheck/Makefile.in
+++ kdeaddons-3.5.10/kate/xmlcheck/Makefile.in
@@ -362,7 +362,7 @@ kde_module_LTLIBRARIES = katexmlcheckplu
 # This is all standard.  Remove the LIB_KHTML reference if you are not
 # using the KHTML Part
 katexmlcheckplugin_la_SOURCES = plugin_katexmlcheck.cpp
-katexmlcheckplugin_la_LIBADD = -lkateinterfaces
+katexmlcheckplugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI)
 katexmlcheckplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 
 # Install the .rc file in the Part's directory (in this case, the part
Index: kdeaddons-3.5.10/kate/xmltools/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/kate/xmltools/Makefile.in
+++ kdeaddons-3.5.10/kate/xmltools/Makefile.in
@@ -370,7 +370,7 @@ kde_module_LTLIBRARIES = katexmltoolsplu
 # This is all standard.  Remove the LIB_KHTML reference if you are not
 # using the KHTML Part
 katexmltoolsplugin_la_SOURCES = pseudo_dtd.cpp plugin_katexmltools.cpp
-katexmltoolsplugin_la_LIBADD = -lkateinterfaces
+katexmltoolsplugin_la_LIBADD = -lkateinterfaces $(LIB_QT) $(LIB_KDEUI) $(LIB_KIO) -lkatepartinterfaces
 katexmltoolsplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 
 # Install the .rc file in the Part's directory (in this case, the part
Index: kdeaddons-3.5.10/konq-plugins/arkplugin/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/konq-plugins/arkplugin/Makefile.in
+++ kdeaddons-3.5.10/konq-plugins/arkplugin/Makefile.in
@@ -359,7 +359,7 @@ INCLUDES = $(all_includes)
 # Install this plugin in the KDE modules directory
 kde_module_LTLIBRARIES = libarkplugin.la
 libarkplugin_la_SOURCES = arkplugin.cpp
-libarkplugin_la_LIBADD = -lkonq
+libarkplugin_la_LIBADD = -lkonq $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) $(LIB_KIO)
 libarkplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 pluginsdir = $(kde_servicesdir)
 plugins_DATA = ark_plugin.desktop
Index: kdeaddons-3.5.10/konq-plugins/dirfilter/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/konq-plugins/dirfilter/Makefile.in
+++ kdeaddons-3.5.10/konq-plugins/dirfilter/Makefile.in
@@ -358,7 +358,7 @@ kde_module_LTLIBRARIES = libdirfilterplu
 AM_CPPFLAGS = $(all_includes)
 #>- METASOURCES = AUTO
 libdirfilterplugin_la_SOURCES = dirfilterplugin.cpp
-libdirfilterplugin_la_LIBADD = -lkonq
+libdirfilterplugin_la_LIBADD = -lkonq  $(LIB_QT) $(LIB_KDECORE) $(LIB_KDEUI) $(LIB_KPARTS)
 libdirfilterplugin_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 iconviewdir = $(kde_datadir)/konqiconview/kpartplugins
 iconview_DATA = dirfilterplugin.rc dirfilterplugin.desktop
Index: kdeaddons-3.5.10/konq-plugins/kimgalleryplugin/Makefile.in
===================================================================
--- kdeaddons-3.5.10.orig/konq-plugins/kimgalleryplugin/Makefile.in
+++ kdeaddons-3.5.10/konq-plugins/kimgalleryplugin/Makefile.in
@@ -361,7 +361,7 @@ INCLUDES = $(all_includes)
 #>- METASOURCES = AUTO
 kde_module_LTLIBRARIES = libkimgallery.la
 libkimgallery_la_SOURCES = imgalleryplugin.cpp imgallerydialog.cpp
-libkimgallery_la_LIBADD = -lkonq
+libkimgallery_la_LIBADD = -lkonq  $(LIB_QT) $(LIB_KDEUI) $(LIB_KIO) $(LIB_KPARTS)
 libkimgallery_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
 iconviewdir = $(kde_datadir)/konqiconview/kpartplugins
 iconview_DATA = kimgalleryplugin.rc kimgalleryplugin.desktop
