diff -Nur kdeaddons-3.5.10-old//konq-plugins/imagerotation/imageconverter.desktop kdeaddons-3.5.10/konq-plugins/imagerotation/imageconverter.desktop
--- kdeaddons-3.5.10-old//konq-plugins/imagerotation/imageconverter.desktop	2011-02-12 18:46:10.824000024 +0200
+++ kdeaddons-3.5.10/konq-plugins/imagerotation/imageconverter.desktop	2011-02-12 18:47:42.131000025 +0200
@@ -58,26 +58,26 @@
 Name[hi]=जेपीईजी
 Name[ne]=जेपीईजी
 Icon=image
-Exec=convert %f "`echo %f | perl -pe 's/\.[^.]+$//'`.jpg"
+Exec=convert "%f" `echo "%f" | sed -e 's/\(.*\)\.\([^.]*$\)/\1.jpg/'`
 
 [Desktop Action convToPNG]
 Name=PNG
 Name[hi]=पीएनजी
 Name[ne]=पीएनजी
 Icon=image
-Exec=convert %f "`echo %f | perl -pe 's/\.[^.]+$//'`.png"
+Exec=convert "%f" `echo "%f" | sed -e 's/\(.*\)\.\([^.]*$\)/\1.png/'`
 
 [Desktop Action convToTIF]
 Name=TIF
 Name[hi]=टिफ़
 Name[ne]=टीआईएफ
 Icon=image
-Exec=convert %f "`echo %f | perl -pe 's/\.[^.]+$//'`.tif"
+Exec=convert "%f" `echo "%f" | sed -e 's/\(.*\)\.\([^.]*$\)/\1.tif/'`
 
 [Desktop Action convToGIF]
 Name=GIF
 Name[hi]=जिफ़
 Name[ne]=जीआईएफ
 Icon=image
-Exec=convert %f "`echo %f | perl -pe 's/\.[^.]+$//'`.gif"
+Exec=convert "%f" `echo "%f" | sed -e 's/\(.*\)\.\([^.]*$\)/\1.gif/'`
 
