diff --git a/man/radeon.man b/man/radeon.man
index 09239cf..2501926 100644
--- a/man/radeon.man
+++ b/man/radeon.man
@@ -346,9 +346,8 @@ and
 .B EXA.
 XAA is the traditional acceleration architecture and support for it is very
 stable.  EXA is a newer acceleration architecture with better performance for
-the Render and Composite extensions, but the rendering code for it is newer and
-possibly unstable.  The default is
-.B XAA.
+the Render and Composite extensions.  The default is
+.B EXA.
 .TP
 .BI "Option \*qAccelDFS\*q \*q" boolean \*q
 Use or don't use accelerated EXA DownloadFromScreen hook when possible (only
diff --git a/src/radeon_driver.c b/src/radeon_driver.c
index c726f8f..1afa181 100644
--- a/src/radeon_driver.c
+++ b/src/radeon_driver.c
@@ -2055,6 +2055,7 @@ static Bool RADEONPreInitAccel(ScrnInfoPtr pScrn)
 #if defined(USE_EXA) && defined(USE_XAA)
     char *optstr;
 #endif
+    int maxy = info->FbMapSize / (pScrn->displayWidth * info->CurrentLayout.pixel_bytes);
 
     if (!(info->accel_state = xcalloc(1, sizeof(struct radeon_accel_state)))) {
 	xf86DrvMsg(pScrn->scrnIndex, X_ERROR, "Unable to allocate accel_state rec!\n");
@@ -2075,16 +2076,20 @@ static Bool RADEONPreInitAccel(ScrnInfoPtr pScrn)
 	info->accel_state->has_tcl = TRUE;
     }
 
-    info->useEXA = FALSE;
     /* if we have shadow fb bail */
     if (info->r600_shadow_fb) 
 	return TRUE;
 
-    if (info->ChipFamily >= CHIP_FAMILY_R600) {
-	xf86DrvMsg(pScrn->scrnIndex, X_DEFAULT,
-	    "Will attempt to use R6xx/R7xx EXA support if DRI is enabled.\n");
+#ifdef XF86DRI
+    if ((!info->directRenderingEnabled) ||
+	(maxy <= pScrn->virtualY * 3) ||
+	(pScrn->videoRam <= 32768))
+	info->useEXA = FALSE;
+    else
 	info->useEXA = TRUE;
-    }
+#else
+	info->useEXA = FALSE;
+#endif
 
     if (!xf86ReturnOptValBool(info->Options, OPTION_NOACCEL, FALSE)) {
 	int errmaj = 0, errmin = 0;
@@ -2099,6 +2104,8 @@ static Bool RADEONPreInitAccel(ScrnInfoPtr pScrn)
 		info->useEXA = TRUE;
 	    } else if (xf86NameCmp(optstr, "XAA") == 0) {
 		from = X_CONFIG;
+		if (info->ChipFamily < CHIP_FAMILY_R600)
+		    info->useEXA = FALSE;
 	    }
 	}
 #else /* USE_XAA */
@@ -2109,6 +2116,9 @@ static Bool RADEONPreInitAccel(ScrnInfoPtr pScrn)
 	    xf86DrvMsg(pScrn->scrnIndex, from,
 		       "Using %s acceleration architecture\n",
 		       info->useEXA ? "EXA" : "XAA");
+	else
+	    xf86DrvMsg(pScrn->scrnIndex, X_DEFAULT,
+		       "Will attempt to use R6xx/R7xx EXA support if DRI is enabled.\n");
 
 #ifdef USE_EXA
 	if (info->useEXA) {
