diff -up ppp-2.4.4/pppd/chap_ms.c.response_len ppp-2.4.4/pppd/chap_ms.c
--- ppp-2.4.4/pppd/chap_ms.c.response_len	2008-03-06 23:07:18.000000000 +0100
+++ ppp-2.4.4/pppd/chap_ms.c	2008-03-06 23:07:49.000000000 +0100
@@ -852,7 +852,7 @@ ChapMS2(u_char *rchallenge, u_char *Peer
     u_char *p = &response[MS_CHAP2_PEER_CHALLENGE];
     int i;
 
-    BZERO(response, sizeof(*response));
+    BZERO(response, MS_CHAP2_RESPONSE_LEN);
 
     /* Generate the Peer-Challenge if requested, or copy it if supplied. */
     if (!PeerChallenge)
