--- boost_1_37_0/boost/function/function_template.hpp.orig	2008-10-16 15:21:50.000000000 +0200
+++ boost_1_37_0/boost/function/function_template.hpp	2008-11-13 23:27:43.000000000 +0100
@@ -950,10 +950,10 @@
           f.vtable->manager(f.functor, this->functor,
                             boost::detail::function::move_functor_tag);
 		  f.vtable = 0;
-#if !defined(BOOST_NO_EXCEPTIONS)      
         } else {
           clear();
         }
+#if !defined(BOOST_NO_EXCEPTIONS)      
       } catch (...) {
         vtable = 0;
         throw;
