--- boost_1_37_0/boost/dynamic_bitset/dynamic_bitset.hpp.debug	2008-10-22 03:13:59.000000000 +0900
+++ boost_1_37_0/boost/dynamic_bitset/dynamic_bitset.hpp	2009-03-23 03:36:40.000000000 +0900
@@ -1017,8 +1017,11 @@
                           ? access_by_bytes
                           : access_by_blocks;
 
+    if (mode)
+       return do_count(m_bits.begin(), num_blocks(), Block(0),
+                                          static_cast<value_to_type<true> *>(0));
     return do_count(m_bits.begin(), num_blocks(), Block(0),
-                                       static_cast<value_to_type<mode> *>(0));
+                                       static_cast<value_to_type<false> *>(0));
 }
 
 
