Index: k3b-1.0.5/admin/cvs.sh
===================================================================
--- k3b-1.0.5.orig/admin/cvs.sh
+++ k3b-1.0.5/admin/cvs.sh
@@ -68,7 +68,7 @@ case $AUTOMAKE_STRING in
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10*)
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.1[0-5]*)
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
Index: k3b-1.0.5/libk3b/core/Makefile.am
===================================================================
--- k3b-1.0.5.orig/libk3b/core/Makefile.am
+++ k3b-1.0.5/libk3b/core/Makefile.am
@@ -13,7 +13,7 @@ libk3bcore_la_SOURCES = k3bcore.cpp k3bg
 include_HEADERS = k3bcore.h k3bdefaultexternalprograms.h k3bexternalbinmanager.h \
 	k3bprocess.h k3bversion.h k3bglobals.h k3bjob.h k3bthread.h \
 	k3bthreadjob.h k3bglobalsettings.h k3bjobhandler.h \
-	k3b_export.h k3bjobhandler.h k3bsimplejobhandler.h
+	k3b_export.h k3bsimplejobhandler.h
 
 METASOURCES = AUTO
 
