diff -up k3b-1.0.5/src/Makefile.in.hidden k3b-1.0.5/src/Makefile.in
--- k3b-1.0.5/src/Makefile.in.hidden	2008-05-27 03:34:44.000000000 -0500
+++ k3b-1.0.5/src/Makefile.in	2008-08-27 14:18:54.000000000 -0500
@@ -512,9 +512,9 @@ k3b_LDADD = ./option/liboption.la ./rip/
 	-lkio -lkparts $(CAM_LIB) $(MUSICBRAINZ_LIBS)
 
 SUBDIRS = option misc rip projects fastscale pics icons konqi mimetypes sounds
-xdg_apps_DATA = k3b.desktop
-silent_DATA = k3b-iso.desktop k3b-cue.desktop
-silentdir = $(kde_appsdir)/.hidden/
+xdg_apps_DATA = k3b.desktop k3b-iso.desktop k3b-cue.desktop
+#silent_DATA = k3b-iso.desktop k3b-cue.desktop
+#silentdir = $(kde_appsdir)/.hidden/
 
 ###################
 # this 10 paths are KDE specific. Use them:
