--- k3b-1.0.5-guppak/configure.in	2008-06-22 13:43:55.000000000 +0300
+++ k3b-1.0.5/configure.in	2008-06-22 13:45:42.000000000 +0300
@@ -576,8 +576,8 @@
 	AC_LANG_CPLUSPLUS
 	AC_COMPILE_IFELSE( 
 		extern "C" {
-		#include <ffmpeg/avformat.h>
-		#include <ffmpeg/avcodec.h>
+		#include <libavformat/avformat.h>
+		#include <libavcodec/avcodec.h>
 		}
 
 		int main() {
@@ -591,8 +591,8 @@
 	LIBS="-lavformat -lavcodec $LIBS"
 	AC_LINK_IFELSE( 
 		extern "C" {
-		#include <ffmpeg/avformat.h>
-		#include <ffmpeg/avcodec.h>
+		#include <libavformat/avformat.h>
+		#include <libavcodec/avcodec.h>
 		}
 
 		int main() {
diff -Naur k3b-1.0.5-guppak/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp
--- k3b-1.0.5-guppak/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp	2008-06-22 14:08:34.000000000 +0300
+++ k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp	2008-06-22 14:10:07.000000000 +0300
@@ -22,7 +22,7 @@
 #include <k3bpluginfactory.h>
 
 extern "C" {
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 }
 
 #include <math.h>
diff -Naur k3b-1.0.5-guppak/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
--- k3b-1.0.5-guppak/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2008-06-22 14:08:34.000000000 +0300
+++ k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2008-06-22 14:09:37.000000000 +0300
@@ -18,8 +18,8 @@
 #include "k3bffmpegwrapper.h"
 
 extern "C" {
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 }
 
 #include <string.h>
