Index: texlive-20080816-source/texk/web2c/xetexdir/xetex.mk
===================================================================
--- texlive-20080816-source.orig/texlive-20080816-source/texk/web2c/xetexdir/xetex.mk
+++ texlive-20080816-source/texlive-20080816-source/texk/web2c/xetexdir/xetex.mk
@@ -168,7 +168,7 @@ pngimage.o: $(srcdir)/xetexdir/pngimage.
 	$(compile) $(ALL_CFLAGS) $(FTFLAGS) $(LIBPNGCPPFLAGS) $(ZLIBCPPFLAGS) -c $< -o $@
 
 pdfimage.o: $(srcdir)/xetexdir/pdfimage.cpp $(srcdir)/xetexdir/pdfimage.h
-	$(CXX) $(ALL_CFLAGS) $(FTFLAGS) $(LIBXPDFCPPFLAGS) -c $< -o $@
+	$(CXX) $(ALL_CXXFLAGS) $(FTFLAGS) $(LIBXPDFCPPFLAGS) -c $< -o $@
 
 XeTeX_pic.o: $(srcdir)/xetexdir/XeTeX_pic.c $(srcdir)/xetexdir/XeTeX_ext.h $(XeTeXImageHdrs)
 	$(compile) $(TECKITFLAGS) $(FTFLAGS) $(ALL_CFLAGS) $(XETEX_DEFINES) -c $< -o $@
