Index: gle-3.1.0/doc/Makefile.am
===================================================================
--- gle-3.1.0/doc/Makefile.am
+++ gle-3.1.0/doc/Makefile.am	2008-05-23 19:33:55.000000000 +0200
@@ -7,5 +7,5 @@
 
 EXTRA_DIST = .cvsignore COPYING.artistic gle-3.0.lsm
 
-SUFFIXES += .lsm
+SUFFIXES = .lsm
 
Index: gle-3.1.0/doc/html/Makefile.am
===================================================================
--- gle-3.1.0/doc/html/Makefile.am
+++ gle-3.1.0/doc/html/Makefile.am	2008-05-23 19:34:24.000000000 +0200
@@ -87,5 +87,5 @@
 
 EXTRA_DIST = .cvsignore $(doc_DATA)
 
-SUFFIXES += .html .gif .jpg
+SUFFIXES = .html .gif .jpg
 
Index: gle-3.1.0/man/Makefile.am
===================================================================
--- gle-3.1.0/man/Makefile.am
+++ gle-3.1.0/man/Makefile.am	2008-05-23 19:36:53.000000000 +0200
@@ -22,7 +22,7 @@
   $(man_RAW) \
   Makefile.sgi
 
-SUFFIXES += .man .3gle
+SUFFIXES = .man .3gle
 
 # for the ordinary install, just copy the raw files ....
 .man.3gle:
@@ -38,4 +38,4 @@
 	psroff -t -man < $< > $*.ps
 
 
-CLEANFILES += $(man_MANS)
+CLEANFILES = $(man_MANS)
