From 09c86c50bf72501f764c0201d1df89566afe8fca Mon Sep 17 00:00:00 2001
From: Dhaval Giani <dhaval.giani@gmail.com>
Date: Tue, 18 May 2010 00:30:12 +0200
Subject: [PATCH] libcgroup tests: Fix build issue with the named hierarchy test case

Signed-off-by: Dhaval Giani <dhaval.giani@gmail.com>
---
 tests/test_named_hierarchy.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/tests/test_named_hierarchy.c b/tests/test_named_hierarchy.c
index c9375ed..56e9a1c 100644
--- a/tests/test_named_hierarchy.c
+++ b/tests/test_named_hierarchy.c
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <stdlib.h>
 #include <libcgroup.h>
 
 /*
-- 
1.6.3

