diff -Nur kvm-82.orig/kernel/Makefile kvm-82/kernel/Makefile
--- Makefile	2009-01-11 01:09:39.000000000 +0200
+++ Makefile	2009-01-11 01:11:53.000000000 +0200
@@ -103,11 +103,6 @@
 install:
 	mkdir -p $(DESTDIR)/$(INSTALLDIR)
 	cp $(ARCH_DIR)/*.ko $(DESTDIR)/$(INSTALLDIR)
-	for i in $(ORIGMODDIR)/drivers/kvm/*.ko \
-		 $(ORIGMODDIR)/arch/$(ARCH_DIR)/kvm/*.ko; do \
-		if [ -f "$$i" ]; then mv "$$i" "$$i.orig"; fi; \
-	done
-	/sbin/depmod -a $(DEPMOD_VERSION)
 
 tmpspec = .tmp.kvm-kmod.spec
 
