diff --git a/Makefile b/Makefile
index 76abff5..82c37eb 100644
--- a/Makefile
+++ b/Makefile
@@ -42,7 +42,7 @@ BOBJECTS = $(BTARGET) \
 	mbr/mbr_c.bin mbr/altmbr_c.bin mbr/gptmbr_c.bin \
 	mbr/mbr_f.bin mbr/altmbr_f.bin mbr/gptmbr_f.bin \
 	core/pxelinux.0 core/isolinux.bin core/isolinux-debug.bin \
-	gpxe/gpxelinux.0 dos/syslinux.com win32/syslinux.exe \
+	gpxe/gpxelinux.0 dos/syslinux.com \
 	$(MODULES)
 
 # BSUBDIRs build the on-target binary components.
@@ -52,7 +52,7 @@ BOBJECTS = $(BTARGET) \
 # files that depend only on the B phase, but may have to be regenerated
 # for "make installer".
 BSUBDIRS = codepage core memdisk modules com32 mbr memdump gpxe sample \
-	   libinstaller dos win32
+	   libinstaller dos 
 ITARGET  =
 IOBJECTS = $(ITARGET) dos/copybs.com \
 	utils/gethostip utils/isohybrid utils/mkdiskimage \
@@ -66,9 +66,9 @@ INSTALL_SBIN  = extlinux/extlinux
 # Things to install in /usr/lib/syslinux
 INSTALL_AUX   =	core/pxelinux.0 gpxe/gpxelinux.0 core/isolinux.bin \
 		core/isolinux-debug.bin \
-		dos/syslinux.com dos/copybs.com win32/syslinux.exe \
+		dos/syslinux.com dos/copybs.com \
 		mbr/*.bin $(MODULES)
-INSTALL_AUX_OPT = win32/syslinux.exe
+INSTALL_AUX_OPT =
 
 # These directories manage their own installables
 INSTALLSUBDIRS = com32 utils
