Index: pam_pkcs11-0.6.6/src/pam_pkcs11/pam_config.c
===================================================================
--- pam_pkcs11-0.6.6.orig/src/pam_pkcs11/pam_config.c
+++ pam_pkcs11-0.6.6/src/pam_pkcs11/pam_config.c
@@ -271,6 +271,10 @@ struct configuration_st *pk_configure( i
       		configuration.wait_for_card = 0;
 		continue;
 	   }
+    	   if (strcmp("card_only", argv[i]) == 0) {
+      		configuration.card_only = 1;
+		continue;
+	   }
     	   if (strcmp("debug", argv[i]) == 0) {
       		configuration.debug = 1;
 		set_debug_level(1);
