--- common/mp4ff/Makefile.am	2007-01-23 20:37:04.208889058 +0200
+++ common/mp4ff/Makefile.am	2007-01-23 20:50:10.014882401 +0200
@@ -1,4 +1,4 @@
-noinst_LTLIBRARIES = libmp4ff.la
+lib_LTLIBRARIES = libmp4ff.la
 
 libmp4ff_la_CFLAGS = -DUSE_TAGGING=1 -fpic
 
--- common/mp4ff/Makefile.am	2007-01-23 20:51:02.454611671 +0200
+++ common/mp4ff/Makefile.am	2007-01-23 20:52:48.047354522 +0200
@@ -1,4 +1,5 @@
 lib_LTLIBRARIES = libmp4ff.la
+include_HEADERS = mp4ff.h
 
 libmp4ff_la_CFLAGS = -DUSE_TAGGING=1 -fpic
 
