--- xscreensaver-5.10/hacks/analogtv.c.debug1	2008-08-06 10:59:35.000000000 +0900
+++ xscreensaver-5.10/hacks/analogtv.c	2009-12-11 02:55:42.000000000 +0900
@@ -1030,11 +1030,11 @@
       it->leveltable[height][0].index=0;
     }
     if (avgheight>=5) {
-      it->leveltable[height][height-1].index=0;
+      if (height >= 1) it->leveltable[height][height-1].index=0;
     }
     if (avgheight>=7) {
       it->leveltable[height][1].index=1;
-      it->leveltable[height][height-2].index=1;
+      if (height >= 2) it->leveltable[height][height-2].index=1;
     }
 
     for (i=0; i<height; i++) {
