Index: gdb/ia64-tdep.c
===================================================================
RCS file: /cvs/src/src/gdb/ia64-tdep.c,v
retrieving revision 1.172
retrieving revision 1.172.2.1
diff -u -a -p -u -p -a -r1.172 -r1.172.2.1
--- gdb/ia64-tdep.c	20 Feb 2008 14:31:40 -0000	1.172
+++ gdb/ia64-tdep.c	31 Mar 2008 03:40:42 -0000	1.172.2.1
@@ -1234,7 +1234,7 @@ examine_prologue (CORE_ADDR pc, CORE_ADD
 	      spill_reg   = rN;
 	      last_prologue_pc = next_pc;
 	    }
-	  else if (qp == 0 && rM >= 32 && rM < 40 && !instores[rM] && 
+	  else if (qp == 0 && rM >= 32 && rM < 40 && !instores[rM-32] && 
 		   rN < 256 && imm == 0)
 	    {
 	      /* mov rN, rM where rM is an input register */
