--- gdb/dwarf2read.c.~1.217.~	2007-04-12 11:24:48.000000000 +0200
+++ gdb/dwarf2read.c	2007-04-13 15:37:15.000000000 +0200
@@ -1194,6 +1194,9 @@ dwarf2_locate_sections (bfd *abfd, asect
 void
 dwarf2_build_psymtabs (struct objfile *objfile, int mainline)
 {
+  if (objfile->sect_index_text == -1)
+    return;
+
   /* We definitely need the .debug_info and .debug_abbrev sections */
 
   dwarf2_per_objfile->info_buffer = dwarf2_read_section (objfile, dwarf_info_section);
--- gdb/ia64-tdep.c	28 Mar 2007 18:56:06 -0000	1.148
+++ gdb/ia64-tdep.c	13 Apr 2007 12:30:47 -0000
@@ -2692,6 +2692,10 @@ ia64_get_dyn_info_list (unw_addr_space_t
     {
       void *buf = NULL;
 
+      if (objfile->sect_index_text == -1)
+	/* No text section.  */
+	continue;
+
       text_sec = objfile->sections + SECT_OFF_TEXT (objfile);
       ip = text_sec->addr;
       ret = ia64_find_unwind_table (objfile, ip, &di, &buf);
