Index: linux-2.6.35/net/irda/af_irda.c
===================================================================
--- linux-2.6.35.orig/net/irda/af_irda.c
+++ linux-2.6.35/net/irda/af_irda.c
@@ -2284,8 +2284,7 @@ static int __irda_getsockopt(struct sock
 			sizeof(struct irda_device_info);
 
 		if (len < offset) {
-			err = -EINVAL;
-			goto out;
+			return -EINVAL;
 		}
 
 		/* Ask lmp for the current discovery log */
