Index: hunspell-1.2.7/src/tools/hunspell.cxx
===================================================================
--- hunspell-1.2.7.orig/src/tools/hunspell.cxx
+++ hunspell-1.2.7/src/tools/hunspell.cxx
@@ -67,9 +67,13 @@
     "/usr/share/myspell:" \
     "/usr/share/myspell/dicts"
 #define USEROOODIR \
+    ".openoffice.org/3/user/wordbook:" \
     ".openoffice.org2/user/wordbook:" \
     ".openoffice.org2.0/user/wordbook"
 #define OOODIR \
+    "/opt/OpenOffice.org/lib/ooo-2.4/share/dict/ooo:" \
+    "/usr/lib/openoffice/basis3.0/share/dict/ooo:" \
+    "/usr/lib/openoffice/share/dict/ooo:" \
     "/opt/openoffice.org/basis3.0/share/dict/ooo:" \
     "/usr/lib/openoffice.org/basis3.0/share/dict/ooo:" \
     "/opt/openoffice.org2.4/share/dict/ooo:" \
