Index: xf86-input-synaptics-1.1.0/src/synaptics.c
===================================================================
--- xf86-input-synaptics-1.1.0.orig/src/synaptics.c
+++ xf86-input-synaptics-1.1.0/src/synaptics.c
@@ -477,8 +477,8 @@ static void set_default_parameters(Local
     pars->press_motion_min_z = xf86SetIntOption(opts, "PressureMotionMinZ", pressureMotionMinZ);
     pars->press_motion_max_z = xf86SetIntOption(opts, "PressureMotionMaxZ", pressureMotionMaxZ);
 
-    pars->min_speed = xf86SetRealOption(opts, "MinSpeed", 0.4);
-    pars->max_speed = xf86SetRealOption(opts, "MaxSpeed", 0.7);
+    pars->min_speed = xf86SetRealOption(opts, "MinSpeed", 0.2);
+    pars->max_speed = xf86SetRealOption(opts, "MaxSpeed", 0.5);
     pars->accl = xf86SetRealOption(opts, "AccelFactor", accelFactor);
     pars->trackstick_speed = xf86SetRealOption(opts, "TrackstickSpeed", 40);
     pars->scroll_dist_circ = xf86SetRealOption(opts, "CircScrollDelta", 0.1);
