diff -ur ktorrent-2.2.4.orig/plugins/search/searchenginelist.cpp ktorrent-2.2.4/plugins/search/searchenginelist.cpp
--- ktorrent-2.2.4.orig/plugins/search/searchenginelist.cpp	2007-11-20 21:09:08.000000000 +0200
+++ ktorrent-2.2.4/plugins/search/searchenginelist.cpp	2008-01-14 21:56:36.000000000 +0200
@@ -113,6 +113,7 @@
 		out << "bytenova.org http://www.bitenova.org/search.php?search=FOOBAR&start=0&start=0&ie=utf-8&oe=utf-8" << ::endl; 
 		out << "torrentspy.com http://torrentspy.com/search.asp?query=FOOBAR" << ::endl; 
 		out << "torrentz.com http://www.torrentz.com/search_FOOBAR" << ::endl; 
+		out << "seedpeer.com http://www.seedpeer.com/search.php?search=FOOBAR" << ::endl; 
 	}
 		
 	KURL SearchEngineList::getSearchURL(bt::Uint32 engine) const
diff -ur ktorrent-2.2.4.orig/plugins/search/searchprefpage.cpp ktorrent-2.2.4/plugins/search/searchprefpage.cpp
--- ktorrent-2.2.4.orig/plugins/search/searchprefpage.cpp	2007-11-20 21:09:08.000000000 +0200
+++ ktorrent-2.2.4/plugins/search/searchprefpage.cpp	2008-01-14 21:57:05.000000000 +0200
@@ -176,6 +176,8 @@
 		se = new QListViewItem(m_engines, "torrentspy.com", "http://torrentspy.com/search.asp?query=FOOBAR");
 
 		se = new QListViewItem(m_engines, "torrentz.com", "http://www.torrentz.com/search_FOOBAR");
+
+		se = new QListViewItem(m_engines, "seedpeer.com", "http://www.seedpeer.com/search.php?search=FOOBAR");
 	}
  
 	void SearchPrefPageWidget::removeAllClicked()
