Index: ktorrent-2.2.7/apps/ktorrent/ktorrent.cpp
===================================================================
--- ktorrent-2.2.7.orig/apps/ktorrent/ktorrent.cpp
+++ ktorrent-2.2.7/apps/ktorrent/ktorrent.cpp
@@ -512,31 +512,31 @@ void KTorrent::setupActions()
 	KAction* pref = KStdAction::preferences(this, SLOT(optionsPreferences()), actionCollection());
 
 	m_start = new KAction(
-			i18n("to start", "Start"), "ktstart",0,this, SLOT(startDownload()),
+			i18n("to start", "Start"), "1rightarrow",0,this, SLOT(startDownload()),
 			actionCollection(), "Start");
 
 	m_stop = new KAction(
-			i18n("to stop", "Stop"), "ktstop",0,this, SLOT(stopDownload()),
+			i18n("to stop", "Stop"), "stop",0,this, SLOT(stopDownload()),
 			actionCollection(), "Stop");
 
 	m_remove = new KAction(
-			i18n("Remove"), "ktremove",0,this, SLOT(removeDownload()),
+			i18n("Remove"), "edittrash",0,this, SLOT(removeDownload()),
 			actionCollection(), "Remove");
 	
 	m_startall = new KAction(
-			i18n("to start all", "Start All"), "ktstart_all",0,this, SLOT(startAllDownloadsCurrentView()),
+			i18n("to start all", "Start All"), "2rightarrow",0,this, SLOT(startAllDownloadsCurrentView()),
 			actionCollection(), "Start all");
 	
-	m_startall_systray = new KAction(i18n("to start all", "Start All"), "ktstart_all",0,this, SLOT(startAllDownloads()),actionCollection());
+	m_startall_systray = new KAction(i18n("to start all", "Start All"), "2rightarrow",0,this, SLOT(startAllDownloads()),actionCollection());
 	
 	m_stopall = new KAction(
-			i18n("to stop all", "Stop All"), "ktstop_all",0,this, SLOT(stopAllDownloadsCurrentView()),
+			i18n("to stop all", "Stop All"), "stop",0,this, SLOT(stopAllDownloadsCurrentView()),
 			actionCollection(), "Stop all");
 	
-	m_stopall_systray = new KAction(i18n("to stop all", "Stop All"), "ktstop_all",0,this, SLOT(stopAllDownloads()),actionCollection());
+	m_stopall_systray = new KAction(i18n("to stop all", "Stop All"), "stop",0,this, SLOT(stopAllDownloads()),actionCollection());
 	
 	m_pasteurl = new KAction(
-			i18n("to paste torrent URL", "Paste Torrent URL..."), "ktstart",0,this, SLOT(torrentPaste()),
+			i18n("to paste torrent URL", "Paste Torrent URL..."), "1rightarrow",0,this, SLOT(torrentPaste()),
 			actionCollection(), "paste_url");
 	
 	m_queuemgr = new KAction(
Index: ktorrent-2.2.7/apps/ktorrent/ktorrentviewmenu.cpp
===================================================================
--- ktorrent-2.2.7.orig/apps/ktorrent/ktorrentviewmenu.cpp
+++ ktorrent-2.2.7/apps/ktorrent/ktorrentviewmenu.cpp
@@ -34,19 +34,19 @@ KTorrentViewMenu::KTorrentViewMenu (KTor
 	KIconLoader* iload = KGlobal::iconLoader();
 	
 	stop_id = insertItem(
-			iload->loadIconSet("ktstop",KIcon::Small),i18n("to stop", "Stop"),
+			iload->loadIconSet("stop",KIcon::Small),i18n("to stop", "Stop"),
 			parent,SLOT(stopDownloads()));
 
 	start_id = insertItem(
-			iload->loadIconSet("ktstart",KIcon::Small),i18n("to start", "Start"),
+			iload->loadIconSet("1rightarrow",KIcon::Small),i18n("to start", "Start"),
 			parent,SLOT(startDownloads()));
 
 	remove_id = insertItem(
-			iload->loadIconSet("ktremove",KIcon::Small),i18n("Remove Torrent"),
+			iload->loadIconSet("edittrash",KIcon::Small),i18n("Remove Torrent"),
 			parent,SLOT(removeDownloads()));
 	
 	remove_all_id = insertItem(
-			iload->loadIconSet("ktremove",KIcon::Small),i18n("Remove Torrent and Data"),
+			iload->loadIconSet("edittrash",KIcon::Small),i18n("Remove Torrent and Data"),
 			parent,SLOT(removeDownloadsAndData()));
 	
 	queue_id = insertItem(
