diff -Nur yali4.orig/yali4/constants.py yali4/yali4/constants.py
--- yali4.orig/yali4/constants.py	2009-10-02 19:15:28.000000000 +0300
+++ yali4/yali4/constants.py	2009-11-16 00:40:22.590676634 +0200
@@ -89,8 +89,8 @@
 consts.cd_repo_uri = join(consts.source_dir, "repo/pisi-index.xml.bz2")
 
 # pardus repository
-consts.pardus_repo_name = "pardus-2009"
-consts.pardus_repo_uri = "http://packages.pardus.org.tr/pardus-2009/pisi-index.xml.bz2"
+consts.pardus_repo_name = "pardus-corporate2"
+consts.pardus_repo_uri = "http://packages.pardus.org.tr/corporate2-test/pisi-index.xml.bz2"
 consts.pardus_release_path = "etc/pardus-release"
 
 # min root partition size
