--- Lib/proto.h	1998-11-30 18:26:50.000000000 +0200
+++ Lib/proto.h	2006-01-10 00:34:14.000000000 +0200
@@ -459,7 +459,6 @@
 void srand48(long);
 int ispow2(int);
 void InitRandom(int);
-int log2(int);
 
 
 
--- Lib/rename.h	1998-11-30 18:26:50.000000000 +0200
+++ Lib/rename.h	2006-01-10 00:34:57.000000000 +0200
@@ -410,7 +410,6 @@
 #define RandomPermute			__RandomPermute
 #define ispow2				__ispow2
 #define InitRandom			__InitRandom
-#define log2				__log2
 
 
 
--- Lib/util.c	1998-11-30 18:26:50.000000000 +0200
+++ Lib/util.c	2006-01-10 00:34:43.000000000 +0200
@@ -506,14 +506,3 @@
   }
 }
 
-/*************************************************************************
-* This function returns the log2(x)
-**************************************************************************/
-int log2(int a)
-{
-  int i;
-
-  for (i=1; a > 1; i++, a = a>>1);
-  return i-1;
-}
-
