Index: linux-2.6.30/fs/fuse/file.c
===================================================================
--- linux-2.6.30.orig/fs/fuse/file.c
+++ linux-2.6.30/fs/fuse/file.c
@@ -1054,7 +1054,8 @@ static ssize_t fuse_direct_io(struct fil
 				break;
 		}
 	}
-	fuse_put_request(fc, req);
+	if (!IS_ERR(req))
+		fuse_put_request(fc, req);
 	if (res > 0) {
 		if (write)
 			fuse_write_update_size(inode, pos);
