Index: arts-1.5.4/acinclude.m4
===================================================================
--- arts-1.5.4.orig/acinclude.m4
+++ arts-1.5.4/acinclude.m4
@@ -12040,28 +12040,12 @@ fi
 ])# AC_LTDL_SYS_DLOPEN_DEPLIBS
 
 AC_DEFUN([AC_LTDL_SHLIBEXT],
-[AC_REQUIRE([AC_LTDL_SNARF_CONFIG])dnl
+[AC_REQUIRE([AC_LIBTOOL_SYS_DYNAMIC_LINKER])dnl
 AC_CACHE_CHECK([which extension is used for shared libraries],
   libltdl_cv_shlibext, [dnl
-(
-  last=
-	case "$host_os" in
-    cygwin* | mingw*) 
-      last=".dll" 
-      ;;
-    *)
-      for spec in $library_names_spec; do
-        last="$spec"
-      done
-      ;;
-  esac
-changequote(, )
-  echo "$last" | sed 's/\[.*\]//;s/^[^.]*//;s/\$.*$//;s/\.$//' > conftest
-changequote([, ])
-)
-libltdl_cv_shlibext=`cat conftest`
-rm -f conftest
-])
+module=yes
+eval libltdl_cv_shlibext=$shrext
+  ])
 if test -n "$libltdl_cv_shlibext"; then
   AC_DEFINE_UNQUOTED(LTDL_SHLIB_EXT, "$libltdl_cv_shlibext",
     [Define to the extension used for shared libraries, say, ".so". ])
Index: arts-1.5.4/libltdl/ltdl.m4
===================================================================
--- arts-1.5.4.orig/libltdl/ltdl.m4
+++ arts-1.5.4/libltdl/ltdl.m4
@@ -118,27 +118,11 @@ fi
 ])# AC_LTDL_SYS_DLOPEN_DEPLIBS
 
 AC_DEFUN([AC_LTDL_SHLIBEXT],
-[AC_REQUIRE([AC_LTDL_SNARF_CONFIG])dnl
+[AC_REQUIRE([AC_LIBTOOL_SYS_DYNAMIC_LINKER])dnl
 AC_CACHE_CHECK([which extension is used for shared libraries],
   libltdl_cv_shlibext, [dnl
-(
-  last=
-	case "$host_os" in
-    cygwin* | mingw*) 
-      last=".dll" 
-      ;;
-    *)
-      for spec in $library_names_spec; do
-        last="$spec"
-      done
-      ;;
-  esac
-changequote(, )
-  echo "$last" | sed 's/\[.*\]//;s/^[^.]*//;s/\$.*$//;s/\.$//' > conftest
-changequote([, ])
-)
-libltdl_cv_shlibext=`cat conftest`
-rm -f conftest
+module=yes
+eval libltdl_cv_shlibext=$shrext
 ])
 if test -n "$libltdl_cv_shlibext"; then
   AC_DEFINE_UNQUOTED(LTDL_SHLIB_EXT, "$libltdl_cv_shlibext",
