--- kdelibs-3.5.0/kioslave/http/http.cc.101956	2005-11-08 16:39:17.000000000 -0600
+++ kdelibs-3.5.0/kioslave/http/http.cc	2005-12-01 13:23:39.000000000 -0600
@@ -2103,6 +2103,8 @@
     return false;
   }
 
+  bool bCacheOnly = (m_request.cache == KIO::CC_CacheOnly);
+
   m_request.fcache = 0;
   m_request.bCachedRead = false;
   m_request.bCachedWrite = false;
@@ -2114,7 +2116,6 @@
   {
      m_request.fcache = checkCacheEntry( );
 
-     bool bCacheOnly = (m_request.cache == KIO::CC_CacheOnly);
      bool bOffline = isOffline(m_request.doProxy ? m_proxyURL : m_request.url);
      if (bOffline && (m_request.cache != KIO::CC_Reload))
         m_request.cache = KIO::CC_CacheOnly;
@@ -2156,6 +2157,12 @@
         return false;
      }
   }
+  else if (bCacheOnly)
+  {
+     // Instructed to only check cache, but _not_ to use cache. Oops.
+     error( ERR_DOES_NOT_EXIST, m_request.url.url() );
+     return false;
+  }
 
   QString header;
   QString davHeader;
