Index: digikam-0.9.6/digikam/libs/widgets/common/splashscreen.cpp
===================================================================
--- digikam-0.9.6.orig/digikam/libs/widgets/common/splashscreen.cpp
+++ digikam-0.9.6/digikam/libs/widgets/common/splashscreen.cpp
@@ -52,7 +52,7 @@ public:
         state           = 0;
         progressBarSize = 3;
         state           = 0;
-        color           = Qt::black;
+        color           = Qt::white;
         alignment       = Qt::AlignLeft;
     }
 
