diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur yakuake-orig/src/yakuake.kcfg yakuake/src/yakuake.kcfg
--- yakuake-orig/src/yakuake.kcfg	2009-02-13 14:10:08.000000000 +0200
+++ yakuake/src/yakuake.kcfg	2009-02-13 14:10:19.000000000 +0200
@@ -32,7 +32,7 @@
       <whatsthis></whatsthis>
     </entry>
     <entry key="height" type="Int">
-      <default>50</default>
+      <default>90</default>
       <label></label>
       <whatsthis></whatsthis>
     </entry>
