diff -Nuar ahenk-lider-1.9.34.orig/lider/plugins/plugin_firewall/main.py ahenk-lider-1.9.34/lider/plugins/plugin_firewall/main.py
--- ahenk-lider-1.9.34.orig/lider/plugins/plugin_firewall/main.py	2011-11-22 16:11:53.585000014 +0200
+++ ahenk-lider-1.9.34/lider/plugins/plugin_firewall/main.py	2011-11-22 16:13:25.234000015 +0200
@@ -207,8 +207,9 @@
 
         firewallRules = rules_xml + ":" + rules_compiled
 
+        # No support for firewall state for now
+        # "firewallState": [firewallState],
         policy = {
-            "firewallState": [firewallState],
             "firewallRules": [firewallRules],
         }
         return policy
