diff -Nuar ahenk-lider-1.9.34.orig/lider/formmain.py ahenk-lider-1.9.34/lider/formmain.py
--- ahenk-lider-1.9.34.orig/lider/formmain.py	2011-11-21 03:11:37.775000100 +0200
+++ ahenk-lider-1.9.34/lider/formmain.py	2011-11-21 03:14:12.599000100 +0200
@@ -1413,6 +1413,7 @@
         else:
             item = self.items[0]
 
+            widget.policy = self.__load_policy(item)
             classes_now, policy_now, classes_new, policy_new = widget.mod_policy(remove=remove)
 
             try:
