diff -up evolution-2.30.1.2/mail/e-mail-reader.c.notjunk-icon evolution-2.30.1.2/mail/e-mail-reader.c
--- evolution-2.30.1.2/mail/e-mail-reader.c.notjunk-icon	2010-04-26 09:13:17.000000000 -0400
+++ evolution-2.30.1.2/mail/e-mail-reader.c	2010-05-01 15:28:20.750200600 -0400
@@ -1292,8 +1292,8 @@ static GtkActionEntry mail_reader_entrie
 	  N_("Mark the selected messages as junk"),
 	  G_CALLBACK (action_mail_mark_junk_cb) },
 
-	{ "mail-mark-not-junk",
-	  "mail-mark-not-junk",
+	{ "mail-mark-notjunk",
+	  "mail-mark-notjunk",
 	  N_("_Not Junk"),
 	  "<Shift><Control>j",
 	  N_("Mark the selected messages as not being junk"),
@@ -1602,9 +1602,9 @@ static EPopupActionEntry mail_reader_pop
 	  N_("Mark as _Junk"),
 	  "mail-mark-junk" },
 
-	{ "mail-popup-mark-not-junk",
+	{ "mail-popup-mark-notjunk",
 	  N_("Mark as _Not Junk"),
-	  "mail-mark-not-junk" },
+	  "mail-mark-notjunk" },
 
 	{ "mail-popup-mark-read",
 	  N_("Mar_k as Read"),
@@ -2316,7 +2316,7 @@ mail_reader_update_actions (EMailReader 
 	action = e_mail_reader_get_action (reader, action_name);
 	gtk_action_set_sensitive (action, sensitive);
 
-	action_name = "mail-mark-not-junk";
+	action_name = "mail-mark-notjunk";
 	sensitive = selection_has_junk_messages;
 	action = e_mail_reader_get_action (reader, action_name);
 	gtk_action_set_sensitive (action, sensitive);
diff -up evolution-2.30.1.2/modules/mail/e-mail-shell-view-actions.c.notjunk-icon evolution-2.30.1.2/modules/mail/e-mail-shell-view-actions.c
--- evolution-2.30.1.2/modules/mail/e-mail-shell-view-actions.c.notjunk-icon	2010-04-26 09:13:17.000000000 -0400
+++ evolution-2.30.1.2/modules/mail/e-mail-shell-view-actions.c	2010-05-01 15:28:20.751200588 -0400
@@ -1223,7 +1223,7 @@ static GtkRadioActionEntry mail_filter_e
 	  MAIL_FILTER_LAST_5_DAYS_MESSAGES },
 
 	{ "mail-filter-messages-not-junk",
-	  "mail-mark-not-junk",
+	  "mail-mark-notjunk",
 	  N_("Messages Not Junk"),
 	  NULL,
 	  NULL,  /* XXX Add a tooltip! */
diff -up evolution-2.30.1.2/modules/mail/e-mail-shell-view-actions.h.notjunk-icon evolution-2.30.1.2/modules/mail/e-mail-shell-view-actions.h
--- evolution-2.30.1.2/modules/mail/e-mail-shell-view-actions.h.notjunk-icon	2010-04-26 09:13:17.000000000 -0400
+++ evolution-2.30.1.2/modules/mail/e-mail-shell-view-actions.h	2010-05-01 15:28:20.752201064 -0400
@@ -115,8 +115,8 @@
 	E_SHELL_WINDOW_ACTION ((window), "mail-mark-important")
 #define E_SHELL_WINDOW_ACTION_MAIL_MARK_JUNK(window) \
 	E_SHELL_WINDOW_ACTION ((window), "mail-mark-junk")
-#define E_SHELL_WINDOW_ACTION_MAIL_MARK_NOT_JUNK(window) \
-	E_SHELL_WINDOW_ACTION ((window), "mail-mark-not-junk")
+#define E_SHELL_WINDOW_ACTION_MAIL_MARK_NOTJUNK(window) \
+	E_SHELL_WINDOW_ACTION ((window), "mail-mark-notjunk")
 #define E_SHELL_WINDOW_ACTION_MAIL_MARK_READ(window) \
 	E_SHELL_WINDOW_ACTION ((window), "mail-mark-read")
 #define E_SHELL_WINDOW_ACTION_MAIL_MARK_UNIMPORTANT(window) \
diff -up evolution-2.30.1.2/ui/evolution-mail-reader.ui.notjunk-icon evolution-2.30.1.2/ui/evolution-mail-reader.ui
--- evolution-2.30.1.2/ui/evolution-mail-reader.ui.notjunk-icon	2010-04-22 18:40:18.000000000 -0400
+++ evolution-2.30.1.2/ui/evolution-mail-reader.ui	2010-05-01 15:28:20.754202368 -0400
@@ -82,7 +82,7 @@
           <menuitem action="mail-mark-unimportant"/>
           <separator/>
           <menuitem action="mail-mark-junk"/>
-          <menuitem action="mail-mark-not-junk"/>
+          <menuitem action="mail-mark-notjunk"/>
           <separator/>
           <menuitem action="mail-flag-for-followup"/>
           <menuitem action="mail-flag-clear"/>
@@ -127,7 +127,7 @@
 #endif
       <toolitem action='mail-delete'/>
       <toolitem action='mail-mark-junk'/>
-      <toolitem action='mail-mark-not-junk'/>
+      <toolitem action='mail-mark-notjunk'/>
     </placeholder>
     <separator/>
     <placeholder name='mail-toolbar-navigation'>
diff -up evolution-2.30.1.2/ui/evolution-mail.ui.notjunk-icon evolution-2.30.1.2/ui/evolution-mail.ui
--- evolution-2.30.1.2/ui/evolution-mail.ui.notjunk-icon	2010-04-26 09:13:18.000000000 -0400
+++ evolution-2.30.1.2/ui/evolution-mail.ui	2010-05-01 15:28:20.754202368 -0400
@@ -109,7 +109,7 @@
     <menuitem action='mail-popup-mark-important'/>
     <menuitem action='mail-popup-mark-unimportant'/>
     <menuitem action='mail-popup-mark-junk'/>
-    <menuitem action='mail-popup-mark-not-junk'/>
+    <menuitem action='mail-popup-mark-notjunk'/>
     <menuitem action='mail-popup-flag-for-followup'/>
     <menuitem action="mail-popup-flag-clear"/>
     <menuitem action="mail-popup-flag-completed"/>
