diff -Naur apollon-1.0.2.1/apollon/apollonqueryview.cpp apollon-1.0.2.1-patched/apollon/apollonqueryview.cpp
--- apollon-1.0.2.1/apollon/apollonqueryview.cpp	2005-04-24 03:40:13.000000000 +0300
+++ apollon-1.0.2.1-patched/apollon/apollonqueryview.cpp	2008-10-13 15:20:37.000000000 +0300
@@ -207,7 +207,9 @@
 
 void ApollonQueryView::stopSearch()
 {
-	emit giftCancelSearchRequest(m_id);
+    if (this)
+	    emit giftCancelSearchRequest(m_id);
+    return;
 }
 
 void ApollonQueryView::moreSearch()
