diff -Nur xine-lib-1.1.13-old/src/demuxers/demux_qt.c xine-lib-1.1.13/src/demuxers/demux_qt.c
--- xine-lib-1.1.13-old/src/demuxers/demux_qt.c	2008-06-24 16:23:11.000000000 +0300
+++ xine-lib-1.1.13/src/demuxers/demux_qt.c	2008-06-24 16:32:25.000000000 +0300
@@ -383,7 +383,7 @@
 
 /* define DEBUG_ATOM_LOAD as 1 to get a verbose parsing of the relevant 
  * atoms */
-#define DEBUG_ATOM_LOAD 0
+#define DEBUG_ATOM_LOAD 1
 
 /* define DEBUG_EDIT_LIST as 1 to get a detailed look at how the demuxer is
  * handling edit lists */
@@ -1206,6 +1206,7 @@
           if ((current_stsd_atom_size > 0x24) &&
               (trak->stsd_atoms[k].audio.codec_fourcc != TWOS_FOURCC) &&
               (trak->stsd_atoms[k].audio.codec_fourcc != SOWT_FOURCC) &&
+              (trak->stsd_atoms[k].audio.codec_fourcc != SAMR_FOURCC) &&
               (trak->stsd_atoms[k].audio.codec_fourcc != RAW_FOURCC)) {
 
             if (_X_BE_32(&trak_atom[atom_pos + 0x20]))
@@ -1237,8 +1238,10 @@
           if (trak->stsd_atoms[k].audio.codec_fourcc == MP4A_FOURCC)
             trak->stsd_atoms[k].audio.vbr = 1;
 
-          if (trak->stsd_atoms[k].audio.codec_fourcc == SAMR_FOURCC)
+          if (trak->stsd_atoms[k].audio.codec_fourcc == SAMR_FOURCC) {
             trak->stsd_atoms[k].audio.vbr = 1;
+	    trak->stsd_atoms[k].audio.channels = 1;
+	  }
 
           if (trak->stsd_atoms[k].audio.codec_fourcc == ALAC_FOURCC) {
             trak->stsd_atoms[k].audio.vbr = 1;
