Index: arrayprobe-2.0/probe.c
===================================================================
--- arrayprobe-2.0.orig/probe.c
+++ arrayprobe-2.0/probe.c
@@ -512,7 +512,7 @@ main (int argc, char *argv[])
 			/* printf (" -s             : simultion mode (use with -f)\n"); */
 			printf (" -r             : report (verbose) mode\n");
 			printf (" -o             : only read new events (since last run, CCISS devices only)\n");
-			printf (" -i             : force ida ioctls. (use with -f if the device is supported by the ida driver)");
+			printf (" -i             : force ida ioctls. (use with -f if the device is supported by the ida driver)\n");
 			exit (1);
 		}
 	}

