--- src/transkode/transkode.desktop.orig	2007-06-14 05:32:32.000000000 +0300
+++ src/transkode/transkode.desktop	2008-09-21 14:34:21.000000000 +0300
@@ -5,6 +5,9 @@
 Icon=transkode
 Type=Application
 DocPath=transkode/transkode.html
-Comment=KDE frontend for various audio transcoding tools Application
+GenericName[en]=Audio Converter
+GenericName[tr]=Ses Dönüştürücü
+Comment[en]=KDE frontend for various audio transcoding tools Application
+Comment[tr]=KDE/QT arayüzlü ses dönüştürme uygulaması
 Terminal=false
-Categories=Qt;KDE;Audio;
\ Dosya sonunda yenisatır yok.
+Categories=Qt;KDE;Application;AudioVideo;
\ Dosya sonunda yenisatır yok.
