--- pynicotine/config.py.org	2006-09-14 14:21:27.000000000 +0300
+++ pynicotine/config.py	2006-09-14 14:21:55.000000000 +0300
@@ -64,7 +64,7 @@
  	"ui":{"chatme":"FOREST GREEN", "chatremote":"","chatlocal":"BLUE", \
  	"chathilite":"red", "search":"","searchq":"GREY", "decimalsep":",", \
  	"roomlistcollapsed": 0, "tabclosers": 1}, \
- 	"urls":{"urlcatching":1,"protocols":{"http":"mozilla \"%s\" &"}, \
+ 	"urls":{"urlcatching":1,"protocols":{"http":"firefox \"%s\" &"}, \
  	"humanizeurls":1}, "interests": {"likes":[], "dislikes":[]}, \
 	"ticker": {"default": "", "rooms": {}, "hide": 0}}
 
