--- mozilla/mailnews/base/prefs/resources/content/aw-identity.js	2004-09-12 12:38:50.000000000 +0300
+++ mozilla/mailnews/base/prefs/resources/content/aw-identity.js	2006-11-25 17:39:09.132179511 +0200
@@ -241,17 +241,86 @@
 
   // save in global variable
   gCurrentDomain = accountData.domain;
-  
+
   var postEmailText = document.getElementById("postEmailText");
   postEmailText.setAttribute("value", "@" + gCurrentDomain);
 }
 
+/**
+ *
+ *  UTF-8 data encode / decode
+ *  http://www.webtoolkit.info/
+ *
+ **/
+
+var Utf8 = {
+
+  // public method for url encoding
+ encode : function (string) {
+    string = string.replace(/\r\n/g,"\n");
+    var utftext = "";
+
+    for (var n = 0; n < string.length; n++) {
+
+      var c = string.charCodeAt(n);
+
+      if (c < 128) {
+        utftext += String.fromCharCode(c);
+      }
+      else if((c > 127) && (c < 2048)) {
+        utftext += String.fromCharCode((c >> 6) | 192);
+        utftext += String.fromCharCode((c & 63) | 128);
+      }
+      else {
+        utftext += String.fromCharCode((c >> 12) | 224);
+        utftext += String.fromCharCode(((c >> 6) & 63) | 128);
+        utftext += String.fromCharCode((c & 63) | 128);
+      }
+
+    }
+
+    return utftext;
+  },
+
+ // public method for url decoding
+ decode : function (utftext) {
+    var string = "";
+    var i = 0;
+    var c = c1 = c2 = 0;
+
+    while ( i < utftext.length ) {
+
+      c = utftext.charCodeAt(i);
+
+      if (c < 128) {
+        string += String.fromCharCode(c);
+        i++;
+      }
+      else if((c > 191) && (c < 224)) {
+        c2 = utftext.charCodeAt(i+1);
+        string += String.fromCharCode(((c & 31) << 6) | (c2 & 63));
+        i += 2;
+      }
+      else {
+        c2 = utftext.charCodeAt(i+1);
+        c3 = utftext.charCodeAt(i+2);
+        string += String.fromCharCode(((c & 15) << 12) | ((c2 & 63) << 6) | (c3 & 63));
+        i += 3;
+      }
+
+    }
+
+    return string;
+  }
+
+}
+
 function checkForFullName() {
     var name = document.getElementById("fullName");
     if (name.value=="") {
         try {
             var userInfo = Components.classes["@mozilla.org/userinfo;1"].getService(Components.interfaces.nsIUserInfo);
-            name.value = userInfo.fullname;
+            name.value = Utf8.decode(userInfo.fullname);
         }
         catch (ex) {
             // dump ("checkForFullName failed: " + ex + "\n");
