diff -up tzdata2011h/europe\~ tzdata2011h/europe
--- tzdata2011l/europe~	2011-06-15 13:31:00.000000000 +0200
+++ tzdata2011l/europe	2011-06-27 16:49:39.323493224 +0200
@@ -2158,8 +2158,8 @@ Zone Asia/Novokuznetsk	 5:48:48 -	NMT	19
 			 7:00	Russia	KRA%sT	1991 Mar 31 2:00s
 			 6:00	Russia	KRA%sT	1992 Jan 19 2:00s
 			 7:00	Russia	KRA%sT	2010 Mar 28 2:00s
-			 6:00	Russia	NOV%sT	2011 Mar 27 2:00s
-			 7:00	-	NOVT # Novosibirsk/Novokuznetsk Time
+			 6:00	Russia	OMS%sT	2011 Mar 27 2:00s
+			 7:00	-	OMST # Novosibirsk/Novokuznetsk Time
 
 #
 # From Oscar van Vlijmen (2001-08-25): [This region consists of]

Diff finished.  Mon Jun 27 16:49:45 2011
