Index: alsa-driver/acore/wrappers.c
===================================================================
--- alsa-driver.orig/acore/wrappers.c
+++ alsa-driver/acore/wrappers.c
@@ -340,6 +340,8 @@ EXPORT_SYMBOL(compat_skip_spaces);
 #endif /* < 2.6.33 */
 
 #ifndef CONFIG_GCD
+#include <linux/gcd.h>
+#ifdef CONFIG_SND_COMPAT_GCD
 /* Greatest common divisor */
 unsigned long gcd(unsigned long a, unsigned long b)
 {
@@ -356,4 +358,5 @@ unsigned long gcd(unsigned long a, unsig
 	return b;
 }
 EXPORT_SYMBOL(gcd);
+#endif /* CONFIG_SND_COMPAT_GCD */
 #endif /* !CONFIG_GCD */
Index: alsa-driver/include/gcd_compat.h
===================================================================
--- alsa-driver.orig/include/gcd_compat.h
+++ alsa-driver/include/gcd_compat.h
@@ -1,6 +1,8 @@
 #ifndef _LINUX_GCD_H
 #define _LINUX_GCD_H
 
+#define CONFIG_SND_COMPAT_GCD
+
 unsigned long gcd(unsigned long a, unsigned long b);
 
 #endif /* _LINUX_GCD_H */
