Index: yali4-2.3.1/yali4/constants.py
===================================================================
--- yali4-2.3.1.orig/yali4/constants.py
+++ yali4-2.3.1/yali4/constants.py
@@ -90,7 +90,7 @@ consts.cd_repo_uri = join(consts.source_
 
 # pardus repository
 consts.pardus_repo_name = "corporate2"
-consts.pardus_repo_uri = "http://packages.pardus.org.tr/corporate2/pisi-index.xml.bz2"
+consts.pardus_repo_uri = "@REPO_URI@"
 consts.pardus_release_path = "etc/pardus-release"
 
 # min root partition size
