diff -Nuar smb4k-0.9.9-orjinal/smb4k/smb4k.desktop smb4k-0.9.9/smb4k/smb4k.desktop
--- smb4k-0.9.9-orjinal/smb4k/smb4k.desktop	2011-03-23 15:27:12.151000029 +0200
+++ smb4k-0.9.9/smb4k/smb4k.desktop	2011-03-23 15:32:49.220000031 +0200
@@ -21,7 +21,7 @@
 GenericName[ru]=The SMB/CIFS Share Browser
 GenericName[sk]=Prehliadač SMB/CIFS zdieľaní
 GenericName[sv]=SMB Katalog programmed
-GenericName[tr]=The SMB/CIFS Share Browser
+GenericName[tr]=SMB/CIFS Paylaşım Tarayıcısı
 GenericName[uk]=Навігатор спільних ресурсів SMB/CIFS
 GenericName[zh_CN]=SMB 共享浏览器
 #
@@ -69,7 +69,7 @@
 Comment[ru]=The SMB/CIFS Share Browser
 Comment[sk]=Prehliadač SMB/CIFS zdieľaní
 Comment[sv]=SMB Katalog programmed
-Comment[tr]=The SMB/CIFS Share Browser
+Comment[tr]=SMB/CIFS Paylaşım Tarayıcısı
 Comment[uk]=Навігатор спільних ресурсів SMB/CIFS
 Comment[zh_CN]=SMB 共享浏览器
 Exec=smb4k