Index: smb4k-0.9.10/po/Makefile.in
===================================================================
--- smb4k-0.9.10.orig/po/Makefile.in
+++ smb4k-0.9.10/po/Makefile.in
@@ -302,8 +302,8 @@ xdg_directorydir = @xdg_directorydir@
 xdg_menudir = @xdg_menudir@
 #>- POFILES = AUTO
 #>+ 2
-POFILES = uk.po sv.po it.po de.po ja.po es.po
-GMOFILES = uk.gmo sv.gmo it.gmo de.gmo ja.gmo es.gmo
+POFILES = uk.po sv.po it.po de.po ja.po es.po tr.po
+GMOFILES = uk.gmo sv.gmo it.gmo de.gmo ja.gmo es.gmo tr.gmo
 EXTRA_DIST = smb4k.pot
 #>- all: all-am
 #>+ 1
@@ -497,7 +497,7 @@ uninstall-am: uninstall-info-am
 .NOEXPORT:
 
 #>+ 2
-KDE_DIST=sv.gmo uk.gmo uk.po de.po ja.gmo Makefile.in ja.po it.gmo sv.po de.gmo es.po es.gmo it.po Makefile.am 
+KDE_DIST=sv.gmo uk.gmo uk.po de.po ja.gmo Makefile.in ja.po it.gmo sv.po de.gmo es.po es.gmo it.po tr.po tr.gmo Makefile.am 
 
 #>+ 19
 uk.gmo: uk.po
@@ -525,7 +525,7 @@ clean-nls:
 
 #>+ 10
 install-nls:
-	@for base in uk sv it de ja es ; do \
+	@for base in uk sv it de ja tr es ; do \
 	  echo $(INSTALL_DATA) $$base.gmo $(DESTDIR)$(kde_locale)/$$base/LC_MESSAGES/$(PACKAGE).mo ;\
 	  $(mkinstalldirs) $(DESTDIR)$(kde_locale)/$$base/LC_MESSAGES ; \
 	  if test -f $$base.gmo; then $(INSTALL_DATA) $$base.gmo $(DESTDIR)$(kde_locale)/$$base/LC_MESSAGES/$(PACKAGE).mo ;\
