diff -p -up ZynAddSubFX-2.4.0/src/Makefile.inc.makefile ZynAddSubFX-2.4.0/src/Makefile.inc
--- ZynAddSubFX-2.4.0/src/Makefile.inc.makefile	2009-07-15 21:48:41.000000000 +0200
+++ ZynAddSubFX-2.4.0/src/Makefile.inc	2009-07-16 19:41:35.000000000 +0200
@@ -31,7 +31,7 @@ LINUX_MIDIIN=ALSA
 #LINUX_AUDIOOUT=OSS_AND_JACK
 LINUX_AUDIOOUT=OSS
 #LINUX_AUDIOOUT=NONE
-#LINUX_AUDIOOUT=JACK
+LINUX_AUDIOOUT=JACK
 #LINUX_AUDIOOUT=JACK_RT     JACK_RT support is broken
 #for PortAudio (PA)
 #LINUX_AUDIOOUT=PA
diff -p -up ZynAddSubFX-2.4.0/src/Makefile.makefile ZynAddSubFX-2.4.0/src/Makefile
--- ZynAddSubFX-2.4.0/src/Makefile.makefile	2009-07-15 21:48:41.000000000 +0200
+++ ZynAddSubFX-2.4.0/src/Makefile	2009-07-16 19:43:18.000000000 +0200
@@ -1,10 +1,6 @@
 include Makefile.inc
 
-ifneq ($(MAKECMDGOALS),debug)
-  CXXFLAGS= -O6 -Wall -g 
-else 
-  CXXFLAGS= -O0 -Wall -Wpointer-arith 
-endif
+CXXFLAGS+=
 
 CXXFLAGS += -DOS_$(OS_PORT) -D$(MIDIIN)MIDIIN -DFFTW_VERSION_$(FFTW_VERSION) -DASM_F2I_$(ASM_F2I) -ggdb
 
