diff -Nur x264-snapshot-20080402-2245-old/configure x264-snapshot-20080402-2245/configure
--- x264-snapshot-20080402-2245-old/configure	2008-04-04 10:45:15.000000000 +0300
+++ x264-snapshot-20080402-2245/configure	2008-04-04 10:45:33.000000000 +0300
@@ -372,7 +372,7 @@
     LDFLAGS="$LDFLAGS $libpthread"
 fi
 
-MP4_LDFLAGS="-lgpac_static"
+MP4_LDFLAGS="-lgpac"
 if [ $SYS = MINGW ]; then
     MP4_LDFLAGS="$MP4_LDFLAGS -lwinmm"
 fi
