--- DriverManager/SQLAllocHandle.c
+++ DriverManager/SQLAllocHandle.c
@@ -224,6 +224,7 @@
  *
  **********************************************************************/
 
+
 #include "drivermanager.h"
 #if defined ( COLLECT_STATS ) && defined( HAVE_SYS_SEM_H )
 #include "__stats.h"
--- DriverManager/SQLConnect.c
+++ DriverManager/SQLConnect.c
@@ -764,6 +764,7 @@
     }
 
     mutex_lib_exit();
+    return 0;
 }
 
 /*
--- cur/SQLAllocHandle.c
+++ cur/SQLAllocHandle.c
@@ -139,4 +139,5 @@
            }
            break;
     }
+    return 0;
 }
--- cur/SQLEndTran.c
+++ cur/SQLEndTran.c
@@ -70,4 +70,5 @@
       default:
         return SQL_ERROR;
     }
+    return 0;
 }
--- cur/SQLError.c
+++ cur/SQLError.c
@@ -149,4 +149,5 @@
 
         return SQL_NO_DATA;
     }
+    return 0;
 }
--- cur/SQLFreeHandle.c
+++ cur/SQLFreeHandle.c
@@ -102,5 +102,6 @@
          */
         return SQL_ERROR;
     }
+    return 0;
 }
 
--- odbcinst/SQLSetConfigMode.c
+++ odbcinst/SQLSetConfigMode.c
@@ -23,6 +23,7 @@
 __set_config_mode( int mode )
 {
     __config_mode = mode;
+    return 0;
 }
 
 __get_config_mode()
