Index: poppler-0.10.6/Makefile.am
===================================================================
--- poppler-0.10.6.orig/Makefile.am
+++ poppler-0.10.6/Makefile.am
@@ -29,7 +29,7 @@ if BUILD_UTILS
 utils_subdir = utils
 endif
 
-SUBDIRS = goo fofi $(splash_subdir) poppler $(utils_subdir) $(glib_subdir) $(qt_subdir) test $(qt4_subdir)
+SUBDIRS = $(glib_subdir) $(qt_subdir) $(qt4_subdir)
 
 EXTRA_DIST =					\
 	README-XPDF				\
@@ -40,9 +40,7 @@ EXTRA_DIST =					\
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA =				\
-	poppler.pc				\
 	$(cairo_pc_file)			\
-	$(splash_pc_file)			\
 	$(glib_pc_file)				\
 	$(qt_pc_file)                           \
 	$(qt4_pc_file)
