Index: mail/components/shell/nsMailGNOMEIntegration.cpp
===================================================================
RCS file: /cvsroot/mozilla/mail/components/shell/nsMailGNOMEIntegration.cpp,v
retrieving revision 1.8.2.1
diff -u -p -6 -r1.8.2.1 nsMailGNOMEIntegration.cpp
--- mail/components/shell/nsMailGNOMEIntegration.cpp	23 Oct 2006 03:32:47 -0000	1.8.2.1
+++ mail/components/shell/nsMailGNOMEIntegration.cpp	19 Nov 2006 11:54:48 -0000
@@ -94,12 +94,18 @@ nsMailGNOMEIntegration::Init()
   NS_ENSURE_SUCCESS(rv, rv);
 
   rv = appPath->AppendNative(NS_LITERAL_CSTRING("thunderbird"));
   NS_ENSURE_SUCCESS(rv, rv);
 
   rv = appPath->GetNativePath(mAppPath);
+
+  /* This path of thunderbird is hardcoded for the default path setting 
+   * And the above code is not used.
+   */
+  mAppPath.Assign ("/usr/bin/thunderbird");
+
   return rv;
 }
 
 NS_IMPL_ISUPPORTS1(nsMailGNOMEIntegration, nsIShellService)
 
 
