Index: omnibook-20090720/init.c
===================================================================
--- omnibook-20090720.orig/init.c
+++ omnibook-20090720/init.c
@@ -533,4 +533,5 @@ MODULE_PARM_DESC(ectype, "Type of embedd
 MODULE_PARM_DESC(userset, "Use 0 to disable, 1 to enable users to set parameters");
 
 MODULE_ALIAS("dmi:*:svnTOSHIBA:pnSatelliteA200:pvrPSAECE-01U00XTE:*");
+MODULE_ALIAS("dmi:*:svnTOSHIBA:pnSatelliteA210:pvrPSAEGE-01M00KTE:*");
 /* End of file */
Index: omnibook-20090720/laptop.h
===================================================================
--- omnibook-20090720.orig/laptop.h
+++ omnibook-20090720/laptop.h
@@ -646,6 +646,15 @@ static struct  dmi_system_id omnibook_id
 	},
 	{
 		.callback = dmi_matched,
+		.ident = "Toshiba Satellite A210",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A210"),
+		},
+		.driver_data = (void*) TSM70
+	},
+	{
+		.callback = dmi_matched,
 		.ident = "Toshiba Satellite P100",
 		.matches = {
 			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
