Index: kmplayer-0.11.1b/src/kmplayerprocess.cpp
===================================================================
--- kmplayer-0.11.1b.orig/src/kmplayerprocess.cpp
+++ kmplayer-0.11.1b/src/kmplayerprocess.cpp
@@ -675,7 +675,7 @@ bool MPlayer::run (const char * args, co
     }
 
     fprintf (stderr, " %s\n", args);
-    *m_process << " " << args;
+    *m_process << "-nomsgmodule -nomsgcolor " << args;
 
     m_process->start (K3Process::NotifyOnExit, K3Process::All);
 
