diff -Nur kmplayer-0.9.3-old/src/kmplayertvsource.cpp kmplayer-0.9.3/src/kmplayertvsource.cpp
--- kmplayer-0.9.3-old/src/kmplayertvsource.cpp	2006-10-04 20:35:44.044339268 +0300
+++ kmplayer-0.9.3/src/kmplayertvsource.cpp	2006-10-04 20:36:46.480847477 +0300
@@ -156,7 +156,7 @@
     driver = new QLineEdit ("", general, 0);
     QWhatsThis::add (driver, i18n ("dummy, v4l or bsdbt848"));
     QLabel *deviceLabel = new QLabel (i18n ("Device:"), general, 0);
-    device = new KURLRequester ("/dev/video", general);
+    device = new KURLRequester ("/dev/video0", general);
     QWhatsThis::add(device, i18n("Path to your video device, eg. /dev/video0"));
     scan = new QPushButton (i18n ("Scan..."), general);
     gridlayout->addWidget (driverLabel, 0, 0);
