Index: VirtualBox-3.1.2_OSE/LocalConfig.kmk
===================================================================
--- /dev/null
+++ VirtualBox-3.1.2_OSE/LocalConfig.kmk
@@ -0,0 +1,8 @@
+# Do not build tests
+VBOX_WITH_TESTCASES     :=
+VBOX_WITH_TESTSUITE     :=
+
+VBOX_WITH_ORIGIN            :=
+VBOX_WITH_RUNPATH           := __VBOXLIBDIR__
+VBOX_PATH_APP_PRIVATE_ARCH  := __VBOXLIBDIR__
+VBOX_PATH_APP_PRIVATE       := __VBOXDATADIR__
