Index: VirtualBox-3.1.2_OSE/src/VBox/Frontends/VirtualBox/Makefile.kmk
===================================================================
--- VirtualBox-3.1.2_OSE.orig/src/VBox/Frontends/VirtualBox/Makefile.kmk
+++ VirtualBox-3.1.2_OSE/src/VBox/Frontends/VirtualBox/Makefile.kmk
@@ -32,11 +32,11 @@ VBOX_WITH_REGISTRATION := 1
 
 # Ask the user to register at Sun. If this setting is disabled the user can
 # still register using the menu if desired.
-VBOX_WITH_REGISTRATION_REQUEST := 1
+VBOX_WITH_REGISTRATION_REQUEST :=
 
 # Show the update notifier dialog during startup. If this setting is disabled
 # the user can still update using the menu if desired.
-# VBOX_WITH_UPDATE_REQUEST := 1
+VBOX_WITH_UPDATE_REQUEST :=
 
 
 #
Index: VirtualBox-3.1.2_OSE/src/VBox/Frontends/VirtualBox/src/VBoxUpdateDlg.cpp
===================================================================
--- VirtualBox-3.1.2_OSE.orig/src/VBox/Frontends/VirtualBox/src/VBoxUpdateDlg.cpp
+++ VirtualBox-3.1.2_OSE/src/VBox/Frontends/VirtualBox/src/VBoxUpdateDlg.cpp
@@ -104,7 +104,7 @@ QStringList VBoxUpdateData::list()
 
 VBoxUpdateData::VBoxUpdateData (const QString &aData)
     : mData (aData)
-    , mPeriodIndex (Period1Day)
+    , mPeriodIndex (PeriodNever)
     , mBranchIndex (BranchStable)
 {
     decode();
