Index: VirtualBox-3.0.0_OSE/configure
===================================================================
--- VirtualBox-3.0.0_OSE.orig/configure
+++ VirtualBox-3.0.0_OSE/configure
@@ -2196,7 +2196,7 @@ check_curl
 # TODO check for libxdamange-dev (X11/extensions/Xdamage.h, additions only)
 [ $WITH_X11    -eq 1 ] && check_xcursor
 [ $WITH_OPENGL -eq 1 ] && check_xmu
-[ "$OS" != "darwin" -a $WITH_OPENGL -eq 1 ] && check_mesa
+# [ "$OS" != "darwin" -a $WITH_OPENGL -eq 1 ] && check_mesa
 [ $WITH_QT4    -eq 1 ] && check_qt4
 [ $WITH_PYTHON -eq 1 -a "$OS" != "darwin" ] && check_python
 
