From 9dce131a70eba9f670021b86db8c379a9e69a4e0 Mon Sep 17 00:00:00 2001
From: Manoj Iyer <manoj.iyer@canonical.com>
Date: Wed, 4 Mar 2009 11:33:32 -0600
Subject: [PATCH 3/6] UBUNTU: SAUCE: Added quirk for Linksys WUSB600N USB wifi-n networking adapter

Bug: #323473

Signed-off-by: Manoj Iyer <manoj.iyer@canonical.com>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 drivers/staging/rt2870/rt2870.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/staging/rt2870/rt2870.h b/drivers/staging/rt2870/rt2870.h
index a69cf33..fcf00ee 100644
--- a/drivers/staging/rt2870/rt2870.h
+++ b/drivers/staging/rt2870/rt2870.h
@@ -146,6 +146,7 @@
 	{USB_DEVICE(0x0789,0x0163)}, /* Logitec */		\
 	{USB_DEVICE(0x0789,0x0164)}, /* Logitec */		\
 	{USB_DEVICE(0x7392,0x7717)}, /* Edimax */		\
+	{USB_DEVICE(0x1737,0x0071)}, /* Linksys */		\
 	{ }/* Terminating entry */                      \
 }
 
-- 
1.6.3

