--- src/pngcodec.c.orig	2010-07-05 01:32:36.261098430 +0300
+++ src/pngcodec.c	2010-07-05 01:34:28.822103038 +0300
@@ -352,11 +352,7 @@
 						info_ptr->palette[i].blue,
 						info_ptr->palette[i].green,
 						info_ptr->palette[i].red,
-#if PNG_LIBPNG_VER > 10243
-						info_ptr->trans_alpha [i]); /* alpha */
-#else
 						info_ptr->trans[i]); /* alpha */
-#endif
 			}
 		}
 
@@ -422,11 +418,7 @@
 		}
 
 		if ((color_type == PNG_COLOR_TYPE_GRAY) && (bit_depth < 8)) {
-#if PNG_LIBPNG_VER > 10243
 			png_set_expand_gray_1_2_4_to_8 (png_ptr);
-#else
-			png_set_gray_1_2_4_to_8(png_ptr);
-#endif
 		}
 
 		if (png_get_valid(png_ptr, info_ptr, PNG_INFO_tRNS)) {
