Index: util-linux-ng-2.18/Makefile.am
===================================================================
--- util-linux-ng-2.18.orig/Makefile.am
+++ util-linux-ng-2.18/Makefile.am
@@ -2,29 +2,12 @@ include $(top_srcdir)/config/include-Mak
 
 SUBDIRS = \
 	include \
-	disk-utils \
-	fdisk \
-	getopt \
-	lib \
-	shlibs \
-	login-utils \
-	misc-utils \
-	po \
-	schedutils \
-	sys-utils \
-	text-utils \
-	tests
+	shlibs
 
 
 RCS_FIND_IGNORE := \( -name SCCS -o -name BitKeeper -o -name .svn -o \
 		-name CVS -o -name .pc -o -name .hg -o -name .git \) -prune -o
 
-if LINUX
-SUBDIRS += \
-	hwclock \
-	partx
-endif
-
 if BUILD_MOUNT
 SUBDIRS += mount
 endif
