diff --git a/hardware/library/libx86/actions.py b/hardware/library/libx86/actions.py
index 9ab8fa4..f87b123 100644
--- a/hardware/library/libx86/actions.py
+++ b/hardware/library/libx86/actions.py
@@ -13,7 +13,7 @@ def setup():
     pisitools.dosed("Makefile", "CFLAGS \?=.*", "CFLAGS = %s" % get.CFLAGS())
 
 def build():
-    autotools.make()
+    autotools.make("BACKEND=x86emu")
 
 def install():
     autotools.rawInstall("DESTDIR=%s" % get.installDIR())
