diff --git a/multimedia/graphics/imagemagick/actions.py b/multimedia/graphics/imagemagick/actions.py
index 3d56e9f..f932365 100644
--- a/multimedia/graphics/imagemagick/actions.py
+++ b/multimedia/graphics/imagemagick/actions.py
@@ -46,5 +46,5 @@ def build():
 def install():
     autotools.rawInstall("DESTDIR=%s" % get.installDIR())
     pisitools.dodoc("AUTHORS.txt", "ChangeLog", "LICENSE", "NEWS.txt")
-    pisitools.remove("/usr/lib/perl5/vendor_perl/%s/i686-linux-thread-multi/auto/Image/Magick/.packlist" % get.curPERL())
+    pisitools.remove("/usr/lib/perl5/vendor_perl/%s/%s-linux-thread-multi/auto/Image/Magick/.packlist" % (get.curPERL(),get.ARCH()))
 
