diff --git a/multimedia/library/atlas/actions.py b/multimedia/library/atlas/actions.py
index d558ad1..de4e31c 100644
--- a/multimedia/library/atlas/actions.py
+++ b/multimedia/library/atlas/actions.py
@@ -19,7 +19,7 @@ def setup():
     shelltools.system("../configure \
                        -Si cputhrchk 0 \
                        -Fa alg -fPIC \
-                       -b 32 \
+                       -b 64 \
                        --with-netlib-lapack=/usr/lib/liblapack.a")
 
 def build():
