diff --git a/multimedia/library/imlib2/actions.py b/multimedia/library/imlib2/actions.py
index e2b30c5..d951025 100644
--- a/multimedia/library/imlib2/actions.py
+++ b/multimedia/library/imlib2/actions.py
@@ -20,7 +20,7 @@ def setup():
                          --with-zlib \
                          --with-bzip2 \
                          --with-id3 \
-                         --disable-amd64 \
+                         --enable-amd64 \
                          --enable-mmx")
 
 def build():
