diff --git a/multimedia/library/libsdl/actions.py b/multimedia/library/libsdl/actions.py
index 3e49f96..b69987d 100644
--- a/multimedia/library/libsdl/actions.py
+++ b/multimedia/library/libsdl/actions.py
@@ -27,7 +27,7 @@ def setup():
                          --enable-file \
                          --enable-alsa \
                          --enable-oss \
-                         --enable-nasm \
+                         --disable-nasm \
                          --enable-video-aalib \
                          --enable-video-caca \
                          --enable-video-directfb \
