diff --git a/multimedia/library/sdl-gfx/actions.py b/multimedia/library/sdl-gfx/actions.py
index d580aef..b901158 100644
--- a/multimedia/library/sdl-gfx/actions.py
+++ b/multimedia/library/sdl-gfx/actions.py
@@ -19,7 +19,7 @@ def setup():
 
     autotools.autoreconf("-fi")
     libtools.libtoolize("--force --install")
-    autotools.configure("--enable-mmx \
+    autotools.configure("--disable-mmx \
                          --disable-static")
 
 def build():
