diff --git a/programming/language/python/pygame/actions.py b/programming/language/python/pygame/actions.py
index 6958e12..5a95baf 100644
--- a/programming/language/python/pygame/actions.py
+++ b/programming/language/python/pygame/actions.py
@@ -30,7 +30,7 @@ def setup():
 
 def build():
     pythonmodules.compile()
-    shelltools.copy("lib/pygame_icon.bmp", "build/lib.linux-i686-%s/pygame/" % get.curPYTHON().replace("python", ""))
+    shelltools.copy("lib/pygame_icon.bmp", "build/lib.linux-x86_64-%s/pygame/" % get.curPYTHON().replace("python", ""))
 
 def install():
     pythonmodules.install()
