diff --git a/programming/library/metis/actions.py b/programming/library/metis/actions.py
index 5a44d3e..0a9bd51 100644
--- a/programming/library/metis/actions.py
+++ b/programming/library/metis/actions.py
@@ -12,6 +12,9 @@ from pisi.actionsapi import get
 
 WorkDir="metis-4.0"
 
+def setup():
+    pisitools.dosed("Makefile.in", "COPTIONS = ",  "COPTIONS = -fPIC")"
+
 def build():
     autotools.make("CC=%s" % get.CC())
 
