diff --git a/programming/profiler/valgrind/actions.py b/programming/profiler/valgrind/actions.py
index 0d4288a..9b57ec0 100644
--- a/programming/profiler/valgrind/actions.py
+++ b/programming/profiler/valgrind/actions.py
@@ -15,6 +15,7 @@ def setup():
 
     autotools.autoreconf("-vfi")
     autotools.configure("--with-x \
+                         --enable-only64bit \
                          --without-mpicc")
 
 def build():
