diff --git a/programming/vcs/git/actions.py b/programming/vcs/git/actions.py
index be9bff5..ab46306 100644
--- a/programming/vcs/git/actions.py
+++ b/programming/vcs/git/actions.py
@@ -46,7 +46,7 @@ def install():
 
     # Remove useless perl directories
     pisitools.removeDir("/usr/lib/perl5/%s" % get.curPERL())
-    pisitools.removeDir("/usr/lib/perl5/vendor_perl/%s/i686-linux-thread-multi" % get.curPERL())
+    pisitools.removeDir("/usr/lib/perl5/vendor_perl/%s/x86_64-linux-thread-multi" % get.curPERL())
 
     # Some docs
     pisitools.dodoc("README", "COPYING", "Documentation/SubmittingPatches")
