diff --git a/system/base/ncompress/actions.py b/system/base/ncompress/actions.py
index 941f7cc..b8ae36f 100644
--- a/system/base/ncompress/actions.py
+++ b/system/base/ncompress/actions.py
@@ -11,7 +11,8 @@ from pisi.actionsapi import shelltools
 from pisi.actionsapi import get
 
 def setup():
-    shelltools.move("Makefile.def", "Makefile")
+    shelltools.move("Makefile.def", "Makefile") 
+    pisitools.dosed("Makefile","options= ","options= $(CFLAGS) -DNOFUNCDEF -DUTIME_H $(LDFLAGS) ")
 
 def build():
     autotools.make()
