diff --git a/system/base/python/actions.py b/system/base/python/actions.py
index 39735f3..ea5fca7 100644
--- a/system/base/python/actions.py
+++ b/system/base/python/actions.py
@@ -28,6 +28,8 @@ def setup():
                          --with-threads \
                          --with-libc='' \
                          --enable-unicode=ucs4 \
+                         --enable-universalsdk \
+                         --with-universal-archs=64-bit  \
                          --with-wctype-functions \
                          --without-system-ffi")
 
