diff --git a/system/boot/gfxboot/actions.py b/system/boot/gfxboot/actions.py
index 1062efc..08b8637 100644
--- a/system/boot/gfxboot/actions.py
+++ b/system/boot/gfxboot/actions.py
@@ -14,6 +14,7 @@ from pisi.actionsapi import get
 def setup():
     pisitools.dosed("Makefile", "^CC.*", "CC = %s" % get.CC())
     pisitools.dosed("doc/Makefile", "xmlto", "xmlto --skip-validation")
+    pisitools.dosed("Makefile", "32", "64")
 
 def build():
     shelltools.export("HOME", get.workDIR())
