diff --git a/system/devel/gmp/actions.py b/system/devel/gmp/actions.py
index 66de128..bd5af87 100644
--- a/system/devel/gmp/actions.py
+++ b/system/devel/gmp/actions.py
@@ -14,7 +14,6 @@ def setup():
     shelltools.export("CCAS","%s -c -Wa,--noexecstack" % get.CC())
 
     autotools.configure("--enable-cxx \
-                         --enable-fat \
                          --disable-mpbsd \
                          --localstatedir=/var/state/gmp")
 
