diff --git a/system/devel/icecream/actions.py b/system/devel/icecream/actions.py
index b324ea2..ea8f9ca 100644
--- a/system/devel/icecream/actions.py
+++ b/system/devel/icecream/actions.py
@@ -24,6 +24,6 @@ def build():
 def install():
     autotools.rawInstall("DESTDIR=%s" % get.installDIR())
 
-    for chost in ["", "i686-pc-linux-gnu-"]:
+    for chost in ["", "x86_64-pc-linux-gnu--"]:
         for i in ["c++", "cc", "g++", "gcc"]:
             pisitools.dosym("icecc", "/opt/icecream/bin/%s%s" % (chost, i))
