diff --git a/system/devel/klibc/actions.py b/system/devel/klibc/actions.py
index c4bdce1..67370fc 100644
--- a/system/devel/klibc/actions.py
+++ b/system/devel/klibc/actions.py
@@ -47,7 +47,6 @@ def build():
     autotools.make('EXTRA_KLIBCAFLAGS="-Wa,--noexecstack" \
                     EXTRA_KLIBCLDFLAGS="-z,noexecstack" \
                     HOSTCC="%s" CC="%s" \
-                    KLIBCARCH=i386 \
                     KLIBCASMARCH=x86 \
                     libdir=/usr/lib \
                     SHLIBDIR=/lib \
@@ -59,7 +58,6 @@ def install():
     autotools.rawInstall('EXTRA_KLIBCAFLAGS="-Wa,--noexecstack" \
                           EXTRA_KLIBCLDFLAGS="-z,noexecstack" \
                           HOSTCC="%s" CC="%s" \
-                          KLIBCARCH=i386 \
                           KLIBCASMARCH=x86 \
                           libdir=/usr/lib \
                           SHLIBDIR=/lib \
