diff --git a/util/crypt/johntheripper/actions.py b/util/crypt/johntheripper/actions.py
index bacc411..5928113 100644
--- a/util/crypt/johntheripper/actions.py
+++ b/util/crypt/johntheripper/actions.py
@@ -26,7 +26,7 @@ def build():
                     CFLAGS="-c -Wall %(CFLAGS)s -DJOHN_SYSTEMWIDE -DJOHN_SYSTEMWIDE_HOME=\\\"\\\\\\\"/etc/john\\\\\\\"\\\"" \
                     LDFLAGS="%(LDFLAGS)s" \
                     OPT_NORMAL="" \
-                    linux-x86-sse2' % conf)
+                    linux-x86-64' % conf)
 
 def install():
     pisitools.dosbin("run/john")
