Index: devel/tex/tool/luatex/actions.py
===================================================================
--- devel.orig/tex/tool/luatex/actions.py
+++ devel/tex/tool/luatex/actions.py
@@ -17,7 +17,7 @@ GetWorkdir = "%s/%s" % (get.workDIR(), W
 
 def setup():
     shelltools.cd("source")
-    autotools.autoreconf()
+    autotools.autoreconf("-fi")
     libtools.libtoolize()
 
     shelltools.export("LC_ALL","C")
