Index: devel/office/openoffice/openoffice/actions.py
===================================================================
--- devel.orig/office/openoffice/openoffice/actions.py
+++ devel/office/openoffice/openoffice/actions.py
@@ -84,11 +84,11 @@
     pisitools.domove("/opt/OpenOffice.org/lib/ooo-3.1/basis3.1/program/unohelper.py", "/usr/lib/%s/site-packages" % get.curPYTHON())

     #install pdfimport, report-builder, wiki-publisher and presenter screen as extensions
-    pisitools.insinto("/opt/OpenOffice.org/lib/ooo-3.1/share/extension/install/", "build/%s/solver/310/unxlngi6.pro/bin/sun-report-builder.oxt" % UpstreamVersion)
-    pisitools.insinto("/opt/OpenOffice.org/lib/ooo-3.1/share/extension/install/", "build/%s/solver/310/unxlngi6.pro/bin/swext/wiki-publisher.oxt" % UpstreamVersion)
-    pisitools.insinto("/opt/OpenOffice.org/lib/ooo-3.1/share/extension/install/", "build/%s/solver/310/unxlngi6.pro/bin/minimizer/sun-presentation-minimizer.oxt" % UpstreamVersion)
-    pisitools.insinto("/opt/OpenOffice.org/lib/ooo-3.1/share/extension/install/", "build/%s/solver/310/unxlngi6.pro/bin/presenter/presenter-screen.oxt" % UpstreamVersion)
-    pisitools.insinto("/opt/OpenOffice.org/lib/ooo-3.1/share/extension/install/", "build/%s/solver/310/unxlngi6.pro/bin/pdfimport/pdfimport.oxt" % UpstreamVersion)
+    pisitools.insinto("/opt/OpenOffice.org/lib/ooo-3.1/share/extension/install/", "build/%s/solver/310/unxlngx6.pro/bin/sun-report-builder.oxt" % UpstreamVersion)
+    pisitools.insinto("/opt/OpenOffice.org/lib/ooo-3.1/share/extension/install/", "build/%s/solver/310/unxlngx6.pro/bin/swext/wiki-publisher.oxt" % UpstreamVersion)
+    pisitools.insinto("/opt/OpenOffice.org/lib/ooo-3.1/share/extension/install/", "build/%s/solver/310/unxlngx6.pro/bin/minimizer/sun-presentation-minimizer.oxt" % UpstreamVersion)
+    pisitools.insinto("/opt/OpenOffice.org/lib/ooo-3.1/share/extension/install/", "build/%s/solver/310/unxlngx6.pro/bin/presenter/presenter-screen.oxt" % UpstreamVersion)
+    pisitools.insinto("/opt/OpenOffice.org/lib/ooo-3.1/share/extension/install/", "build/%s/solver/310/unxlngx6.pro/bin/pdfimport/pdfimport.oxt" % UpstreamVersion)

     #Fix mktemp directory in unopkg
     pisitools.dosed("%s/opt/OpenOffice.org/lib/ooo-3.1/program/unopkg" % get.installDIR(), "/bin/mktemp", "/usr/bin/mktemp")
