diff -Nur opencore-amr-old/amrnb/Makefile opencore-amr/amrnb/Makefile
--- opencore-amr-old/amrnb/Makefile	2009-07-06 11:24:58.360729232 +0300
+++ opencore-amr/amrnb/Makefile	2009-07-06 11:26:54.013359985 +0300
@@ -77,9 +77,8 @@
 libopencore-amrnb.a: $(OBJS)
 	ar rcs $@ $+
 
-install: libopencore-amrnb.a $(SHLIB)
+install: $(SHLIB)
 	install -d $(DESTDIR)$(PREFIX)/$(LIBDIR)
-	install -m 644 libopencore-amrnb.a $(DESTDIR)$(PREFIX)/$(LIBDIR)
 	install $(SHLIB) $(DESTDIR)$(PREFIX)/$(LIBDIR)
 ifneq ($(shell uname), Darwin)
 	ln -sf $(SHLIB) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(SONAME)
diff -Nur opencore-amr-old/amrwb/Makefile opencore-amr/amrwb/Makefile
--- opencore-amr-old/amrwb/Makefile	2009-07-06 11:24:58.406728542 +0300
+++ opencore-amr/amrwb/Makefile	2009-07-06 11:27:13.816495606 +0300
@@ -63,9 +63,8 @@
 libopencore-amrwb.a: $(OBJS)
 	ar rcs $@ $+
 
-install: libopencore-amrwb.a $(SHLIB)
+install: $(SHLIB)
 	install -d $(DESTDIR)$(PREFIX)/$(LIBDIR)
-	install -m 644 libopencore-amrwb.a $(DESTDIR)$(PREFIX)/$(LIBDIR)
 	install $(SHLIB) $(DESTDIR)$(PREFIX)/$(LIBDIR)
 ifneq ($(shell uname), Darwin)
 	ln -sf $(SHLIB) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(SONAME)
