diff -Nur mpeg4ip-1.5.0.1-old/server/mp4live/Makefile.am mpeg4ip-1.5.0.1/server/mp4live/Makefile.am
--- mpeg4ip-1.5.0.1-old/server/mp4live/Makefile.am	2006-10-24 23:15:28.031150174 +0300
+++ mpeg4ip-1.5.0.1/server/mp4live/Makefile.am	2006-10-24 23:16:34.564035638 +0300
@@ -169,6 +169,7 @@
 	@SRTPLIB@ \
 	-lpthread \
 	@SDL_LIBS@ $(GUILIBS) @FFMPEG_LIB@ @LIBVORBIS_LIB@ $(XVID_LIB) \
+	@ALSA_LIBS@ \
 	@X264_LIB@ 
 
 EXTRA_DIST = $(ed2) video_v4l_source.cpp 
diff -Nur mpeg4ip-1.5.0.1-old/server/mp4live/Makefile.in mpeg4ip-1.5.0.1/server/mp4live/Makefile.in
--- mpeg4ip-1.5.0.1-old/server/mp4live/Makefile.in	2006-10-24 23:15:28.031150174 +0300
+++ mpeg4ip-1.5.0.1/server/mp4live/Makefile.in	2006-10-24 23:17:03.193683270 +0300
@@ -482,6 +482,7 @@
 	@SRTPLIB@ \
 	-lpthread \
 	@SDL_LIBS@ $(GUILIBS) @FFMPEG_LIB@ @LIBVORBIS_LIB@ $(XVID_LIB) \
+	@ALSA_LIBS@ \
 	@X264_LIB@ 
 
 EXTRA_DIST = $(ed2) video_v4l_source.cpp 
