diff -Nur mpeg4ip-1.5.0.1-old/common/video/iso-mpeg4/configure.in mpeg4ip-1.5.0.1/common/video/iso-mpeg4/configure.in
--- mpeg4ip-1.5.0.1-old/common/video/iso-mpeg4/configure.in	2008-04-11 14:23:03.000000000 +0300
+++ mpeg4ip-1.5.0.1/common/video/iso-mpeg4/configure.in	2008-04-11 14:25:11.000000000 +0300
@@ -134,7 +134,6 @@
 MY_CHECK_CC_OPT_SET([-Wformat=2],BILLS_CWARNINGS, [c compiler])
 BILLS_CPPWARNINGS="$BASE_ERR"
 AC_LANG_PUSH(C++)
-MY_CHECK_CC_OPT_SET([-Wmissing-prototypes], BILLS_CPPWARNINGS,[c++ compiler])
 MY_CHECK_CC_OPT_SET([-Wno-char-subscripts], BILLS_CPPWARNINGS,[c++ compiler])
 MY_CHECK_CC_OPT_SET([-Woverloaded-virtual], BILLS_CPPWARNINGS,[c++ compiler])
 MY_CHECK_CC_OPT_SET([-Wno-unknown-pragmas], BILLS_CPPWARNINGS,[c++ compiler])
