diff -Nur mpeg4ip-1.5.0.1-old/configure.in mpeg4ip-1.5.0.1/configure.in
--- mpeg4ip-1.5.0.1-old/configure.in	2008-04-11 14:10:21.000000000 +0300
+++ mpeg4ip-1.5.0.1/configure.in	2008-04-11 14:11:03.000000000 +0300
@@ -469,8 +469,8 @@
 if test x$disable_ffmpeg = xfalse; then
   if test x$have_ffmpeg_arg = xtrue; then
      AC_MSG_NOTICE([have enable ffmpeg $enable_ffmpeg])
-     FFMPEG_INC=-I${enable_ffmpeg}/libavcodec
-     FFMPEG_LIB="${enable_ffmpeg}/libavcodec/libavcodec.a -lz"
+     FFMPEG_INC=-I${enable_ffmpeg}
+     FFMPEG_LIB="-L/usr/lib -lavcodec -lz"
      saveCFLAGS="$CFLAGS"
      CFLAGS="$CFLAGS $FFMPEG_INC"
      AC_CHECK_TYPES([AVRational], , , [#include <avcodec.h>])
@@ -495,8 +495,8 @@
         AC_DEFINE(HAVE_FFMPEG_INSTALLED, 1, [defined if ffmpeg is installed])
         FFMPEG_INC=
         FFMPEG_LIB="-lavcodec -lz $ffmpeg_add_libs"
-        AC_CHECK_TYPES([AVRational], , , [#include <ffmpeg/avcodec.h>])
-	AC_CHECK_MEMBERS(AVCodecContext.time_base, , , [#include <ffmpeg/avcodec.h>])
+        AC_CHECK_TYPES([AVRational], , , [#include <libavcodec/avcodec.h>])
+	AC_CHECK_MEMBERS(AVCodecContext.time_base, , , [#include <libavcodec/avcodec.h>])
         AC_DEFINE(HAVE_FFMPEG, 1, [enable ffmpeg for mp4live])
      fi
   fi
@@ -779,7 +779,7 @@
 AC_SUBST(BILLS_CWARNINGS)
 AC_SUBST(BILLS_CPPWARNINGS)
 
-PLAYER_PLUGIN_DIR=$libdir/mp4player_plugin
+PLAYER_PLUGIN_DIR=$libdir/mp4player
 AC_SUBST(PLAYER_PLUGIN_DIR)
 
 SDL_CFLAGS=`sdl-config --cflags`
diff -Nur mpeg4ip-1.5.0.1-old/player/plugin/audio/ffmpeg/ffmpeg.cpp mpeg4ip-1.5.0.1/player/plugin/audio/ffmpeg/ffmpeg.cpp
--- mpeg4ip-1.5.0.1-old/player/plugin/audio/ffmpeg/ffmpeg.cpp	2008-04-11 14:10:21.000000000 +0300
+++ mpeg4ip-1.5.0.1/player/plugin/audio/ffmpeg/ffmpeg.cpp	2008-04-11 14:10:28.000000000 +0300
@@ -166,7 +166,7 @@
     break;
   }
   if (userdata) {
-    ffmpeg->m_c->extradata = (void *)userdata;
+    ffmpeg->m_c->extradata = (uint8_t *)userdata;
     ffmpeg->m_c->extradata_size = ud_size;
   }
   if (avcodec_open(ffmpeg->m_c, ffmpeg->m_codec) < 0) {
diff -Nur mpeg4ip-1.5.0.1-old/player/plugin/audio/ffmpeg/ffmpeg.h mpeg4ip-1.5.0.1/player/plugin/audio/ffmpeg/ffmpeg.h
--- mpeg4ip-1.5.0.1-old/player/plugin/audio/ffmpeg/ffmpeg.h	2008-04-11 14:10:21.000000000 +0300
+++ mpeg4ip-1.5.0.1/player/plugin/audio/ffmpeg/ffmpeg.h	2008-04-11 14:10:28.000000000 +0300
@@ -25,7 +25,7 @@
 #define __MPEG4IP_FFMPEG_H__ 1
 #include "codec_plugin.h"
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
diff -Nur mpeg4ip-1.5.0.1-old/player/plugin/video/ffmpeg/ffmpeg.cpp mpeg4ip-1.5.0.1/player/plugin/video/ffmpeg/ffmpeg.cpp
--- mpeg4ip-1.5.0.1-old/player/plugin/video/ffmpeg/ffmpeg.cpp	2008-04-11 14:10:21.000000000 +0300
+++ mpeg4ip-1.5.0.1/player/plugin/video/ffmpeg/ffmpeg.cpp	2008-04-11 14:10:28.000000000 +0300
@@ -255,7 +255,7 @@
   }
     break;
   case CODEC_ID_SVQ3:
-    ffmpeg->m_c->extradata = (void *)userdata;
+    ffmpeg->m_c->extradata = (uint8_t *)userdata;
     ffmpeg->m_c->extradata_size = ud_size;
     if (vinfo != NULL) {
       ffmpeg->m_c->width = vinfo->width;
diff -Nur mpeg4ip-1.5.0.1-old/player/plugin/video/ffmpeg/ffmpeg.h mpeg4ip-1.5.0.1/player/plugin/video/ffmpeg/ffmpeg.h
--- mpeg4ip-1.5.0.1-old/player/plugin/video/ffmpeg/ffmpeg.h	2008-04-11 14:10:21.000000000 +0300
+++ mpeg4ip-1.5.0.1/player/plugin/video/ffmpeg/ffmpeg.h	2008-04-11 14:10:28.000000000 +0300
@@ -25,7 +25,7 @@
 #define __MPEG4IP_FFMPEG_H__ 1
 #include "codec_plugin.h"
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
diff -Nur mpeg4ip-1.5.0.1-old/server/mp4live/audio_ffmpeg.h mpeg4ip-1.5.0.1/server/mp4live/audio_ffmpeg.h
--- mpeg4ip-1.5.0.1-old/server/mp4live/audio_ffmpeg.h	2008-04-11 14:10:21.000000000 +0300
+++ mpeg4ip-1.5.0.1/server/mp4live/audio_ffmpeg.h	2008-04-11 14:10:28.000000000 +0300
@@ -25,7 +25,7 @@
 #include "audio_encoder.h"
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 }
diff -Nur mpeg4ip-1.5.0.1-old/server/mp4live/video_encoder_class.cpp mpeg4ip-1.5.0.1/server/mp4live/video_encoder_class.cpp
--- mpeg4ip-1.5.0.1-old/server/mp4live/video_encoder_class.cpp	2008-04-11 14:10:21.000000000 +0300
+++ mpeg4ip-1.5.0.1/server/mp4live/video_encoder_class.cpp	2008-04-11 14:10:28.000000000 +0300
@@ -27,7 +27,7 @@
 extern "C" {
 #ifdef HAVE_FFMPEG
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
diff -Nur mpeg4ip-1.5.0.1-old/server/mp4live/video_ffmpeg.h mpeg4ip-1.5.0.1/server/mp4live/video_ffmpeg.h
--- mpeg4ip-1.5.0.1-old/server/mp4live/video_ffmpeg.h	2008-04-11 14:10:21.000000000 +0300
+++ mpeg4ip-1.5.0.1/server/mp4live/video_ffmpeg.h	2008-04-11 14:10:28.000000000 +0300
@@ -25,7 +25,7 @@
 #include "video_encoder.h"
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
