diff -Nur mpeg4ip-1.5.0.1-old/configure mpeg4ip-1.5.0.1/configure
--- mpeg4ip-1.5.0.1-old/configure	2008-04-11 12:19:46.000000000 +0300
+++ mpeg4ip-1.5.0.1/configure	2008-04-11 12:20:13.000000000 +0300
@@ -28881,7 +28881,7 @@
    if test x$USENASM = xtrue; then
 	echo "$as_me:$LINENO: checking nasm version..." >&5
 echo $ECHO_N "checking nasm version...... $ECHO_C" >&6
-	BILLSNASM_arg=`nasm -r | tr '.' ' '`
+	BILLSNASM_arg=`nasm -v | tr '.' ' '`
         BILLSNASM_ret=`util/testnasm.sh $BILLSNASM_arg`
 	if test x$BILLSNASM_ret = x"yes"; then
 	   USENASM=true
diff -Nur mpeg4ip-1.5.0.1-old/configure.in mpeg4ip-1.5.0.1/configure.in
--- mpeg4ip-1.5.0.1-old/configure.in	2008-04-11 12:19:46.000000000 +0300
+++ mpeg4ip-1.5.0.1/configure.in	2008-04-11 12:20:19.000000000 +0300
@@ -682,7 +682,7 @@
 
    if test x$USENASM = xtrue; then
 	AC_MSG_CHECKING(nasm version...)
-	BILLSNASM_arg=`nasm -r | tr '.' ' '`
+	BILLSNASM_arg=`nasm -v | tr '.' ' '`
         BILLSNASM_ret=`util/testnasm.sh $BILLSNASM_arg`
 	if test x$BILLSNASM_ret = x"yes"; then
 	   USENASM=true
diff -Nur mpeg4ip-1.5.0.1-old/util/testlame.sh mpeg4ip-1.5.0.1/util/testlame.sh
--- mpeg4ip-1.5.0.1-old/util/testlame.sh	2008-04-11 12:19:46.000000000 +0300
+++ mpeg4ip-1.5.0.1/util/testlame.sh	2008-04-11 12:20:44.000000000 +0300
@@ -1,6 +1,6 @@
 #
 # shell to verify nasm version
-# nasm -r has format "NASM version <foo> <extra stuff>"
+# nasm -v has format "NASM version <foo> <extra stuff>"
 #
 # This shell looks for version, then sees if we're 0.98.19 or greater
 #
diff -Nur mpeg4ip-1.5.0.1-old/util/testnasm.sh mpeg4ip-1.5.0.1/util/testnasm.sh
--- mpeg4ip-1.5.0.1-old/util/testnasm.sh	2008-04-11 12:19:46.000000000 +0300
+++ mpeg4ip-1.5.0.1/util/testnasm.sh	2008-04-11 12:21:02.000000000 +0300
@@ -1,6 +1,6 @@
 #
 # shell to verify nasm version
-# nasm -r has format "NASM version <foo> <extra stuff>"
+# nasm -v has format "NASM version <foo> <extra stuff>"
 #
 # This shell looks for version, then sees if we're 0.98.19 or greater
 #
