diff -Nur mpeg4ip-1.5.0.1-old/server/mp4live/video_x264.cpp mpeg4ip-1.5.0.1/server/mp4live/video_x264.cpp
--- mpeg4ip-1.5.0.1-old/server/mp4live/video_x264.cpp	2009-10-20 20:25:31.982964891 +0300
+++ mpeg4ip-1.5.0.1/server/mp4live/video_x264.cpp	2009-10-20 20:25:54.474932722 +0300
@@ -205,7 +205,7 @@
   m_param.pf_log = x264_log;
 
   m_param.i_threads = Profile()->GetIntegerValue(CFG_X264_THREADS);
-  m_pts_add = m_param.i_bframe ? (m_param.b_bframe_pyramid ? 2 : 1) : 0;
+  m_pts_add = m_param.i_bframe ? (m_param.i_bframe_pyramid ? 2 : 1) : 0;
   m_pts_add *= m_frame_time;
   debug_message("pts add "D64, m_pts_add);
 
