Index: linux-2.6.32/arch/x86/kernel/aperture_64.c
===================================================================
--- linux-2.6.32.orig/arch/x86/kernel/aperture_64.c
+++ linux-2.6.32/arch/x86/kernel/aperture_64.c
@@ -30,6 +30,7 @@
 #include <asm/k8.h>
 
 int gart_iommu_aperture;
+EXPORT_SYMBOL_GPL(gart_iommu_aperture);
 int gart_iommu_aperture_disabled __initdata;
 int gart_iommu_aperture_allowed __initdata;
 
Index: linux-2.6.32/drivers/staging/Kconfig
===================================================================
--- linux-2.6.32.orig/drivers/staging/Kconfig
+++ linux-2.6.32/drivers/staging/Kconfig
@@ -101,7 +101,7 @@ source "drivers/staging/p9auth/Kconfig"
 
 source "drivers/staging/line6/Kconfig"
 
-source "drivers/gpu/drm/radeon/Kconfig"
+source "drivers/gpu/drm/nouveau/Kconfig"
 
 source "drivers/staging/octeon/Kconfig"
 
