diff -up ConsoleKit-0.3.0/src/ck-manager.c.null-warning ConsoleKit-0.3.0/src/ck-manager.c
--- ConsoleKit-0.3.0/src/ck-manager.c.null-warning	2009-04-20 23:55:53.248884043 -0400
+++ ConsoleKit-0.3.0/src/ck-manager.c	2009-04-20 23:56:27.736126951 -0400
@@ -1213,7 +1213,7 @@ session_is_real_user (CkSession *session
 
         g_object_get (session,
                       "unix-user", &uid,
-                      "session-type", session_type,
+                      "session-type", &session_type,
                       NULL);
 
         username = get_user_name (uid);
