Index: pavucontrol-0.9.8/src/pavucontrol.cc
===================================================================
--- pavucontrol-0.9.8.orig/src/pavucontrol.cc
+++ pavucontrol-0.9.8/src/pavucontrol.cc
@@ -213,18 +213,18 @@ void ext_stream_restore_read_cb(
     if (eol < 0) {
         dec_outstanding(w);
         g_debug(_("Failed to initialize stream_restore extension: %s"), pa_strerror(pa_context_errno(context)));
-        w->deleteEventRoleWidget();
+        /*w->deleteEventRoleWidget();*/
         return;
     }
 
-    w->createEventRoleWidget();
+    /*w->createEventRoleWidget();*/
 
     if (eol > 0) {
         dec_outstanding(w);
         return;
     }
 
-    w->updateRole(*i);
+    /*w->updateRole(*i);*/
 }
 
 static void ext_stream_restore_subscribe_cb(pa_context *c, void *userdata) {
