Index: kdeaddons-3.5.10/konq-plugins/dirfilter/dirfilterplugin.desktop
===================================================================
--- kdeaddons-3.5.10.orig/konq-plugins/dirfilter/dirfilterplugin.desktop
+++ kdeaddons-3.5.10/konq-plugins/dirfilter/dirfilterplugin.desktop
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Type=Service
 X-KDE-PluginInfo-Author=Dawit Alemayehu,Martin Koller
 X-KDE-PluginInfo-Email=adawit@kde.org,m.koller@surfeu.at
 X-KDE-PluginInfo-Name=DirFilter
