--- procps-3.2.7/top.c
+++ procps-3.2.7/top.c	2006-06-26 16:26:56.000000000 +0200
@@ -2480,21 +2480,21 @@
    static const char err_num_cpus[] = "\aSorry, terminal is not big enough";
 #ifdef WARN_NOT_SMP
    // standardized 'smp' errors
-   static const char err_smp[] = "\aSorry, only 1 cpu detected";
+   static const char err_smp[] = "\aSorry, only one cpu detected";
 #endif
 
    switch (c) {
       case '1':
          if (Cpu_tot+7 > Screen_rows && !CHKw(Curwin, View_CPUSUM)) {
             show_msg(err_num_cpus);
-            break;
-         }
+         } else {
 #ifdef WARN_NOT_SMP
          if (Cpu_tot > 1) TOGw(Curwin, View_CPUSUM);
          else show_msg(err_smp);
 #else
          TOGw(Curwin, View_CPUSUM);
 #endif
+         }
          break;
 
       case 'a':
