diff -up pilot-link-0.12.3/include/pi-md5.h.md5 pilot-link-0.12.3/include/pi-md5.h
--- pilot-link-0.12.3/include/pi-md5.h.md5	2006-10-17 15:24:07.000000000 +0200
+++ pilot-link-0.12.3/include/pi-md5.h	2007-12-11 08:47:56.000000000 +0100
@@ -13,8 +13,10 @@
 #ifndef MD5_H
 #define MD5_H
 
+#include <stdint.h>
+
 #define UINT8 unsigned char
-#define UINT32 unsigned long
+#define UINT32 uint32_t
 
 struct MD5Context {
 	UINT32 buf[4];
