diff -rup notify-python-0.1.1-orig/src/pynotify.c notify-python-0.1.1/src/pynotify.c
--- notify-python-0.1.1-orig/src/pynotify.c	2006-10-08 19:59:03.000000000 +0200
+++ notify-python-0.1.1/src/pynotify.c	2007-03-01 12:17:24.000000000 +0100
@@ -215,6 +215,19 @@ _wrap_notify_notification_attach_to_widg
 }
 
 static PyObject *
+_wrap_notify_notification_attach_to_status_icon(PyGObject *self, PyObject *args, PyObject *kwargs)
+{
+    static char *kwlist[] = { "attach", NULL };
+    PyGObject *attach;
+    
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O!:NotifyNotification.attach_to_status_icon", kwlist, &PyGtkWidget_Type, &attach))
+        return NULL;
+    notify_notification_attach_to_status_icon(NOTIFY_NOTIFICATION(self->obj), GTK_STATUS_ICON(attach->obj));
+    Py_INCREF(Py_None);
+    return Py_None;
+}
+
+static PyObject *
 _wrap_notify_notification_show(PyGObject *self)
 {
     int ret;
@@ -499,6 +512,7 @@ _wrap_notify_notification_set_hint(PyGOb
 static PyMethodDef _PyNotifyNotification_methods[] = {
     { "update", (PyCFunction)_wrap_notify_notification_update, METH_VARARGS|METH_KEYWORDS },
     { "attach_to_widget", (PyCFunction)_wrap_notify_notification_attach_to_widget, METH_VARARGS|METH_KEYWORDS },
+    { "attach_to_status_icon", (PyCFunction)_wrap_notify_notification_attach_to_status_icon, METH_VARARGS|METH_KEYWORDS },
     { "show", (PyCFunction)_wrap_notify_notification_show, METH_NOARGS },
     { "set_timeout", (PyCFunction)_wrap_notify_notification_set_timeout, METH_VARARGS|METH_KEYWORDS },
     { "set_category", (PyCFunction)_wrap_notify_notification_set_category, METH_VARARGS|METH_KEYWORDS },
