
--- a/src/settingsdialog.cpp
+++ b/src/settingsdialog.cpp
@@ -96,6 +96,10 @@
     ui_scrobbling.groupBox->hide();
 #endif
 
+#ifdef NBREAKPAD
+    ui_connection.crashReportCheck->hide();
+#endif
+
     // Add icons to user icon dropdown
     QPixmap pixmap( MooseUtils::dataPath( "icons/user_red.png" ) );
     ui_account.colourCombo->setItemIcon( 0, pixmap );
@@ -225,7 +229,9 @@
     connect( ui_connection.automaticProxyButton, SIGNAL( toggled( bool ) ), this, SLOT( configChanged() ) );
     connect( ui_connection.manualProxyButton, SIGNAL( toggled( bool ) ), this, SLOT( configChanged() ) );
     connect( ui_connection.downloadMetadataCheck, SIGNAL( toggled( bool ) ), this, SLOT( configChanged() ) );
+#ifndef NBREAKPAD
     connect( ui_connection.crashReportCheck, SIGNAL( toggled( bool ) ), this, SLOT( configChanged() ) );
+#endif
 
     #ifndef Q_WS_X11
     connect( ui_mediadevices.enableIPodScrobbling, SIGNAL( toggled( bool ) ), SLOT( configChanged() ) );
@@ -401,7 +407,9 @@
     ui_connection.proxyUsernameEdit->setText( The::settings().getProxyUser() );
     ui_connection.proxyPasswordEdit->setText( The::settings().getProxyPassword() );
     ui_connection.downloadMetadataCheck->setChecked( The::settings().currentUser().isMetaDataEnabled() );
+#ifndef NBREAKPAD
     ui_connection.crashReportCheck->setChecked( The::settings().currentUser().crashReportingEnabled() );
+#endif
     ui_connection.manualProxyButton->setChecked( The::settings().isUseProxy() );
 }
 
@@ -592,7 +600,9 @@
     The::settings().setProxyPort( ui_connection.proxyPortEdit->text().toInt() );
     The::settings().setUseProxy( ui_connection.manualProxyButton->isChecked() );
     The::settings().currentUser().setMetaDataEnabled( ui_connection.downloadMetadataCheck->isChecked() );
+#ifndef NBREAKPAD
     The::settings().currentUser().setCrashReportingEnabled( ui_connection.crashReportCheck->isChecked() );
+#endif
 
     m_reconnect =  ui_connection.proxyHostEdit->text()              != originalProxyHost     ||
                    ui_connection.proxyUsernameEdit->text()          != originalProxyUsername ||

