--- a/src/playcontrols.ui
+++ b/src/playcontrols.ui
@@ -146,7 +146,7 @@
   <customwidget>
    <class>VolumeSlider</class>
    <extends>QFrame</extends>
-   <header>volumeslider.h</header>
+   <header>./volumeslider.h</header>
    <container>1</container>
   </customwidget>
  </customwidgets>
--- a/src/SideBarView.cpp
+++ b/src/SideBarView.cpp
@@ -133,7 +133,7 @@
     else
     {
         setStyle( new TreeStyle<QPlastiqueStyle> );
-        setStyleSheet( "QTreeView { border: 0px }" );
+        setStyleSheet( "QTreeView { border: 0px; background: white; }" );
     }
 
 #endif
