diff -Naur qt-x11-free-3.3.8b-orig/src/kernel/qapplication.cpp qt-x11-free-3.3.8b/src/kernel/qapplication.cpp
--- qt-x11-free-3.3.8b-orig/src/kernel/qapplication.cpp	2008-08-08 14:26:07.000000000 +0300
+++ qt-x11-free-3.3.8b/src/kernel/qapplication.cpp	2008-08-08 14:28:05.000000000 +0300
@@ -1933,7 +1933,7 @@
 	}
     }
     if ( !app_font ) {
-	app_font = new QFont( "Helvetica" );
+	app_font = new QFont( "Sans Serif", 10 );
 	Q_CHECK_PTR( app_font );
     }
     return *app_font;
