--- hunspell-1.2.4~/src/hunspell/suggestmgr.cxx	2008-06-17 03:57:54.000000000 -0500
+++ hunspell-1.2.4/src/hunspell/suggestmgr.cxx	2008-07-07 15:50:44.000000000 -0500
@@ -120,7 +120,7 @@
       return ns;
 }
 
-// generate suggestions for a mispelled word
+// generate suggestions for a misspelled word
 //    pass in address of array of char * pointers
 // onlycompoundsug: probably bad suggestions (need for ngram sugs, too)
 
--- hunspell-1.2.4~/src/tools/hunspell.cxx	2008-06-17 06:09:13.000000000 -0500
+++ hunspell-1.2.4/src/tools/hunspell.cxx	2008-07-07 15:50:58.000000000 -0500
@@ -1434,8 +1434,8 @@
 			fprintf(stderr,gettext("  -h, --help\tdisplay this help and exit\n"));
 			fprintf(stderr,gettext("  -H\t\tHTML input file format\n"));
 			fprintf(stderr,gettext("  -i enc\tinput encoding\n"));
-			fprintf(stderr,gettext("  -l\t\tprint mispelled words\n"));
-			fprintf(stderr,gettext("  -L\t\tprint lines with mispelled words\n"));
+			fprintf(stderr,gettext("  -l\t\tprint misspelled words\n"));
+			fprintf(stderr,gettext("  -L\t\tprint lines with misspelled words\n"));
 			fprintf(stderr,gettext("  -m \t\tanalyze the words of the input text\n"));
 			fprintf(stderr,gettext("  -n\t\tnroff/troff input file format\n"));
 			fprintf(stderr,gettext("  -p dict\tset dict custom dictionary\n"));
@@ -1448,7 +1448,7 @@
 //			fprintf(stderr,gettext("  -U\t\tautomatic correction of typical misspellings to stdout\n"));
 			fprintf(stderr,gettext("  -v, --version\tprint version number\n"));
 			fprintf(stderr,gettext("  -vv\t\tprint Ispell compatible version number\n"));
-			fprintf(stderr,gettext("  -w\t\tprint mispelled words (= lines) from one word/line input.\n\n"));
+			fprintf(stderr,gettext("  -w\t\tprint misspelled words (= lines) from one word/line input.\n\n"));
 			fprintf(stderr,gettext("Example: hunspell -d en_US file.txt    # interactive spelling\n"
                                                "         hunspell -l file.txt          # print misspelled words\n"
                                                "         hunspell -i utf-8 file.txt    # check UTF-8 encoded file\n\n"));
