diff -ur kdemultimedia-3.5.0.orig/kaudiocreator/encoder.cpp kdemultimedia-3.5.0/kaudiocreator/encoder.cpp
--- kdemultimedia-3.5.0.orig/kaudiocreator/encoder.cpp	2005-09-10 11:20:05.000000000 +0300
+++ kdemultimedia-3.5.0/kaudiocreator/encoder.cpp	2006-01-25 23:15:05.000000000 +0200
@@ -55,7 +55,9 @@
 	QString currentEncoderGroup = QString("Encoder_%1").arg(encoder);
 	prefs = EncoderPrefs::prefs(currentEncoderGroup);
 	if ( !EncoderPrefs::hasPrefs(currentEncoderGroup) ) {
+    /*
 		KMessageBox::sorry(0, i18n("No encoder has been selected.\nPlease select an encoder in the configuration."), i18n("No Encoder Selected"));
+    */
 		prefs->setCommandLine(QString::null);
 	}				
 }
