diff -ur kdemultimedia-3.5.1.orig/kscd/kscd.kcfg kdemultimedia-3.5.1/kscd/kscd.kcfg
--- kdemultimedia-3.5.1.orig/kscd/kscd.kcfg	2005-09-10 11:19:57.000000000 +0300
+++ kdemultimedia-3.5.1/kscd/kscd.kcfg	2006-02-05 00:31:13.000000000 +0200
@@ -26,7 +26,7 @@
         <entry name="Autoplay" type="Bool" key="AUTOPLAY">
             <label>Start playing when a CD is inserted.</label>
             <whatsthis>When this option is selected the CD will start playing automatically upon being inserted into the CD-ROM.</whatsthis>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="StopExit" type="Bool" key="STOPEXIT">
             <label>Stop playing the CD on program exit.</label>
