diff -Nur kdemultimedia-3.5.9-old/configure.in kdemultimedia-3.5.9/configure.in
--- kdemultimedia-3.5.9-old/configure.in	2008-04-24 16:22:27.000000000 +0300
+++ kdemultimedia-3.5.9/configure.in	2008-04-24 16:24:33.000000000 +0300
@@ -776,7 +776,7 @@
 have_musicbrainz=no
 
 if test "x$with_musicbrainz" != xno; then
-  	KDE_CHECK_HEADER(tunepimp/tp_c.h, have_musicbrainz=yes)
+  	KDE_CHECK_HEADER(tunepimp-0.5/tp_c.h, have_musicbrainz=yes)
 
   	if test "x$with_musicbrainz" != xcheck && test "x$have_musicbrainz" != xyes; then
     		AC_MSG_ERROR([--with-musicbrainz was given, but test for MusicBrainz failed])
diff -Nur kdemultimedia-3.5.9-old/juk/configure.in.in kdemultimedia-3.5.9/juk/configure.in.in
--- kdemultimedia-3.5.9-old/juk/configure.in.in	2008-04-24 16:22:27.000000000 +0300
+++ kdemultimedia-3.5.9/juk/configure.in.in	2008-04-24 16:24:53.000000000 +0300
@@ -10,7 +10,7 @@
 have_musicbrainz=no
 
 if test "x$with_musicbrainz" != xno; then
-  	KDE_CHECK_HEADER(tunepimp/tp_c.h, have_musicbrainz=yes)
+  	KDE_CHECK_HEADER(tunepimp-0.5/tp_c.h, have_musicbrainz=yes)
 
   	if test "x$with_musicbrainz" != xcheck && test "x$have_musicbrainz" != xyes; then
     		AC_MSG_ERROR([--with-musicbrainz was given, but test for MusicBrainz failed])
diff -Nur kdemultimedia-3.5.9-old/juk/ktrm.cpp kdemultimedia-3.5.9/juk/ktrm.cpp
--- kdemultimedia-3.5.9-old/juk/ktrm.cpp	2008-04-24 16:22:27.000000000 +0300
+++ kdemultimedia-3.5.9/juk/ktrm.cpp	2008-04-24 16:25:07.000000000 +0300
@@ -35,7 +35,7 @@
 #include <qobject.h>
 #include <qfile.h>
 
-#include <tunepimp/tp_c.h>
+#include <tunepimp-0.5/tp_c.h>
 #include <fixx11h.h>
 
 class KTRMLookup;
