Properly detect if device-mapper is loaded.

--- cryptsetup-1.0.6.old/lib/libdevmapper.c	2008-09-23 15:18:05.000000000 +0200
+++ cryptsetup-1.0.6/lib/libdevmapper.c	2008-09-23 15:18:25.000000000 +0200
@@ -32,9 +32,16 @@ static void set_dm_error(int level, cons
 	va_end(va);
 }
 
+static int _dm_simple(int task, const char *name);
+
 static int dm_init(void)
 {
 	dm_log_init(set_dm_error);
+	if (!_dm_simple(DM_DEVICE_LIST_VERSIONS, "test")) {
+		set_error("Cannot communicate with device-mapper. Is the dm_mod module loaded?");
+		return -1;
+	}
+
 	return 1;	/* unsafe memory */
 }
 
--- cryptsetup-1.0.6.old/lib/setup.c	2008-09-23 15:18:05.000000000 +0200
+++ cryptsetup-1.0.6/lib/setup.c	2008-09-23 15:18:25.000000000 +0200
@@ -695,7 +695,10 @@ static int crypt_job(int (*job)(int arg,
 
 	backend = get_setup_backend(default_backend);
 
-	setup_enter(backend,options->icb->log);
+	if (setup_enter(backend,options->icb->log) < 0) {
+		r = -ENOSYS;
+		goto out;
+	}
 
 	if (!backend) {
 		set_error("No setup backend available");
