diff -Nuar zarafa-6.40.0.orig/src/ECtools/zarafa-stats/Makefile.am zarafa-6.40.0/ECtools/zarafa-stats/Makefile.am
--- zarafa-6.40.0.orig/src/ECtools/zarafa-stats/Makefile.am	2010-04-26 11:37:49.561680215 +0300
+++ zarafa-6.40.0/src/ECtools/zarafa-stats/Makefile.am	2010-04-26 11:38:31.435806562 +0300
@@ -13,7 +13,7 @@
 	${top_builddir}/mapi4linux/src/libmapi.la \
 	${top_builddir}/common/libcommon_mapi.la \
 	${top_builddir}/common/libcommon_util.la \
-	-lncurses
+	-lncurses -luuid
 
 zarafa_stats_SOURCES = zarafa-stats.cpp
 
diff -Nuar zarafa-6.40.0.orig/src/ECtools/zarafa-stats/Makefile.in zarafa-6.40.0/ECtools/zarafa-stats/Makefile.in
--- zarafa-6.40.0.orig/src/ECtools/zarafa-stats/Makefile.in	2010-04-26 11:37:49.561680215 +0300
+++ zarafa-6.40.0/src/ECtools/zarafa-stats/Makefile.in	2010-04-26 11:38:22.475687815 +0300
@@ -250,7 +250,7 @@
 	${top_builddir}/mapi4linux/src/libmapi.la \
 	${top_builddir}/common/libcommon_mapi.la \
 	${top_builddir}/common/libcommon_util.la \
-	-lncurses
+	-lncurses -luuid
 
 zarafa_stats_SOURCES = zarafa-stats.cpp
 all: all-am
