Index: libsdl1.2-1.2.13/sdl.pc.in
===================================================================
--- libsdl1.2-1.2.13.orig/sdl.pc.in	2007-12-31 05:48:39.000000000 +0100
+++ libsdl1.2-1.2.13/sdl.pc.in	2008-01-02 03:20:00.000000000 +0100
@@ -10,5 +10,5 @@
 Version: @SDL_VERSION@
 Requires:
 Conflicts:
-Libs: -L${libdir} @SDL_RLD_FLAGS@ @SDL_LIBS@
+Libs: -L${libdir} @SDL_RLD_FLAGS@ -lSDL
 Cflags: -I${includedir}/SDL @SDL_CFLAGS@
Index: libsdl1.2-1.2.13/sdl-config.in
===================================================================
--- libsdl1.2-1.2.13.orig/sdl-config.in	2007-12-31 05:48:39.000000000 +0100
+++ libsdl1.2-1.2.13/sdl-config.in	2008-01-02 03:20:00.000000000 +0100
@@ -45,7 +45,7 @@
       echo -I@includedir@/SDL @SDL_CFLAGS@
       ;;
 @ENABLE_SHARED_TRUE@    --libs)
-@ENABLE_SHARED_TRUE@      echo -L@libdir@ @SDL_RLD_FLAGS@ @SDL_LIBS@
+@ENABLE_SHARED_TRUE@      echo -L@libdir@ @SDL_RLD_FLAGS@ -lSDL
 @ENABLE_SHARED_TRUE@      ;;
 @ENABLE_STATIC_TRUE@@ENABLE_SHARED_TRUE@    --static-libs)
 @ENABLE_STATIC_TRUE@@ENABLE_SHARED_FALSE@    --libs|--static-libs)
