Index: libsdl1.2-1.2.13/src/video/SDL_blit_N.c
===================================================================
--- libsdl1.2-1.2.13.orig/src/video/SDL_blit_N.c	2007-12-31 05:48:14.000000000 +0100
+++ libsdl1.2-1.2.13/src/video/SDL_blit_N.c	2008-01-02 03:20:00.000000000 +0100
@@ -2333,6 +2333,8 @@
     { 0x00FF0000,0x0000FF00,0x000000FF, 2, 0x0000001F,0x000003E0,0x00007C00,
       0, ConvertX86p32_16BGR555, ConvertX86, NO_ALPHA },
     { 0x00FF0000,0x0000FF00,0x000000FF, 3, 0x00FF0000,0x0000FF00,0x000000FF,
+      1, ConvertMMXpII32_24RGB888, ConvertMMX, NO_ALPHA },
+    { 0x00FF0000,0x0000FF00,0x000000FF, 3, 0x00FF0000,0x0000FF00,0x000000FF,
       0, ConvertX86p32_24RGB888, ConvertX86, NO_ALPHA },
     { 0x00FF0000,0x0000FF00,0x000000FF, 3, 0x000000FF,0x0000FF00,0x00FF0000,
       0, ConvertX86p32_24BGR888, ConvertX86, NO_ALPHA },
