Index: omnibook-20090720/laptop.h
===================================================================
--- omnibook-20090720.orig/laptop.h
+++ omnibook-20090720/laptop.h
@@ -591,41 +591,41 @@ static struct  dmi_system_id omnibook_id
 		.driver_data = (void*) XE3GF,
 	},
 	{
-                .callback = dmi_matched,
-                .ident = "Toshiba Satellite A70",
-                .matches = {
-                        DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
-                        DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A70"),
-                },
-                .driver_data = (void*) TSM70
-        },
+		.callback = dmi_matched,
+		.ident = "Toshiba Satellite A70",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A70"),
+		},
+		.driver_data = (void*) TSM70
+	},
 	{
-                .callback = dmi_matched,
-                .ident = "Toshiba Satellite A75",
-                .matches = {
-                        DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
-                        DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A75"),
-                },
-                .driver_data = (void*) TSM70
-        },
+		.callback = dmi_matched,
+		.ident = "Toshiba Satellite A75",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A75"),
+		},
+		.driver_data = (void*) TSM70
+	},
 	{
-                .callback = dmi_matched,
-                .ident = "Toshiba Satellite A80",
-                .matches = {
-                        DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
-                        DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A80"),
-                },
-                .driver_data = (void*) TSM70
-        },
-        {
-                .callback = dmi_matched,
-                .ident = "Toshiba Satellite A105",
-                .matches = {
-                        DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
-                        DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A105"),
-                },
-                .driver_data = (void*) TSA105
-        },
+		.callback = dmi_matched,
+		.ident = "Toshiba Satellite A80",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A80"),
+		},
+		.driver_data = (void*) TSM70
+	},
+	{
+		.callback = dmi_matched,
+		.ident = "Toshiba Satellite A105",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A105"),
+		},
+		.driver_data = (void*) TSA105
+	},
 	{
 		.callback = dmi_matched,
 		.ident = "Toshiba Satellite A100",
@@ -637,6 +637,15 @@ static struct  dmi_system_id omnibook_id
 	},
 	{
 		.callback = dmi_matched,
+		.ident = "Toshiba Satellite A200",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A200"),
+		},
+		.driver_data = (void*) TSM70
+	},
+	{
+		.callback = dmi_matched,
 		.ident = "Toshiba Satellite P100",
 		.matches = {
 			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
Index: omnibook-20090720/init.c
===================================================================
--- omnibook-20090720.orig/init.c
+++ omnibook-20090720/init.c
@@ -532,4 +532,6 @@ module_param_named(userset, omnibook_use
 MODULE_PARM_DESC(ectype, "Type of embedded controller firmware");
 MODULE_PARM_DESC(userset, "Use 0 to disable, 1 to enable users to set parameters");
 
+MODULE_ALIAS("dmi:*:svnTOSHIBA:pnSatelliteA200:pvrPSAECE-01U00XTE:*");
+MODULE_ALIAS("dmi:*:svnTOSHIBA:pnSatelliteA200:pvrPSAE0E-02G01STE:*");
 /* End of file */
