Index: omnibook-20090720/init.c
===================================================================
--- omnibook-20090720.orig/init.c
+++ omnibook-20090720/init.c
@@ -534,4 +534,6 @@ MODULE_PARM_DESC(userset, "Use 0 to disa
 
 MODULE_ALIAS("dmi:*:svnTOSHIBA:pnSatelliteA200:pvrPSAECE-01U00XTE:*");
 MODULE_ALIAS("dmi:*:svnTOSHIBA:pnSatelliteA210:pvrPSAEGE-01M00KTE:*");
+/* Toshiba Satellite A300 */
+MODULE_ALIAS("dmi:*:svnTOSHIBA:pnSatelliteA300:pvrPSAJ0E-01300QTE:*");
 /* End of file */
Index: omnibook-20090720/laptop.h
===================================================================
--- omnibook-20090720.orig/laptop.h
+++ omnibook-20090720/laptop.h
@@ -653,6 +653,15 @@ static struct  dmi_system_id omnibook_id
 		},
 		.driver_data = (void*) TSM70
 	},
+	{ /* Toshiba Satellite A300 (#11300) */
+		.callback = dmi_matched,
+		.ident = "Toshiba Satellite A300",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Satellite A300"),
+		},
+		.driver_data = (void*) TSM70
+	},
 	{
 		.callback = dmi_matched,
 		.ident = "Toshiba Satellite P100",
