diff --exclude=.svn --exclude=.svnignore --exclude=CVS -Naur PolicyKit-0.8-orig/src/polkit-grant/polkit-grant-helper.c PolicyKit-0.8/src/polkit-grant/polkit-grant-helper.c
--- PolicyKit-0.8-orig/src/polkit-grant/polkit-grant-helper.c	2008-06-25 11:58:53.000000000 +0300
+++ PolicyKit-0.8/src/polkit-grant/polkit-grant-helper.c	2008-06-25 11:59:30.000000000 +0300
@@ -379,9 +379,6 @@
                                                         fprintf (stderr, "polkit-grant-helper: examining member '%s' of group '%s'\n", user, groups[n]);
 #endif /* PGH_DEBUG */
 
-                                                        /* skip user 'root' since he is often member of 'wheel' etc. */
-                                                        if (strcmp (user, "root") == 0)
-                                                                continue;
                                                         /* TODO: we should probably only consider users with an uid
                                                          * in a given "safe" range, e.g. between 500 and 32000 or
                                                          * something like that...
