diff --exclude=.svn --exclude=.svnignore --exclude=CVS -pruN PolicyKit-0.8-orig/policy/org.freedesktop.policykit.policy.in PolicyKit-0.8/policy/org.freedesktop.policykit.policy.in
--- PolicyKit-0.8-orig/policy/org.freedesktop.policykit.policy.in	2008-05-09 09:43:43.000000000 +0300
+++ PolicyKit-0.8/policy/org.freedesktop.policykit.policy.in	2008-05-09 09:48:31.000000000 +0300
@@ -1,10 +1,9 @@
-<?xml version="1.0" encoding="utf-8"?>
-
+<?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE policyconfig PUBLIC
  "-//freedesktop//DTD PolicyKit Policy Configuration 1.0//EN"
  "http://www.freedesktop.org/standards/PolicyKit/1.0/policyconfig.dtd">
 
-<!-- 
+<!--
 Policy definitions for core PolicyKit actions
 
 Copyright (c) 2007 David Zeuthen <david@fubar.dk>
@@ -20,7 +19,9 @@ file are instantly applied.
 
   <action id="org.freedesktop.policykit.read">
     <_description>Read authorizations of other users</_description>
+    <_description xml:lang="tr">Diğer kullanıcıların yetkilerini oku</_description>
     <_message>Authentication is required to read authorizations of other users</_message>
+    <_message xml:lang="tr">Diğer kullanıcıların yetkilerini okumak için kimlik doğrulama gerekiyor</_message>
     <defaults>
       <allow_any>no</allow_any>
       <allow_inactive>no</allow_inactive>
@@ -30,7 +31,9 @@ file are instantly applied.
 
   <action id="org.freedesktop.policykit.revoke">
     <_description>Revoke authorizations from other users</_description>
-    <_message>Authentication is required to revoke authorizations other users</_message>
+    <_description xml:lang="tr">Diğer kullanıcıların yetkilerini kaldır</_description>
+    <_message>Authentication is required to revoke authorizations of other users</_message>
+    <_message xml:lang="tr">Diğer kullanıcıların yetkilerini kaldırmak için kimlik doğrulama gerekiyor</_message>
     <defaults>
       <allow_any>no</allow_any>
       <allow_inactive>no</allow_inactive>
@@ -40,7 +43,9 @@ file are instantly applied.
 
   <action id="org.freedesktop.policykit.grant">
     <_description>Grant authorizations to other users</_description>
+    <_description xml:lang="tr">Diğer kullanıcılara yetkiler ver</_description>
     <_message>Authentication is required to grant authorizations to other users</_message>
+    <_message xml:lang="tr">Diğer kullanıcılara yetkiler vermek için kimlik doğrulama gerekiyor</_message>
     <defaults>
       <allow_any>no</allow_any>
       <allow_inactive>no</allow_inactive>
@@ -50,7 +55,9 @@ file are instantly applied.
 
   <action id="org.freedesktop.policykit.modify-defaults">
     <_description>Modify defaults for implicit authorizations</_description>
+    <_description xml:lang="tr">Öntanımlı yetkilendirme varsayılanlarını değiştir</_description>
     <_message>Authentication is required to modify the defaults for implicit authorizations</_message>
+    <_message xml:lang="tr">Öntanımlı yetkilendirme varsayılanlarını değiştirmek için kimlik doğrulama gerekiyor</_message>
     <defaults>
       <allow_any>no</allow_any>
       <allow_inactive>no</allow_inactive>
