Index: hw/dmx/Makefile.am
===================================================================
--- hw/dmx/Makefile.am.orig
+++ hw/dmx/Makefile.am
@@ -1,6 +1,6 @@
 DIST_SUBDIRS = input vnc config glxProxy examples doc
 
-SUBDIRS = input vnc config examples
+SUBDIRS = input config examples
 bin_PROGRAMS = Xdmx
 
 if XINERAMA
Index: hw/dmx/dmx-config.h
===================================================================
--- hw/dmx/dmx-config.h.orig
+++ hw/dmx/dmx-config.h
@@ -72,8 +72,8 @@
 /* Enable the DMX extension */
 #define DMXEXT
 
-/* Enable VNC ability */
-#define DMXVNC 1
+/* Disable VNC ability */
+#undef DMXVNC
 
 /* Disable the extensions that are not currently supported */
 #undef MULTIBUFFER
Index: hw/dmx/input/Makefile.am
===================================================================
--- hw/dmx/input/Makefile.am.orig
+++ hw/dmx/input/Makefile.am
@@ -65,7 +65,6 @@ AM_CFLAGS = $(DIX_CFLAGS) \
             -I$(top_srcdir)/hw/dmx \
             $(GLX_INCS) \
             -DHAVE_DMX_CONFIG_H \
-            -DDMXVNC=1 \
             $(GLX_DEFS) \
             @DMXMODULES_CFLAGS@
 
Index: hw/dmx/vnc/Makefile.am
===================================================================
--- hw/dmx/vnc/Makefile.am.orig
+++ hw/dmx/vnc/Makefile.am
@@ -37,7 +37,6 @@ AM_CFLAGS = \
             -I$(top_srcdir)/hw/xfree86/common \
             -DHAVE_DMX_CONFIG_H \
 	    $(DIX_CFLAGS) \
-            -DDMXVNC=1 \
             @DMXMODULES_CFLAGS@
 
 ###EXTRA_DIST = dmxdetach.c
