diff -Naur QtCurve-KDE3-1.4.0/ChangeLog QtCurve-KDE3-1.4.1/ChangeLog
--- QtCurve-KDE3-1.4.0/ChangeLog	2010-05-14 22:06:45.000000000 +0300
+++ QtCurve-KDE3-1.4.1/ChangeLog	2010-05-18 11:48:20.000000000 +0300
@@ -1,3 +1,7 @@
+1.4.1
+-----
+1. Fix config dialog compilation.
+
 1.4.0
 -----
 1. Add striped window and menu background options.
diff -Naur QtCurve-KDE3-1.4.0/CMakeLists.txt QtCurve-KDE3-1.4.1/CMakeLists.txt
--- QtCurve-KDE3-1.4.0/CMakeLists.txt	2010-05-14 22:06:45.000000000 +0300
+++ QtCurve-KDE3-1.4.1/CMakeLists.txt	2010-05-18 11:48:20.000000000 +0300
@@ -8,7 +8,7 @@
 set(CPACK_SOURCE_GENERATOR "TBZ2")
 set(CPACK_PACKAGE_VERSION_MAJOR "1")
 set(CPACK_PACKAGE_VERSION_MINOR "4")
-set(CPACK_PACKAGE_VERSION_PATCH "0")
+set(CPACK_PACKAGE_VERSION_PATCH "1")
 set(CPACK_PACKAGE_CONTACT "Craig Drummond <craig.p.drummond@googlemail.com>")
 set(QTCURVE_VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}")
 set(QTCURVE_VERSION_FULL "${QTCURVE_VERSION}.${CPACK_PACKAGE_VERSION_PATCH}")
diff -Naur QtCurve-KDE3-1.4.0/common/common.h QtCurve-KDE3-1.4.1/common/common.h
--- QtCurve-KDE3-1.4.0/common/common.h	2010-05-14 22:06:45.000000000 +0300
+++ QtCurve-KDE3-1.4.1/common/common.h	2010-05-18 11:48:20.000000000 +0300
@@ -1139,8 +1139,6 @@
 #define MAX(a, b) ((b) < (a) ? (a) : (b))
 #endif
 
-#ifndef CONFIG_DIALOG
-
 /* Taken from rgb->hsl routines taken from KColor
     Copyright 2007 Matthew Woehlke <mw_triad@users.sourceforge.net>
 */
@@ -1453,7 +1451,6 @@
     cb->pixel = ca->pixel;
 #endif
 }
-#endif
 
 #if (!defined CONFIG_DIALOG)
 static unsigned char checkBounds(int num)
diff -Naur QtCurve-KDE3-1.4.0/common/config_file.c QtCurve-KDE3-1.4.1/common/config_file.c
--- QtCurve-KDE3-1.4.0/common/config_file.c	2010-05-14 22:06:45.000000000 +0300
+++ QtCurve-KDE3-1.4.1/common/config_file.c	2010-05-18 11:48:20.000000000 +0300
@@ -40,6 +40,23 @@
 #define OLD_CONFIG_FILE           "qtcurvestylerc"
 #define VERSION_KEY               "version"
 
+#ifdef __cplusplus
+
+#if QT_VERSION >= 0x040000
+#include <QMap>
+#include <QFile>
+#include <QTextStream>
+#define TO_LATIN1(A) A.toLatin1().constData()
+#else
+#define TO_LATIN1(A) A.latin1()
+
+#include <qmap.h>
+#include <qfile.h>
+#include <qtextstream.h>
+#endif
+
+#endif // __cplusplus
+
 #ifdef CONFIG_READ
 static int c2h(char ch)
 {
@@ -839,19 +856,6 @@
 
 #ifdef __cplusplus
 
-#if QT_VERSION >= 0x040000
-#include <QMap>
-#include <QFile>
-#include <QTextStream>
-#define TO_LATIN1(A) A.toLatin1().constData()
-#else
-#define TO_LATIN1(A) A.latin1()
-
-#include <qmap.h>
-#include <qfile.h>
-#include <qtextstream.h>
-#endif
-
 class QtCConfig
 {
     public:
diff -Naur QtCurve-KDE3-1.4.0/config/qtcurveconfig.cpp QtCurve-KDE3-1.4.1/config/qtcurveconfig.cpp
--- QtCurve-KDE3-1.4.0/config/qtcurveconfig.cpp	2010-05-14 22:06:45.000000000 +0300
+++ QtCurve-KDE3-1.4.1/config/qtcurveconfig.cpp	2010-05-18 11:48:20.000000000 +0300
@@ -1809,7 +1809,7 @@
          mapKdeIcons->isChecked()!=currentStyle.mapKdeIcons ||
          framelessGroupBoxes->isChecked()!=currentStyle.framelessGroupBoxes ||
 
-         getSquareFlags()!=opts.square ||
+         getSquareFlags()!=currentStyle.square ||
 
          toInt(passwordChar->text())!=currentStyle.passwordChar ||
 
