Index: QtCurve-KDE3-1.1.1/config/CMakeLists.txt
===================================================================
--- QtCurve-KDE3-1.1.1.orig/config/CMakeLists.txt
+++ QtCurve-KDE3-1.1.1/config/CMakeLists.txt
@@ -8,5 +8,6 @@ include_directories(${CMAKE_SOURCE_DIR} 
 kde3_add_ui_files(kstyle_qtcurve_config_PART_SRCS qtcurveconfigbase.ui)
 kde3_add_kpart(kstyle_qtcurve_config ${kstyle_qtcurve_config_PART_SRCS})
 kde3_automoc(${kstyle_qtcurve_config_PART_SRCS})
-target_link_libraries(kstyle_qtcurve_config ${QT_AND_KDECORE_LIBS} kio kdeui)
+target_link_libraries(kstyle_qtcurve_config ${QT_AND_KDECORE_LIBS}
+    -L${KDE3_LIB_DIR} kio kdeui)
 install(TARGETS kstyle_qtcurve_config DESTINATION ${KDE3_LIB_DIR}/kde3)
