Index: QtCurve-KDE3-1.0.0/CMakeLists.txt
===================================================================
--- QtCurve-KDE3-1.0.0.orig/CMakeLists.txt
+++ QtCurve-KDE3-1.0.0/CMakeLists.txt
@@ -41,12 +41,12 @@ else (NOT QTC_QT_ONLY)
 endif (NOT QTC_QT_ONLY)
 
 if (NOT QTC_QT_ONLY)
-exec_program(kde-config
+exec_program(${KDE3PREFIX}/bin/kde-config
              ARGS --prefix --expandvars
              OUTPUT_VARIABLE CMAKE_INSTALL_PREFIX)
 endif (NOT QTC_QT_ONLY)
 
-exec_program(kde-config
+exec_program(${KDE3PREFIX}/bin/kde-config
              ARGS --install locale --expandvars
              OUTPUT_VARIABLE KDE3_LOCALE_PREFIX)
 
