diff -Nur QtCurve-KDE3-0.69.1.orig/CMakeLists.txt QtCurve-KDE3-0.69.1/CMakeLists.txt
--- QtCurve-KDE3-0.69.1.orig/CMakeLists.txt	2009-10-14 21:32:50.000000000 +0000
+++ QtCurve-KDE3-0.69.1/CMakeLists.txt	2009-11-15 16:23:02.350078972 +0000
@@ -21,11 +21,11 @@
 find_package(KDE3 REQUIRED)
 include(KDE3Macros)
 
-exec_program(kde-config
+exec_program(${KDE3PREFIX}/bin/kde-config
              ARGS --prefix --expandvars
              OUTPUT_VARIABLE CMAKE_INSTALL_PREFIX)
 
-exec_program(kde-config
+exec_program(${KDE3PREFIX}/bin/kde-config
              ARGS --install locale --expandvars
              OUTPUT_VARIABLE KDE3_LOCALE_PREFIX)
 
