--- common/config_file.c.orig	2009-03-11 23:14:56.000000000 +0200
+++ common/config_file.c	2009-03-11 23:15:08.000000000 +0200
@@ -1363,7 +1363,7 @@
 
 static const char * getSystemConfigFile()
 {
-    static const char * constFiles[]={ "/etc/qt4/"QTC_FILE, "/etc/qt3/"QTC_FILE, "/etc/qt/"QTC_FILE, "/etc/"QTC_FILE, NULL };
+    static const char * constFiles[]={ "/etc/qt4/"QTC_FILE, "/etc/qt3/"QTC_FILE, "/etc/qt/"QTC_FILE, "/etc/"QTC_FILE, "/usr/kde/3.5/share/apps/QtCurve/Pardus.qtcurve", NULL };
 
     int i;
 
