Index: common/config_file.c
===================================================================
--- common/config_file.c.orig
+++ common/config_file.c
@@ -1987,7 +1987,7 @@ static bool fileExists(const char *path)
 
 static const char * getSystemConfigFile()
 {
-    static const char * constFiles[]={ /*"/etc/qt4/"QTC_OLD_FILE, "/etc/qt3/"QTC_OLD_FILE, "/etc/qt/"QTC_OLD_FILE,*/ "/etc/"QTC_OLD_FILE, NULL };
+    static const char * constFiles[]={ /*"/etc/qt4/"QTC_OLD_FILE, "/etc/qt3/"QTC_OLD_FILE, "/etc/qt/"QTC_OLD_FILE,*/ "/etc/"QTC_OLD_FILE, "/usr/kde/3.5/share/apps/QtCurve/Pardus.qtcurve", NULL };
 
     int i;
 
