diff -up sane-backends-1.0.20/doc/descriptions/epson2.desc.epson-expression800 sane-backends-1.0.20/doc/descriptions/epson2.desc
--- sane-backends-1.0.20/doc/descriptions/epson2.desc.epson-expression800	2009-03-05 03:27:35.000000000 +0100
+++ sane-backends-1.0.20/doc/descriptions/epson2.desc	2009-06-17 11:30:17.231323638 +0200
@@ -156,7 +156,7 @@
 :status :complete
 
 :model "Expression 800"
-:interface "SCSI"
+:interface "SCSI" "EPSON" "Expression800" "processor"
 :status :complete
 
 :model "Expression 1600"
diff -up sane-backends-1.0.20/doc/descriptions/epson.desc.epson-expression800 sane-backends-1.0.20/doc/descriptions/epson.desc
--- sane-backends-1.0.20/doc/descriptions/epson.desc.epson-expression800	2009-03-05 02:54:33.000000000 +0100
+++ sane-backends-1.0.20/doc/descriptions/epson.desc	2009-06-17 11:30:17.230323577 +0200
@@ -167,7 +167,7 @@
 :status :complete
 
 :model "Expression 800"
-:interface "SCSI"
+:interface "SCSI" "EPSON" "Expression800" "processor"
 :status :complete
 
 :model "Expression 1600"
