diff --git a/Assemble.c b/Assemble.c
index e4d6181..72f7942 100644
--- a/Assemble.c
+++ b/Assemble.c
@@ -434,13 +434,6 @@ int Assemble(struct supertype *st, char *mddev,
 				}
 			}
 			st = tst; tst = NULL;
-			if (!auto_assem && tmpdev->next != NULL) {
-				fprintf(stderr, Name ": %s is a container, but is not "
-					"only device given: confused and aborting\n",
-					devname);
-				st->ss->free_super(st);
-				return 1;
-			}
 			if (verbose > 0)
 				fprintf(stderr, Name ": found match on member %s in %s\n",
 					content->text_version, devname);
