Index: tuxonice-userui-1.0/Makefile
===================================================================
--- tuxonice-userui-1.0.orig/Makefile
+++ tuxonice-userui-1.0/Makefile
@@ -5,7 +5,7 @@ DESTDIR :=
 PREFIX := /usr/local
 INSTDIR := $(DESTDIR)$(PREFIX)/sbin
 
-MODULES = text fbsplash usplash
+MODULES = text fbsplash
 TARGETS = $(patsubst %,tuxoniceui_%,$(MODULES))
 CORE_OBJECTS = userui_core.o
 
