Index: foo2zjs-0.0_20090623/hplj10xx.rules
===================================================================
--- foo2zjs-0.0_20090623.orig/hplj10xx.rules
+++ foo2zjs-0.0_20090623/hplj10xx.rules
@@ -34,3 +34,7 @@ KERNEL=="lp*", BUS=="usb", SYSFS{idVendo
 KERNEL=="lp*", BUS=="usb", SYSFS{idVendor}=="03f0", \
 	SYSFS{product}=="HP LaserJet P1505", NAME="usb/%k", \
 	SYMLINK+="hpljP1505-%n", MODE="0666", RUN+="/etc/hotplug/usb/hpljP1505" 
+#Own udev rule for HP Laserjet P1505n
+KERNEL=="lp*", BUS=="usb", SYSFS{idVendor}=="03f0", \
+	SYSFS{product}=="HP LaserJet P1505n", NAME="usb/%k", \
+	SYMLINK+="hpljP1505n-%n", MODE="0666", RUN+="/etc/hotplug/usb/hpljP1505n" 
