Index: foo2zjs/foo2zjs-wrapper.in
===================================================================
--- foo2zjs.orig/foo2zjs-wrapper.in
+++ foo2zjs/foo2zjs-wrapper.in
@@ -397,7 +397,7 @@ Custom*)
 		;;
 1|letter)	PAPER=1;	paper=letter;    XDIM="10200"; YDIM="6600"
 		case "$MODEL" in
-		0)	set_clipping 204 102	204 106;;
+		0)	set_clipping 108 94	108 98;;
 		1)	set_clipping 192 96	192 96;;
 		# 1)	set_clipping 96 96	288 96;;
 		esac
