--- CImg.h.orig	2010-05-28 16:50:54.631800743 +0000
+++ CImg.h	2010-05-28 16:52:52.671799843 +0000
@@ -256,9 +256,9 @@
 // Using FFMPEG libraries is not mandatory.
 #ifdef cimg_use_ffmpeg
 extern "C" {
-#include "avformat.h"
-#include "avcodec.h"
-#include "swscale.h"
+#include <libavformat/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libswscale/swscale.h>
 }
 #endif
 
@@ -284,7 +284,7 @@
 // to get a builtin support of various image formats (PNG,JPEG,TIFF,...).
 // Using Magick++ is not mandatory.
 #ifdef cimg_use_magick
-#include "Magick++.h"
+#include <ImageMagick/Magick++.h>
 #endif
 
 // FFTW3 configuration.
@@ -323,11 +323,11 @@
 //
 // OpenEXR can be used to read/write .exr file formats.
 #ifdef cimg_use_openexr
-#include "ImfRgbaFile.h"
-#include "ImfInputFile.h"
-#include "ImfChannelList.h"
-#include "ImfMatrixAttribute.h"
-#include "ImfArray.h"
+#include <OpenEXR/ImfRgbaFile.h>
+#include <OpenEXR/ImfInputFile.h>
+#include <OpenEXR/ImfChannelList.h>
+#include <OpenEXR/ImfMatrixAttribute.h>
+#include <OpenEXR/ImfArray.h>
 #endif
 
 // Lapack configuration.
