--- modules/amr_float_dec/Makefile	24 May 2006 09:52:54 -0000	1.4
+++ modules/amr_float_dec/Makefile	6 Mar 2007 14:56:24 -0000
@@ -19,10 +19,10 @@
 OBJS=../amr_dec/amr_in.o amr_float_dec.o
 
 #AMR NB obj
-CFLAGS+=-DGPAC_HAS_AMR_FT
-ifeq ($(CONFIG_AMR_NB_FT), yes)
-OBJS+=./amr_nb_ft/interf_dec.o ./amr_nb_ft/interf_enc.o ./amr_nb_ft/sp_dec.o ./amr_nb_ft/sp_enc.o
-endif
+#CFLAGS+=-DGPAC_HAS_AMR_FT
+#ifeq ($(CONFIG_AMR_NB_FT), yes)
+#OBJS+=./amr_nb_ft/interf_dec.o ./amr_nb_ft/interf_enc.o ./amr_nb_ft/sp_dec.o ./amr_nb_ft/sp_enc.o
+#endif
 
 
 #AMR WB obj
