Index: gdb-6.8.50.20090302/gdb/symfile.c
===================================================================
--- gdb-6.8.50.20090302.orig/gdb/symfile.c
+++ gdb-6.8.50.20090302/gdb/symfile.c
@@ -1768,6 +1768,8 @@ debug_print_executable_changed (void)
 void
 debug_print_missing (const char *binary, const char *debug)
 {
+    /* Don't mess people with debug messages */
+    return;
   size_t binary_len0 = strlen (binary) + 1;
   size_t debug_len0 = debug ? strlen (debug) + 1 : 0;
   struct missing_filepair *missing_filepair;
