Index: system-config-printer-1.1.10/Makefile.in
===================================================================
--- system-config-printer-1.1.10.orig/Makefile.in
+++ system-config-printer-1.1.10/Makefile.in
@@ -376,7 +376,7 @@ bin_SCRIPTS = \
 	system-config-printer-applet		\
 	my-default-printer
 
-@UDEV_RULES_TRUE@udevrulesdir = $(sysconfdir)/udev/rules.d
+@UDEV_RULES_TRUE@udevrulesdir = /lib/udev/rules.d
 @UDEV_RULES_TRUE@udevrules_DATA = udev/70-printers.rules
 @UDEV_RULES_TRUE@udev_udev_configure_printer_SOURCES = \
 @UDEV_RULES_TRUE@	udev/udev-configure-printer.c
