Index: polkit-gnome-0.96/src/polkit-gnome-authentication-agent-1.desktop.in.in
===================================================================
--- polkit-gnome-0.96.orig/src/polkit-gnome-authentication-agent-1.desktop.in.in
+++ polkit-gnome-0.96/src/polkit-gnome-authentication-agent-1.desktop.in.in
@@ -5,4 +5,3 @@ Exec=@FULL_LIBEXECDIR@/polkit-gnome-auth
 Terminal=false
 Type=Application
 Categories=
-OnlyShowIn=GNOME;
