Index: mysql-5.5.8/scripts/mysql_install_db.pl.in
===================================================================
--- mysql-5.5.8.orig/scripts/mysql_install_db.pl.in
+++ mysql-5.5.8/scripts/mysql_install_db.pl.in
@@ -261,7 +261,7 @@ elsif ( $opt->{basedir} )
 }
 else
 {
-  $print_defaults='@bindir@/my_print_defaults';
+  $print_defaults='/usr/bin/my_print_defaults';
 }
 
 -x $print_defaults or -f "$print_defaults.exe"
Index: mysql-5.5.8/scripts/mysql_install_db.sh
===================================================================
--- mysql-5.5.8.orig/scripts/mysql_install_db.sh
+++ mysql-5.5.8/scripts/mysql_install_db.sh
@@ -230,7 +230,7 @@ then
     exit 1
   fi
 else
-  print_defaults="@bindir@/my_print_defaults"
+  print_defaults="/usr/bin/my_print_defaults"
 fi
 
 if test ! -x "$print_defaults"
