Index: boost_1_38_0/libs/wave/build/Jamfile.v2
===================================================================
--- boost_1_38_0.orig/libs/wave/build/Jamfile.v2
+++ boost_1_38_0/libs/wave/build/Jamfile.v2
@@ -33,9 +33,7 @@ SOURCES =
 lib boost_wave
     :
     $(SOURCES)
-    ../../filesystem/build//boost_filesystem
     ../../thread/build//boost_thread
-    ../../date_time/build//boost_date_time
     ;
 
 for local source in $(SOURCES)
