diff -Naur avahi-0.6.22/avahi-autoipd/avahi-autoipd.action.linux avahi-0.6.22.orig/avahi-autoipd/avahi-autoipd.action.linux
--- avahi-0.6.22.orig/avahi-autoipd/avahi-autoipd.action.linux	2008-08-07 15:41:53.000000000 +0300
+++ avahi-0.6.22/avahi-autoipd/avahi-autoipd.action.linux	2008-08-07 15:42:19.000000000 +0300
@@ -36,11 +36,11 @@
 
     case "$1" in
         BIND)
-            ip addr add "$3"/16 brd 169.254.255.255 label "$2:avahi" scope link dev "$2" 
+            ip addr add "$3"/16 brd 169.254.255.255 label "$2" scope link dev "$2" 
             ;;
 
         CONFLICT|UNBIND|STOP)
-            ip addr del "$3"/16 brd 169.254.255.255 label "$2:avahi" scope link dev "$2" 
+            ip addr del "$3"/16 brd 169.254.255.255 label "$2" scope link dev "$2" 
             ;;
 
         *)
@@ -55,11 +55,11 @@
 
     case "$1" in
         BIND)
-            ifconfig "$2:3" inet "$3" netmask 255.255.0.0 broadcast 169.254.255.255 up
+            ifconfig "$2" inet "$3" netmask 255.255.0.0 broadcast 169.254.255.255 up
             ;;
 
         CONFLICT|STOP|UNBIND)
-            ifconfig "$2:3" down
+            ifconfig "$2" down
             ;;
 
         *)
