--- kaffeine/src/instwizard.cpp.orig	2007-07-28 19:45:37.000000000 +0300
+++ kaffeine/src/instwizard.cpp	2007-07-28 23:40:06.000000000 +0300
@@ -107,12 +107,12 @@
 	*/
 	//WIN32-CODECS
 	info << "<hr><b>" << i18n("WIN32 Codecs") << "...</b><br>";
-	QDir d("/usr/lib/win32");
+	QDir d("/usr/lib/essential");
 	QStringList entries = d.entryList("wmv*");
 	if (entries.count())
 		info << "<font color=\"DarkGreen\">" << i18n("Ok.") << "</font>";
 	else
-		info << "<font color=\"DarkRed\">" << i18n("No WIN32 codecs found in /usr/lib/win32. You're not able to play Windows Media 9 files, newer Real Media files and some less common formats. Download the codecs here:")
+		info << "<font color=\"DarkRed\">" << i18n("No WIN32 codecs found in /usr/lib/essential. You're not able to play Windows Media 9 files, newer Real Media files and some less common formats. Download the codecs here:")
 		<< " <a href=\"http://www1.mplayerhq.hu/homepage/design7/codecs.html\">http://www.mplayerhq.hu</a>.</font>";
 
 	//LIBDVDCSS
