Index: notification-daemon-0.4.0/src/capplet/notification-properties.desktop.in
===================================================================
--- notification-daemon-0.4.0.orig/src/capplet/notification-properties.desktop.in
+++ notification-daemon-0.4.0/src/capplet/notification-properties.desktop.in
@@ -9,3 +9,4 @@ Type=Application
 StartupNotify=true
 Categories=GNOME;GTK;Settings;
 OnlyShowIn=GNOME;
+NoDisplay=true
