--- alleggl/include/alleggl.h~	2007-07-21 12:02:43.000000000 +0200
+++ alleggl/include/alleggl.h	2007-08-19 14:24:57.000000000 +0200
@@ -72,10 +72,10 @@
 #   ifndef HAVE_LIBPTHREAD
 #      error AllegroGL requires Allegro to have pthread support enabled!
 #   endif
-#else
+/*#else
 #   ifndef ALLEGRO_HAVE_LIBPTHREAD
 #      error AllegroGL requires Allegro to have pthread support enabled!
-#   endif
+#   endif*/
 #endif
 #include <alleggl_config.h>
 #endif
