Index: mozilla-1.9.2/modules/libpref/src/init/all.js
===================================================================
--- mozilla-1.9.2.orig/modules/libpref/src/init/all.js
+++ mozilla-1.9.2/modules/libpref/src/init/all.js
@@ -972,7 +972,7 @@ pref("intl.charsetmenu.browser.cache.siz
 pref("intl.charset.detector",               "chrome://global/locale/intl.properties");
 pref("intl.charset.default",                "chrome://global-platform/locale/intl.properties");
 pref("intl.ellipsis",                       "chrome://global-platform/locale/intl.properties");
-pref("intl.locale.matchOS",                 false);
+pref("intl.locale.matchOS",                 true);
 // fallback charset list for Unicode conversion (converting from Unicode)
 // currently used for mail send only to handle symbol characters (e.g Euro, trademark, smartquotes)
 // for ISO-8859-1
