diff -Nur kdeutils-3.5.10-old/ark/sevenzip.cpp kdeutils-3.5.10/ark/sevenzip.cpp
--- kdeutils-3.5.10-old/ark/sevenzip.cpp	2008-11-03 15:25:58.000000000 +0200
+++ kdeutils-3.5.10/ark/sevenzip.cpp	2008-11-03 15:38:50.000000000 +0200
@@ -207,6 +207,9 @@
   // extract (and maybe overwrite)
   *kp << m_unarchiver_program << "x";
 
+  if ( !m_password.isEmpty() )
+    *kp << "-p"+ m_password;
+
   if ( ArkSettings::extractOverwrite() )
   {
     //*kp << "-ao";
