--- kdeutils-3.5.10-old/ark/sevenzip.cpp	2008-11-04 20:31:58.000000000 +0200
+++ kdeutils-3.5.10/ark/sevenzip.cpp	2008-11-05 00:54:01.000000000 +0200
@@ -251,7 +251,7 @@
   unsigned int pos = 0;
   int strpos, len;
 
-  columns[ 0 ] = line.right( line.length() - m_nameColumnPos +1);
+  columns[ 0 ] = line.right( line.length() - m_nameColumnPos);
   line.truncate( m_nameColumnPos );
 
   // Go through our columns, try to pick out data, return silently on failure
