--- kdeutils-3.5.10/ark/arch.cpp	2008-02-13 11:38:15.000000000 +0200
+++ kdeutils-3.5.10-patches/ark/arch.cpp	2009-04-07 11:47:30.000000000 +0300
@@ -105,12 +105,30 @@
   if ( _kp->normalExit() )
     exitStatus = _kp->exitStatus();
 
+  // Open could fail on some archives which don't give a volume list without the password (#8889).
+  if ( exitStatus != 0 && passwordRequired())
+  {
+      QString msg;
+      if ( !m_password.isEmpty() )
+          msg = i18n("The password was incorrect. ");
+      if (KPasswordDialog::getPassword( m_password, msg+i18n("You must enter a password to extract the file:") ) == KPasswordDialog::Accepted )
+      {
+          delete _kp;
+          _kp = m_currentProcess = 0;
+          clearShellOutput();
+          open(); // try to open the file again with a password
+          return;
+      }
+      m_password = "";
+  }
+
   if ( exitStatus == 1 )
   {
     exitStatus = 0;    // because 1 means empty archive - not an error.
                        // Is this a safe assumption?
   }
 
+
   if ( !exitStatus )
     emit sigOpen( this, true, m_filename,
                   Arch::Extract | Arch::Delete | Arch::Add | Arch::View );
