--- kdeutils-3.5.10/ark/rar.cpp	2008-02-13 11:38:15.000000000 +0200
+++ kdeutils-3.5.10-patches/ark/rar.cpp	2009-04-07 11:46:02.000000000 +0300
@@ -119,7 +119,14 @@
   m_finished = false;
 
   KProcess *kp = m_currentProcess = new KProcess;
-  *kp << m_unarchiver_program << "v" << "-c-" << m_filename;
+  *kp << m_unarchiver_program << "v" << "-c-";
+
+  if ( !m_password.isEmpty() )
+    *kp << "-p" + m_password;
+  else
+    *kp << "-p-";
+
+  *kp << m_filename;
 
   connect( kp, SIGNAL( receivedStdout(KProcess*, char*, int) ),
            SLOT( slotReceivedTOC(KProcess*, char*, int) ) );
