--- kdeutils-3.5.10/ark/sevenzip.h	2006-05-22 21:08:38.000000000 +0300
+++ kdeutils-3.5.10-patches/ark/sevenzip.h	2009-04-07 11:31:02.000000000 +0300
@@ -41,6 +41,7 @@
 
     virtual void remove( QStringList * );
     virtual void unarchFileInternal( );
+    virtual bool passwordRequired();
 
   protected slots:
     virtual bool processLine( const QCString& line );
--- kdeutils-3.5.10/ark/sevenzip.cpp	2009-04-07 11:30:51.000000000 +0300
+++ kdeutils-3.5.10-patches/ark/sevenzip.cpp	2009-04-07 11:31:02.000000000 +0300
@@ -163,6 +163,11 @@
   }
 }
 
+bool SevenZipArch::passwordRequired()
+{
+    return m_lastShellOutput.findRev("Wrong password?")+1;
+}
+
 void SevenZipArch::remove( QStringList *list )
 {
   if ( !list )
