Index: linux-2.6.32/drivers/staging/rt2870/2870_main_dev.c
===================================================================
--- linux-2.6.32.orig/drivers/staging/rt2870/2870_main_dev.c
+++ linux-2.6.32/drivers/staging/rt2870/2870_main_dev.c
@@ -110,6 +110,7 @@ struct usb_device_id rtusb_usb_id[] = {
 	{ USB_DEVICE(0x0CDE, 0x0022) }, /* ZCOM */
 	{ USB_DEVICE(0x0586, 0x3416) }, /* Zyxel */
 	{ USB_DEVICE(0x0CDE, 0x0025) }, /* Zyxel */
+	{ USB_DEVICE(0x0586, 0x341a) }, /* Zyxel */
 	{ USB_DEVICE(0x1740, 0x9701) }, /* EnGenius */
 	{ USB_DEVICE(0x1740, 0x9702) }, /* EnGenius */
 	{ USB_DEVICE(0x1740, 0x9703) }, /* EnGenius 3070 */
