diff --git a/hardware/library/obexftp/actions.py b/hardware/library/obexftp/actions.py
index 61c5c18..2490b43 100644
--- a/hardware/library/obexftp/actions.py
+++ b/hardware/library/obexftp/actions.py
@@ -27,5 +27,5 @@ def install():
     autotools.rawInstall("DESTDIR=%s" % get.installDIR())
 
     for f in [".packlist", "OBEXFTP.bs"]:
-        pisitools.remove("/usr/lib/perl5/vendor_perl/%s/i686-linux-thread-multi/auto/OBEXFTP/%s" % (get.curPERL(), f))
+        pisitools.remove("/usr/lib/perl5/vendor_perl/%s/x86_64-linux-thread-multi/auto/OBEXFTP/%s" % (get.curPERL(), f))
 
