diff --git a/hardware/printer/foomatic-db-engine/actions.py b/hardware/printer/foomatic-db-engine/actions.py
index 25e8051..5ed4aa8 100644
--- a/hardware/printer/foomatic-db-engine/actions.py
+++ b/hardware/printer/foomatic-db-engine/actions.py
@@ -32,7 +32,7 @@ def install():
 
     # Clean .packlists
     pisitools.removeDir("/usr/lib/perl5/%s" % get.curPERL())
-    pisitools.removeDir("/usr/lib/perl5/vendor_perl/%s/i686-linux-thread-multi" % get.curPERL())
+    pisitools.removeDir("/usr/lib/perl5/vendor_perl/%s/x86_64-linux-thread-multi" % get.curPERL())
 
     shelltools.cd("..")
     pisitools.dodoc("ChangeLog", "COPYING", "README", "TODO", "USAGE")
