diff --git a/multimedia/sound/lame/actions.py b/multimedia/sound/lame/actions.py
index b53d138..caa5758 100644
--- a/multimedia/sound/lame/actions.py
+++ b/multimedia/sound/lame/actions.py
@@ -18,7 +18,7 @@ def setup():
     libtools.libtoolize("--copy --force")
     autotools.autoreconf("-fi")
 
-    shelltools.makedirs("libmp3lame/i386/.libs")
+    shelltools.makedirs("libmp3lame/%s/.libs" %(get.HOST()))
     autotools.configure("--disable-mp3x \
                          --disable-static \
                          --enable-shared \
