diff --git a/multimedia/sound/mpg123/actions.py b/multimedia/sound/mpg123/actions.py
index 77afa01..0cb1246 100644
--- a/multimedia/sound/mpg123/actions.py
+++ b/multimedia/sound/mpg123/actions.py
@@ -14,7 +14,7 @@ KeepSpecial = ["libtool"]
 def setup():
     pisitools.dosed("configure", "-faltivec")
     autotools.configure('--with-audio="alsa oss" \
-                         --with-cpu=sse \
+                         --with-cpu=x86-64 \
                          --with-optimization=2 \
                          --enable-network=yes \
                          --disable-ltdl-install')
