diff --git a/multimedia/video/ffmpeg/actions.py b/multimedia/video/ffmpeg/actions.py
index e7cf56d..810896e 100644
--- a/multimedia/video/ffmpeg/actions.py
+++ b/multimedia/video/ffmpeg/actions.py
@@ -23,8 +23,7 @@ def setup():
     shelltools.chmod("version.sh", 0755)
 
     # CPU thing is just used for CMOV detection
-    autotools.rawConfigure("--cpu=i686 \
-                            --prefix=/usr \
+    autotools.rawConfigure("--prefix=/usr \
                             --mandir=/usr/share/man \
                             --enable-runtime-cpudetect \
                             --enable-gpl \
@@ -50,7 +49,6 @@ def setup():
                             --enable-mmx \
                             --enable-mmx2 \
                             --enable-sse \
-                            --enable-vaapi \
                             --enable-vdpau \
                             --enable-yasm \
                             --disable-stripping \
