diff --git a/network/library/nss/actions.py b/network/library/nss/actions.py
index 79a1c0b..b3de9b2 100644
--- a/network/library/nss/actions.py
+++ b/network/library/nss/actions.py
@@ -12,7 +12,8 @@ from pisi.actionsapi import get
 
 WorkDir="mozilla"
 
-def build():
+def build(): 
+    shelltools.export("USE_64","1")
     shelltools.export("BUILD_OPT","1")
     shelltools.export("NSS_ENABLE_ECC","1")
     shelltools.export("OPT_FLAGS","%s -g -fno-strict-aliasing" % get.CFLAGS())
