diff --git a/programming/language/java/sun-java/actions.py b/programming/language/java/sun-java/actions.py
index 85c469b..f299efc 100644
--- a/programming/language/java/sun-java/actions.py
+++ b/programming/language/java/sun-java/actions.py
@@ -15,7 +15,7 @@ Name = "6u17"
 
 def setup():
     shelltools.system("chmod +x construct.sh")
-    shelltools.system("sh jdk-%s-dlj-linux-i586.bin --accept-license" % Name)
+    shelltools.system("sh jdk-%s-dlj-linux-amd64.bin --accept-license" % Name)
 
 def install():
     pisitools.dodir("/opt")
@@ -23,7 +23,7 @@ def install():
 
     # Install mozilla plugin
     pisitools.dodir("/usr/lib/nsbrowser/plugins")
-    pisitools.dosym("/opt/sun-jre/plugin/i386/ns7/libjavaplugin_oji.so", "/usr/lib/nsbrowser/plugins/javaplugin.so")
+    pisitools.dosym("/opt/sun-jre/plugin/amd64/ns7/libjavaplugin_oji.so", "/usr/lib/nsbrowser/plugins/javaplugin.so")
 
     for doc in ["COPYRIGHT", "LICENSE", "README.html", "README_ja.html", "README_zh_CN.html", "THIRDPARTYLICENSEREADME.txt"]:
         file = "%s/opt/sun-jdk/%s" % (get.installDIR(),doc)
diff --git a/programming/language/java/sun-java/pspec.xml b/programming/language/java/sun-java/pspec.xml
index 1769924..f81291a 100644
--- a/programming/language/java/sun-java/pspec.xml
+++ b/programming/language/java/sun-java/pspec.xml
@@ -13,7 +13,7 @@
         <IsA>library</IsA>
         <Summary>Sun's J2SE Development Kit</Summary>
         <Description>Java Platform, Standard Edition (also known as Java 2 Platform) lets you develop and deploy Java applications on desktops and servers, as well as today's demanding Embedded and Real-Time environments.</Description>
-        <Archive sha1sum="bca59cb275f37b492e69ceecbda75a573d94b553" type="binary">http://download.java.net/dlj/binaries/jdk-6u17-dlj-linux-i586.bin</Archive>
+        <Archive sha1sum="a42bba8a2efe2a86403702507b99ff63086acea5" type="binary">http://download.java.net/dlj/binaries/jdk-6u17-dlj-linux-amd64.bin</Archive>
         <BuildDependencies>
             <Dependency>alsa-lib</Dependency>
             <Dependency>unixODBC</Dependency>
