diff --git a/programming/library/libstdc++/actions.py b/programming/library/libstdc++/actions.py
index 73bdad8..ceeab9b 100644
--- a/programming/library/libstdc++/actions.py
+++ b/programming/library/libstdc++/actions.py
@@ -18,8 +18,8 @@ LIBPATH = "/usr/lib/gcc-lib/%s/3.3.6" % get.HOST()
 DATAPATH = "/usr/share/gcc-data/%s/3.3.6" % get.HOST()
 STDCXX_INCDIR = "/usr/lib/gcc-lib/%s/3.3.6/include/g++-v3" % get.HOST()
 
-cflags = get.CFLAGS().replace("mtune", "mcpu").replace("-fstack-protector", "").replace("-D_FORTIFY_SOURCE=2", "").replace("-mcpu=generic", "-mcpu=i686")
-cxxflags =  get.CXXFLAGS().replace("mtune", "mcpu").replace("-fstack-protector", "").replace("-D_FORTIFY_SOURCE=2", "").replace("-mcpu=generic", "-mcpu=i686")
+cflags = get.CFLAGS().replace("mtune", "mcpu").replace("-fstack-protector", "").replace("-D_FORTIFY_SOURCE=2", "").replace("-mcpu=generic", "-mcpu=x86-64")
+cxxflags =  get.CXXFLAGS().replace("mtune", "mcpu").replace("-fstack-protector", "").replace("-D_FORTIFY_SOURCE=2", "").replace("-mcpu=generic", "-mcpu=x86-64")
 
 def setup():
     shelltools.export("CFLAGS", cflags)
