diff --git a/programming/library/qca/actions.py b/programming/library/qca/actions.py
index 00ac0b3..d54f47b 100644
--- a/programming/library/qca/actions.py
+++ b/programming/library/qca/actions.py
@@ -10,7 +10,7 @@ WorkDir = "qca-1.0"
 def setup():
     autotools.rawConfigure("--prefix=/usr/qt/3")
     pisitools.dosed("Makefile", "(?m)^(CFLAGS.*)",  "CFLAGS =" + get.CFLAGS())
-    pisitools.dosed("Makefile", "(?m)^(CXXFLAGS.*)", "CXXFLAGS =" + get.CXXFLAGS())
+    pisitools.dosed("Makefile", "(?m)^(CXXFLAGS.*)", "CXXFLAGS =" + get.CXXFLAGS() + " -fPIC ")
 
 def build():
     autotools.make()
