diff --git a/programming/tool/dev86/actions.py b/programming/tool/dev86/actions.py
index b1e630e..020617c 100644
--- a/programming/tool/dev86/actions.py
+++ b/programming/tool/dev86/actions.py
@@ -11,6 +11,10 @@ from pisi.actionsapi import get
 
 WorkDir="dev86-%s" % get.srcVERSION()
 
+def setup():
+    pisitools.dosed("makefile.in", "alt-libs elksemu", "alt-libs")
+    pisitools.dosed("makefile.in", "install-lib install-emu", "install-lib")
+
 def build():
     autotools.make("-j1 < /dev/null")
 
