diff --git a/system/devel/binutils/actions.py b/system/devel/binutils/actions.py
index d58c12c..598a6e0 100644
--- a/system/devel/binutils/actions.py
+++ b/system/devel/binutils/actions.py
@@ -29,7 +29,7 @@ def install():
 
     # Rebuild libbfd.a and libiberty.a with -fPIC
     pisitools.remove("/usr/lib/libbfd.a")
-    pisitools.remove("/usr/lib/libiberty.a")
+    pisitools.remove("/usr/lib64/libiberty.a")
     # pisitools.remove("/usr/include/libiberty.h")
 
     autotools.make("-C libiberty clean")
@@ -39,7 +39,7 @@ def install():
     autotools.make('CFLAGS="-fPIC %s" -C bfd' % get.CFLAGS())
 
     pisitools.insinto("/usr/lib", "bfd/libbfd.a")
-    pisitools.insinto("/usr/lib", "libiberty/libiberty.a")
+    pisitools.insinto("/usr/lib64", "libiberty/libiberty.a")
     pisitools.insinto("/usr/include", "include/libiberty.h")
 
     # Prevent programs to link against libbfd and libopcodes dynamically,
diff --git a/system/devel/binutils/pspec.xml b/system/devel/binutils/pspec.xml
index 3382022..d1b1d81 100644
--- a/system/devel/binutils/pspec.xml
+++ b/system/devel/binutils/pspec.xml
@@ -56,7 +56,8 @@
         <Name>binutils</Name>
         <Files>
             <Path fileType="executable">/usr/bin</Path>
-            <Path fileType="library">/usr/lib</Path>
+             <Path fileType="library">/usr/lib</Path> 
+             <Path fileType="library">/usr/lib64</Path>
             <Path fileType="header">/usr/include</Path>
             <Path fileType="info">/usr/share/info</Path>
             <Path fileType="man">/usr/share/man</Path>
