diff --git a/util/archive/p7zip/actions.py b/util/archive/p7zip/actions.py
index d22f1ed..a0fef40 100644
--- a/util/archive/p7zip/actions.py
+++ b/util/archive/p7zip/actions.py
@@ -7,11 +7,13 @@
 
 from pisi.actionsapi import autotools
 from pisi.actionsapi import pisitools
+from pisi.actionsapi import shelltools
 from pisi.actionsapi import get
 
 WorkDir = "p7zip_%s" % get.srcVERSION()
 
 def build():
+    shelltools.copy("makefile.linux_amd64", "makefile.machine")
     autotools.make('OPTFLAGS="%s -DHAVE_GCCVISIBILITYPATCH -fvisibility=hidden -fvisibility-inlines-hidden" \
                     all3' % get.CFLAGS())
 
