Index: devel/kernel/default/drivers/module-fglrx/actions.py
===================================================================
--- devel.orig/kernel/default/drivers/module-fglrx/actions.py
+++ devel/kernel/default/drivers/module-fglrx/actions.py
@@ -42,8 +42,8 @@ def install():
             "common/usr/share/doc/fglrx/examples/etc/acpi/events":  "/etc/acpi",
             "common/etc":               "/",
             "common/usr/X11R6/include": "/usr",
-            "arch/%s/usr/X11R6/lib/*" % Target: "/usr/lib",
-            "arch/%s/usr/lib/*" % Target:       "/usr/lib",
+            "arch/%s/usr/X11R6/lib64/*" % Target: "/usr/lib",
+            "arch/%s/usr/lib64/*" % Target:       "/usr/lib",
             "common/usr/include/GL/":   "/usr/lib/xorg/fglrx/include",
             "common/usr/share":         "/usr"
             }
