Index: devel/science/library/djbfft/actions.py
===================================================================
--- devel.orig/science/library/djbfft/actions.py
+++ devel/science/library/djbfft/actions.py
@@ -13,6 +13,11 @@ from pisi.actionsapi import get
 lib_version=get.srcVERSION()[0] + "." + get.srcVERSION()[2] + "." + get.srcVERSION()[3]
 
 def setup():
+    # Not overwrite this files.
+    shelltools.unlink('conf-cc')
+    shelltools.unlink('conf-ld')
+    shelltools.unlink('conf-home')
+    
     shelltools.echo('conf-cc',"%s %s -O1 -fomit-frame-pointer -malign-double -fPIC -DPIC" % (get.CC(),get.CFLAGS()))
     shelltools.echo('conf-ld',"%s %s" % (get.CC(),get.LDFLAGS()))
     shelltools.echo('conf-home',"%s/usr" % get.installDIR())
