diff -Nur ffmpeg-old/cmdutils.c ffmpeg/cmdutils.c
--- ffmpeg-old/cmdutils.c	2010-02-11 00:17:10.314692869 +0200
+++ ffmpeg/cmdutils.c	2010-02-11 00:21:23.948695784 +0200
@@ -370,6 +370,7 @@
 {
     fprintf(stderr, "%s version " FFMPEG_VERSION ", Copyright (c) %d-%d Fabrice Bellard, et al.\n",
             program_name, program_birth_year, this_year);
+    /*
     fprintf(stderr, "  built on %s %s with %s %s\n",
             __DATE__, __TIME__, CC_TYPE, CC_VERSION);
     fprintf(stderr, "  configuration: " FFMPEG_CONFIGURATION "\n");
@@ -381,6 +382,7 @@
     PRINT_LIB_CONFIG(SWSCALE,  "libswscale",  swscale_configuration());
     PRINT_LIB_CONFIG(POSTPROC, "libpostproc", postproc_configuration());
     print_all_lib_versions(stderr, 1);
+    */
 }
 
 void show_version(void) {
@@ -391,7 +393,7 @@
 void show_license(void)
 {
     printf(
-#if CONFIG_NONFREE
+#if 0
     "This version of %s has nonfree parts compiled in.\n"
     "Therefore it is not legally redistributable.\n",
     program_name
