Index: ptsp-client-rootfs/var/db/comar3/scripts/System.Service/ptsp_client.py
===================================================================
--- ptsp-client-rootfs.orig/var/db/comar3/scripts/System.Service/ptsp_client.py
+++ ptsp-client-rootfs/var/db/comar3/scripts/System.Service/ptsp_client.py
@@ -46,6 +46,7 @@ def start():
         fail(MSG_ERR_NOTTHIN)
 
     startDependencies("acpid")
+    startDependencies("hal")
     configure()
     loadEnvironment()
     os.environ["XAUTHLOCALHOSTNAME"]=os.uname()[1]
