diff -Nuar ahenk-agent-1.9.30.orig/modules/mod_firewall.py ahenk-agent-1.9.30/modules/mod_firewall.py
--- ahenk-agent-1.9.30.orig/modules/mod_firewall.py	2011-11-22 16:17:04.105000014 +0200
+++ ahenk-agent-1.9.30/modules/mod_firewall.py	2011-11-22 16:18:32.334000014 +0200
@@ -60,11 +60,13 @@
         stack.reverse()
         first = True
         logging.info("Firewall: Loading rule sets.")
+
+        # firewallState policy not supported for now
         for policy in stack:
-            firewallState = policy.get("firewallState", [""])[0]
+            # firewallState = policy.get("firewallState", [""])[0]
             firewallRules = policy.get("firewallRules", [""])[0]
-            if firewallState == "on":
-                enable_firewall(firewallRules, options, first)
-            elif firewallState == "off" and first:
-                disable_firewall(options)
+            # if firewallState == "on":
+            enable_firewall(firewallRules, options, first)
+            # elif firewallState == "off" and first:
+            #    disable_firewall(options)
             first = False
