diff -Nur util-linux-ng-2.15-old/libs/blkid/src/probers/ntfs.c util-linux-ng-2.15/libs/blkid/src/probers/ntfs.c
--- util-linux-ng-2.15-old/libs/blkid/src/probers/ntfs.c	2009-05-21 11:17:36.712563135 +0300
+++ util-linux-ng-2.15/libs/blkid/src/probers/ntfs.c	2009-05-21 11:20:16.013689359 +0300
@@ -172,7 +172,7 @@
 
 const struct blkid_idinfo ntfs_idinfo =
 {
-	.name		= "ntfs",
+	.name		= "ntfs-3g",
 	.usage		= BLKID_USAGE_FILESYSTEM,
 	.probefunc	= probe_ntfs,
 	.magics		=
