--- xterm-222/charproc.c.can-2003-0063	2006-09-28 02:06:28.000000000 +0200
+++ xterm-222/charproc.c	2006-11-22 18:38:28.000000000 +0100
@@ -4587,8 +4587,13 @@
 	break;
 
     case 21:			/* Report the window's title */
+/* This code block is disabled to fix a security issue reported
+ * in CAN-2003-0063
+ */
+#if 0
 	report_win_label(xw, 'l', &text,
 			 XGetWMName(screen->display, VShellWindow, &text));
+#endif
 	break;
 
     default:			/* DECSLPP (24, 25, 36, 48, 72, 144) */
