--- libmovtar-0.1.3/configure.in.old	2003-12-08 13:38:16.000000000 +0100
+++ libmovtar-0.1.3/configure.in	2003-12-08 13:38:44.000000000 +0100
@@ -8,7 +8,7 @@
 ac_default_prefix=/usr/local
 
 dnl Check for GLib
-AM_PATH_GLIB(1.2.0, , AC_MSG_ERROR("You need to install GLib - you can find it on http://www.gtk.org"))
+PKG_CHECK_MODULES(GLIB, glib-2.0, , AC_MSG_ERROR("You need to install GLib - you can find it on http://www.gtk.org"))
 
 dnl Checks for programs.
 AC_PROG_CC
