Index: VirtualBox-3.1.2_OSE/src/VBox/Additions/x11/Installer/98vboxadd-xclient
===================================================================
--- VirtualBox-3.1.2_OSE.orig/src/VBox/Additions/x11/Installer/98vboxadd-xclient
+++ VirtualBox-3.1.2_OSE/src/VBox/Additions/x11/Installer/98vboxadd-xclient
@@ -26,7 +26,9 @@ for i in $HOME/.vboxclient-*.pid; do
   fi
 done
 
-/usr/bin/VBoxClient --clipboard
-/usr/bin/VBoxClient --checkhostversion
-/usr/bin/VBoxClient --display
-/usr/bin/VBoxClient --seamless
+if [ -c /dev/vboxuser ]; then
+    /usr/bin/VBoxClient --clipboard
+    /usr/bin/VBoxClient --checkhostversion
+    /usr/bin/VBoxClient --display
+    /usr/bin/VBoxClient --seamless
+fi
