Index: VirtualBox-2.2.2_OSE/src/VBox/Additions/x11/Installer/98vboxadd-xclient
===================================================================
--- VirtualBox-2.2.2_OSE.orig/src/VBox/Additions/x11/Installer/98vboxadd-xclient
+++ VirtualBox-2.2.2_OSE/src/VBox/Additions/x11/Installer/98vboxadd-xclient
@@ -17,7 +17,9 @@
 # additional information or have any questions.
 #
 
-killall -q /usr/bin/VBoxClient
-/usr/bin/VBoxClient --clipboard
-/usr/bin/VBoxClient --autoresize
-/usr/bin/VBoxClient --seamless
+if [ -c /dev/vboxuser ]; then
+    killall -q /usr/bin/VBoxClient
+    /usr/bin/VBoxClient --clipboard
+    /usr/bin/VBoxClient --autoresize
+    /usr/bin/VBoxClient --seamless
+fi
