diff -ur PyKDE-3.16.0.orig/sip/kio/kmimetype.sip PyKDE-3.16.0/sip/kio/kmimetype.sip
--- PyKDE-3.16.0.orig/sip/kio/kmimetype.sip	2006-09-22 16:11:53.000000000 +0300
+++ PyKDE-3.16.0/sip/kio/kmimetype.sip	2006-11-09 23:38:36.000000000 +0200
@@ -101,7 +101,7 @@
 %If ( KDE_3_2_0 -  )
     static KMimeType::Ptr defaultMimeTypePtr ();
     QString              parentMimeType () const;
-    bool                 is (const QString&) const;
+    bool                 is (const QString&) const /PyName=is_/; //'is' is a reserved keyword in python
     static KMimeType::Ptr diagnoseFileName (const QString&, QString&);
 %End
 
