--- a/src/package-manager.py	(revision 26916)
+++ b/src/package-manager.py	(revision 26917)
@@ -121,7 +121,7 @@
         self.showInstalledAction = KToggleAction(i18n("Show Installed Packages"),"package",KShortcut.null(),
                                                  self.mainwidget.removeState,self.actionCollection(),
                                                  "show_installed_action")
-        self.showNewAction = KToggleAction(i18n("Show New Packages"),"edit_add",KShortcut.null(),
+        self.showNewAction = KToggleAction(i18n("Show New Packages"),"list-add",KShortcut.null(),
                                            self.mainwidget.installState,self.actionCollection(),"show_new_action")
         self.showUpgradeAction = KToggleAction(i18n("Show Upgradable Packages"),"reload",KShortcut.null(),
                                                self.mainwidget.updateCheck ,self.actionCollection(),"show_upgradable_action")
