Index: src/grouplist.py
===================================================================
--- src/grouplist.py	(revision 29672)
+++ src/grouplist.py	(working copy)
@@ -67,7 +67,9 @@
     def currentGroup(self):
         if not self.count():
             return None
-        return unicode(self.currentItem().data(Qt.UserRole).toString())
+        if self.currentItem():
+           return unicode(self.currentItem().data(Qt.UserRole).toString())
+        return None
 
     def groupChanged(self):
         self.emit(SIGNAL("groupChanged()"))
