Index: src/tray.py
===================================================================
--- src/tray.py	(revision 36047)
+++ src/tray.py	(working copy)
@@ -194,5 +194,5 @@
 
     def showPopup(self):
         if self._ready_to_popup():
-            self.showMessage(i18n('Updates'), i18n("There are %1 updates available!", self.unread))
+            Pds.notify(i18n('Updates'), i18n("There are %1 updates available!", self.unread))
 
