diff -Nur gutenprint-5.1.7-old/src/ghost/ijsgutenprint.c gutenprint-5.1.7/src/ghost/ijsgutenprint.c
--- gutenprint-5.1.7-old/src/ghost/ijsgutenprint.c	2008-04-08 12:08:05.000000000 +0300
+++ gutenprint-5.1.7/src/ghost/ijsgutenprint.c	2008-04-08 12:08:10.000000000 +0300
@@ -35,8 +35,8 @@
 #include <string.h>
 #include <unistd.h>
 #include <locale.h>
-#include <ijs.h>
-#include <ijs_server.h>
+#include <ijs/ijs.h>
+#include <ijs/ijs_server.h>
 #include <errno.h>
 #include <gutenprint/gutenprint-intl-internal.h>
 
