Index: libproxy-0.4.3/CMakeLists.txt
===================================================================
--- libproxy-0.4.3.orig/CMakeLists.txt
+++ libproxy-0.4.3/CMakeLists.txt
@@ -41,7 +41,7 @@ if(${WIN32})
 else()
   set(rlibdir     lib${LIB_SUFFIX}/)
   set(rmoduledir  ${rlibdir}${CMAKE_PROJECT_NAME}/${CMAKE_PROJECT_VERSION}/modules/)
-  set(rlibexecdir libexec/)
+  set(rlibexecdir lib/${CMAKE_PROJECT_NAME})
   set(rbindir     bin/)
   set(rincludedir include/)
   set(rdatadir    share/)
