diff -Nuar rtorrent-0.8.9.orig/doc//rtorrent.rc rtorrent-0.8.9/doc//rtorrent.rc
--- rtorrent-0.8.9.orig/doc//rtorrent.rc	2011-11-10 09:55:23.885000101 +0200
+++ rtorrent-0.8.9/doc//rtorrent.rc	2011-11-10 10:00:20.001000102 +0200
@@ -106,3 +106,6 @@
 # before forcing. Overworked systems might need lower values to get a
 # decent hash checking rate.
 #hash_max_tries = 10
+
+# Configure scgi port to use rtorrent through web gui
+#scgi_port = localhost:5000
