Index: kdebase-3.5.10/kcontrol/clock/dtime.cpp
===================================================================
--- kdebase-3.5.10.orig/kcontrol/clock/dtime.cpp
+++ kdebase-3.5.10/kcontrol/clock/dtime.cpp
@@ -234,11 +234,13 @@ void Dtime::load()
   KConfig config("kcmclockrc", true, false);
   config.setGroup("NTP");
   timeServerList->insertStringList(QStringList::split(',', config.readEntry("servers",
-    i18n("Public Time Server (pool.ntp.org),\
+"Public Time Server (pool.ntp.org),\
 asia.pool.ntp.org,\
 europe.pool.ntp.org,\
+time.ume.tubitak.gov.tr, \
+time.deu.edu.tr, \
 north-america.pool.ntp.org,\
-oceania.pool.ntp.org"))));
+oceania.pool.ntp.org")));
   setDateTimeAuto->setChecked(config.readBoolEntry("enabled", false));
 
   // Reset to the current date and time
