Index: kdebase-3.5.10/startkde
===================================================================
--- kdebase-3.5.10.orig/startkde
+++ kdebase-3.5.10/startkde
@@ -415,9 +415,6 @@ echo 'startkde: Shutting down...'  1>&2
 # Clean up
 kdeinit_shutdown
 dcopserver_shutdown --wait
-artsshell -q terminate
-# KDE4 support
-kde4 kdeinit4_shutdown 2>/dev/null
 
 echo 'startkde: Running shutdown scripts...'  1>&2
 
