Index: kdebase-3.5.10/kicker/libkicker/kickerSettings.kcfg
===================================================================
--- kdebase-3.5.10.orig/kicker/libkicker/kickerSettings.kcfg
+++ kdebase-3.5.10/kicker/libkicker/kickerSettings.kcfg
@@ -35,7 +35,7 @@
 <entry name="UseBackgroundTheme" type="Bool" >
       <label>Enable background image</label>
       <whatsthis>When this option is enabled, the panel will display a tiled image as its background</whatsthis>
-      <default>true</default>
+      <default>false</default>
    </entry>
 
 <entry name="ColorizeBackground" type="Bool" >
