Index: kdebase-3.5.10/kcontrol/icons/iconthemes.cpp
===================================================================
--- kdebase-3.5.10.orig/kcontrol/icons/iconthemes.cpp
+++ kdebase-3.5.10/kcontrol/icons/iconthemes.cpp
@@ -353,8 +353,10 @@ void IconThemesConfig::themeSelected(QLi
   if (!icon.isValid()) {
 #ifdef HAVE_LIBART
 	  icon=icontheme.iconPath("exec.svg", size, KIcon::MatchBest);
-	  if(engine.load(size, size, icon.path))
-              m_previewExec->setPixmap(*engine.image());
+      if(!icon.path.isEmpty()){
+          if(engine.load(size, size, icon.path))
+                  m_previewExec->setPixmap(*engine.image());
+          }
           else {
               icon=icontheme.iconPath("exec.svgz", size, KIcon::MatchBest);
               if(engine.load(size, size, icon.path))
@@ -369,8 +371,10 @@ void IconThemesConfig::themeSelected(QLi
   if (!icon.isValid()) {
 #ifdef HAVE_LIBART
 	  icon=icontheme.iconPath("folder.svg", size, KIcon::MatchBest);
-	  if(engine.load(size, size, icon.path))
-              m_previewFolder->setPixmap(*engine.image());
+      if(!icon.path.isEmpty()){
+          if(engine.load(size, size, icon.path))
+                  m_previewFolder->setPixmap(*engine.image());
+          }
           else {
               icon=icontheme.iconPath("folder.svgz", size, KIcon::MatchBest);
               if(engine.load(size, size, icon.path))
@@ -385,8 +389,10 @@ void IconThemesConfig::themeSelected(QLi
   if (!icon.isValid()) {
 #ifdef HAVE_LIBART
 	  icon=icontheme.iconPath("txt.svg", size, KIcon::MatchBest);
-	  if(engine.load(size, size, icon.path))
-              m_previewDocument->setPixmap(*engine.image());
+      if(!icon.path.isEmpty()){
+          if(engine.load(size, size, icon.path))
+                  m_previewDocument->setPixmap(*engine.image());
+          }
           else {
               icon=icontheme.iconPath("txt.svgz", size, KIcon::MatchBest);
               if(engine.load(size, size, icon.path))
