diff -ur kdebase-3.4.92.orig/kcontrol/konq/desktopbehavior_impl.cpp kdebase-3.4.92/kcontrol/konq/desktopbehavior_impl.cpp
--- kdebase-3.4.92.orig/kcontrol/konq/desktopbehavior_impl.cpp	2005-10-10 18:03:47.000000000 +0300
+++ kdebase-3.4.92/kcontrol/konq/desktopbehavior_impl.cpp	2005-10-24 02:18:10.000000000 +0300
@@ -301,7 +301,7 @@
     g_pConfig->setGroup( "General" );
     vrootBox->setChecked( g_pConfig->readBoolEntry( "SetVRoot", false ) );
     iconsEnabledBox->setChecked( g_pConfig->readBoolEntry( "Enabled", true ) );
-    autoLineupIconsBox->setChecked( g_pConfig->readBoolEntry( "AutoLineUpIcons", false ) );
+    autoLineupIconsBox->setChecked( g_pConfig->readBoolEntry( "AutoLineUpIcons", true ) );
 
     //
     g_pConfig->setGroup( "Mouse Buttons" );
diff -ur kdebase-3.4.92.orig/kdesktop/kdesktop.kcfg kdebase-3.4.92/kdesktop/kdesktop.kcfg
--- kdebase-3.4.92.orig/kdesktop/kdesktop.kcfg	2005-10-10 18:04:11.000000000 +0300
+++ kdebase-3.4.92/kdesktop/kdesktop.kcfg	2005-10-24 02:18:33.000000000 +0300
@@ -51,7 +51,7 @@
       <!--     if (!config->readBoolEntry("CopyDesktopLinks", true)) -->
     </entry>
     <entry key="AutoLineUpIcons" type="Bool">
-      <default>false</default>
+      <default>true</default>
       <label>Automatically line up icons</label>
       <whatsthis>Check this option if you want to see your icons automatically aligned to the grid when you move them.</whatsthis>
     </entry>
