diff -Nur kdebase-3.4.92.orig/kcontrol/ebrowsing/plugins/ikws/searchproviders/uluzilla.desktop kdebase-3.4.92/kcontrol/ebrowsing/plugins/ikws/searchproviders/uluzilla.desktop
--- kdebase-3.4.92.orig/kcontrol/ebrowsing/plugins/ikws/searchproviders/uluzilla.desktop	1970-01-01 02:00:00.000000000 +0200
+++ kdebase-3.4.92/kcontrol/ebrowsing/plugins/ikws/searchproviders/uluzilla.desktop	2005-10-24 02:29:07.000000000 +0300
@@ -0,0 +1,8 @@
+[Desktop Entry]
+Encoding=UTF-8
+Keys=ulubug,uluzilla
+Name=Uludağ Bug Database Bug Number Search
+Name[tr]=Uludağ Hata Veritabanı Hata Numarasıyla Arama
+Query=http://bugs.uludag.org.tr/show_bug.cgi?id=\\{@}
+ServiceTypes=SearchProvider
+Type=Service
