Index: kcontrol/arts/arts.cpp
===================================================================
--- kcontrol/arts/arts.cpp	(revision 583458)
+++ kcontrol/arts/arts.cpp	(working copy)
@@ -69,7 +69,7 @@
 	config->setGroup("Arts");
 	bool startServer = config->readBoolEntry("StartServer",true);
 	bool startRealtime = config->readBoolEntry("StartRealtime",true);
-	QString args = config->readEntry("Arguments","-F 10 -S 4096 -s 60 -m artsmessage -c drkonqi -l 3 -f");
+	QString args = config->readEntry("Arguments","-F 10 -S 4096 -s 3 -m artsmessage -c drkonqi -l 3 -f");
 
 	delete config;
 
@@ -261,7 +261,7 @@
 	networkTransparent->setChecked(config->readBoolEntry("NetworkTransparent",false));
 	fullDuplex->setChecked(config->readBoolEntry("FullDuplex",false));
 	autoSuspend->setChecked(config->readBoolEntry("AutoSuspend",true));
-	suspendTime->setValue(config->readNumEntry("SuspendTime",60));
+	suspendTime->setValue(config->readNumEntry("SuspendTime",3));
 	deviceName->setText(config->readEntry("DeviceName",QString::null));
 	customDevice->setChecked(!deviceName->text().isEmpty());
 	hardware->addOptions->setText(config->readEntry("AddOptions",QString::null));
