Index: kcontrol/style/kcmstyle.cpp
===================================================================
--- kcontrol/style/kcmstyle.cpp	(revision 583461)
+++ kcontrol/style/kcmstyle.cpp	(working copy)
@@ -874,9 +874,9 @@
 	// Load effects.
 	config.setGroup("KDE");
 
-	cbEnableEffects->setChecked( config.readBoolEntry( "EffectsEnabled", false) );
+	cbEnableEffects->setChecked( config.readBoolEntry( "EffectsEnabled", true) );
 
-	if ( config.readBoolEntry( "EffectAnimateCombo", false) )
+	if ( config.readBoolEntry( "EffectAnimateCombo", true) )
 		comboComboEffect->setCurrentItem( 1 );
 	else
 		comboComboEffect->setCurrentItem( 0 );
@@ -929,7 +929,7 @@
 	slOpacity->setValue( (int)(100 * settings.readDoubleEntry("/KStyle/Settings/MenuOpacity", 0.90)) );
 
 	// Menu Drop-shadows...
-	cbMenuShadow->setChecked( settings.readBoolEntry("/KStyle/Settings/MenuDropShadow", false) );
+	cbMenuShadow->setChecked( settings.readBoolEntry("/KStyle/Settings/MenuDropShadow", true) );
 
 	if (cbEnableEffects->isChecked()) {
 		containerFrame->setEnabled( true );
@@ -1000,7 +1000,7 @@
 		comboToolbarIcons->setCurrentItem(0);
 
 	config.setGroup("KDE");
-	cbIconsOnButtons->setChecked(config.readBoolEntry("ShowIconsOnPushButtons", false));
+	cbIconsOnButtons->setChecked(config.readBoolEntry("ShowIconsOnPushButtons", true));
 	cbEnableTooltips->setChecked(!config.readBoolEntry("EffectNoTooltip", false));
 	cbTearOffHandles->setChecked(config.readBoolEntry("InsertTearOffHandle", false));
 
