--- kdebase-3.5.0.orig/kdm/kfrontend/kgreeter.cpp	2005-12-06 13:55:33.000000000 +0200
+++ kdebase-3.5.0/kdm/kfrontend/kgreeter.cpp	2005-12-19 01:24:40.000000000 +0200
@@ -120,8 +120,8 @@
 		userView = new UserListView( this );
 		connect( userView, SIGNAL(clicked( QListViewItem * )),
 		         SLOT(slotUserClicked( QListViewItem * )) );
-		connect( userView, SIGNAL(doubleClicked( QListViewItem * )),
-		         SLOT(accept()) );
+		//		connect( userView, SIGNAL(doubleClicked( QListViewItem * )),
+		//		         SLOT(accept()) );
 	}
 	if (_userCompletion)
 		userList = new QStringList;
