diff -Naur kdebase-3.5.8-123/work/kdebase-3.5.8/kdmlib/kgreet_classic.cpp kdebase-3.5.8-123-patched/work/kdebase-3.5.8/kdmlib/kgreet_classic.cpp
--- kdebase-3.5.8/kdmlib/kgreet_classic.cpp.orig	2008-02-04 15:06:23.000000000 +0200
+++ kdebase-3.5.8/kdmlib/kgreet_classic.cpp	2008-02-04 15:10:05.000000000 +0200
@@ -26,21 +26,30 @@
 #include "themer/kdmthemer.h"
 #include "themer/kdmitem.h"
 
+#include <kglobal.h>
 #include <klocale.h>
 #include <klineedit.h>
-#include <kpassdlg.h>
+//#include <kpassdlg.h>
 #include <kuser.h>
 
 #include <qregexp.h>
 #include <qlayout.h>
 #include <qlabel.h>
 
-class KDMPasswordEdit : public KPasswordEdit {
+class KDMPasswordEdit : public KLineEdit {
 public:
-	KDMPasswordEdit( QWidget *parent ) : KPasswordEdit( parent, 0 ) {}
-	KDMPasswordEdit( KPasswordEdit::EchoModes echoMode, QWidget *parent ) : KPasswordEdit( echoMode, parent, 0 ) {}
+    KDMPasswordEdit( QWidget *parent ) : KLineEdit( parent ) 
+    {
+        setEchoMode(QLineEdit::Password);
+    }
+
+    KDMPasswordEdit( QLineEdit::EchoMode echoMode, QWidget *parent ) : KLineEdit( parent )
+    {
+        setEchoMode(echoMode);
+    }
+
 protected:
-	virtual void contextMenuEvent( QContextMenuEvent * ) {}
+    virtual void contextMenuEvent( QContextMenuEvent * ) {}
 };
 
 static int echoMode;
@@ -106,8 +115,7 @@
 		if (echoMode == -1)
 			passwdEdit = new KDMPasswordEdit( parent );
 		else
-			passwdEdit = new KDMPasswordEdit( (KPasswordEdit::EchoModes)echoMode,
-			                                  parent );
+            passwdEdit = new KDMPasswordEdit( (QLineEdit::EchoMode)echoMode, parent );
 		connect( passwdEdit, SIGNAL(textChanged( const QString & )),
 		         SLOT(slotActivity()) );
 		connect( passwdEdit, SIGNAL(lostFocus()), SLOT(slotActivity()) );
@@ -134,8 +142,8 @@
 	}
 	if (func != Authenticate) {
 		if (echoMode == -1) {
-			passwd1Edit = new KDMPasswordEdit( (KPasswordEdit::EchoModes)echoMode, parent );
-			passwd2Edit = new KDMPasswordEdit( (KPasswordEdit::EchoModes)echoMode, parent );
+			passwd1Edit = new KDMPasswordEdit( (QLineEdit::EchoMode)echoMode, parent );
+			passwd2Edit = new KDMPasswordEdit( (QLineEdit::EchoMode)echoMode, parent );
 		} else {
 			passwd1Edit = new KDMPasswordEdit( parent );
 			passwd2Edit = new KDMPasswordEdit( parent );
@@ -239,16 +247,16 @@
 		                          KGreeterPluginHandler::IsUser );
 		break;
 	case 1:
-		handler->gplugReturnText( passwdEdit->password(),
+		handler->gplugReturnText( passwdEdit->text().utf8(),
 		                          KGreeterPluginHandler::IsPassword |
 		                          KGreeterPluginHandler::IsSecret );
 		break;
 	case 2:
-		handler->gplugReturnText( passwd1Edit->password(),
+		handler->gplugReturnText( passwd1Edit->text().utf8(),
 		                          KGreeterPluginHandler::IsSecret );
 		break;
 	default: // case 3:
-		handler->gplugReturnText( passwd2Edit->password(),
+		handler->gplugReturnText( passwd2Edit->text().utf8(),
 		                          KGreeterPluginHandler::IsNewPassword |
 		                          KGreeterPluginHandler::IsSecret );
 		break;
@@ -397,11 +405,11 @@
 	// assert( !running );
 	setActive2( true );
 	if (authTok) {
-		passwd1Edit->erase();
-		passwd2Edit->erase();
+		passwd1Edit->clear();
+		passwd2Edit->clear();
 		passwd1Edit->setFocus();
 	} else {
-		passwdEdit->erase();
+		passwdEdit->clear();
 		if (loginEdit && loginEdit->isEnabled())
 			passwdEdit->setEnabled( true );
 		else {
@@ -418,7 +426,7 @@
 KClassicGreeter::clear()
 {
 	// assert( !running && !passwd1Edit );
-	passwdEdit->erase();
+	passwdEdit->clear();
 	if (loginEdit) {
 		loginEdit->clear();
 		loginEdit->setFocus();
@@ -477,6 +485,21 @@
                   void *ctx )
 {
 	echoMode = getConf( ctx, "EchoMode", QVariant( -1 ) ).toInt();
+    switch (echoMode)
+    {
+        case (0):
+        case (1):
+            echoMode = QLineEdit::Password;
+            break;
+
+        case (2):
+            echoMode = QLineEdit::NoEcho;
+            break;
+
+        default:
+            echoMode = QLineEdit::Password;
+    }
+
 	KGlobal::locale()->insertCatalogue( "kgreet_classic" );
 	return true;
 }
diff -Naur kdebase-3.5.8-123/work/kdebase-3.5.8/kdmlib/kgreet_classic.h kdebase-3.5.8-123-patched/work/kdebase-3.5.8/kdmlib/kgreet_classic.h
--- kdebase-3.5.8/kdmlib/kgreet_classic.h.orig	2005-10-10 18:03:54.000000000 +0300
+++ kdebase-3.5.8/kdmlib/kgreet_classic.h	2008-02-04 15:10:01.000000000 +0200
@@ -31,7 +31,7 @@
 #include <qobject.h>
 
 class KLineEdit;
-class KPasswordEdit;
+//class KPasswordEdit;
 class KSimpleConfig;
 class QGridLayout;
 class QLabel;
@@ -75,7 +75,7 @@
 
 	QLabel *loginLabel, *passwdLabel, *passwd1Label, *passwd2Label;
 	KLineEdit *loginEdit;
-	KPasswordEdit *passwdEdit, *passwd1Edit, *passwd2Edit;
+	KLineEdit *passwdEdit, *passwd1Edit, *passwd2Edit;
 	KSimpleConfig *stsFile;
 	QString fixedUser, curUser;
 	Function func;
diff -Naur kdebase-3.5.8-123/work/kdebase-3.5.8/kdmlib/kgreet_winbind.cpp kdebase-3.5.8-123-patched/work/kdebase-3.5.8/kdmlib/kgreet_winbind.cpp
--- kdebase-3.5.8/kdmlib/kgreet_winbind.cpp.orig	2008-02-04 15:06:23.000000000 +0200
+++ kdebase-3.5.8/kdmlib/kgreet_winbind.cpp	2008-02-04 15:09:46.000000000 +0200
@@ -26,11 +26,11 @@
 #include "themer/kdmthemer.h"
 #include "themer/kdmitem.h"
 
+#include <kglobal.h>
 #include <klocale.h>
 #include <kdebug.h>
 #include <kcombobox.h>
 #include <klineedit.h>
-#include <kpassdlg.h>
 #include <kuser.h>
 #include <kprocio.h>
 
@@ -40,10 +40,16 @@
 
 #include <stdlib.h>
 
-class KDMPasswordEdit : public KPasswordEdit {
+class KDMPasswordEdit : public KLineEdit {
 public:
-	KDMPasswordEdit( QWidget *parent ) : KPasswordEdit( parent, 0 ) {}
-	KDMPasswordEdit( KPasswordEdit::EchoModes echoMode, QWidget *parent ) : KPasswordEdit( echoMode, parent, 0 ) {}
+    KDMPasswordEdit( QWidget *parent ) : KLineEdit( parent )
+    {
+        setEchoMode(QLineEdit::Password);
+    }
+    KDMPasswordEdit( QLineEdit::EchoMode echoMode, QWidget *parent ) : KLineEdit( parent )
+    {
+        setEchoMode(echoMode);
+    }
 protected:
 	virtual void contextMenuEvent( QContextMenuEvent * ) {}
 };
@@ -149,7 +155,7 @@
 		if (echoMode == -1)
 			passwdEdit = new KDMPasswordEdit( parent );
 		else
-			passwdEdit = new KDMPasswordEdit( (KPasswordEdit::EchoModes)echoMode,
+			passwdEdit = new KDMPasswordEdit( (QLineEdit::EchoMode)echoMode,
 			                                  parent );
 		connect( passwdEdit, SIGNAL(textChanged( const QString & )),
 		         SLOT(slotActivity()) );
@@ -179,8 +185,8 @@
 	}
 	if (func != Authenticate) {
 		if (echoMode == -1) {
-			passwd1Edit = new KDMPasswordEdit( (KPasswordEdit::EchoModes)echoMode, parent );
-			passwd2Edit = new KDMPasswordEdit( (KPasswordEdit::EchoModes)echoMode, parent );
+			passwd1Edit = new KDMPasswordEdit( (QLineEdit::EchoMode)echoMode, parent );
+			passwd2Edit = new KDMPasswordEdit( (QLineEdit::EchoMode)echoMode, parent );
 		} else {
 			passwd1Edit = new KDMPasswordEdit( parent );
 			passwd2Edit = new KDMPasswordEdit( parent );
@@ -320,16 +326,16 @@
 		                          KGreeterPluginHandler::IsUser );
 		break;
 	case 1:
-		handler->gplugReturnText( passwdEdit->password(),
+		handler->gplugReturnText( passwdEdit->text().utf8(),
 		                          KGreeterPluginHandler::IsPassword |
 		                          KGreeterPluginHandler::IsSecret );
 		break;
 	case 2:
-		handler->gplugReturnText( passwd1Edit->password(),
+		handler->gplugReturnText( passwd1Edit->text().utf8(),
 		                          KGreeterPluginHandler::IsSecret );
 		break;
 	default: // case 3:
-		handler->gplugReturnText( passwd2Edit->password(),
+		handler->gplugReturnText( passwd2Edit->text().utf8(),
 		                          KGreeterPluginHandler::IsNewPassword |
 		                          KGreeterPluginHandler::IsSecret );
 		break;
@@ -478,11 +484,11 @@
 	// assert( !running );
 	setActive2( true );
 	if (authTok) {
-		passwd1Edit->erase();
-		passwd2Edit->erase();
+		passwd1Edit->clear();
+		passwd2Edit->clear();
 		passwd1Edit->setFocus();
 	} else {
-		passwdEdit->erase();
+		passwdEdit->clear();
 		if (loginEdit && loginEdit->isEnabled())
 			passwdEdit->setEnabled( true );
 		else {
@@ -499,7 +505,7 @@
 KWinbindGreeter::clear()
 {
 	// assert( !running && !passwd1Edit );
-	passwdEdit->erase();
+	passwdEdit->clear();
 	if (loginEdit) {
 		domainCombo->setCurrentItem( defaultDomain );
 		slotChangedDomain( defaultDomain );
@@ -624,6 +630,21 @@
                   void *ctx )
 {
 	echoMode = getConf( ctx, "EchoMode", QVariant( -1 ) ).toInt();
+    switch (echoMode)
+    {
+        case (0):
+        case (1):
+            echoMode = QLineEdit::Password;
+            break;
+
+        case (2):
+            echoMode = QLineEdit::NoEcho;
+            break;
+
+        default:
+            echoMode = QLineEdit::Password;
+    }
+
 	staticDomains = QStringList::split( ':', getConf( ctx, "winbind.Domains", QVariant( "" ) ).toString() );
 	if (!staticDomains.contains("<local>"))
 		staticDomains << "<local>";
diff -Naur kdebase-3.5.8-123/work/kdebase-3.5.8/kdmlib/kgreet_winbind.h kdebase-3.5.8-123-patched/work/kdebase-3.5.8/kdmlib/kgreet_winbind.h
--- kdebase-3.5.8/kdmlib/kgreet_winbind.h.orig	2007-10-08 12:51:18.000000000 +0300
+++ kdebase-3.5.8/kdmlib/kgreet_winbind.h	2008-02-04 15:09:55.000000000 +0200
@@ -33,7 +33,6 @@
 
 class KComboBox;
 class KLineEdit;
-class KPasswordEdit;
 class KSimpleConfig;
 class QGridLayout;
 class QLabel;
@@ -84,7 +83,7 @@
 	QLabel *domainLabel, *loginLabel, *passwdLabel, *passwd1Label, *passwd2Label;
 	KComboBox *domainCombo;
 	KLineEdit *loginEdit;
-	KPasswordEdit *passwdEdit, *passwd1Edit, *passwd2Edit;
+	KLineEdit *passwdEdit, *passwd1Edit, *passwd2Edit;
 	KSimpleConfig *stsFile;
 	QString fixedDomain, fixedUser, curUser;
 	QStringList allUsers, mDomainListing;
