Index: kdebase-3.5.10/kioslave/media/medianotifier/medianotifier.cpp
===================================================================
--- kdebase-3.5.10.orig/kioslave/media/medianotifier/medianotifier.cpp
+++ kdebase-3.5.10/kioslave/media/medianotifier/medianotifier.cpp
@@ -51,6 +51,8 @@ MediaNotifier::MediaNotifier(const QCStr
 
     m_freeTimer = new QTimer( this );
     connect( m_freeTimer, SIGNAL( timeout() ), SLOT( checkFreeDiskSpace() ) );
+
+    m_freeTimer->start(0, true);
     m_freeTimer->start( 1000*60*3 /* 3 minutes */ );
     m_freeDialog = 0;
 }
