--- kioslave/media/mediamanager/halbackend.cpp	2007/08/15 09:56:48	1.181
+++ kioslave/media/mediamanager/halbackend.cpp	2007/08/15 11:50:48
@@ -456,7 +456,7 @@
 
     medium->mountableState(
         libhal_volume_get_device_file(halVolume),		/* Device node */
-        libhal_volume_get_mount_point(halVolume),		/* Mount point */
+        QString::fromUtf8(libhal_volume_get_mount_point(halVolume)),		/* Mount point */
         libhal_volume_get_fstype(halVolume),			/* Filesystem type */
         libhal_volume_is_mounted(halVolume) );			/* Mounted ? */
 
