diff -ur kdebase-3.5.5.orig/kioslave/media/services/media_safelyremove.desktop kdebase-3.5.5/kioslave/media/services/media_safelyremove.desktop
--- kdebase-3.5.5.orig/kioslave/media/services/media_safelyremove.desktop	2006-10-01 20:31:53.000000000 +0300
+++ kdebase-3.5.5/kioslave/media/services/media_safelyremove.desktop	2006-11-04 14:21:52.000000000 +0200
@@ -60,5 +60,5 @@
 Name[vi]=Gỡ ra An toàn
 Name[zh_CN]=安全删除
 Name[zh_TW]=安全的移除
-Exec=kio_media_mounthelper -s %u
+Exec=kio_media_mounthelper -u %u
 
