Index: kicker/kicker/ui/service_mnu.cpp
===================================================================
--- kicker/kicker/ui/service_mnu.cpp	(revision 665122)
+++ kicker/kicker/ui/service_mnu.cpp	(working copy)
@@ -311,7 +311,7 @@
             {
                 if (comment.find(serviceName, 0, true) == -1)
                 {
-                    serviceName = i18n("Entries in K-menu: %1 app name, %2 description", "%1 - %2").arg(serviceName, comment);
+                    serviceName = i18n("Entries in K-menu: %1 app name, %2 description", "%1 (%2)").arg(serviceName, comment);
                 }
                 else
                 {
