Index: kdenlive-0.7.4/src/kdenlive.desktop
===================================================================
--- kdenlive-0.7.4.orig/src/kdenlive.desktop
+++ kdenlive-0.7.4/src/kdenlive.desktop
@@ -4,9 +4,11 @@ Name=Kdenlive
 GenericName=Video Editor
 GenericName[de]=Video-Editor
 GenericName[it]=Editor Video
+GenericName[tr]=Video Düzenleyici
 Comment=Nonlinear video editor for KDE
 Comment[de]=Nicht-linearer Video-Editor für KDE
 Comment[it]=Editor Video non lineare per KDE
+Comment[tr]=KDE için Nonlineer video düzenleyici
 Type=Application
 Exec=kdenlive %i -caption "%c" %u
 Icon=kdenlive
