diff -Nur openmotif-2.3.2-old/clients/mwm/Makefile.am openmotif-2.3.2/clients/mwm/Makefile.am
--- openmotif-2.3.2-old/clients/mwm/Makefile.am	2009-05-02 05:20:56.000000000 +0300
+++ openmotif-2.3.2/clients/mwm/Makefile.am	2009-05-02 05:21:19.000000000 +0300
@@ -14,7 +14,7 @@
 INCLUDES = -DLARGECURSORS -DR2_COMPAT -DUNMAP_ON_RESTART \
            -DCDE_INSTALLATION_TOP=\"@CDE_INSTALLATION_TOP@\" \
            -DCDE_CONFIGURATION_TOP=\"@CDE_CONFIGURATION_TOP@\" \
-           -DBITMAPDIR=\"@INCDIR@/bitmaps\" \
+           -DBITMAPDIR=\"@INCDIR@/bitmaps\"  -DLIBDIR=\"$(rcdir)\" \
            -I$(top_builddir)/lib \
            -I$(srcdir)/../../lib ${X_CFLAGS}
 
