Index: mozilla/toolkit/library/Makefile.in
===================================================================
--- mozilla.orig/toolkit/library/Makefile.in
+++ mozilla/toolkit/library/Makefile.in
@@ -227,7 +227,7 @@ EXTRA_DSO_LDOPTS += $(LIBCONIC_LIBS)
 endif
 
 ifdef MOZ_ENABLE_DBUS
-EXTRA_DSO_LDOPTS += $(MOZ_DBUS_LIBS)
+EXTRA_DSO_LDOPTS += $(MOZ_DBUS_LIBS) -ldbus-glib-1
 endif
 
 ifdef MOZ_ENABLE_GCONF
