Index: tools/tools.pro
===================================================================
--- tools/tools.pro
+++ tools/tools.pro	2010-01-24 15:41:03.000000000 +0100
@@ -14,6 +14,7 @@
          SUBDIRS += designer
      }
      SUBDIRS     += linguist
+     SUBDIRS     += qdoc3
      symbian: SUBDIRS = designer
      wince*: SUBDIRS = qtestlib designer
      unix:!mac:!embedded:contains(QT_CONFIG, qt3support):SUBDIRS += qtconfig
Index: tools/qdoc3/qdoc3.pro
===================================================================
--- tools/qdoc3/qdoc3.pro
+++ tools/qdoc3/qdoc3.pro	2010-01-24 15:41:27.000000000 +0100
@@ -107,3 +107,7 @@
 
 target.path = $$[QT_INSTALL_BINS]
 INSTALLS += target
+
+TARGET= qdoc3
+target.path = $$[QT_INSTALL_BINS]
+INSTALLS += target
