diff -Nur baselayout-old/etc/skel/.bashrc baselayout/etc/skel/.bashrc
--- baselayout-old/etc/skel/.bashrc	2008-04-30 16:03:45.375762283 +0300
+++ baselayout/etc/skel/.bashrc	2008-04-30 16:04:12.934765662 +0300
@@ -3,8 +3,6 @@
 # including some apparently interactive shells such as scp and rcp
 # that can't tolerate any output.
 
-source /etc/profile
-
 # Test for an interactive shell.  There is no need to set anything
 # past this point for scp and rcp, and it's important to refrain from
 # outputting anything in those cases.
