diff -ur baselayout.orig/etc/skel/.gtkrc-2.0-kde baselayout/etc/skel/.gtkrc-2.0-kde
--- baselayout.orig/etc/skel/.gtkrc-2.0-kde	2007-03-09 03:26:48.000000000 +0200
+++ baselayout/etc/skel/.gtkrc-2.0-kde	2008-05-06 14:43:16.000000000 +0300
@@ -2,6 +2,7 @@
 # You can edit it in the KDE control center, under "GTK Styles and Fonts"
 
 include "/usr/share/themes/Qt/gtk-2.0/gtkrc"
+include "/etc/gtk-2.0/gtkrc"
 
 style "user-font"
 {
