--- etc/mailcap.orig	2009-05-02 04:34:13.000000000 +0300
+++ etc/mailcap	2009-05-02 04:34:27.000000000 +0300
@@ -55,8 +55,8 @@
 application/vnd.ms-excel; oocalc '%s'; -n "$DISPLAY"
 
 # PDF and PS & DVI
-application/pdf; kpdf '%s'; -n "$DISPLAY"
-application/x-pdf; kpdf '%s'; -n "$DISPLAY"
+application/pdf; okular '%s'; -n "$DISPLAY"
+application/x-pdf; okular '%s'; -n "$DISPLAY"
 application/postscript; ghostview '%s'; -n "$DISPLAY"
 application/x-dvi; ghostview '%s'; -n "$DISPLAY"
 
