diff -Nur ifuse-0.9.4-old/src/Makefile.am ifuse-0.9.4/src/Makefile.am
--- ifuse-0.9.4-old/src/Makefile.am	2009-11-18 20:30:59.652237175 +0200
+++ ifuse-0.9.4/src/Makefile.am	2009-11-18 20:44:27.296963698 +0200
@@ -1,5 +1,5 @@
-AM_CFLAGS = $(libglib2_CFLAGS) $(libfuse_CFLAGS) $(libiphone_CFLAGS) -g
-AM_LDFLAGS =  $(libglib2_LIBS) $(libfuse_LIBS) $(libiphone_LIBS)
+INCLUDES = $(libiphone_CFLAGS) $(libfuse_CFLAGS) $(libglib2_CFLAGS)
+ifuse_LDADD =  $(libiphone_LIBS) $(libfuse_LIBS) $(libglib2_LIBS)
 
 bin_PROGRAMS = ifuse
 
