Index: kjsembed/bindwizard/doxygen2imp_cpp.xsl
===================================================================
RCS file: /home/kde/kdebindings/kjsembed/bindwizard/doxygen2imp_cpp.xsl,v
retrieving revision 1.9
diff -u -3 -p -r1.9 doxygen2imp_cpp.xsl
--- kjsembed/bindwizard/doxygen2imp_cpp.xsl	20 Jun 2004 22:49:29 -0000	1.9
+++ kjsembed/bindwizard/doxygen2imp_cpp.xsl	23 Aug 2004 09:42:42 -0000
@@ -284,6 +284,7 @@ KJS::Object <xsl:value-of select="$clazz
 
     <xsl:if test="($clazz = 'QCanvasItem') or ($clazz = 'QCanvasPolygonalItem')">
 #endif // This constructor has been disabled by the XSL template
+    return KJS::Object();
     </xsl:if>
 }
     </xsl:when>
