--- setup.py.orig	2009-07-03 03:20:30.375323310 +0300
+++ setup.py	2009-07-03 03:21:19.974717198 +0300
@@ -39,7 +39,7 @@
 <http://bitbucket.org/birkenfeld/sphinx/get/tip.gz#egg=Sphinx-dev>`_.
 '''
 
-requires = ['Pygments>=0.8', 'Jinja2>=2.1', 'docutils>=0.4']
+requires = ['Pygments>=0.8', 'Jinja2>=2.1']
 
 if sys.version_info < (2, 4):
     print 'ERROR: Sphinx requires at least Python 2.4 to run.'
