diff -up numactl-2.0.3/libnuma.c.orig numactl-2.0.3/libnuma.c
--- numactl-2.0.3/libnuma.c.orig	2009-08-10 07:02:12.000000000 -0400
+++ numactl-2.0.3/libnuma.c	2009-08-10 07:02:39.000000000 -0400
@@ -464,7 +464,7 @@ set_thread_constraints(void)
 			nodes_allowed_list = malloc(strlen(buffer)-18);
 			strncpy(nodes_allowed_list, buffer + 19,
 				strlen(buffer) - 19);
-			nodes_allowed_list[strlen(nodes_allowed_list)-1] = '\0';
+			nodes_allowed_list[strlen(buffer)-19] = '\0';
 		}
 	}
 	fclose(f);
