Index: corporate2/data.qrc
===================================================================
--- corporate2/data.qrc	(revision 33882)
+++ corporate2/data.qrc	(revision 33896)
@@ -1,5 +1,9 @@
 <RCC>
   <qresource prefix="/">
+    <file>gui/pics/bootloader.png</file>
+    <file>gui/pics/bootloader-big.png</file>
+    <file>gui/pics/summary.png</file>
+    <file>gui/pics/summary-big.png</file>
     <file>gui/pics/corporate.png</file>
     <file>gui/pics/corporate2.png</file>
     <file>gui/pics/kafa.png</file>
