Index: setup.py
===================================================================
--- setup.py	(revision 33706)
+++ setup.py	(revision 33810)
@@ -21,7 +21,10 @@
         build.run(self)
 
         self.mkpath(self.build_base)
-        self.spawn(["rcc", "-binary", "data.qrc", "-o", "%s/data.rcc" % self.build_base])
+        try:
+            self.spawn(["rcc", "-binary", "data.qrc", "-o", "%s/data.rcc" % self.build_base])
+        except:
+            self.spawn(["rcc-qt4", "-binary", "data.qrc", "-o", "%s/data.rcc" % self.build_base])
 
 
 class Install(install):
