Index: metis-4.0/Lib/Makefile
===================================================================
--- metis-4.0.orig/Lib/Makefile
+++ metis-4.0/Lib/Makefile
@@ -21,7 +21,7 @@ OBJS = coarsen.o fm.o initpart.o match.o
 	$(CC) $(CFLAGS) -c $*.c
 
 ../libmetis.so.0.0: $(OBJS)
-	$(CC) $(CFLAGS) -fPIC -shared -Wl,-soname=libmetis.so.0 -o $@ $(OBJS)
+	$(CC) $(CFLAGS) -fPIC -shared -Wl,-soname=libmetis.so.0 -lm -o $@ $(OBJS)
 	ln -s libmetis.so.0.0 ../libmetis.so.0
 	ln -s libmetis.so.0.0 ../libmetis.so
 
