Index: pysvn-1.7.2/Tests/pysvn_test_common.mak
===================================================================
--- pysvn-1.7.2.orig/Tests/pysvn_test_common.mak
+++ pysvn-1.7.2/Tests/pysvn_test_common.mak
@@ -4,7 +4,7 @@
 #	include this mak file after defining the variables it needs
 #
 KNOWN_GOOD_VERSION=py$(PY_VERSION_MAJ)-svn$(SVN_VERSION_MAJ_MIN)
-all: test-01.unix.new.log.clean test-04.unix.new.log test-05.unix.new.log test-06.unix.new.log test-07.unix.new.log
+all: test-01.unix.new.log.clean test-04.unix.new.log test-05.unix.new.log test-07.unix.new.log
 
 # the extra tests rely on having a repos over HTTP.
 extratests: test-03.unix.new.log
@@ -16,7 +16,7 @@ help:
 	@echo "make diff-01       - run wb-diff on the outcome of test 01"
 	@echo "make new-01        - use the new log as the known-good log of test 01"
 
-clean: clean-01 clean-03 clean-04 clean-05 clean-06 clean-07
+clean: clean-01 clean-03 clean-04 clean-05 clean-07
 
 #
 #	Helpers while developing and verifying tests
@@ -87,22 +87,6 @@ diff-05: test-05.unix.new.log
 new-05: test-05.unix.new.log
 	cp  test-05.unix.new.log test-05.unix.known_good-$(KNOWN_GOOD_VERSION).log
 
-test-06.unix.new.log: test-06.sh test-06.unix.known_good-$(KNOWN_GOOD_VERSION).log
-	-rm -rf testroot-06
-	if PYTHONPATH=../Source $(PYTHON) svn_min_version.py 1 3 0; then PATH=$(SVN_BIN_DIR):$(PATH) PYTHON=$(PYTHON) ./test-06.sh >test-06.unix.new.log 2>&1; fi
-	if PYTHONPATH=../Source $(PYTHON) svn_min_version.py 1 3 0; then $(PYTHON) benchmark_diff.py test-06.unix.known_good-$(KNOWN_GOOD_VERSION).log test-06.unix.new.log; fi
-
-clean-06:
-	-rm -f test-06.unix.new.log
-	-rm -f test-06.unix.new.log.clean
-	-rm -rf testroot-06
-
-diff-06: test-06.unix.new.log
-	wb-diff test-06.unix.known_good-$(KNOWN_GOOD_VERSION).log.clean test-06.unix.new.log.clean
-
-new-06: test-06.unix.new.log
-	cp  test-06.unix.new.log test-06.unix.known_good-$(KNOWN_GOOD_VERSION).log
-
 test-07.unix.new.log: test-07.sh test-07.unix.known_good-$(KNOWN_GOOD_VERSION).log
 	-rm -rf testroot-07
 	if PYTHONPATH=../Source $(PYTHON) svn_min_version.py 1 5 0; then PATH=$(SVN_BIN_DIR):$(PATH) PYTHON=$(PYTHON) ./test-07.sh >test-07.unix.new.log 2>&1; fi
