--- ../l110n/tr/editor/ui/chrome/composer/editorOverlay.dtd.orig	2008-12-25 14:54:36.000000000 +0200
+++ ../l10n/tr/editor/ui/chrome/composer/editorOverlay.dtd	2008-12-25 14:52:49.000000000 +0200
@@ -146,7 +146,7 @@
 <!ENTITY findToolbarCmd.label "Bul">
 <!ENTITY findToolbarCmd.tooltip "Sayfada metin bul">
 <!ENTITY fontCourier.label "Courier">
-<!ENTITY fontFixedWidth.label "Sabit en">
+<!ENTITY fontFixedWidth.label "Sabit genişlik">
 <!ENTITY fontHelveticaFont.label "Helvetica, Arial">
 <!ENTITY fontStyleMenu.label "Metin Stili">
 <!ENTITY fontTimes.label "Times">
