diff -Nur smplayer-0.6.7_20090703-old/src/mplayerprocess.cpp smplayer-0.6.7_20090703/src/mplayerprocess.cpp
--- smplayer-0.6.7_20090703-old/src/mplayerprocess.cpp	2009-07-17 13:15:57.058701342 +0300
+++ smplayer-0.6.7_20090703/src/mplayerprocess.cpp	2009-07-17 13:19:15.392504588 +0300
@@ -102,18 +102,18 @@
 static QRegExp rx_ao("^AO: \\[(.*)\\]");
 static QRegExp rx_paused("^ID_PAUSED");
 #if !CHECK_VIDEO_CODEC_FOR_NO_VIDEO
-static QRegExp rx_novideo("^Video: no video");
+static QRegExp rx_novideo("^Video: video yok");
 #endif
-static QRegExp rx_cache("^Cache fill:.*");
-static QRegExp rx_create_index("^Generating Index:.*");
-static QRegExp rx_play("^Starting playback...");
-static QRegExp rx_connecting("^Connecting to .*");
-static QRegExp rx_resolving("^Resolving .*");
+static QRegExp rx_cache("^Önbellek doyumu:.*");
+static QRegExp rx_create_index("^İndeks yenileniyor:.*");
+static QRegExp rx_play("^Çalma başlatılıyor...");
+static QRegExp rx_connecting("^Bağlantı kuruluyor .*");
+static QRegExp rx_resolving("^Adres çözülüyor .*");
 static QRegExp rx_screenshot("^\\*\\*\\* screenshot '(.*)'");
-static QRegExp rx_endoffile("^Exiting... \\(End of file\\)|^ID_EXIT=EOF");
+static QRegExp rx_endoffile("^Çıkılıyor... \\(Dosya sonu\\)|^ID_EXIT=EOF");
 static QRegExp rx_mkvchapters("\\[mkv\\] Chapter (\\d+) from");
-static QRegExp rx_aspect2("^Movie-Aspect is ([0-9,.]+):1");
-static QRegExp rx_fontcache("^\\[ass\\] Updating font cache|^\\[ass\\] Init");
+static QRegExp rx_aspect2("^Video-Görünümü ([0-9,.]+):1");
+static QRegExp rx_fontcache("^\\[ass\\] Yazıtipi tampon belleği güncelleniyor|^\\[ass\\] Init");
 #if DVDNAV_SUPPORT
 static QRegExp rx_dvdnav_switch_title("^DVDNAV, switched to title: (\\d+)");
 static QRegExp rx_dvdnav_length("^ANS_length=(.*)");
