Index: xf86-video-intel-2.12.0/src/i830_driver.c
===================================================================
--- xf86-video-intel-2.12.0.orig/src/i830_driver.c
+++ xf86-video-intel-2.12.0/src/i830_driver.c
@@ -373,9 +373,6 @@ static Bool i830_kernel_mode_enabled(Scr
 		if (xf86LoadKernelModule("i915"))
 			ret = drmCheckModesettingSupported(busIdString);
 	}
-	/* Be nice to the user and load fbcon too */
-	if (!ret)
-		(void)xf86LoadKernelModule("fbcon");
 	free(busIdString);
 	if (ret)
 		return FALSE;
