--- Makefile.orig	2009-01-03 17:04:38.000000000 +0200
+++ Makefile	2009-06-02 00:39:26.000000000 +0300
@@ -50,7 +50,7 @@
 
 AR = ar
 INSTALL = install
-STRIP = strip
+STRIP =
 RM = rm -f
 LN = ln -sf
 ifeq ($(MAKECMDGOALS), debug)
@@ -59,7 +59,7 @@
 OPT = $(POPT)
 endif
 INCLUDE = -Iinclude
-CFLAGS = $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
+CFLAGS += $(OPT) $(WARN) $(INCLUDE) $(CFLAGS.EXTRA)
 
 LIB.SRCS = src/glew.c
 LIB.OBJS = $(LIB.SRCS:.c=.o)
@@ -75,16 +75,10 @@
 VISUALINFO.BIN.OBJS = $(VISUALINFO_BIN.SRCS:.c=.o)
 BIN.LIBS = -Llib $(LDFLAGS.DYNAMIC) -l$(NAME) $(LDFLAGS.EXTRA) $(LDFLAGS.GL)
 
-all debug: lib/$(LIB.SHARED) lib/$(LIB.STATIC) bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN)
-
-lib:
-	mkdir lib
-
-lib/$(LIB.STATIC): $(LIB.OBJS)
-	$(AR) cr $@ $^
+all debug: lib/$(LIB.SHARED) bin/$(GLEWINFO.BIN) bin/$(VISUALINFO.BIN)
 
 lib/$(LIB.SHARED): $(LIB.SOBJS)
-	$(LD) $(LDFLAGS.SO) -o $@ $^ $(LIB.LDFLAGS) $(LIB.LIBS)
+	$(LD) $(LDFLAGS) $(LDFLAGS.SO) $^ $(LIB.LDFLAGS) $(LIB.LIBS) -o $@
 ifeq ($(patsubst mingw%,mingw,$(SYSTEM)), mingw)
 else
 	$(LN) $(LIB.SHARED) lib/$(LIB.SONAME)
@@ -92,10 +86,10 @@
 endif
 
 bin/$(GLEWINFO.BIN): $(GLEWINFO.BIN.SRCS) lib/$(LIB.SHARED)
-	$(CC) $(CFLAGS) -o $@ $(GLEWINFO.BIN.SRCS) $(BIN.LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(GLEWINFO.BIN.SRCS) $(BIN.LIBS) -o $@
 
 bin/$(VISUALINFO.BIN): $(VISUALINFO.BIN.SRCS) lib/$(LIB.SHARED)
-	$(CC) $(CFLAGS) -o $@ $(VISUALINFO.BIN.SRCS) $(BIN.LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(VISUALINFO.BIN.SRCS) $(BIN.LIBS) -o $@
 
 %.o: %.c
 	$(CC) -c $(CFLAGS) -o $@ $<
@@ -113,11 +107,10 @@
 	$(INSTALL) -d -m 0755 $(LIBDIR)
 # runtime
 ifeq ($(patsubst mingw%,mingw,$(SYSTEM)), mingw)
-	$(STRIP) -x lib/$(LIB.SHARED)
+xxxxxxx
 	$(INSTALL) -m 0644 lib/$(LIB.SHARED) $(BINDIR)/
 else
-	$(STRIP) -x lib/$(LIB.SHARED)
-	$(INSTALL) -m 0644 lib/$(LIB.SHARED) $(LIBDIR)/
+	$(INSTALL) -m 0755 lib/$(LIB.SHARED) $(LIBDIR)/
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.SONAME)
 endif
 # development files
@@ -127,8 +120,6 @@
 ifeq ($(patsubst mingw%,mingw,$(SYSTEM)), mingw)
 	$(INSTALL) -m 0644 lib/$(LIB.DEVLNK) $(LIBDIR)/
 else
-	strip -x lib/$(LIB.STATIC)
-	$(INSTALL) -m 0644 lib/$(LIB.STATIC) $(LIBDIR)/
 	$(LN) $(LIB.SHARED) $(LIBDIR)/$(LIB.DEVLNK)
 endif
 # utilities
