diff -pruN akode-2.0.2-orig/akode/lib/auto_sink.cpp akode-2.0.2/akode/lib/auto_sink.cpp
--- akode-2.0.2-orig/akode/lib/auto_sink.cpp	2008-04-04 16:37:46.000000000 +0300
+++ akode-2.0.2/akode/lib/auto_sink.cpp	2008-04-04 16:38:10.000000000 +0300
@@ -22,6 +22,7 @@
 #include "auto_sink.h"
 
 #include <iostream>
+#include <cstdlib>
 
 namespace aKode {
 
diff -pruN akode-2.0.2-orig/akode/lib/magic.cpp akode-2.0.2/akode/lib/magic.cpp
--- akode-2.0.2-orig/akode/lib/magic.cpp	2008-04-04 16:41:02.000000000 +0300
+++ akode-2.0.2/akode/lib/magic.cpp	2008-04-04 16:42:59.000000000 +0300
@@ -22,6 +22,7 @@
 #include "magic.h"
 #include "file.h"
 #include <iostream>
+#include <cstring>
 using std::cerr;
 
 namespace aKode {
diff -pruN akode-2.0.2-orig/akode/lib/player.cpp akode-2.0.2/akode/lib/player.cpp
--- akode-2.0.2-orig/akode/lib/player.cpp	2008-04-04 16:41:02.000000000 +0300
+++ akode-2.0.2/akode/lib/player.cpp	2008-04-04 16:41:58.000000000 +0300
@@ -21,6 +21,7 @@
 #include <pthread.h>
 #include <semaphore.h>
 #include <assert.h>
+#include <cstring>
 
 #include "audioframe.h"
 #include "audiobuffer.h"
diff -pruN akode-2.0.2-orig/akode/akodeplay/akodeplay.cpp akode-2.0.2/akode/akodeplay/akodeplay.cpp
--- akode-2.0.2-orig/akode/akodeplay/akodeplay.cpp	2008-04-04 16:46:51.000000000 +0300
+++ akode-2.0.2/akode/akodeplay/akodeplay.cpp	2008-04-04 16:47:46.000000000 +0300
@@ -19,6 +19,7 @@
 */
 
 #include <iostream>
+#include <cstdlib>
 
 #include "../lib/akodelib.h"
 #include "../lib/player.h"
