Index: ufraw-0.15/ufraw.desktop
===================================================================
--- ufraw-0.15.orig/ufraw.desktop
+++ ufraw-0.15/ufraw.desktop
@@ -13,6 +13,7 @@ GenericName[pt]=Conversor de imagens cru
 GenericName[ru]=Проявитель цифровых негативов
 GenericName[sr]=Преводилац Raw слика
 GenericName[sr@latin]=Prevodilac Raw slika
+GenericName[tr]=Ham Resim Çevirici
 
 Comment=A utility to read and manipulate raw images from digital cameras
 Comment[cs]=Program ke čtení a manipulaci raw snímků z digitálních fotoaparátů
@@ -27,6 +28,7 @@ Comment[pt]=Programa para ler e manipula
 Comment[ru]=Проявка цифровых негативов и сохранение в TIFF/JPEG
 Comment[sr]=Алат за читање и обраду raw слика из дигиталних камера
 Comment[sr@latin]=Alat za čitanje i obradu raw slika iz digitalnih kamera
+Comment[tr]=Sayısal kameralardan alınan ham resimleri okumak ve işlemek için yardımcı bir uygulama
 
 Encoding=UTF-8
 Version=1.0
@@ -39,6 +41,6 @@ TryExec=ufraw
 Terminal=false
 StartupNotify=true
 
-MimeType=application/x-ufraw;image/x-dcraw
+MimeType=application/x-ufraw;image/x-dcraw;
 
 Icon=ufraw.png
