--- bash_completion	2006-03-01 18:20:18.000000000 +0200
+++ pardus_comp	2006-09-18 10:47:52.000000000 +0300
@@ -564,9 +564,9 @@
 
 	return 0
 } &&
-complete -F _service service
-[ -d /etc/init.d/ ] && complete -F _service $default \
-	$(for i in /etc/init.d/*; do echo ${i##*/}; done)
+#complete -F _service service
+#[ -d /etc/init.d/ ] && complete -F _service $default \
+#	$(for i in /etc/init.d/*; do echo ${i##*/}; done)
 
 # chown(1) completion
 #
@@ -685,7 +685,7 @@
 
 	return 0
 }
-complete -F _mount $default $filenames mount
+#complete -F _mount $default $filenames mount
 
 # Linux rmmod(8) completion. This completes on a list of all currently
 # installed kernel modules.
@@ -9083,7 +9083,7 @@
 
 	return 0
 }
-complete -F _svn $default svn
+complete -F _svn $default $filenames svn
 
 _svnadmin()
 {
