--- rules/72_active.cf.orig	2010-01-04 11:03:13.000000000 +0200
+++ rules/72_active.cf	2010-01-04 11:03:54.000000000 +0200
@@ -540,7 +540,7 @@
 ##} FH_DATE_IS_19XX
 
 ##{ FH_DATE_PAST_20XX
-header   FH_DATE_PAST_20XX	Date =~ /20[1-9][0-9]/ [if-unset: 2006]
+header   FH_DATE_PAST_20XX	Date =~ /20[2-9][0-9]/ [if-unset: 2006]
 describe FH_DATE_PAST_20XX	The date is grossly in the future.
 ##} FH_DATE_PAST_20XX
 
