Index: work/xorg-cf-files-1.0.2/linux.cf
===================================================================
--- work.orig/xorg-cf-files-1.0.2/linux.cf
+++ work/xorg-cf-files-1.0.2/linux.cf
@@ -578,7 +578,7 @@ InstallNamedTargetNoClobber(install,file
 #define MkdirHierCmd		mkdir -p
 
 #ifndef HaveLib64
-# if defined (AMD64Architecture) || defined (s390xArchitecture) || defined (Ppc64Architecture)
+# if 0
 #  define HaveLib64	YES
 # else
 #  define HaveLib64	NO
