Index: work/xorg-cf-files-1.0.2/X11.tmpl
===================================================================
--- work.orig/xorg-cf-files-1.0.2/X11.tmpl
+++ work/xorg-cf-files-1.0.2/X11.tmpl
@@ -1571,7 +1571,7 @@ FCHOWN_DEFINES = -DHAS_FCHOWN
 #endif
 #endif
 #ifndef XAppLoadDir
-#define XAppLoadDir $(LIBDIR)/app-defaults
+#define XAppLoadDir $(SHAREDIR)/app-defaults
 #endif
 #ifndef XtErrorPrefix
 #define XtErrorPrefix /* as empty string */
