Index: qt-x11-free-3.3.8b/src/kernel/qfont.cpp
===================================================================
--- qt-x11-free-3.3.8b.orig/src/kernel/qfont.cpp
+++ qt-x11-free-3.3.8b/src/kernel/qfont.cpp
@@ -1386,7 +1386,7 @@ static void initFontSubst()
     // default substitutions
     static const char *initTbl[] = {
 
-#if defined(Q_WS_X11)
+#if 0
 	"arial",        "helvetica",
 	"helv",         "helvetica",
 	"tms rmn",      "times",
