Index: src/3rdparty/opentype/ftglue.c
===================================================================
RCS file: /cvs/fontconfig/fontconfig/src/Attic/ftglue.c,v
retrieving revision 1.1.2.3
retrieving revision 1.1.2.4
diff -u -r1.1.2.3 -r1.1.2.4
--- src/3rdparty/opentype/ftglue.c	8 Mar 2006 02:30:43 -0000	1.1.2.3
+++ src/3rdparty/opentype/ftglue.c	2 Jun 2006 18:48:30 -0000	1.1.2.4
@@ -290,7 +290,7 @@
       if ( tag == the_tag )
       {
         LOG(( "TrueType table (start: %ld) (size: %ld)\n", start, size ));
-        error = ftglue_stream_seek( stream, offset+start );
+        error = ftglue_stream_seek( stream, start );
         goto FoundIt;
       }
     }
