Index: tools/designer/uic/form.cpp
===================================================================
--- tools/designer/uic/form.cpp	(revision 464760)
+++ tools/designer/uic/form.cpp	(working copy)
@@ -726,15 +726,6 @@
 		}
 		n2 = n2.nextSibling().toElement();
 	    }
-        } else if ( n.tagName() == "includehints" ) {
-            QDomElement n2 = n.firstChild().toElement();
-            while ( !n2.isNull() ) {
-                if ( n2.tagName() == "includehint" ) {
-                    QString file = n2.firstChild().toText().data();
-                    localIncludes += file;
-                }
-                n2 = n2.nextSibling().toElement();
-            }
         }
     }
 
