--- configure.old	2008-04-15 05:27:51.000000000 +0200
+++ configure	2008-04-15 05:28:31.000000000 +0200
@@ -32021,7 +32021,7 @@
 echo $ECHO_N "checking to see if compiler understands -Werror... $ECHO_C" >&6; }
 
   save_CFLAGS="$CFLAGS"
-  CFLAGS="$CFLAGS -Werror"
+  CFLAGS="$CFLAGS"
 
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
@@ -32067,7 +32067,7 @@
   CFLAGS="$save_CFLAGS"
 
   if test "X$flag_ok" = Xyes ; then
-    ERROR_CFLAGS="$ERROR_CFLAGS -Werror"
+    ERROR_CFLAGS="$ERROR_CFLAGS"
     true
   else
 
@@ -32390,7 +32390,7 @@
 echo $ECHO_N "checking to see if c++ compiler understands -Werror... $ECHO_C" >&6; }
 
   save_CPPFLAGS="$CPPFLAGS"
-  CPPFLAGS="$CPPFLAGS -Werror"
+  CPPFLAGS="$CPPFLAGS"
 
   ac_ext=cpp
 ac_cpp='$CXXCPP $CPPFLAGS'
@@ -32462,7 +32462,7 @@
 
 
     if test "x$werror_supported" = "xyes"; then
-        ERROR_CXXFLAGS="$ERROR_CXXFLAGS -Werror"
+        ERROR_CXXFLAGS="$ERROR_CXXFLAGS"
 
                 for f in '-Wno-non-virtual-dtor'
         do
