From f430752857621871af1d09acab68fbc617e8c165 Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Thu, 23 Dec 2010 23:42:44 +0200
Subject: [PATCH 2/6] Dont ignore if user want to set user id by self

BUG:FIXED:15786
---
 yali/postinstall.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/yali/postinstall.py b/yali/postinstall.py
index da32356..f6ae76a 100644
--- a/yali/postinstall.py
+++ b/yali/postinstall.py
@@ -179,7 +179,7 @@ def addUsers():
     obj = bus.get_object("tr.org.pardus.comar", "/package/baselayout")
     for user in yali.users.PENDING_USERS:
         ctx.logger.debug("User %s adding to system" % user.username)
-        uid = obj.addUser(-1, user.username,user.realname, "", "",
+        uid = obj.addUser(user.uid, user.username, user.realname, "", "",
                           unicode(user.passwd), user.groups, [], [],
                           dbus_interface="tr.org.pardus.comar.User.Manager")
         ctx.logger.debug("New user's id is %s" % uid)
-- 
1.7.3.4

