From 392815b07c10785119f4264cda0e2134a7fa0810 Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Fri, 24 Dec 2010 00:44:57 +0200
Subject: [PATCH 5/6] Fixes inconsistentLVM question message's line break

---
 yali/gui/interface.py |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/yali/gui/interface.py b/yali/gui/interface.py
index 18180ac..913f1d3 100644
--- a/yali/gui/interface.py
+++ b/yali/gui/interface.py
@@ -122,11 +122,11 @@ class Interface(object):
         na = {'msg': message, 'pvs': ", ".join(pv_names)}
         rc = self.messageWindow(_("Warning"),
                                 _("Error processing LVM.\n"
-                                  "There is inconsistent LVM data on %(msg)s.  You can "
-                                  "reinitialize all related PVs (%(pvs)s) which will erase "
-                                  "the LVM metadata, or ignore which will preserve the "
+                                  "There is inconsistent LVM data on %(msg)s. You\ncan "
+                                  "reinitialize all related PVs (%(pvs)s) which will\nerase "
+                                  "the LVM metadata, or ignore which will preserve the\n"
                                   "contents.  This action may also be applied to all other "
-                                  "PVs with inconsistent metadata.") % na, type="custom",
+                                  "PVs with\ninconsistent metadata.") % na, type="custom",
                                 customButtons = [_("Ignore"), _("Ignore all"),
                                                  _("Re-initialize"), _("Re-initialize all") ],
                                 customIcon="question")
-- 
1.7.3.4

