From 81c39ad9f5976f4acb3f846ebd0e21e7480750d4 Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Fri, 10 Dec 2010 15:07:32 +0200
Subject: [PATCH 5/6] Change installPartition label If /boot directory is used by RAID.

---
 yali/gui/ScrBootloader.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/yali/gui/ScrBootloader.py b/yali/gui/ScrBootloader.py
index a0c6bf4..836b379 100644
--- a/yali/gui/ScrBootloader.py
+++ b/yali/gui/ScrBootloader.py
@@ -87,7 +87,7 @@ class Widget(QWidget, ScreenWidget):
                 self.ui.installMBR.setText("The first sector of")
                 self.boot_disk = self.bootloader.choices[BOOT_TYPE_MBR][0]
             elif choice == BOOT_TYPE_RAID:
-                self.ui.installMBR.setText("The RAID array where Pardus is installed")
+                self.ui.installPartition.setText("The RAID array where Pardus is installed")
                 self.boot_partition = self.bootloader.choices[BOOT_TYPE_RAID][0]
             elif choice == BOOT_TYPE_PARTITION:
                 self.ui.installPartition.setText("The partition where Pardus is installed")
-- 
1.7.3.2

