From d91490d2e08a6d50aa0dbd09404fcac7d0e4ce00 Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Fri, 10 Dec 2010 15:10:42 +0200
Subject: [PATCH 6/6] Revert root password restrictions.

---
 yali/gui/ScrAdmin.py |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/yali/gui/ScrAdmin.py b/yali/gui/ScrAdmin.py
index b7bb5c5..c653fde 100644
--- a/yali/gui/ScrAdmin.py
+++ b/yali/gui/ScrAdmin.py
@@ -128,8 +128,6 @@ class Widget(QWidget, ScreenWidget):
             if len(password) < 4:
                 self.intf.informationWindow.update(_('Password is too short.'), type="error")
                 self.pass_valid = False
-            elif filter(lambda x: not x in string.ascii_letters, password):
-                self.intf.informationWindow.update(_("Don't use invalid characters"), type="error")
             else:
                 self.intf.informationWindow.hide()
                 self.pass_valid = True
-- 
1.7.3.2

