Index: yali-2.4.0_20101021/yali/data/default.xml
===================================================================
--- yali-2.4.0_20101021.orig/yali/data/default.xml
+++ yali-2.4.0_20101021/yali/data/default.xml
@@ -11,13 +11,13 @@
             <username>pars</username>
             <realname>Panthera Pardus Tulliana</realname>
             <password>pardus</password>
-            <groups>audio,dialout,disk,pnp,pnpadmin,users,video,wheel,lp,lpadmin,cdrom,floppy</groups>
+            <groups>audio,dialout,disk,pnp, users,video,wheel,lp,lpadmin,cdrom,floppy</groups>
         </user>
         <user>
             <username>caddy</username>
             <realname>Panthera Pardus Caddy</realname>
             <password>pardus</password>
-            <groups>audio,dialout,disk,pnp,pnpadmin,users,wheel,lp,lpadmin,cdrom,floppy</groups>
+            <groups>audio,dialout,disk,pnp, users,wheel,lp,lpadmin,cdrom,floppy</groups>
         </user>
     </users>
     <!--<reponame>pardus-2008</reponame>-->
Index: yali-2.4.0_20101021/yali/gui/ScrUsers.py
===================================================================
--- yali-2.4.0_20101021.orig/yali/gui/ScrUsers.py
+++ yali-2.4.0_20101021/yali/gui/ScrUsers.py
@@ -266,8 +266,7 @@ Proceed with the installation after you
         # ignore last character. see bug #887
         u.realname = unicode(self.ui.realname.text())
         u.passwd = unicode(self.ui.pass1.text())
-        u.groups = ["users", "pnp", "pnpadmin", "removable",
-                    "disk", "audio", "video", "power",
+        u.groups = ["users", "pnp", "disk", "audio", "video", "power",
                     "dialout", "lp", "lpadmin", "cdrom", "floppy"]
         pix = self.normalUserIcon
         if self.ui.admin.isChecked():
Index: yali-2.4.0_20101021/yali/kahya.py
===================================================================
--- yali-2.4.0_20101021.orig/yali/kahya.py
+++ yali-2.4.0_20101021/yali/kahya.py
@@ -124,7 +124,7 @@ class kahya:
     def __init__(self):
         self.fileSystems=["swap","ext4","ext3","ntfs","reiserfs","xfs"]
         self.fileSystems2=["ext4","ext3","xfs"]
-        self.defaultGroups=["audio","dialout","disk","pnp","pnpadmin","power","removable","users","video","lp","lpadmin","cdrom","floppy"]
+        self.defaultGroups=["audio","dialout","disk","pnp","power","users","video","lp","lpadmin","cdrom","floppy"]
         self.errorList=[]
         self.RatioList=[]
         self.correctData=yaliReadPiks.kahyaData()
