diff -up tzdata2011l/southamerica\~ tzdata2011l/southamerica
--- tzdata2011l/southamerica~	2011-10-10 06:15:43.000000000 +0200
+++ tzdata2011l/southamerica	2011-10-14 16:40:35.679270087 +0200
@@ -1054,10 +1054,8 @@ Zone America/Maceio	-2:22:52 -	LMT	1914
 # of America/Salvador.
 Zone America/Bahia	-2:34:04 -	LMT	1914
 			-3:00	Brazil	BR%sT	2003 Sep 24
-			-3:00	-	BRT
-# as noted above, not yet in operation.
-#			-3:00	-	BRT	2011 Oct 16
-#			-3:00	Brazil	BR%sT
+			-3:00	-	BRT	2011 Oct 16
+			-3:00	Brazil	BR%sT
 #
 # Goias (GO), Distrito Federal (DF), Minas Gerais (MG),
 # Espirito Santo (ES), Rio de Janeiro (RJ), Sao Paulo (SP), Parana (PR),

Diff finished.  Fri Oct 14 16:40:40 2011
