diff -Nur libcap-2.16-old/Make.Rules libcap-2.16/Make.Rules
--- libcap-2.16-old/Make.Rules	2009-04-29 12:57:21.000000000 +0300
+++ libcap-2.16/Make.Rules	2009-04-29 12:58:57.000000000 +0300
@@ -46,7 +46,7 @@
 IPATH += -I$(topdir)/libcap/include -I$(KERNEL_HEADERS)
 
 CC := gcc
-CFLAGS := -O2
+CFLAGS := pardusCFLAGS
 BUILD_CC := $(CC)
 BUILD_CFLAGS := $(CFLAGS) $(IPATH)
 AR := ar
@@ -57,7 +57,7 @@
         -Wstrict-prototypes -Wmissing-prototypes \
         -Wnested-externs -Winline -Wshadow
 LD=$(CC) -Wl,-x -shared
-LDFLAGS := #-g
+LDFLAGS := pardusLDFLAGS
 
 SYSTEM_HEADERS = /usr/include
 INCS=$(topdir)/libcap/include/sys/capability.h
