--- data/soundconverter.desktop.in.in	2008-08-01 23:48:25.000000000 +0300
+++ data/soundconverter.desktop.patch.in.in	2008-12-30 02:17:50.000000000 +0200
@@ -4,7 +4,9 @@
 Name=No name
-_Name=Sound Converter
+_Name=soundconverter
 _GenericName=Audio Format Converter
+GenericName[tr]=Ses Biçimi Çeviricisi
 _Comment=Convert audio files into other formats
+Comment[tr]=Ses dosyalarını diğer biçimlere çevirin
 Icon=soundconverter
 MimeType=application/x-ogg;application/ogg;audio/x-mp3;audio/x-scpls;audio/x-mp3;audio/x-mpeg;audio/mpeg;audio/x-mpegurl;application/x-flac;audio/x-wav;
 Exec=soundconverter %U
