Index: ChangeLog
===================================================================
--- ChangeLog	(revision 551)
+++ ChangeLog	(revision 552)
@@ -1,7 +1,10 @@
+2009-08-06 Fatih Aşıcı <fatih@pardus.org.tr>
+	* Added vboxnetadp as an optional module for VirtualBox.
+
 2009-06-06 Fatih Aşıcı <fatih@pardus.org.tr>
-		* Added console interface. Fallback if
-		  - PyQt4 is not available,
-		  - not executed under X server
+	* Added console interface. Fallback if
+	  - PyQt4 is not available,
+	  - not executed under X server
 
 2009-05-31 Fatih Aşıcı <fatih@pardus.org.tr>
-		* Initial import.
+	* Initial import.
Index: virt-wrapper.py
===================================================================
--- virt-wrapper.py	(revision 551)
+++ virt-wrapper.py	(revision 552)
@@ -16,7 +16,7 @@
 class VBox:
     def __init__(self):
         self.requiredModules = ("vboxdrv",)
-        self.optionalModules = ("vboxnetflt",)
+        self.optionalModules = ("vboxnetadp", "vboxnetflt")
         self.conflictedModules = ("kvm-amd", "kvm-intel", "kvm")
 
 class KVM:
