Index: blueman-1.21/data/blueman-manager.desktop.in
===================================================================
--- blueman-1.21.orig/data/blueman-manager.desktop.in
+++ blueman-1.21/data/blueman-manager.desktop.in
@@ -7,4 +7,4 @@ Exec=blueman-manager
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GTK;GNOME;Settings;HardwareSettings;
+Categories=Qt;KDE;System;Settings;HardwareSettings;
