Index: blueman-1.21/blueman/Constants.py
===================================================================
--- blueman-1.21.orig/blueman/Constants.py
+++ blueman-1.21/blueman/Constants.py
@@ -10,7 +10,7 @@ DHCP_CONFIG_FILE = "/etc/dhcp3/dhcpd.con
 POLKIT = "yes" == "yes"
 HAL_ENABLED = "no" == "yes"
 
-DEF_BROWSE_COMMAND = "nautilus --browser obex://[%d]"
+DEF_BROWSE_COMMAND = "konqueror obex://[%d]"
 
 import os
 import gettext
