diff -ur amarok-1.4.4.orig/amarok/src/amarokcore/amarok.kcfg amarok-1.4.4/amarok/src/amarokcore/amarok.kcfg
--- amarok-1.4.4.orig/amarok/src/amarokcore/amarok.kcfg	2006-10-26 23:40:20.000000000 +0300
+++ amarok-1.4.4/amarok/src/amarokcore/amarok.kcfg	2006-11-18 01:29:57.000000000 +0200
@@ -277,7 +277,7 @@
     </entry>
     <entry key="Use Custom Icon Theme" type="Bool">
         <label>Whether Amarok's custom icon theme or the system icon theme is used.</label>
-        <default>true</default>
+        <default>false</default>
     </entry>
     <entry key="Collection Browser View Mode" type="Int">
         <label>Whether the collection view is in the flat or tree view mode</label>
