diff -Nur kdeutils-3.5.10-old/ark/archiveformatinfo.cpp kdeutils-3.5.10/ark/archiveformatinfo.cpp
--- kdeutils-3.5.10-old/ark/archiveformatinfo.cpp	2008-11-02 02:51:17.000000000 +0200
+++ kdeutils-3.5.10/ark/archiveformatinfo.cpp	2008-11-02 02:52:43.000000000 +0200
@@ -59,8 +59,8 @@
   addFormatInfo( LHA_FORMAT, "application/x-lha", ".lha" );
 
   addFormatInfo( ZIP_FORMAT, "application/x-jar", ".jar" );
-  addFormatInfo( ZIP_FORMAT, "application/x-zip", ".zip" );
-  addFormatInfo( ZIP_FORMAT, "application/x-zip-compressed", ".zip" );
+  addFormatInfo( SEVENZIP_FORMAT, "application/x-zip", ".zip" );
+  addFormatInfo( SEVENZIP_FORMAT, "application/x-zip-compressed", ".zip" );
 
   addFormatInfo( COMPRESSED_FORMAT, "application/x-gzip", ".gz" );
   addFormatInfo( COMPRESSED_FORMAT, "application/x-bzip", ".bz" );
