---
 dejagnu.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: dejagnu-1.4.4.cvs20060709/dejagnu.h
===================================================================
--- dejagnu-1.4.4.cvs20060709.orig/dejagnu.h	2007-04-01 17:44:41.000000000 -0400
+++ dejagnu-1.4.4.cvs20060709/dejagnu.h	2007-04-01 17:45:00.000000000 -0400
@@ -156,12 +156,12 @@ totals (void)
   printf ("\nTotals:\n");
   printf ("\t#passed:\t\t%d\n", passed);
   printf ("\t#real failed:\t\t%d\n", failed);
-  if (xfail)
+  if (xfailed)
     printf ("\t#expected failures:\t\t%d\n", xfailed);
   if (untest)
     printf ("\t#untested:\t\t%d\n", untest);
   if (unresolve)
-    printf ("\t#unresolved:\t\t%d\n", unresolved);
+    printf ("\t#unresolved:\t\t%d\n", unresolve);
 }
 
 #ifdef __cplusplus
