diff -Nur slmodem-2.9.11-20080126.orig/drivers/amrmo_init.c slmodem-2.9.11-20080126/drivers/amrmo_init.c
--- slmodem-2.9.11-20080126.orig/drivers/amrmo_init.c	2008-01-24 08:30:03.000000000 +0200
+++ slmodem-2.9.11-20080126/drivers/amrmo_init.c	2008-03-24 01:03:48.000000000 +0200
@@ -609,7 +609,11 @@
 		}
 	}
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24)
         ret = request_irq(amrmo->irq, &amrmo_pci_interrupt,SA_SHIRQ,
+#else
+        ret = request_irq(amrmo->irq, &amrmo_pci_interrupt,IRQF_SHARED,
+#endif
 			  amrmo->name,amrmo);
 	if(ret) {
 		printk(KERN_ERR "slamr: failed request_irq\n");
