Index: connection.py
===================================================================
--- connection.py	(revision 30105)
+++ connection.py	(revision 30106)
@@ -645,7 +645,7 @@
             self.device.setText(self.new_conn[1])
             if "net" in self.link.modes:
                 self.r1.setChecked(True)
-                self.dns1.setChecked(True)
+                self.dns2.setChecked(True)
         if "net" in self.link.modes:
             self.slotFields()
 
