--- exp_inter.c	2007-11-11 05:43:07.000000000 +0200
+++ exp_inter.c	2007-11-11 05:43:46.000000000 +0200
@@ -35,7 +35,7 @@
 
 #include <ctype.h>
 
-#include "tcl.h"
+#include "tclInt.h"
 #include "string.h"
 
 #include "exp_tty_in.h"
--- expect_cf.h.in	2004-05-07 20:48:02.000000000 +0300
+++ expect_cf.h.in	2007-12-24 04:40:13.000000000 +0200
@@ -4,7 +4,7 @@
 #ifndef __EXPECT_CF_H__
 #define __EXPECT_CF_H__
 
-#undef NO_STDLIB_H		/* Tcl requires this name */
+#undef NO_STDLIB_H
 #undef NO_UNION_WAIT
 #undef HAVE_STDARG_H
 #undef HAVE_VARARGS_H
@@ -12,12 +12,12 @@
 #undef HAVE_SYSCONF_H
 #undef HAVE_SYS_FCNTL_H
 #undef HAVE_SYS_WAIT_H
-#undef HAVE_SYS_BSDTYPES_H	/* nice ISC special */
-#undef HAVE_SYS_SELECT_H	/* nice ISC special */
-#undef HAVE_SYS_TIME_H		/* nice ISC special */
-#undef HAVE_SYS_PTEM_H		/* SCO needs this for window size */
-#undef HAVE_STRREDIR_H		/* Solaris needs this for console redir */
-#undef HAVE_STRPTY_H		/* old-style Dynix ptys need this */
+#undef HAVE_SYS_BSDTYPES_H
+#undef HAVE_SYS_SELECT_H
+#undef HAVE_SYS_TIME_H
+#undef HAVE_SYS_PTEM_H
+#undef HAVE_STRREDIR_H
+#undef HAVE_STRPTY_H
 #undef HAVE_UNISTD_H
 #undef HAVE_SYSMACROS_H
 #undef HAVE_INTTYPES_H
@@ -26,8 +26,8 @@
 
 #undef pid_t
 #undef RETSIGTYPE
-#undef TIME_WITH_SYS_TIME	/* ok to include both time.h and sys/time.h */
-#undef SETPGRP_VOID		/* if setpgrp takes 0 args */
+#undef TIME_WITH_SYS_TIME
+#undef SETPGRP_VOID
 
 /*
  * This section is for compile macros needed by
@@ -42,7 +42,7 @@
 #undef SIMPLE_EVENT
 #undef HAVE_STRFTIME
 #undef HAVE_MEMMOVE
-#undef HAVE_TIMEZONE		/* timezone() a la Pyramid */
+#undef HAVE_TIMEZONE
 #undef HAVE_SIGLONGJMP
 #undef HAVE_STRCHR
 
