diff -Nur xine-lib-1.1.16.2-old/src/demuxers/demux_matroska.c xine-lib-1.1.16.2/src/demuxers/demux_matroska.c
--- xine-lib-1.1.16.2-old/src/demuxers/demux_matroska.c	2009-02-22 11:57:54.000000000 +0200
+++ xine-lib-1.1.16.2/src/demuxers/demux_matroska.c	2009-02-22 11:58:06.000000000 +0200
@@ -2078,6 +2078,31 @@
   return 1;
 }
 
+static int parse_simpleblock(demux_matroska_t *this, size_t block_len, uint64_t cluster_timecode, uint64_t block_duration)
+{
+  int has_block           = 0;
+  off_t block_pos         = 0;
+  off_t file_len          = 0;
+  int normpos             = 0;
+  int is_key              = 1;
+
+  lprintf("simpleblock\n");
+  block_pos = this->input->get_current_pos(this->input);
+  file_len = this->input->get_length(this->input);
+  if( file_len )
+    normpos = (int) ( (double) block_pos * 65535 / file_len );
+
+  if (!read_block_data(this, block_len))
+    return 0;
+
+  has_block = 1;
+    /* we have the duration, we can parse the block now */
+  if (!parse_block(this, block_len, cluster_timecode, block_duration,
+                   normpos, is_key))
+    return 0;
+  return 1;
+}
+
 static int parse_block_group(demux_matroska_t *this,
                              uint64_t cluster_timecode,
                              uint64_t cluster_duration) {
@@ -2142,7 +2167,8 @@
 
 static int parse_cluster(demux_matroska_t *this) {
   ebml_parser_t *ebml = this->ebml;
-  int next_level = 2;
+  int this_level = ebml->level;
+  int next_level = this_level;
   uint64_t timecode = 0;
   uint64_t duration = 0;
 
@@ -2159,7 +2185,7 @@
     this->first_cluster_found = 1;
   }
 
-  while (next_level == 2) {
+  while (next_level == this_level) {
     ebml_elem_t elem;
 
     if (!ebml_read_elem_head(ebml, &elem))
@@ -2183,6 +2209,11 @@
         if ((elem.len > 0) && !parse_block_group(this, timecode, duration))
           return 0;
         break;
+      case MATROSKA_ID_CL_SIMPLEBLOCK:
+        lprintf("simpleblock\n");
+        if (!parse_simpleblock(this, elem.len, timecode, duration))
+          return 0;
+        break;
       case MATROSKA_ID_CL_BLOCK:
         lprintf("block\n");
         if (!ebml_skip(ebml, &elem))
diff -Nur xine-lib-1.1.16.2-old/src/demuxers/matroska.h xine-lib-1.1.16.2/src/demuxers/matroska.h
--- xine-lib-1.1.16.2-old/src/demuxers/matroska.h	2009-02-22 11:57:54.000000000 +0200
+++ xine-lib-1.1.16.2/src/demuxers/matroska.h	2009-02-22 11:58:06.000000000 +0200
@@ -62,6 +62,7 @@
 #define MATROSKA_ID_CL_BLOCKGROUP                 0xA0
 #define MATROSKA_ID_CL_BLOCK                      0xA1
 #define MATROSKA_ID_CL_BLOCKVIRTUAL               0xA2
+#define MATROSKA_ID_CL_SIMPLEBLOCK                0xA3
 #define MATROSKA_ID_CL_BLOCKADDITIONS             0x75A1
 #define MATROSKA_ID_CL_BLOCKMORE                  0xA6
 #define MATROSKA_ID_CL_BLOCKADDID                 0xEE
