diff -Nur xine-lib-1.1.15-old/src/xine-engine/xine.c xine-lib-1.1.15/src/xine-engine/xine.c
--- xine-lib-1.1.15-old/src/xine-engine/xine.c	2008-09-28 00:36:55.000000000 +0300
+++ xine-lib-1.1.15/src/xine-engine/xine.c	2008-09-28 00:37:23.000000000 +0300
@@ -1700,7 +1700,7 @@
   _x_scan_plugins(this);
 
 #ifdef HAVE_SETLOCALE
-  if (!setlocale(LC_CTYPE, ""))
+  if (!setlocale(LC_CTYPE, "C"))
     xprintf(this, XINE_VERBOSITY_LOG, _("xine: locale not supported by C library\n"));
 #endif
 
diff -Nur xine-lib-1.1.15-old/src/xine-utils/utils.c xine-lib-1.1.15/src/xine-utils/utils.c
--- xine-lib-1.1.15-old/src/xine-utils/utils.c	2008-09-28 00:36:55.000000000 +0300
+++ xine-lib-1.1.15/src/xine-utils/utils.c	2008-09-28 00:37:15.000000000 +0300
@@ -564,7 +564,7 @@
   char *codeset = NULL;
   
 #ifdef HAVE_NL_LANGINFO
-  setlocale(LC_ALL, "");
+  setlocale(LC_ALL, "C");
   codeset = nl_langinfo(CODESET);
 #endif
   /*
