Index: ghostscript-8.70/lib/bdftops
===================================================================
--- ghostscript-8.70.orig/lib/bdftops
+++ ghostscript-8.70/lib/bdftops
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -dBATCH -dNODISPLAY -- bdftops.ps "$@"
+exec "$gs" -q -dBATCH -dNODISPLAY -- bdftops.ps "$@"
Index: ghostscript-8.70/lib/dumphint
===================================================================
--- ghostscript-8.70.orig/lib/dumphint
+++ ghostscript-8.70/lib/dumphint
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 
 OPTIONS="-dSAFER -dDELAYSAFER"
@@ -27,4 +26,4 @@ if [ $# -ne 1 ]; then
 	exit 1
 fi
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY $OPTIONS -- "`dirname $0`/dumphint.ps" "$1"
+exec "$gs" -q -dNODISPLAY $OPTIONS -- "`dirname $0`/dumphint.ps" "$1"
Index: ghostscript-8.70/lib/eps2eps
===================================================================
--- ghostscript-8.70.orig/lib/eps2eps
+++ ghostscript-8.70/lib/eps2eps
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 OPTIONS="-dDEVICEWIDTH=250000 -dDEVICEHEIGHT=250000"
 while true
@@ -26,4 +25,4 @@ if [ $# -ne 2 ]; then
 	exit 1
 fi
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=epswrite -sstdout=%stderr "-sOutputFile=$2" -dNOPAUSE -dBATCH -dSAFER $OPTIONS "$1"
+exec "$gs" -q -sDEVICE=epswrite -sstdout=%stderr "-sOutputFile=$2" -dNOPAUSE -dBATCH -dSAFER $OPTIONS "$1"
Index: ghostscript-8.70/lib/font2c
===================================================================
--- ghostscript-8.70.orig/lib/font2c
+++ ghostscript-8.70/lib/font2c
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -dWRITESYSTEMDICT -- font2c.ps "$@"
+exec "$gs" -q -dNODISPLAY -dWRITESYSTEMDICT -- font2c.ps "$@"
Index: ghostscript-8.70/lib/gsbj
===================================================================
--- ghostscript-8.70.orig/lib/gsbj
+++ ghostscript-8.70/lib/gsbj
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=bj10e -r180 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
+exec "$gs" -q -sDEVICE=bj10e -r180 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
Index: ghostscript-8.70/lib/gsdj500
===================================================================
--- ghostscript-8.70.orig/lib/gsdj500
+++ ghostscript-8.70/lib/gsdj500
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=djet500 -r300 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
+exec "$gs" -q -sDEVICE=djet500 -r300 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
Index: ghostscript-8.70/lib/gsdj
===================================================================
--- ghostscript-8.70.orig/lib/gsdj
+++ ghostscript-8.70/lib/gsdj
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=deskjet -r300 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
+exec "$gs" -q -sDEVICE=deskjet -r300 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
Index: ghostscript-8.70/lib/gslj
===================================================================
--- ghostscript-8.70.orig/lib/gslj
+++ ghostscript-8.70/lib/gslj
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=laserjet -r300 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
+exec "$gs" -q -sDEVICE=laserjet -r300 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
Index: ghostscript-8.70/lib/gslp
===================================================================
--- ghostscript-8.70.orig/lib/gslp
+++ ghostscript-8.70/lib/gslp
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=epson -r180 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
+exec "$gs" -q -sDEVICE=epson -r180 -dNOPAUSE -sPROGNAME=$0 -- gslp.ps --heading-center "`date`" "$@"
Index: ghostscript-8.70/lib/gsnd
===================================================================
--- ghostscript-8.70.orig/lib/gsnd
+++ ghostscript-8.70/lib/gsnd
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -dNODISPLAY "$@"
+exec "$gs" -dNODISPLAY "$@"
Index: ghostscript-8.70/lib/pdf2dsc
===================================================================
--- ghostscript-8.70.orig/lib/pdf2dsc
+++ ghostscript-8.70/lib/pdf2dsc
@@ -11,9 +11,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 me=`basename $0`
 
@@ -31,5 +30,5 @@ pdffile=$1
 dscfile=$2
 : ${dscfile:=`echo $pdffile | sed 's,\.[^/.]*,,'`.dsc}
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -dSAFER -dDELAYSAFER\
+exec "$gs" -q -dNODISPLAY -dSAFER -dDELAYSAFER\
     -sPDFname="$pdffile" -sDSCname="$dscfile" pdf2dsc.ps -c quit
Index: ghostscript-8.70/lib/pdf2ps
===================================================================
--- ghostscript-8.70.orig/lib/pdf2ps
+++ ghostscript-8.70/lib/pdf2ps
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 OPTIONS=""
 while true
@@ -35,4 +34,4 @@ fi
 # Doing an initial 'save' helps keep fonts from being flushed between pages.
 # We have to include the options twice because -I only takes effect if it
 # appears before other options.
-exec "$GS_EXECUTABLE" $OPTIONS -q -dNOPAUSE -dBATCH -dSAFER -sDEVICE=pswrite "-sOutputFile=$outfile" $OPTIONS -c save pop -f "$1"
+exec "$gs" $OPTIONS -q -dNOPAUSE -dBATCH -dSAFER -sDEVICE=pswrite "-sOutputFile=$outfile" $OPTIONS -c save pop -f "$1"
Index: ghostscript-8.70/lib/pdfopt
===================================================================
--- ghostscript-8.70.orig/lib/pdfopt
+++ ghostscript-8.70/lib/pdfopt
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 OPTIONS="-dSAFER -dDELAYSAFER"
 while true
@@ -26,4 +25,4 @@ if [ $# -ne 2 ]; then
 	exit 1
 fi
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY $OPTIONS -- pdfopt.ps "$1" "$2"
+exec "$gs" -q -dNODISPLAY $OPTIONS -- pdfopt.ps "$1" "$2"
Index: ghostscript-8.70/lib/pf2afm
===================================================================
--- ghostscript-8.70.orig/lib/pf2afm
+++ ghostscript-8.70/lib/pf2afm
@@ -10,8 +10,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -dSAFER -dDELAYSAFER  -- pf2afm.ps "$@"
+exec "$gs" -q -dNODISPLAY -dSAFER -dDELAYSAFER  -- pf2afm.ps "$@"
Index: ghostscript-8.70/lib/pfbtopfa
===================================================================
--- ghostscript-8.70.orig/lib/pfbtopfa
+++ ghostscript-8.70/lib/pfbtopfa
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 if [ $# -eq 2 ] 
 then
@@ -22,4 +21,4 @@ else
     exit 1
 fi
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -- pfbtopfa.ps "$1" "$outfile"
+exec "$gs" -q -dNODISPLAY -- pfbtopfa.ps "$1" "$outfile"
Index: ghostscript-8.70/lib/pphs
===================================================================
--- ghostscript-8.70.orig/lib/pphs
+++ ghostscript-8.70/lib/pphs
@@ -9,8 +9,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -- pphs.ps "$@"
+exec "$gs" -q -dNODISPLAY -- pphs.ps "$@"
Index: ghostscript-8.70/lib/printafm
===================================================================
--- ghostscript-8.70.orig/lib/printafm
+++ ghostscript-8.70/lib/printafm
@@ -9,8 +9,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -- printafm.ps "$@"
+exec "$gs" -q -dNODISPLAY -- printafm.ps "$@"
Index: ghostscript-8.70/lib/ps2epsi
===================================================================
--- ghostscript-8.70.orig/lib/ps2epsi
+++ ghostscript-8.70/lib/ps2epsi
@@ -6,9 +6,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 # try to create a temporary file securely
 if test -z "$TMPDIR"; then
@@ -92,7 +91,7 @@ awk 'F==1	{
 		}
 	' F=1 - F=2 "${outfile}" >>"$tmpfile"
 
-"$GS_EXECUTABLE" -q -dNOPAUSE -dSAFER -dDELAYSAFER -r72 -sDEVICE=bit -sOutputFile=/dev/null "$tmpfile" ps2epsi.ps "$tmpfile" <"${infile}" 1>&2
+"$gs" -q -sPAPERSIZE=a0 -dNOPAUSE -dSAFER -dDELAYSAFER -r72 -sDEVICE=bit -sOutputFile=/dev/null "$tmpfile" ps2epsi.ps "$tmpfile" <"${infile}" 1>&2
 rm -f "$tmpfile"
 rm -rf "$tmpdir"
 
Index: ghostscript-8.70/lib/ps2pdfwr
===================================================================
--- ghostscript-8.70.orig/lib/ps2pdfwr
+++ ghostscript-8.70/lib/ps2pdfwr
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 OPTIONS="-dSAFER"
 while true
@@ -42,4 +41,4 @@ fi
 
 # We have to include the options twice because -I only takes effect if it
 # appears before other options.
-exec "$GS_EXECUTABLE" $OPTIONS -q -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -sstdout=%stderr "-sOutputFile=$outfile" $OPTIONS -c .setpdfwrite -f "$infile"
+exec "$gs" $OPTIONS -q -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -sstdout=%stderr "-sOutputFile=$outfile" $OPTIONS -c .setpdfwrite -f "$infile"
Index: ghostscript-8.70/lib/ps2ps
===================================================================
--- ghostscript-8.70.orig/lib/ps2ps
+++ ghostscript-8.70/lib/ps2ps
@@ -7,9 +7,8 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
 OPTIONS="-dSAFER"
 while true
@@ -27,4 +26,4 @@ if [ $# -ne 2 ]; then
 	exit 1
 fi
 
-exec "$GS_EXECUTABLE" -q -sDEVICE=pswrite -sstdout=%stderr "-sOutputFile=$2" -dNOPAUSE -dBATCH $OPTIONS "$1"
+exec "$gs" -q -sDEVICE=pswrite -sstdout=%stderr "-sOutputFile=$2" -dNOPAUSE -dBATCH $OPTIONS "$1"
Index: ghostscript-8.70/lib/wftopfa
===================================================================
--- ghostscript-8.70.orig/lib/wftopfa
+++ ghostscript-8.70/lib/wftopfa
@@ -6,8 +6,7 @@
 GS_EXECUTABLE=gs
 gs="`dirname $0`/$GS_EXECUTABLE"
 if test ! -x "$gs"; then
-	gs="$GS_EXECUTABLE"
+	gs="$gs"
 fi
-GS_EXECUTABLE="$gs"
 
-exec "$GS_EXECUTABLE" -q -dNODISPLAY -- wftopfa.ps "$@"
+exec "$gs" -q -dNODISPLAY -- wftopfa.ps "$@"
