diff -Naur libupnp-1.6.6.orig/upnp/sample/Makefile.am libupnp-1.6.6/upnp/sample/Makefile.am
--- libupnp-1.6.6.orig/upnp/sample/Makefile.am	2008-02-10 03:55:29.000000000 +0200
+++ libupnp-1.6.6/upnp/sample/Makefile.am	2010-01-04 10:10:34.001304167 +0200
@@ -68,9 +68,9 @@
 
 if WITH_DOCUMENTATION
 examplesdir = $(docdir)/examples
-examples_DATA = \
-		$(upnp_tv_ctrlpt_SOURCES) \
-		$(upnp_tv_device_SOURCES)
+#examples_DATA = \
+#		$(upnp_tv_ctrlpt_SOURCES) \
+#		$(upnp_tv_device_SOURCES)
 endif
 
 
