Index: xz-4.999.9beta-143-g3e49/src/Makefile.in
===================================================================
--- xz-4.999.9beta-143-g3e49.orig/src/Makefile.in
+++ xz-4.999.9beta-143-g3e49/src/Makefile.in
@@ -246,7 +246,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 xz = @xz@
-SUBDIRS = liblzma xz xzdec lzmainfo scripts
+SUBDIRS = liblzma xz xzdec
 EXTRA_DIST = common
 all: all-recursive
 
Index: xz-4.999.9beta-143-g3e49/src/xzdec/Makefile.in
===================================================================
--- xz-4.999.9beta-143-g3e49.orig/src/xzdec/Makefile.in
+++ xz-4.999.9beta-143-g3e49/src/xzdec/Makefile.in
@@ -34,7 +34,7 @@ PRE_UNINSTALL = :
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-bin_PROGRAMS = xzdec$(EXEEXT) lzmadec$(EXEEXT)
+bin_PROGRAMS = xzdec$(EXEEXT)
 @COND_W32_TRUE@am__append_1 = xzdec_w32res.rc
 @COND_GNULIB_TRUE@am__append_2 = $(top_builddir)/lib/libgnu.a
 @COND_W32_TRUE@am__append_3 = lzmadec_w32res.rc
Index: xz-4.999.9beta-143-g3e49/src/xz/Makefile.in
===================================================================
--- xz-4.999.9beta-143-g3e49.orig/src/xz/Makefile.in
+++ xz-4.999.9beta-143-g3e49/src/xz/Makefile.in
@@ -296,7 +296,7 @@ xz_LDFLAGS = $(STATIC_LDFLAGS)
 xz_LDADD = $(top_builddir)/src/liblzma/liblzma.la $(am__append_2) \
 	$(LTLIBINTL)
 dist_man_MANS = xz.1
-xzlinks = unxz xzcat lzma unlzma lzcat
+xzlinks = unxz xzcat
 all: all-am
 
 .SUFFIXES:
