Index: policy/tr.org.pardus.comar.system.packagehandler.policy
===================================================================
--- policy/tr.org.pardus.comar.system.packagehandler.policy	(revision 28288)
+++ policy/tr.org.pardus.comar.system.packagehandler.policy	(revision 28293)
@@ -4,7 +4,7 @@
 <policyconfig>
     <vendor>Pardus</vendor>
     <vendor_url>http://www.pardus.org.tr</vendor_url>
-    <icon_name>preferences-desktop-default-applications</icon_name>
+    <icon_name>package</icon_name>
 
     <action id="tr.org.pardus.comar.system.packagehandler.setuppackage">
         <description>Execute package handler setup scripts</description>
Index: policy/tr.org.pardus.comar.system.package.policy
===================================================================
--- policy/tr.org.pardus.comar.system.package.policy	(revision 28288)
+++ policy/tr.org.pardus.comar.system.package.policy	(revision 28293)
@@ -4,7 +4,7 @@
 <policyconfig>
     <vendor>Pardus</vendor>
     <vendor_url>http://www.pardus.org.tr</vendor_url>
-    <icon_name>preferences-desktop-default-applications</icon_name>
+    <icon_name>package</icon_name>
 
     <action id="tr.org.pardus.comar.system.package.postinstall">
         <description>Execute post-install script of a package</description>
Index: policy/tr.org.pardus.comar.system.settings.policy
===================================================================
--- policy/tr.org.pardus.comar.system.settings.policy	(revision 28288)
+++ policy/tr.org.pardus.comar.system.settings.policy	(revision 28293)
@@ -4,7 +4,7 @@
 <policyconfig>
     <vendor>Pardus</vendor>
     <vendor_url>http://www.pardus.org.tr</vendor_url>
-    <icon_name>preferences-other</icon_name>
+    <icon_name>preferences-system</icon_name>
 
     <action id="tr.org.pardus.comar.system.settings.set">
         <description>Change system settings</description>
Index: policy/tr.org.pardus.comar.time.clock.policy
===================================================================
--- policy/tr.org.pardus.comar.time.clock.policy	(revision 28288)
+++ policy/tr.org.pardus.comar.time.clock.policy	(revision 28293)
@@ -4,7 +4,7 @@
 <policyconfig>
     <vendor>Pardus</vendor>
     <vendor_url>http://www.pardus.org.tr</vendor_url>
-    <icon_name>preferences-system-time</icon_name>
+    <icon_name>date</icon_name>
 
     <action id="tr.org.pardus.comar.time.clock.set">
         <description>Set date/time</description>
Index: policy/tr.org.pardus.comar.system.manager.policy
===================================================================
--- policy/tr.org.pardus.comar.system.manager.policy	(revision 28288)
+++ policy/tr.org.pardus.comar.system.manager.policy	(revision 28293)
@@ -4,7 +4,7 @@
 <policyconfig>
     <vendor>Pardus</vendor>
     <vendor_url>http://www.pardus.org.tr</vendor_url>
-    <icon_name>preferences-desktop-default-applications</icon_name>
+    <icon_name>package</icon_name>
 
     <action id="tr.org.pardus.comar.system.manager.installpackage">
         <description>Install package</description>
Index: models/Time.Clock.xml
===================================================================
--- models/Time.Clock.xml	(revision 28288)
+++ models/Time.Clock.xml	(revision 28293)
@@ -12,10 +12,10 @@
         <method name="getDate" access_label="get">
             <arg name="date" type="s" direction="out"/>
         </method>
-        <method name="loadFromHW">
+        <method name="loadFromHW" access_label="get">
             <annotation name="org.freedesktop.DBus.Method.NoReply" value="true"/>
         </method>
-        <method name="saveToHW">
+        <method name="saveToHW" access_label="set">
             <annotation name="org.freedesktop.DBus.Method.NoReply" value="true"/>
         </method>
         <method name="setTimeZone" access_label="set">
