diff -Nur ReportLab_2_3-old/src/reportlab/rl_config.py ReportLab_2_3/src/reportlab/rl_config.py
--- ReportLab_2_3-old/src/reportlab/rl_config.py	2009-05-05 15:55:54.000000000 +0300
+++ ReportLab_2_3/src/reportlab/rl_config.py	2009-05-05 15:56:09.000000000 +0300
@@ -54,27 +54,27 @@
                 '/usr/lib/Acrobat6/Resource/Font',      #Linux, Acrobat 5?
                 '/usr/lib/Acrobat5/Resource/Font',      #Linux, Acrobat 5?
                 '/usr/lib/Acrobat4/Resource/Font',      #Linux, Acrobat 4
-                '/usr/local/Acrobat9/Resource/Font',    #Linux, Acrobat 5?
-                '/usr/local/Acrobat8/Resource/Font',    #Linux, Acrobat 5?
-                '/usr/local/Acrobat7/Resource/Font',    #Linux, Acrobat 5?
-                '/usr/local/Acrobat6/Resource/Font',    #Linux, Acrobat 5?
-                '/usr/local/Acrobat5/Resource/Font',    #Linux, Acrobat 5?
-                '/usr/local/Acrobat4/Resource/Font',    #Linux, Acrobat 4
+                '/opt/Acrobat9/Resource/Font',    #Linux, Acrobat 5?
+                '/opt/Acrobat8/Resource/Font',    #Linux, Acrobat 5?
+                '/opt/Acrobat7/Resource/Font',    #Linux, Acrobat 5?
+                '/opt/Acrobat6/Resource/Font',    #Linux, Acrobat 5?
+                '/opt/Acrobat5/Resource/Font',    #Linux, Acrobat 5?
+                '/opt/Acrobat4/Resource/Font',    #Linux, Acrobat 4
                 '%(REPORTLAB_DIR)s/fonts',              #special
                 '%(REPORTLAB_DIR)s/../fonts',           #special
                 '%(REPORTLAB_DIR)s/../../fonts',        #special
-                '%(HOME)s/fonts',                       #special
+                '%(HOME)s/.fonts',                       #special
                  )
 
 # places to look for TT Font information
 TTFSearchPath = (
                 'c:/winnt/fonts',
                 'c:/windows/fonts',
-                '/usr/lib/X11/fonts/TrueType/',
+                '/usr/share/fonts/ttf-bitstream-vera/',
                 '%(REPORTLAB_DIR)s/fonts',      #special
                 '%(REPORTLAB_DIR)s/../fonts',   #special
                 '%(REPORTLAB_DIR)s/../../fonts',#special
-                '%(HOME)s/fonts',               #special
+                '%(HOME)s/.fonts',               #special
                 #mac os X - from
                 #http://developer.apple.com/technotes/tn/tn2024.html
                 '~/Library/Fonts',
@@ -92,12 +92,12 @@
                   '/usr/lib/Acrobat6/Resource/CMap',
                   '/usr/lib/Acrobat5/Resource/CMap',
                   '/usr/lib/Acrobat4/Resource/CMap',
-                  '/usr/local/Acrobat9/Resource/CMap',
-                  '/usr/local/Acrobat8/Resource/CMap',
-                  '/usr/local/Acrobat7/Resource/CMap',
-                  '/usr/local/Acrobat6/Resource/CMap',
-                  '/usr/local/Acrobat5/Resource/CMap',
-                  '/usr/local/Acrobat4/Resource/CMap',
+                  '/opt/Acrobat9/Resource/CMap',
+                  '/opt/Acrobat8/Resource/CMap',
+                  '/opt/Acrobat7/Resource/CMap',
+                  '/opt/Acrobat6/Resource/CMap',
+                  '/opt/Acrobat5/Resource/CMap',
+                  '/opt/Acrobat4/Resource/CMap',
                   'C:\\Program Files\\Adobe\\Acrobat\\Resource\\CMap',
                   'C:\\Program Files\\Adobe\\Acrobat 9.0\\Resource\\CMap',
                   'C:\\Program Files\\Adobe\\Acrobat 8.0\\Resource\\CMap',
@@ -108,7 +108,7 @@
                   '%(REPORTLAB_DIR)s/fonts/CMap',       #special
                   '%(REPORTLAB_DIR)s/../fonts/CMap',    #special
                   '%(REPORTLAB_DIR)s/../../fonts/CMap', #special
-                  '%(HOME)s/fonts/CMap',                #special
+                  '%(HOME)s/.fonts/CMap',                #special
                   )
 
 #### Normally don't need to edit below here ####
