diff -Nur bootsplash-theme-pardus-0.7-old/Pardus/config/bootsplash-1024x768.cfg bootsplash-theme-pardus-0.7/Pardus/config/bootsplash-1024x768.cfg
--- bootsplash-theme-pardus-0.7-old/Pardus/config/bootsplash-1024x768.cfg	2009-10-15 15:39:20.529198685 +0300
+++ bootsplash-theme-pardus-0.7/Pardus/config/bootsplash-1024x768.cfg	2009-10-15 15:42:38.435198607 +0300
@@ -13,7 +13,7 @@
 tx=0
 ty=0
 tw=1024
-th=725
+th=721
 
 jpeg=/etc/bootsplash/themes/Pardus/images/verbose-1024x768.jpg
 silentjpeg=/etc/bootsplash/themes/Pardus/images/silent-1024x768.jpg
diff -Nur bootsplash-theme-pardus-0.7-old/Pardus/config/bootsplash-1280x1024.cfg bootsplash-theme-pardus-0.7/Pardus/config/bootsplash-1280x1024.cfg
--- bootsplash-theme-pardus-0.7-old/Pardus/config/bootsplash-1280x1024.cfg	2009-10-15 15:39:20.529198685 +0300
+++ bootsplash-theme-pardus-0.7/Pardus/config/bootsplash-1280x1024.cfg	2009-10-15 15:43:25.932963627 +0300
@@ -13,7 +13,7 @@
 tx=0
 ty=0
 tw=1280
-th=970
+th=977
 
 jpeg=/etc/bootsplash/themes/Pardus/images/verbose-1280x1024.jpg
 silentjpeg=/etc/bootsplash/themes/Pardus/images/silent-1280x1024.jpg
diff -Nur bootsplash-theme-pardus-0.7-old/Pardus/config/bootsplash-1280x800.cfg bootsplash-theme-pardus-0.7/Pardus/config/bootsplash-1280x800.cfg
--- bootsplash-theme-pardus-0.7-old/Pardus/config/bootsplash-1280x800.cfg	2009-10-15 15:39:20.529198685 +0300
+++ bootsplash-theme-pardus-0.7/Pardus/config/bootsplash-1280x800.cfg	2009-10-15 15:44:00.804949004 +0300
@@ -13,7 +13,7 @@
 tx=0
 ty=0
 tw=1280
-th=755
+th=753
 
 jpeg=/etc/bootsplash/themes/Pardus/images/verbose-1280x800.jpg
 silentjpeg=/etc/bootsplash/themes/Pardus/images/silent-1280x800.jpg
diff -Nur bootsplash-theme-pardus-0.7-old/Pardus/config/bootsplash-640x480.cfg bootsplash-theme-pardus-0.7/Pardus/config/bootsplash-640x480.cfg
--- bootsplash-theme-pardus-0.7-old/Pardus/config/bootsplash-640x480.cfg	2009-10-15 15:39:20.529198685 +0300
+++ bootsplash-theme-pardus-0.7/Pardus/config/bootsplash-640x480.cfg	2009-10-15 15:41:20.848303152 +0300
@@ -13,7 +13,7 @@
 tx=0
 ty=0
 tw=640
-th=453
+th=433
 
 jpeg=/etc/bootsplash/themes/Pardus/images/verbose-640x480.jpg
 silentjpeg=/etc/bootsplash/themes/Pardus/images/silent-640x480.jpg
diff -Nur bootsplash-theme-pardus-0.7-old/Pardus/config/bootsplash-800x600.cfg bootsplash-theme-pardus-0.7/Pardus/config/bootsplash-800x600.cfg
--- bootsplash-theme-pardus-0.7-old/Pardus/config/bootsplash-800x600.cfg	2009-10-15 15:39:20.529198685 +0300
+++ bootsplash-theme-pardus-0.7/Pardus/config/bootsplash-800x600.cfg	2009-10-15 15:41:58.916952634 +0300
@@ -13,7 +13,7 @@
 tx=0
 ty=0
 tw=800
-th=566
+th=553
 
 jpeg=/etc/bootsplash/themes/Pardus/images/verbose-800x600.jpg
 silentjpeg=/etc/bootsplash/themes/Pardus/images/silent-800x600.jpg
