Index: pisi/actionsapi/cmaketools.py
===================================================================
--- pisi/actionsapi/cmaketools.py	(revision 26993)
+++ pisi/actionsapi/cmaketools.py	(revision 26998)
@@ -60,7 +60,7 @@
                       -DCMAKE_C_FLAGS="%s" \
                       -DCMAKE_CXX_FLAGS="%s" \
                       -DCMAKE_LD_FLAGS="%s" \
-                      -DCMAKE_BUILD_TYPE=None %s %s' % (installPrefix, get.CFLAGS(), get.CXXFLAGS(), get.LDFLAGS(), parameters, sourceDir)
+                      -DCMAKE_BUILD_TYPE=RelWithDebInfo %s %s' % (installPrefix, get.CFLAGS(), get.CXXFLAGS(), get.LDFLAGS(), parameters, sourceDir)
 
         if system(args):
             raise ConfigureError(_('Configure failed.'))
