Index: pisi-2.4_alpha2/pisi/cli/addrepo.py
===================================================================
--- pisi-2.4_alpha2.orig/pisi/cli/addrepo.py
+++ pisi-2.4_alpha2/pisi/cli/addrepo.py
@@ -56,14 +56,15 @@ NB: We support only local files (e.g., /
 
     def run(self):
 
-        if len(self.args)==2 or len(self.args)==0:
+        if len(self.args) == 2 or len(self.args) == 0:
             self.init()
-            if len(self.args)==2:
+            if len(self.args) == 2:
                 name = self.args[0]
                 indexuri = self.args[1]
             else:
-                name = 'pardus-2009'
-                indexuri = 'http://paketler.pardus.org.tr/pardus-2009/pisi-index.xml.bz2'
+                name = 'Pardus'
+                arch = ctx.config.values.general.architecture
+                indexuri = 'http://paketler.pardus.org.tr/pardus/kurumsal2/stable/%s/pisi-index.xml.xz' % arch
 
             if ctx.get_option('no_fetch'):
                 if not ctx.ui.confirm(_('Add %s repository without updating the database?\nBy confirming '
