diff --git a/pisi/db/repodb.py b/pisi/db/repodb.py
index 0056032..06dd749 100644
--- a/pisi/db/repodb.py
+++ b/pisi/db/repodb.py
@@ -151,11 +151,12 @@ class RepoDB(lazydb.LazyDB):
 
     def get_repo_doc(self, repo_name):
         repo = self.get_repo(repo_name)
-        if repo.indexuri.is_remote_file():
-            index = os.path.basename(repo.indexuri.get_uri())
+
+        index_path = repo.indexuri.get_uri()
+
+        if index_path.endswith("bz2") or repo.indexuri.is_remote_file():
+            index = os.path.basename(index_path)
             index_path = pisi.util.join_path(ctx.config.index_dir(), repo_name, index)
-        else:
-            index_path = repo.indexuri.get_uri()
 
         if index_path.endswith("bz2"):
             index_path = pisi.util.remove_suffix(".bz2", index_path)
