--- a/pisi/component.py	(revision 26810)
+++ b/pisi/component.py	(revision 26812)
@@ -61,11 +61,11 @@
     # LocalName for system.base could be sistem.taban or "Taban Sistem",
     # this could be useful for GUIs
 
-    t_LocalName = [autoxml.LocalText, autoxml.mandatory]
+    t_LocalName = [autoxml.LocalText, autoxml.optional]
 
     # Information about the component
-    t_Summary = [autoxml.LocalText, autoxml.mandatory]
-    t_Description = [autoxml.LocalText, autoxml.mandatory]
+    t_Summary = [autoxml.LocalText, autoxml.optional]
+    t_Description = [autoxml.LocalText, autoxml.optional]
     t_Group = [ autoxml.String, autoxml.optional ]
 
     # the parts of this component.
