From 13072d8ff1b683971ef88f5d3746ec40011dd314 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fatih=20A=C5=9F=C4=B1c=C4=B1?= <fatih@pardus.org.tr>
Date: Mon, 22 Feb 2010 16:54:47 +0200
Subject: [PATCH] revert r27902

---
 ChangeLog                |    6 ------
 pisi/operations/build.py |    2 +-
 2 files changed, 1 insertions(+), 7 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 70f6b75..5d36df6 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -4,12 +4,6 @@
 		doesn't allow a unified repository for multiple
 		architectures.
 
-2010-02-12 Serdar Dalgıç <serdar@pardus.org.tr>
-		* pisi/operations/build.py: Use glob.glob0 and lstrip instead of
-		glob.glob. Typos in filePaths in pspec.xml were sometimes ignored and
-		abandoned files were not succesfully caught in some cases. Now it is
-		fixed. Thanks Fatih for the collaboration.
-
 2010-02-12 Gökçen Eraslan <gokcen@pardus.org.tr>
 		* pisi/actionsapi/shelltools.py: Replace FileError with ctx.ui.error
 		since FileError is defined in pisitoolsfunctions.
diff --git a/pisi/operations/build.py b/pisi/operations/build.py
index 93a60a7..7e8ed76 100644
--- a/pisi/operations/build.py
+++ b/pisi/operations/build.py
@@ -59,7 +59,7 @@ def get_file_type(path, pinfo_list, install_dir):
     """Return the file type of a path according to the given PathInfo
     list"""
 
-    Match = lambda x: [match for match in glob.glob0(install_dir, x.lstrip("/")) if pisi.util.join_path(install_dir, path).find(match) > -1]
+    Match = lambda x: [match for match in glob.glob(install_dir + x) if pisi.util.join_path(install_dir, path).find(match) > -1]
 
     def Sort(x):
         x.sort(reverse=True)
-- 
1.6.5.6

