From 83e1fcf9fa77379e1c11ffac75e40733deb2fb26 Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Mon, 14 Feb 2011 09:04:06 +0200
Subject: [PATCH] Fix typo

---
 yali/gui/ScrInstall.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/yali/gui/ScrInstall.py b/yali/gui/ScrInstall.py
index 64159ab..8c0a192 100644
--- a/yali/gui/ScrInstall.py
+++ b/yali/gui/ScrInstall.py
@@ -122,7 +122,7 @@ class Widget(QWidget, ScreenWidget):
         # Disable mouse handler
         ctx.mainScreen.dontAskCmbAgain = True
         ctx.mainScreen.theme_shortcut.setEnabled(False)
-        ctx.mainScreen.system_menu.setEnabled(False)
+        ctx.mainScreen.ui.system_menu.setEnabled(False)
 
         # start installer thread
         ctx.logger.debug("PkgInstaller is creating...")
-- 
1.7.3.4

