--- Makefile.orig	2008-12-10 18:17:57.000000000 +0200
+++ Makefile	2008-12-10 18:19:12.000000000 +0200
@@ -36,7 +36,7 @@
 mkpasswd_LDADD += -lcrypt
 endif
 
-all: Makefile.depend whois mkpasswd #pos
+all: Makefile.depend whois mkpasswd pos
 
 whois_OBJECTS := whois.o utils.o
 mkpasswd_OBJECTS := mkpasswd.o utils.o
@@ -68,7 +68,7 @@
 	$(PERL) -w make_tld_serv.pl < tld_serv_list > $@
 
 ##############################################################################
-install: install-whois install-mkpasswd #install-pos
+install: install-whois install-mkpasswd install-pos
 
 install-whois: whois
 	$(INSTALL) -d $(BASEDIR)$(prefix)/bin/
