Index: pam_pkcs11-0.6.6/configure
===================================================================
--- pam_pkcs11-0.6.6.orig/configure
+++ pam_pkcs11-0.6.6/configure
@@ -6125,7 +6125,7 @@ $as_echo "#define HAVE_DCGETTEXT 1" >>co
 
 
 
-ALL_LINGUAS="de fr nl pl pt_br ru"
+ALL_LINGUAS="de fr nl pl pt_br ru tr"
 
 # Checks for programs.
 ac_ext=c
Index: pam_pkcs11-0.6.6/configure.in
===================================================================
--- pam_pkcs11-0.6.6.orig/configure.in
+++ pam_pkcs11-0.6.6/configure.in
@@ -16,7 +16,7 @@ AM_GNU_GETTEXT_VERSION([0.17])
 AM_GNU_GETTEXT([external])
 
 dnl Add the languages which your application supports here.
-ALL_LINGUAS="de fr nl pl pt_br ru"
+ALL_LINGUAS="de fr nl pl pt_br ru tr"
 
 # Checks for programs.
 AC_PROG_CC
