Index: pam_pkcs11-0.6.6/src/tools/pkcs11_eventmgr.c
===================================================================
--- pam_pkcs11-0.6.6.orig/src/tools/pkcs11_eventmgr.c
+++ pam_pkcs11-0.6.6/src/tools/pkcs11_eventmgr.c
@@ -174,7 +174,7 @@ static int execute_event (const char *ac
 	while (actionlist) {
 		int res;
 		char *action_cmd= actionlist->data;
-		DBG1("Executiong action: '%s'",action_cmd);
+		DBG1("Executing action: '%s'",action_cmd);
 		/*
 		there are some security issues on using system() in
 		setuid/setgid programs. so we will use an alternate function
@@ -405,7 +405,7 @@ int main(int argc, char *argv[]) {
 	    return 1;
 	}
 	sprintf(moduleSpec,SPEC_TEMPLATE, pkcs11_module);
-	DBG2("loading Module explictly, moduleSpec=<%s> module=%s\n",
+	DBG2("loading Module explicitly, moduleSpec=<%s> module=%s\n",
 						moduleSpec, pkcs11_module);
 	module = SECMOD_LoadUserModule(moduleSpec, NULL, 0);
 	free(moduleSpec);
