Subject: [PATCH] From 8531fcfcc5261231b44a12c0a6d329b3cc4280fa Mon Sep 17 00:00:00 2001
From: unknown author <cooker@mandrivalinux.org>

Subject: [PATCH] retain vt

Signed-off-by: unknown author <cooker@mandrivalinux.org>

---

 src/main.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/src/main.c b/src/main.c
index e622755..05b6769 100644
--- a/src/main.c
+++ b/src/main.c
@@ -109,6 +109,7 @@ typedef struct
   uint32_t should_be_attached : 1;
   uint32_t should_retain_splash : 1;
   uint32_t is_inactive : 1;
+  uint32_t should_retain_vt: 1;
 
   char *kernel_console_tty;
   char *override_splash_path;
@@ -696,6 +697,7 @@ static void
 on_show_splash (state_t *state)
 {
   bool has_display;
+/*  state->should_retain_vt = false;*/
 
   if (state->is_inactive)
     {
@@ -762,7 +764,7 @@ quit_splash (state_t *state)
 
   if (state->terminal != NULL)
     {
-      if (!state->should_retain_splash)
+      if (!state->should_retain_splash && !state->should_retain_vt)
         {
           ply_trace ("Not retaining splash, so deallocating VT");
           ply_terminal_deactivate_vt (state->terminal);
@@ -805,6 +807,7 @@ on_hide_splash (state_t *state)
     return;
 
   ply_trace ("hiding boot splash");
+  state->should_retain_vt = true;
   dump_details_and_quit_splash (state);
 }
 
