--- gstreamer/audiooutput.cpp.orig	2009-02-27 12:40:17.000000000 +0100
+++ gstreamer/audiooutput.cpp	2009-02-27 12:41:48.000000000 +0100
@@ -42,6 +42,7 @@
     static int count = 0;
     m_name = "AudioOutput" + QString::number(count++);
     if (m_backend->isValid()) {
+	g_set_application_name(qApp->applicationName().toUtf8());
         m_audioBin = gst_bin_new (NULL);
         gst_object_ref (GST_OBJECT (m_audioBin));
         gst_object_sink (GST_OBJECT (m_audioBin));     
