diff -up system-config-printer-1.1.15/AdvancedServerSettings.py.cupsd.conf-parser system-config-printer-1.1.15/AdvancedServerSettings.py
--- system-config-printer-1.1.15/AdvancedServerSettings.py.cupsd.conf-parser	2009-09-01 12:08:33.000000000 +0100
+++ system-config-printer-1.1.15/AdvancedServerSettings.py	2009-12-03 17:18:29.015925153 +0000
@@ -154,7 +154,7 @@ class AdvancedServerSettingsDialog:
         self.browse_poll = []
         f.seek (0)
         for line in f.readlines ():
-            l = line.lower ()
+            l = line.lower ().strip ()
             if l.startswith ("preservejobhistory "):
                 try:
                     preserve_job_history = parse_yesno (l)
