diff -up system-config-printer-1.1.15/system-config-printer.py.browsepoll system-config-printer-1.1.15/system-config-printer.py
--- system-config-printer-1.1.15/system-config-printer.py.browsepoll	2009-11-30 12:43:08.000000000 +0000
+++ system-config-printer-1.1.15/system-config-printer.py	2009-12-03 17:09:32.221925957 +0000
@@ -3379,7 +3379,7 @@ class GUI(GtkGUI, monitor.Watcher):
         self.setDataButtonState()
 
     def save_serversettings(self):
-        setting_dict = self.server_settings.copy()
+        setting_dict = dict()
         for widget, setting in [
             (self.chkServerBrowse, cups.CUPS_SERVER_REMOTE_PRINTERS),
             (self.chkServerShare, cups.CUPS_SERVER_SHARE_PRINTERS),
