diff -up system-config-printer-1.1.15/system-config-printer.py.lpd-uri system-config-printer-1.1.15/system-config-printer.py
--- system-config-printer-1.1.15/system-config-printer.py.lpd-uri	2009-12-08 14:00:31.646480878 +0000
+++ system-config-printer-1.1.15/system-config-printer.py	2009-12-08 14:00:35.027606138 +0000
@@ -5767,10 +5767,10 @@ class NewPrinterGUI(GtkGUI):
                 (scheme, rest) = urllib.splittype (device.uri)
                 (hostport, rest) = urllib.splithost (rest)
                 (queue, rest) = urllib.splitquery (rest)
-                if queue[0] == '/':
-                    queue = queue[1:]
-
                 if queue != '':
+                    if queue[0] == '/':
+                        queue = queue[1:]
+
                     device.menuentry = _("LPD/LPR queue '%s'") % queue
                 else:
                     device.menuentry = _("LPD/LPR queue")
