diff -up system-config-printer-1.1.15/jobviewer.py.short-lived-states system-config-printer-1.1.15/jobviewer.py
--- system-config-printer-1.1.15/jobviewer.py.short-lived-states	2009-11-30 12:43:08.000000000 +0000
+++ system-config-printer-1.1.15/jobviewer.py	2009-12-09 10:53:20.936016882 +0000
@@ -1430,7 +1430,13 @@ class JobViewer (GtkGUI, monitor.Watcher
         try:
             notification = self.state_reason_notifications[tuple]
             if notification.get_data ('closed') != True:
-                notification.close ()
+                try:
+                    notification.close ()
+                except glib.GError:
+                    # Can fail if the notification wasn't even shown
+                    # yet (as in bug #545733).
+                    pass
+
             del self.state_reason_notifications[tuple]
             self.set_statusicon_visibility ()
         except KeyError:
