diff -ru mediatomb-0.11.0.orig/src/storage/mysql/mysql_storage.cc mediatomb-0.11.0/src/storage/mysql/mysql_storage.cc
--- mediatomb-0.11.0.orig/src/storage/mysql/mysql_storage.cc	2009-02-16 12:12:35.000000000 +0000
+++ mediatomb-0.11.0/src/storage/mysql/mysql_storage.cc	2009-02-16 12:12:44.000000000 +0000
@@ -48,6 +48,9 @@
     #include <zlib.h>
 #endif
 
+#include <my_global.h>
+#include <my_sys.h>
+
 // updates 1->2
 #define MYSQL_UPDATE_1_2_1 "ALTER TABLE `mt_cds_object` CHANGE `location` `location` BLOB NULL DEFAULT NULL"
 #define MYSQL_UPDATE_1_2_2 "ALTER TABLE `mt_cds_object` CHANGE `metadata` `metadata` BLOB NULL DEFAULT NULL"
diff -ru mediatomb-0.11.0.orig/src/tools.cc mediatomb-0.11.0/src/tools.cc
--- mediatomb-0.11.0.orig/src/tools.cc	2009-02-16 12:12:35.000000000 +0000
+++ mediatomb-0.11.0/src/tools.cc	2009-02-16 12:13:15.000000000 +0000
@@ -303,8 +303,8 @@
     Ref<StringBuffer> buf(new StringBuffer(len / 2));
     for (int i = 0; i < len; i += 2)
     {
-        char *chi = strchr(HEX_CHARS, ptr[i]);
-        char *clo = strchr(HEX_CHARS, ptr[i + 1]);
+        const char *chi = strchr(HEX_CHARS, ptr[i]);
+        const char *clo = strchr(HEX_CHARS, ptr[i + 1]);
         int hi, lo;
         
         if (chi)
@@ -397,7 +397,7 @@
             char clo = data[i++];
             int hi, lo;
 
-            char *pos;
+            const char *pos;
 
             pos = strchr(hex, chi);
             if (!pos)
