diff -Naur expect-5.43.0.orig/pkgIndex.in expect-5.43.0/pkgIndex.in
--- expect-5.43.0.orig/pkgIndex.in	2002-02-12 02:00:56.000000000 +0000
+++ expect-5.43.0/pkgIndex.in	2006-07-10 17:03:27.000000000 +0000
@@ -7,4 +7,4 @@
 # script is sourced, the variable $dir must contain the
 # full path name of this file's directory.
 
-package ifneeded Expect @EXP_VERSION_FULL@ [list load [file join $dir .. @EXP_SHARED_LIB_FILE@]]
+package ifneeded Expect @EXP_VERSION_FULL@ [list load [file join @prefix@ lib libexpect.so.@EXP_VERSION_FULL@]]
