diff -Naurp expect-5.43.orig/exp_command.c expect-5.43/exp_command.c
--- expect-5.43.orig/exp_command.c	2004-08-20 17:18:01.000000000 +0000
+++ expect-5.43/exp_command.c	2007-09-04 19:30:20.000000000 +0000
@@ -310,7 +310,15 @@ expBusy(esPtr)
     fcntl(x,F_DUPFD,esPtr->fdin);
     close(x);
   }
-  expCloseOnExec(esPtr->fdin);
+  if (esPtr->fdout != EXP_NOFD &&
+      esPtr->fdout != esPtr->fdin) {
+    fcntl(esPtr->fdin,F_DUPFD,esPtr->fdout);
+  }
+  if (esPtr->fdin > 2)
+    expCloseOnExec(esPtr->fdin);
+  if (esPtr->fdout != EXP_NOFD &&
+      esPtr->fdout > 2)
+    expCloseOnExec(esPtr->fdout);
   esPtr->fdBusy = TRUE;
 }
 
