diff -Nur glibc-20081113T2206-old/sysdeps/unix/sysv/linux/configure glibc-20081113T2206/sysdeps/unix/sysv/linux/configure
--- glibc-20081113T2206-old/sysdeps/unix/sysv/linux/configure	2010-02-24 23:27:44.897488794 +0200
+++ glibc-20081113T2206/sysdeps/unix/sysv/linux/configure	2010-02-24 23:29:26.834494294 +0200
@@ -371,15 +371,15 @@
 # in /lib and /etc.
 case "$prefix" in
 /usr | /usr/)
-  # 64-bit libraries on bi-arch platforms go in /lib64 instead of /lib.
+  # 64-bit libraries on bi-arch platforms go in /lib instead of /lib.
   # Allow earlier configure scripts to handle libc_cv_slibdir, libdir,
   # and libc_cv_localedir.
   test -n "$libc_cv_slibdir" || \
   case $machine in
   sparc/sparc64 | x86_64 | powerpc/powerpc64 | s390/s390-64)
-    libc_cv_slibdir="/lib64"
+    libc_cv_slibdir="/lib"
     if test "$libdir" = '${exec_prefix}/lib'; then
-      libdir='${exec_prefix}/lib64';
+      libdir='${exec_prefix}/lib';
       # Locale data can be shared between 32bit and 64bit libraries
       libc_cv_localedir='${exec_prefix}/lib/locale'
     fi
diff -Nur glibc-20081113T2206-old/sysdeps/unix/sysv/linux/configure.in glibc-20081113T2206/sysdeps/unix/sysv/linux/configure.in
--- glibc-20081113T2206-old/sysdeps/unix/sysv/linux/configure.in	2010-02-24 23:27:44.902488874 +0200
+++ glibc-20081113T2206/sysdeps/unix/sysv/linux/configure.in	2010-02-24 23:29:12.856498186 +0200
@@ -159,15 +159,15 @@
 # in /lib and /etc.
 case "$prefix" in
 /usr | /usr/)
-  # 64-bit libraries on bi-arch platforms go in /lib64 instead of /lib.
+  # 64-bit libraries on bi-arch platforms go in /lib instead of /lib.
   # Allow earlier configure scripts to handle libc_cv_slibdir, libdir,
   # and libc_cv_localedir.
   test -n "$libc_cv_slibdir" || \
   case $machine in
   sparc/sparc64 | x86_64 | powerpc/powerpc64 | s390/s390-64)
-    libc_cv_slibdir="/lib64"
+    libc_cv_slibdir="/lib"
     if test "$libdir" = '${exec_prefix}/lib'; then
-      libdir='${exec_prefix}/lib64';
+      libdir='${exec_prefix}/lib';
       # Locale data can be shared between 32bit and 64bit libraries
       libc_cv_localedir='${exec_prefix}/lib/locale'
     fi
