--- /dev/null	2009-01-03 15:05:17.165255704 +0200
+++ sox.pc	2009-01-03 23:25:53.000000000 +0200
@@ -0,0 +1,12 @@
+prefix=/usr
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: SoX
+Description: Audio file format and effects library
+Version: 14.2.0
+URL: http://sox.sourceforge.net
+Libs: -L${libdir} -lsox -lsfx
+Libs.private: -lm 
+Cflags: -I${includedir}/sox
