diff -Nuar gnome-doc-utils-0.10.2.orig/doc/gnome-doc-make/Makefile.in gnome-doc-utils-0.10.2/doc/gnome-doc-make/Makefile.in
--- gnome-doc-utils-0.10.2.orig/doc/gnome-doc-make/Makefile.in	2007-04-09 21:33:36.000000000 +0300
+++ gnome-doc-utils-0.10.2/doc/gnome-doc-make/Makefile.in	2007-04-09 21:34:31.000000000 +0300
@@ -807,7 +807,6 @@
 	done
 	@if test "x$(_ENABLE_SK)" = "xtrue"; then \
 	  echo "scrollkeeper-update -p $(DESTDIR)$(_sklocalstatedir) -o $(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
-	  scrollkeeper-update -p "$(DESTDIR)$(_sklocalstatedir)" -o "$(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
 	fi;
 
 install-doc-dsk:
diff -Nuar gnome-doc-utils-0.10.2.orig/doc/xslt/Makefile.in gnome-doc-utils-0.10.2/doc/xslt/Makefile.in
--- gnome-doc-utils-0.10.2.orig/doc/xslt/Makefile.in	2007-04-09 21:33:35.000000000 +0300
+++ gnome-doc-utils-0.10.2/doc/xslt/Makefile.in	2007-04-09 21:34:15.000000000 +0300
@@ -832,7 +832,6 @@
 	done
 	@if test "x$(_ENABLE_SK)" = "xtrue"; then \
 	  echo "scrollkeeper-update -p $(DESTDIR)$(_sklocalstatedir) -o $(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
-	  scrollkeeper-update -p "$(DESTDIR)$(_sklocalstatedir)" -o "$(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
 	fi;
 
 install-doc-dsk:
diff -Nuar gnome-doc-utils-0.10.2.orig/tools/gnome-doc-utils.make gnome-doc-utils-0.10.2/tools/gnome-doc-utils.make
--- gnome-doc-utils-0.10.2.orig/tools/gnome-doc-utils.make	2007-04-09 21:33:36.000000000 +0300
+++ gnome-doc-utils-0.10.2/tools/gnome-doc-utils.make	2007-04-09 21:34:51.000000000 +0300
@@ -527,7 +527,6 @@
 	done
 	@if test "x$(_ENABLE_SK)" = "xtrue"; then \
 	  echo "scrollkeeper-update -p $(DESTDIR)$(_sklocalstatedir) -o $(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
-	  scrollkeeper-update -p "$(DESTDIR)$(_sklocalstatedir)" -o "$(DESTDIR)$(OMF_DIR)/$(DOC_MODULE)"; \
 	fi;
 
 install-doc-dsk:
