diff -ur gnome-doc-utils-0.18.1.orig/tools/Makefile.am gnome-doc-utils-0.18.1/tools/Makefile.am
--- gnome-doc-utils-0.18.1.orig/tools/Makefile.am	2009-11-18 05:38:23.000000000 +0200
+++ gnome-doc-utils-0.18.1/tools/Makefile.am	2010-01-22 20:25:45.393729058 +0200
@@ -10,7 +10,7 @@
 
 endif
 
-pkgconfigdir = $(datadir)/pkgconfig
+pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = gnome-doc-utils.pc
 
 EXTRA_DIST =			\
