Index: src/service-manager.desktop
===================================================================
--- src/service-manager.desktop	(revision 21182)
+++ src/service-manager.desktop	(revision 21183)
@@ -3,7 +3,7 @@
 Name[tr]=Servis Yöneticisi
 Name[es]=Administrador de Servicios
 Name[nl]=Dienstbeheer
-Name[de]=Dienst-Manager
+Name[de]=Dienste-Manager
 Name[pt_BR]=Gerenciador de Serviços
 Name[fr]=Gestionnaire de Services
 Name[it]=Gestore dei servizi
