diff -ur juk.orig/jukui.rc juk/jukui.rc
--- juk.orig/jukui.rc	2005-09-10 11:20:03.000000000 +0300
+++ juk/jukui.rc	2006-01-06 22:22:20.000000000 +0200
@@ -77,25 +77,7 @@
 	</Menu>
 </MenuBar>
 
-<ToolBar name="mainToolBar" hidden="true" noMerge="1"><text>Main Toolbar</text>
-	<Action name="file_new"/>
-	<Action name="file_open"/>
-	<Action name="file_save"/>
-
-	<Separator lineSeparator="true"/>
-
-	<Action name="edit_cut"/>
-	<Action name="edit_copy"/>
-	<Action name="edit_paste"/>
-
-	<Separator lineSeparator="true"/>
-
-	<Action name="showSearch"/>
-	<Action name="showEditor"/>
-	<Action name="showNowPlaying"/>
-</ToolBar>
-
-<ToolBar name="playToolBar" noMerge="1"><text>Play Toolbar</text>
+<ToolBar name="mainToolBar" noMerge="1"><text>Play Toolbar</text>
 	<Action name="play"/>
 	<Action name="pause"/>
 	<Action name="stop"/>
