diff -ur kdemultimedia-3.5.0.orig/kscd/kcompactdisc.h kdemultimedia-3.5.0/kscd/kcompactdisc.h
--- kdemultimedia-3.5.0.orig/kscd/kcompactdisc.h	2005-10-10 17:57:15.000000000 +0300
+++ kdemultimedia-3.5.0/kscd/kcompactdisc.h	2006-01-18 10:59:27.000000000 +0200
@@ -87,7 +87,7 @@
      */
     bool setDevice(
         const QString &device = defaultDevice,
-        unsigned volume = 50,
+        unsigned volume = 75,
         bool digitalPlayback = true,
         const QString &audioSystem = QString::null,
         const QString &audioDevice = QString::null);
