Index: kdemultimedia-3.5.9/kmix/kmixdockwidget.cpp
===================================================================
--- kdemultimedia-3.5.9.orig/kmix/kmixdockwidget.cpp
+++ kdemultimedia-3.5.9/kmix/kmixdockwidget.cpp
@@ -33,6 +33,7 @@
 #include <kiconloader.h>
 #include <kdebug.h>
 #include <kwin.h>
+#include <kprocess.h>
 
 #include <qapplication.h>
 #include <qcursor.h>
@@ -94,6 +95,11 @@ void KMixDockWidget::createActions()
   if (a2) a2->plug( popupMenu );
   }
 
+  (void)new KAction( i18n( "Pulseaudio Config..." ), 0, this, SLOT( runPulse()),
+  actionCollection(), "pulse_config" );
+  KAction *a3 = actionCollection()->action( "pulse_config" );
+  if ( a3 ) a3->plug( popupMenu );
+
    // Setup volume preview
   if ( _playBeepOnVolumeChange ) {
     _audioPlayer = new KAudioPlayer("KDE_Beep_Digital_1.ogg");
@@ -364,6 +370,14 @@ KMixDockWidget::dockMute()
 }
 
 void
+KMixDockWidget::runPulse()
+{
+    KProcess proc;
+    proc << "/usr/bin/pavucontrol";
+    proc.start(KProcess::DontCare);
+}
+
+void
 KMixDockWidget::contextMenuAboutToShow( KPopupMenu* /* menu */ )
 {
     KAction* showAction = actionCollection()->action("minimizeRestore");
Index: kdemultimedia-3.5.9/kmix/kmixdockwidget.h
===================================================================
--- kdemultimedia-3.5.9.orig/kmix/kmixdockwidget.h
+++ kdemultimedia-3.5.9/kmix/kmixdockwidget.h
@@ -77,6 +77,7 @@ class KMixDockWidget : public KSystemTra
  private slots:
    void dockMute();
    void selectMaster();
+   void runPulse();
    void handleNewMaster(int soundcard_id, QString& channel_id);
 };
 
