Index: plug-ins/pygimp/Makefile.in
===================================================================
--- plug-ins/pygimp/Makefile.in.orig
+++ plug-ins/pygimp/Makefile.in
@@ -747,13 +747,6 @@ install-pygimpPYTHON: $(pygimp_PYTHON)
 	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(pygimpdir)'"; \
 	  $(INSTALL_DATA) $$files "$(DESTDIR)$(pygimpdir)" || exit $$?; \
 	done || exit $$?; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pygimpdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygimpdir)" $$dlist; \
-	  fi; \
-	else :; fi
 
 uninstall-pygimpPYTHON:
 	@$(NORMAL_UNINSTALL)
