Index: linux-2.6.34/drivers/staging/rt2860/rt_linux.c
===================================================================
--- linux-2.6.34.orig/drivers/staging/rt2860/rt_linux.c
+++ linux-2.6.34/drivers/staging/rt2860/rt_linux.c
@@ -30,7 +30,7 @@
 #include <linux/slab.h>
 #include "rt_config.h"
 
-unsigned long RTDebugLevel = RT_DEBUG_ERROR;
+unsigned long RTDebugLevel = RT_DEBUG_OFF;
 
 /* for wireless system event message */
 char const *pWirelessSysEventText[IW_SYS_EVENT_TYPE_NUM] = {
Index: linux-2.6.34/drivers/staging/rt2860/rt_linux.h
===================================================================
--- linux-2.6.34.orig/drivers/staging/rt2860/rt_linux.h
+++ linux-2.6.34/drivers/staging/rt2860/rt_linux.h
@@ -425,16 +425,9 @@ do{                                   \
 
 #define DBGPRINT(Level, Fmt)    DBGPRINT_RAW(Level, Fmt)
 
-#define DBGPRINT_ERR(Fmt)           \
-{                                   \
-    printk("ERROR! ");          \
-    printk Fmt;                  \
-}
+#define DBGPRINT_ERR(Fmt)
 
-#define DBGPRINT_S(Status, Fmt)		\
-{									\
-	printk Fmt;					\
-}
+#define DBGPRINT_S(Status, Fmt)
 
 #else
 #define DBGPRINT(Level, Fmt)
