--- src/kdedistutils.py	2008-01-01 01:39:05.000000000 +0200
+++ src/kdedistutils.py	2008-01-01 01:39:35.000000000 +0200
@@ -2064,8 +2064,7 @@
 
 def ask_kde_config(question):
     # Look for the kde-config program
-    kdeconfig = find_executable("kde-config", os.environ['PATH'] + os.pathsep + \
-        os.pathsep.join(['/bin','/usr/bin','/opt/kde3/bin','/opt/kde/bin','/usr/local/bin']))
+    kdeconfig = "/usr/kde/3.5/bin/kde-config"
     if kdeconfig!=None:
         # Ask the kde-config program for the 
         fhandle = os.popen(kdeconfig+' ' + question,'r')
