diff -ur pykdeextensions-0.4.0.orig/src/kdedistutils.py pykdeextensions-0.4.0/src/kdedistutils.py
--- pykdeextensions-0.4.0.orig/src/kdedistutils.py	2006-01-20 20:25:03.000000000 +0200
+++ pykdeextensions-0.4.0/src/kdedistutils.py	2006-01-20 20:27:04.000000000 +0200
@@ -513,8 +513,8 @@
         self.build_dir = None
         self.python_inc_dir = None
         self.python_dir = None
-        self.kde_inc_dir = None
-        self.kde_lib_dir = None
+        self.kde_inc_dir = '/usr/kde/3.5/include'
+        self.kde_lib_dir = '/usr/kde/3.5/lib'
         self.kde_kcm_lib_dir = None
         self.qt_inc_dir = None
         self.qt_lib_dir = None
@@ -1421,8 +1421,8 @@
         self.build_dir = None
         self.python_inc_dir = None
         self.python_dir = None
-        self.kde_inc_dir = None
-        self.kde_lib_dir = None
+        self.kde_inc_dir = '/usr/kde/3.5/include'
+        self.kde_lib_dir = '/usr/kde/3.5/lib'
         self.kde_kioslave_lib_dir = None
         self.kde_protocol_dir = None
         self.qt_inc_dir = None
