diff -ur xorg-server-1.4.0.90.orig/hw/xfree86/common/xf86.h xorg-server-1.4.2/hw/xfree86/common/xf86.h
--- xorg-server-1.4.0.90.orig/hw/xfree86/common/xf86.h	2007-02-18 03:31:36.000000000 +0200
+++ xorg-server-1.4.2/hw/xfree86/common/xf86.h	2007-03-14 23:16:31.444459626 +0200
@@ -86,7 +86,7 @@
 			(((p) == Pix24Use32) ? 32 : 0))
 
 /* variables for debugging */
-#ifdef BUILDDEBUG
+#ifdef DEBUG
 extern char* xf86p8bit[];
 extern CARD32 xf86DummyVar1;
 extern CARD32 xf86DummyVar2;
@@ -367,7 +367,7 @@
 void xf86DeleteModuleInfo(int idx);
 
 /* xf86Debug.c */
-#ifdef BUILDDEBUG
+#ifdef DEBUG
  void xf86Break1(void);
 void xf86Break2(void);
 void xf86Break3(void);
diff -ur xorg-server-1.4.0.90.orig/hw/xfree86/common/xf86str.h xorg-server-1.4.2/hw/xfree86/common/xf86str.h
--- xorg-server-1.4.0.90.orig/hw/xfree86/common/xf86str.h	2006-11-18 05:32:36.000000000 +0200
+++ xorg-server-1.4.2/hw/xfree86/common/xf86str.h	2007-03-14 23:16:35.168891441 +0200
@@ -1099,7 +1099,7 @@
 #define CLK_REG_RESTORE		-2
 
 /* xf86Debug.c */
-#ifdef BUILDDEBUG
+#ifdef DEBUG
 typedef struct {
     long sec;
     long usec;
diff -ur xorg-server-1.4.0.90.orig/hw/xfree86/loader/xf86sym.c xorg-server-1.4.2/hw/xfree86/loader/xf86sym.c
--- xorg-server-1.4.0.90.orig/hw/xfree86/loader/xf86sym.c	2007-02-18 03:31:36.000000000 +0200
+++ xorg-server-1.4.2/hw/xfree86/loader/xf86sym.c	2007-03-14 23:16:42.005848420 +0200
@@ -725,7 +725,7 @@
     SYMFUNC(DPMSSupported)
 #endif
 /* xf86Debug.c */
-#ifdef BUILDDEBUG
+#ifdef DEBUG
     SYMFUNC(xf86Break1)
     SYMFUNC(xf86Break2)
     SYMFUNC(xf86Break3)
@@ -1146,7 +1146,7 @@
     SYMVAR(byte_reversed)
     SYMVAR(xf86inSuspend)
     /* debugging variables */
-#ifdef BUILDDEBUG
+#ifdef DEBUG
     SYMVAR(xf86p8bit)
     SYMVAR(xf86DummyVar1)
     SYMVAR(xf86DummyVar2)
