diff --git a/extras/modem-modeswitch/61-mobile-action.rules b/extras/modem-modeswitch/61-mobile-action.rules
index 9116b25..cba3dba 100644
--- a/extras/modem-modeswitch/61-mobile-action.rules
+++ b/extras/modem-modeswitch/61-mobile-action.rules
@@ -8,7 +8,8 @@ ATTRS{idVendor}=="0df7", ATTRS{idProduct}=="0800", RUN+="modem-modeswitch -v 0x%
 
 # modem-modeswitch does not work with these devices, the fake CD-ROM needs to be ejected
 
-# ZTE MF6xx
-ATTRS{idVendor}=="19d2", ATTRS{idProduct}=="2000", RUN+="/usr/bin/eject %k"
-
 LABEL="mobile_action_end"
+
+# ZTE MF6xx needs the fake CD-ROM ejected
+ENV{ID_CDROM}=="1", ENV{ID_VENDOR_ID}=="19d2", ENV{ID_MODEL_ID}=="2000", RUN+="/usr/bin/eject %k"
+
