Index: udev-141/rules/rules.d/50-udev-default.rules
===================================================================
--- udev-141.orig/rules/rules.d/50-udev-default.rules
+++ udev-141/rules/rules.d/50-udev-default.rules
@@ -65,7 +65,7 @@ SUBSYSTEM=="block", GROUP="disk"
 
 # floppy
 KERNEL=="fd[0-9]", GROUP="floppy"
-KERNEL=="fd[0-9]", ACTION=="add", ATTRS{cmos}=="?*", RUN+="create_floppy_devices -c -t $attr{cmos} -m %M -M 0640 -G floppy $root/%k"
+KERNEL=="fd[0-9]", ACTION=="add", ATTRS{cmos}=="?*", RUN+="create_floppy_devices -c -t $attr{cmos} -m %M -M 0640 -G pnp $root/%k"
 KERNEL=="hd*", SUBSYSTEMS=="ide", ATTRS{media}=="floppy", OPTIONS+="all_partitions"
 
 # cdrom
