diff -ur xfsprogs-2.8.18.orig/install-sh xfsprogs-2.8.18/install-sh
--- xfsprogs-2.8.18.orig/install-sh	2006-12-13 06:57:22.000000000 +0200
+++ xfsprogs-2.8.18/install-sh	2007-01-09 20:33:40.279237187 +0200
@@ -250,7 +250,10 @@
 	;;
     old_lib)
 	install_name=$target/$old_library
-	$CP $old_library $install_name
+	for fold_library in $library_names
+	do
+	    $CP $fold_library $install_name
+    done
 	status=$?
 	$MANIFEST f $FILEMODE $OWNER $GROUP $HERE/$old_library ${install_name#$DIST_ROOT}
 	;;
