diff -up kdevelop-3.5.4/lib/antlr/antlr/CharScanner.hpp.orig kdevelop-3.5.4/lib/antlr/antlr/CharScanner.hpp
--- kdevelop-3.5.4/lib/antlr/antlr/CharScanner.hpp.orig	2009-03-02 13:50:12.000000000 +0100
+++ kdevelop-3.5.4/lib/antlr/antlr/CharScanner.hpp	2009-03-02 13:51:48.000000000 +0100
@@ -12,6 +12,8 @@
 
 #include <map>
 
+#include <cstdio>
+
 #ifdef HAS_NOT_CCTYPE_H
 #include <ctype.h>
 #else
diff -up kdevelop-3.5.4/lib/antlr/src/String.cpp.orig kdevelop-3.5.4/lib/antlr/src/String.cpp
