Index: yakuake-2.8.1/yakuake/src/yakuake.kcfg
===================================================================
--- yakuake-2.8.1.orig/yakuake/src/yakuake.kcfg
+++ yakuake-2.8.1/yakuake/src/yakuake.kcfg
@@ -22,7 +22,7 @@
       <whatsthis></whatsthis>
     </entry>
     <entry key="steps" type="Int">
-      <default>20</default>
+      <default>0</default>
       <label></label>
       <whatsthis></whatsthis>
     </entry>
