--- linux-atm-2.5.0/src/config/Makefile.in.org	2008-03-16 23:40:36.000000000 +0200
+++ linux-atm-2.5.0/src/config/Makefile.in	2008-03-16 23:40:49.000000000 +0200
@@ -338,11 +338,11 @@
 	uninstall uninstall-am uninstall-local
 
 
-install-exec-local:
-	-cp hosts.atm /etc
+#install-exec-local:
+#	-cp hosts.atm /etc
 
-uninstall-local:
-	-rm /etc/hosts.atm
+#uninstall-local:
+#	-rm /etc/hosts.atm
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
