Index: wine-1.1.44/tools/wine.desktop
===================================================================
--- wine-1.1.44.orig/tools/wine.desktop
+++ wine-1.1.44/tools/wine.desktop
@@ -19,6 +19,7 @@ Name[nn]=Wine Programlaster for Windowsa
 Name[sr]=Wine - дизач Windows програма
 Name[sr@latin]=Wine - dizač Windows programa
 Name[hr]=Wine - dizač Windows programa
+Name[tr]=Wine Windows Programı Yükleyicisi
 Exec=wine start /unix %f
 MimeType=application/x-ms-dos-executable;application/x-msi;application/x-win-lnk;
 Icon=wine
