diff -Nur qemu-0.10.5-old/Makefile.target qemu-0.10.5/Makefile.target
--- qemu-0.10.5-old/Makefile.target	2009-05-26 10:48:16.614652393 +0300
+++ qemu-0.10.5/Makefile.target	2009-05-26 10:48:54.187527262 +0300
@@ -78,7 +78,7 @@
 cc-option = $(shell if $(CC) $(1) $(2) -S -o /dev/null -xc /dev/null \
               > /dev/null 2>&1; then echo "$(2)"; else echo "$(3)"; fi ;)
 
-HELPER_CFLAGS=
+HELPER_CFLAGS+=
 
 ifeq ($(ARCH),i386)
 HELPER_CFLAGS+=-fomit-frame-pointer
