diff -Nur qemu-0.10.5-old/Makefile qemu-0.10.5/Makefile
--- qemu-0.10.5-old/Makefile	2009-05-26 10:48:16.623652472 +0300
+++ qemu-0.10.5/Makefile	2009-05-26 10:49:24.844527372 +0300
@@ -232,7 +232,7 @@
 install: all $(if $(BUILD_DOCS),install-doc)
 	mkdir -p "$(DESTDIR)$(bindir)"
 ifneq ($(TOOLS),)
-	$(INSTALL) -m 755 -s $(TOOLS) "$(DESTDIR)$(bindir)"
+	$(INSTALL) -m 755 $(TOOLS) "$(DESTDIR)$(bindir)"
 endif
 ifneq ($(BLOBS),)
 	mkdir -p "$(DESTDIR)$(datadir)"
diff -Nur qemu-0.10.5-old/Makefile.target qemu-0.10.5/Makefile.target
--- qemu-0.10.5-old/Makefile.target	2009-05-26 10:48:16.614652393 +0300
+++ qemu-0.10.5/Makefile.target	2009-05-26 10:49:24.850402452 +0300
@@ -745,7 +745,7 @@
 
 install: all
 ifneq ($(PROGS),)
-	$(INSTALL) -m 755 -s $(PROGS) "$(DESTDIR)$(bindir)"
+	$(INSTALL) -m 755 $(PROGS) "$(DESTDIR)$(bindir)"
 endif
 
 # Include automatically generated dependency files
