diff -p -up qemu-0.10.0/hw/pc.c.kernel-option-vga qemu-0.10.0/hw/pc.c
--- qemu-0.10.0/hw/pc.c.kernel-option-vga	2009-03-04 23:54:45.000000000 +0100
+++ qemu-0.10.0/hw/pc.c	2009-03-05 14:50:06.000000000 +0100
@@ -552,6 +552,7 @@ static void load_linux(uint8_t *option_r
     uint8_t header[1024];
     target_phys_addr_t real_addr, prot_addr, cmdline_addr, initrd_addr;
     FILE *f, *fi;
+    char *vmode;
 
     /* Align to 16 bytes as a paranoia measure */
     cmdline_size = (strlen(kernel_cmdline)+16) & ~15;
@@ -620,6 +621,26 @@ static void load_linux(uint8_t *option_r
 	stw_p(header+0x22, cmdline_addr-real_addr);
     }
 
+
+    /* handle vga= parameter */
+    vmode = strstr(kernel_cmdline, "vga=");
+    if (vmode) {
+        char *space;
+        unsigned int video_mode;
+        /* skip "vga=" */
+        vmode += 4;
+        if (!strncmp(vmode, "normal", 6)) {
+            video_mode = 0xffff;
+        } else if (!strncmp(vmode, "ext", 3)) {
+            video_mode = 0xfffe;
+        } else if (!strncmp(vmode, "ask", 3)) {
+            video_mode = 0xfffd;
+        } else {
+            video_mode = strtol(vmode, NULL, 0);
+        }
+        stw_p(header+0x1fa, video_mode);
+    }
+
     /* loader type */
     /* High nybble = B reserved for Qemu; low nybble is revision number.
        If this code is substantially changed, you may want to consider
