Index: glib-2.26.1/gio/Makefile.am
===================================================================
--- glib-2.26.1.orig/gio/Makefile.am
+++ glib-2.26.1/gio/Makefile.am
@@ -264,7 +264,7 @@ giowin32include_HEADERS = \
 
 endif
 
-SUBDIRS += tests
+SUBDIRS +=
 
 libgio_2_0_la_SOURCES =		\
 	gappinfo.c 		\
Index: glib-2.26.1/Makefile.am
===================================================================
--- glib-2.26.1.orig/Makefile.am
+++ glib-2.26.1/Makefile.am
@@ -3,7 +3,7 @@ include $(top_srcdir)/Makefile.decl
 
 ACLOCAL_AMFLAGS = -I m4macros ${ACLOCAL_FLAGS}
 
-SUBDIRS = . m4macros glib gmodule gthread gobject gio tests po docs
+SUBDIRS = . m4macros glib gmodule gthread gobject gio po docs
 DIST_SUBDIRS = $(SUBDIRS) build
 
 bin_SCRIPTS = glib-gettextize
Index: glib-2.26.1/gio/Makefile.in
===================================================================
--- glib-2.26.1.orig/gio/Makefile.in
+++ glib-2.26.1/gio/Makefile.in
@@ -334,7 +334,7 @@ ETAGS = etags
 CTAGS = ctags
 am__tty_colors = \
 red=; grn=; lgn=; blu=; std=
-DIST_SUBDIRS = libasyncns xdgmime inotify fen win32 . fam tests
+DIST_SUBDIRS = libasyncns xdgmime inotify fen win32 . fam
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 am__relativize = \
   dir0=`pwd`; \
@@ -603,7 +603,7 @@ EXTRA_DIST = strinfo.c gio-marshal.list
 TEST_PROGS = 
 NULL = 
 SUBDIRS = $(am__append_1) $(am__append_3) $(am__append_7) \
-	$(am__append_10) . $(am__append_13) tests
+	$(am__append_10) . $(am__append_13)
 @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@noinst_DATA = gio-2.0.lib
 @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@install_ms_lib_cmd = $(INSTALL) gio-2.0.lib $(DESTDIR)$(libdir)
 @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@uninstall_ms_lib_cmd = -rm $(DESTDIR)$(libdir)/gio-2.0.lib
Index: glib-2.26.1/Makefile.in
===================================================================
--- glib-2.26.1.orig/Makefile.in
+++ glib-2.26.1/Makefile.in
@@ -419,7 +419,7 @@ EXTRA_DIST = ChangeLog.pre-2-20 ChangeLo
 	gio-2.0-uninstalled.pc.in gio-unix-2.0-uninstalled.pc.in
 TEST_PROGS = 
 ACLOCAL_AMFLAGS = -I m4macros ${ACLOCAL_FLAGS}
-SUBDIRS = . m4macros glib gmodule gthread gobject gio tests po docs
+SUBDIRS = . m4macros glib gmodule gthread gobject gio po docs
 DIST_SUBDIRS = $(SUBDIRS) build
 bin_SCRIPTS = glib-gettextize
 AM_CPPFLAGS = \
