--- darkice-0.19/configure.old	2008-07-21 01:41:00.000000000 -0400
+++ darkice-0.19/configure	2008-07-21 01:41:40.000000000 -0400
@@ -6662,7 +6662,7 @@
 
 
   OGG_LIB_LOC=
-    for FILE in libogg.a; do
+    for FILE in libogg.a libogg.so; do
     for DIR in ${CONFIG_VORBIS_PREFIX} ${CONFIG_VORBIS_PREFIX}/lib64 ${CONFIG_VORBIS_PREFIX}/lib ; do
             if test ".OGG_LIB_LOC"="." && test -f "$DIR/$FILE"; then
         OGG_LIB_LOC=$DIR
@@ -6688,7 +6688,7 @@
 
 
   VORBIS_LIB_LOC=
-    for FILE in libvorbis.a; do
+    for FILE in libvorbis.a libvorbis.so; do
     for DIR in ${CONFIG_VORBIS_PREFIX} ${CONFIG_VORBIS_PREFIX}/lib64 ${CONFIG_VORBIS_PREFIX}/lib ; do
             if test ".VORBIS_LIB_LOC"="." && test -f "$DIR/$FILE"; then
         VORBIS_LIB_LOC=$DIR
@@ -6714,7 +6714,7 @@
 
 
   VORBISENC_LIB_LOC=
-    for FILE in libvorbisenc.a; do
+    for FILE in libvorbisenc.a libvorbisenc.so; do
     for DIR in ${CONFIG_VORBIS_PREFIX} ${CONFIG_VORBIS_PREFIX}/lib64 ${CONFIG_VORBIS_PREFIX}/lib ; do
             if test ".VORBISENC_LIB_LOC"="." && test -f "$DIR/$FILE"; then
         VORBISENC_LIB_LOC=$DIR
