Index: libvdpau-0.1_pre20090325/src/Makefile.am
===================================================================
--- libvdpau-0.1_pre20090325.orig/src/Makefile.am
+++ libvdpau-0.1_pre20090325/src/Makefile.am
@@ -7,7 +7,7 @@ lib_LTLIBRARIES = libvdpau.la
 libvdpau_la_SOURCES = \
     vdpau_wrapper.c
 
-libvdpau_la_LDFLAGS = -version-info 1:0:0 -no-undefined
+libvdpau_la_LDFLAGS = -version-info 1:0:0 -no-undefined -ldl
 
 libvdpauincludedir = $(includedir)/vdpau
 libvdpauinclude_HEADERS = \
Index: libvdpau-0.1_pre20090325/trace/Makefile.am
===================================================================
--- libvdpau-0.1_pre20090325.orig/trace/Makefile.am
+++ libvdpau-0.1_pre20090325/trace/Makefile.am
@@ -7,7 +7,7 @@ lib_LTLIBRARIES = libvdpau_trace.la
 libvdpau_trace_la_SOURCES = \
     vdpau_trace.cpp
 
-libvdpau_trace_la_LDFLAGS = -avoid-version -module -no-undefined
+libvdpau_trace_la_LDFLAGS = -avoid-version -module -no-undefined -ldl
 
 libvdpau_traceincludedir = $(includedir)/vdpau
 libvdpau_traceinclude_HEADERS = \
