diff -Nur vlc-1.0.2-old/modules/codec/x264.c vlc-1.0.2/modules/codec/x264.c
--- vlc-1.0.2-old/modules/codec/x264.c	2009-10-20 20:09:56.881214757 +0300
+++ vlc-1.0.2/modules/codec/x264.c	2009-10-20 20:10:21.258919184 +0300
@@ -992,7 +992,7 @@
 
 #if X264_BUILD >= 22
     var_Get( p_enc, SOUT_CFG_PREFIX "bpyramid", &val );
-    p_sys->param.b_bframe_pyramid = val.b_bool;
+    p_sys->param.i_bframe_pyramid = val.b_bool;
 #endif
 
     var_Get( p_enc, SOUT_CFG_PREFIX "ref", &val );
