Index: src/plugin_setup.cpp
===================================================================
--- src/plugin_setup.cpp	(revision 344)
+++ src/plugin_setup.cpp	(working copy)
@@ -65,6 +65,10 @@
         for (i = 0; i < argc; i++) {
             printf("ARG: %s = %s\n", argn[i], argv[i]);
 
+            if (argn[i] == NULL) {
+                continue;
+            }
+
             if (g_ascii_strcasecmp(argn[i], "name") == 0) {
                 instance->name = g_strdup(argv[i]);
             }
