Index: module-init-tools-3.9/modinfo.c
===================================================================
--- module-init-tools-3.9.orig/modinfo.c
+++ module-init-tools-3.9/modinfo.c
@@ -242,7 +242,7 @@ static struct elf_file *grab_module(cons
 			module = grab_elf_file(filename);
 			if (!module)
 				error("modinfo: could not open %s: %s\n",
-					*filename, strerror(errno));
+					filename, strerror(errno));
 			free(filename);
 			return module;
 		}
