--- squashfs-tools/Makefile.orig	2010-12-31 15:36:47.000000000 -0600
+++ squashfs-tools/Makefile	2011-01-01 10:58:33.000000000 -0600
@@ -100,8 +100,8 @@
 UNSQUASHFS_OBJS = unsquashfs.o unsquash-1.o unsquash-2.o unsquash-3.o \
 	unsquash-4.o swap.o compressor.o
 
-CFLAGS = $(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
-	-D_GNU_SOURCE -DCOMP_DEFAULT=\"$(COMP_DEFAULT)\"  -O2 -Wall
+CFLAGS = $(RPM_OPT_FLAGS) $(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
+	-D_GNU_SOURCE -DCOMP_DEFAULT=\"$(COMP_DEFAULT)\"
 
 LIBS = -lpthread -lm
 ifdef GZIP_SUPPORT
