diff -Nur DirectFB-1.2.8-old/configure DirectFB-1.2.8/configure
--- DirectFB-1.2.8-old/configure	2009-04-23 15:39:51.000000000 +0300
+++ DirectFB-1.2.8/configure	2009-04-23 15:40:11.000000000 +0300
@@ -22649,7 +22649,7 @@
 
 if test "$enable_x11" = "yes"; then
   CFLAGS_saved="$CFLAGS"
-  CFLAGS="$CFLAGS -I/usr/X11R6/include"
+  CFLAGS="$CFLAGS "
   if test "${ac_cv_header_X11_X_h+set}" = set; then
   { echo "$as_me:$LINENO: checking for X11/X.h" >&5
 echo $ECHO_N "checking for X11/X.h... $ECHO_C" >&6; }
@@ -22790,8 +22790,8 @@
 echo "$as_me: WARNING:
 *** no X11/X.h found -- building without X11 support." >&2;}
   else
-    X11_LIBS="-L/usr/X11R6/lib -lX11 -lXext"
-    X11_CFLAGS="-I/usr/X11R6/include"
+    X11_LIBS=" -lX11 -lXext"
+    X11_CFLAGS=""
   fi
 fi
 
