Index: LVM2.2.02.67/daemons/dmeventd/Makefile.in
===================================================================
--- LVM2.2.02.67.orig/daemons/dmeventd/Makefile.in
+++ LVM2.2.02.67/daemons/dmeventd/Makefile.in
@@ -53,7 +53,7 @@ include $(top_builddir)/make.tmpl
 all: device-mapper
 device-mapper: $(TARGETS)
 
-LIBS += -ldevmapper
+LIBS += -ldevmapper -ldl
 LVMLIBS += -ldevmapper-event $(PTHREAD_LIBS)
 
 dmeventd: $(LIB_SHARED) dmeventd.o
Index: LVM2.2.02.67/tools/Makefile.in
===================================================================
--- LVM2.2.02.67.orig/tools/Makefile.in
+++ LVM2.2.02.67/tools/Makefile.in
@@ -146,7 +146,7 @@ liblvm2cmd-static.a: $(top_builddir)/lib
 
 liblvm2cmd.$(LIB_SUFFIX): liblvm2cmd.a $(LDDEPS)
 	$(CC) -shared -Wl,-soname,$@.$(LIB_VERSION) \
-	$(CFLAGS) $(CLDFLAGS) -o $@ $(LVMLIBS) $(LIBS) \
+	$(CFLAGS) $(CLDFLAGS) $(LVMLIBS) $(LIBS) -o $@ \
 		@CLDWHOLEARCHIVE@ liblvm2cmd.a @CLDNOWHOLEARCHIVE@
 
 liblvm2cmd.$(LIB_SUFFIX).$(LIB_VERSION): liblvm2cmd.$(LIB_SUFFIX)
