Index: LVM2.2.02.56/daemons/dmeventd/Makefile.in
===================================================================
--- LVM2.2.02.56.orig/daemons/dmeventd/Makefile.in
+++ LVM2.2.02.56/daemons/dmeventd/Makefile.in
@@ -53,7 +53,7 @@ include ../../make.tmpl
 all: device-mapper
 device-mapper: $(TARGETS)
 
-LIBS += -ldl
+LIBS += -ldl -ldevmapper
 LVMLIBS += -ldevmapper-event -lpthread -ldevmapper
 
 $(VERSIONED_SHLIB): $(LIB_SHARED)
Index: LVM2.2.02.56/tools/Makefile.in
===================================================================
--- LVM2.2.02.56.orig/tools/Makefile.in
+++ LVM2.2.02.56/tools/Makefile.in
@@ -139,6 +139,9 @@ liblvm2cmd-static.a: $(top_builddir)/lib
 	$(AR) rs $@ $(OBJECTS) lvmcmdlib.o lvm2cmd-static.o
 
 liblvm2cmd.$(LIB_SUFFIX): liblvm2cmd.a $(LDDEPS)
+	$(CC) -shared -Wl,-soname,$(notdir $@).$(LIB_VERSION) \
+	$(CFLAGS) $(CLDFLAGS) $(LIBS) -o $@ \
+	@CLDWHOLEARCHIVE@ $< @CLDNOWHOLEARCHIVE@ $(LVMLIBS)
 
 .commands: commands.h cmdnames.h Makefile
 	$(CC) -E -P $(srcdir)/cmdnames.h 2> /dev/null | \
