Index: kdeartwork-3.5.10/kwin-styles/glow/Makefile.in
===================================================================
--- kdeartwork-3.5.10.orig/kwin-styles/glow/Makefile.in
+++ kdeartwork-3.5.10/kwin-styles/glow/Makefile.in
@@ -379,7 +379,7 @@ SUBDIRS = . config themes
 kde_module_LTLIBRARIES = kwin3_glow.la
 kwin3_glow_la_SOURCES = glowclient.cpp glowbutton.cpp
 kwin3_glow_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
-kwin3_glow_la_LIBADD = $(LIB_KDECORE) -lkdecorations
+kwin3_glow_la_LIBADD = $(LIB_QT) $(LIB_KDECORE) -lkdecorations
 #>- METASOURCES = AUTO
 noinst_HEADERS = glowclient.h glowbutton.h resources.h
 linkdir = $(kde_datadir)/kwin/
Index: kdeartwork-3.5.10/kwin-styles/cde/Makefile.in
===================================================================
--- kdeartwork-3.5.10.orig/kwin-styles/cde/Makefile.in
+++ kdeartwork-3.5.10/kwin-styles/cde/Makefile.in
@@ -375,7 +375,7 @@ SUBDIRS = . config
 kde_module_LTLIBRARIES = kwin3_cde.la
 kwin3_cde_la_SOURCES = cdeclient.cpp
 kwin3_cde_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
-kwin3_cde_la_LIBADD = -lkdecorations
+kwin3_cde_la_LIBADD = -lkdecorations $(LIB_QT) $(LIB_KDECORE)
 #>- METASOURCES = AUTO
 noinst_HEADERS = cdeclient.h
 lnkdir = $(kde_datadir)/kwin/
Index: kdeartwork-3.5.10/kwin-styles/kde1/Makefile.in
===================================================================
--- kdeartwork-3.5.10.orig/kwin-styles/kde1/Makefile.in
+++ kdeartwork-3.5.10/kwin-styles/kde1/Makefile.in
@@ -356,7 +356,7 @@ INCLUDES = $(all_includes) 
 kde_module_LTLIBRARIES = kwin3_kde1.la
 kwin3_kde1_la_SOURCES = kde1client.cpp
 kwin3_kde1_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
-kwin3_kde1_la_LIBADD = -lkdecorations
+kwin3_kde1_la_LIBADD = -lkdecorations $(LIB_QT) $(LIB_KDECORE)
 #>- METASOURCES = AUTO
 noinst_HEADERS = kde1client.h
 lnkdir = $(kde_datadir)/kwin/
Index: kdeartwork-3.5.10/kwin-styles/openlook/Makefile.in
===================================================================
--- kdeartwork-3.5.10.orig/kwin-styles/openlook/Makefile.in
+++ kdeartwork-3.5.10/kwin-styles/openlook/Makefile.in
@@ -356,7 +356,7 @@ INCLUDES = $(all_includes)
 kde_module_LTLIBRARIES = kwin3_openlook.la
 kwin3_openlook_la_SOURCES = OpenLook.cpp
 kwin3_openlook_la_LDFLAGS = $(all_libraries) $(KDE_PLUGIN) -module
-kwin3_openlook_la_LIBADD = -lkdecorations
+kwin3_openlook_la_LIBADD = -lkdecorations $(LIB_QT) $(LIB_KDECORE)
 noinst_HEADERS = OpenLook.h
 #>- METASOURCES = AUTO
 linkdir = $(kde_datadir)/kwin/
