diff -ur kdenetwork-3.5.2.orig/kopete/kopete/kopetewindow.cpp kdenetwork-3.5.2/kopete/kopete/kopetewindow.cpp
--- kdenetwork-3.5.2.orig/kopete/kopete/kopetewindow.cpp	2006-04-03 16:46:20.000000000 +0300
+++ kdenetwork-3.5.2/kopete/kopete/kopetewindow.cpp	2006-04-03 16:47:06.000000000 +0300
@@ -285,7 +285,7 @@
 		CTRL+SHIFT+Key_I, KKey::QtWIN+CTRL+Key_I, Kopete::ChatSessionManager::self(), SLOT(slotReadMessage()) );
 
 	globalAccel->insert( QString::fromLatin1("Show/Hide Contact List"), i18n("Show/Hide Contact List"), i18n("Show or hide the contact list"),
-		CTRL+SHIFT+Key_S, KKey::QtWIN+CTRL+Key_S, this, SLOT(slotShowHide()) );
+		CTRL+SHIFT+Key_G, KKey::QtWIN+CTRL+Key_G, this, SLOT(slotShowHide()) );
 
 	globalAccel->insert( QString::fromLatin1("Set Away/Back"), i18n("Set Away/Back"), i18n("Sets away from keyboard or sets back"),
 		CTRL+SHIFT+Key_W, KKey::QtWIN+CTRL+SHIFT+Key_W, this, SLOT(slotToggleAway()) );
