diff -ur kdenetwork-3.5.5.orig/kopete/libkopete/kopeteprefs.cpp kdenetwork-3.5.5/kopete/libkopete/kopeteprefs.cpp
--- kdenetwork-3.5.5.orig/kopete/libkopete/kopeteprefs.cpp	2006-10-01 20:27:14.000000000 +0300
+++ kdenetwork-3.5.5/kopete/libkopete/kopeteprefs.cpp	2006-10-03 22:37:15.000000000 +0300
@@ -156,7 +156,7 @@
 	// Load the reconnection setting
 	config->setGroup("General");
 	mReconnectOnDisconnect = config->readBoolEntry("ReconnectOnDisconnect", true);
-	mAutoConnect = config->readBoolEntry("AutoConnect", false);
+	mAutoConnect = config->readBoolEntry("AutoConnect", true);
 
 	// Nothing has changed yet
 	mWindowAppearanceChanged = false;
