diff -Nur kdelibs-3.5.9-old/kdeprint/cups/cupsdconf2/cupsd.conf.template kdelibs-3.5.9/kdeprint/cups/cupsdconf2/cupsd.conf.template
--- kdelibs-3.5.9-old/kdeprint/cups/cupsdconf2/cupsd.conf.template	2008-02-22 17:33:45.614843837 +0200
+++ kdelibs-3.5.9/kdeprint/cups/cupsdconf2/cupsd.conf.template	2008-02-22 17:38:42.460716406 +0200
@@ -256,28 +256,28 @@
 # <b>User (User)</b>
 # <p>
 # The user the server runs under.  Normally this
-# must be <b>lp</b>, however you can configure things for another user
+# must be <b>pnp</b>, however you can configure things for another user
 # as needed.</p>
 # <p>
 # Note: the server must be run initially as root to support the
 # default IPP port of 631.  It changes users whenever an external
 # program is run...</p>
 # <p>
-# <i>ex</i>: lp</p>
+# <i>ex</i>: pnp</p>
 $$
-#User lp
+#User pnp
 @@
 
 %%group
 # <b>Group (Group)</b>
 # <p>
 # The group the server runs under.  Normally this
-# must be <b>sys</b>, however you can configure things for another
+# must be <b>pnpadmin</b>, however you can configure things for another
 # group as needed.</p>
 # <p>
-# <i>ex</i>: sys</p>
+# <i>ex</i>: pnpadmin</p>
 $$
-#Group sys
+#Group pnpadmin
 @@
 
 %%ripcache
@@ -590,11 +590,11 @@
 # <p>
 # The group name for "System" (printer administration)
 # access.  The default varies depending on the operating system, but
-# will be <b>sys</b>, <b>system</b>, or <b>root</b> (checked for in that order).</p>
+# will be <b>pnpadmin</b>, <b>system</b>, or <b>root</b> (checked for in that order).</p>
 # <p>
-# <i>ex</i>: sys</p>
+# <i>ex</i>: pnpadmin</p>
 $$
-#SystemGroup sys
+#SystemGroup pnpadmin
 @@
 
 %%servercertificate
