Index: kdelibs-3.5.10/kdeprint/cups/cupsdconf2/cupsd.conf.template
===================================================================
--- kdelibs-3.5.10.orig/kdeprint/cups/cupsdconf2/cupsd.conf.template
+++ kdelibs-3.5.10/kdeprint/cups/cupsdconf2/cupsd.conf.template
@@ -272,12 +272,12 @@ $$
 # <b>Group (Group)</b>
 # <p>
 # The group the server runs under.  Normally this
-# must be <b>sys</b>, however you can configure things for another
+# must be <b>lpadmin</b>, however you can configure things for another
 # group as needed.</p>
 # <p>
-# <i>ex</i>: sys</p>
+# <i>ex</i>: lpadmin</p>
 $$
-#Group sys
+#Group lpadmin
 @@
 
 %%ripcache
@@ -590,11 +590,11 @@ $$
 # <p>
 # The group name for "System" (printer administration)
 # access.  The default varies depending on the operating system, but
-# will be <b>sys</b>, <b>system</b>, or <b>root</b> (checked for in that order).</p>
+# will be <b>lpadmin</b>, <b>system</b>, or <b>root</b> (checked for in that order).</p>
 # <p>
-# <i>ex</i>: sys</p>
+# <i>ex</i>: lpadmin</p>
 $$
-#SystemGroup sys
+#SystemGroup lpadmin
 @@
 
 %%servercertificate
