diff -Nur kdelibs-3.5.9-old/kdeprint/cups/cupsdconf2/cupsdconf.cpp kdelibs-3.5.9/kdeprint/cups/cupsdconf2/cupsdconf.cpp
--- kdelibs-3.5.9-old/kdeprint/cups/cupsdconf2/cupsdconf.cpp	2008-02-22 17:33:45.592847182 +0200
+++ kdelibs-3.5.9/kdeprint/cups/cupsdconf2/cupsdconf.cpp	2008-02-22 17:40:35.586518670 +0200
@@ -114,8 +114,8 @@
 	maxjobs_ = 0;
 	maxjobsperprinter_ = 0;
 	maxjobsperuser_ = 0;
-	user_ = "lp";
-	group_ = "sys";
+	user_ = "pnp";
+	group_ = "pnpadmin";
 	ripcache_ = "8m";
 	filterlimit_ = 0;
 	browsing_ = true;
