Index: kdeprint/cups/kmcupsmanager.cpp
===================================================================
--- kdeprint/cups/kmcupsmanager.cpp	(Revision 505172)
+++ kdeprint/cups/kmcupsmanager.cpp	(Arbeitskopie)
@@ -83,7 +83,7 @@
 	// change LANG variable so that CUPS is always using
 	// english language: translation may only come from the PPD
 	// itself, or from KDE.
-	setenv("LANG", "en", 1);
+	setenv("LANG", "en_US.UTF-8", 1);
 }
 
 KMCupsManager::~KMCupsManager()
