Index: kio/kio/paste.cpp
===================================================================
--- kio/kio/paste.cpp	(revision 486413)
+++ kio/kio/paste.cpp	(working copy)
@@ -172,6 +172,7 @@
   if ( QTextDrag::canDecode( data ) && QTextDrag::decode( data, text ) )
   {
       QTextStream txtStream( ba, IO_WriteOnly );
+      txtStream.setEncoding(QTextStream::UnicodeUTF8);
       txtStream << text;
   }
   else
