Index: mimetypes/application/vnd.oasis.opendocument.image.desktop
===================================================================
--- mimetypes/application/vnd.oasis.opendocument.image.desktop	(revision 482753)
+++ mimetypes/application/vnd.oasis.opendocument.image.desktop	(working copy)
@@ -3,7 +3,7 @@
 Type=MimeType
 MimeType=application/vnd.oasis.opendocument.image
 Patterns=*.odi;*.ODI
-Icon=krita_kra
+Icon=ooo-draw2.0
 Comment=OASIS OpenDocument Image
 Comment[af]=OASIS oop Dokument beeld
 Comment[bg]=Изображение на OASIS OpenDocument
Index: mimetypes/application/vnd.sun.xml.draw.desktop
===================================================================
--- mimetypes/application/vnd.sun.xml.draw.desktop	(revision 482753)
+++ mimetypes/application/vnd.sun.xml.draw.desktop	(working copy)
@@ -68,7 +68,7 @@
 Comment[zh_TW]=StarOffice.org 繪圖
 MimeType=application/vnd.sun.xml.draw
 Type=MimeType
-Icon=image
+Icon=ooo-draw2.0
 Patterns=*.sxd;*.SXD;
 [Property::X-KDE-NativeExtension]
 Type=QString
Index: mimetypes/application/vnd.sun.xml.draw.template.desktop
===================================================================
--- mimetypes/application/vnd.sun.xml.draw.template.desktop	(revision 482753)
+++ mimetypes/application/vnd.sun.xml.draw.template.desktop	(working copy)
@@ -62,7 +62,7 @@
 Comment[zh_CN]=OpenOffice.org 绘图模板
 MimeType=application/vnd.sun.xml.draw.template
 Type=MimeType
-Icon=image
+Icon=ooo-draw2.0
 Patterns=*.std;*.STD;
 [Property::X-KDE-NativeExtension]
 Type=QString
Index: mimetypes/application/vnd.oasis.opendocument.chart.desktop
===================================================================
--- mimetypes/application/vnd.oasis.opendocument.chart.desktop	(revision 482753)
+++ mimetypes/application/vnd.oasis.opendocument.chart.desktop	(working copy)
@@ -3,7 +3,7 @@
 Type=MimeType
 MimeType=application/vnd.oasis.opendocument.chart
 Patterns=*.odc;*.ODC
-Icon=kchart_chrt
+Icon=ooo-calc2.0
 Comment=OASIS OpenDocument Chart
 Comment[af]=OASIS Oop Dokument kaart
 Comment[bg]=Диаграма на OASIS OpenDocument
