--- kdebluetooth-1.0_beta5/kdebluetooth/kbluetooth/kbluetooth.desktop.org	2007-08-23 17:12:40.000000000 +0300
+++ kdebluetooth-1.0_beta5/kdebluetooth/kbluetooth/kbluetooth.desktop	2007-08-23 17:32:30.000000000 +0300
@@ -31,7 +31,7 @@
 GenericName[sv]=Blåtandserver
 GenericName[tr]=Bluetooth Sunucusu
 GenericName[xx]=xxBluetooth Serverxx
-Categories=Settings;HardwareSettings;
+Categories=KDE;System;Application;
 OnlyShowIn=KDE;
 Comment=KDE Bluetooth Framework Metaserver
 Comment[bg]=Мета сървър за Bluetooth Framework
