Index: kdebluetooth-1.0_beta8/kdebluetooth/libkbluetooth/Makefile.am
===================================================================
--- kdebluetooth-1.0_beta8.orig/kdebluetooth/libkbluetooth/Makefile.am
+++ kdebluetooth-1.0_beta8/kdebluetooth/libkbluetooth/Makefile.am
@@ -12,7 +12,7 @@ libkbluetooth_la_LDFLAGS = $(all_librari
 libkbluetooth_la_LIBADD = $(LIB_KDECORE) $(DBUS_LIBS) $(LIB_KIO) $(BLUETOOTH_LIBS) -ldbus-qt-1
 
 libkbluetoothincludedir = $(includedir)/libkbluetooth
-libkbluetoothinclude_HEADERS = adapter.h dbusinit.h dbussignal.h dbussignal.h manager.h inputdevice.h inputmanager.h \
+libkbluetoothinclude_HEADERS = adapter.h dbusinit.h dbussignal.h manager.h inputdevice.h inputmanager.h \
 			       security.h dbusfilter.h service.h passkeyagent.h servicerecord.h devicemimeconverter.h \
                                deviceaddress.h rfcommsocket.h servicediscovery.h sdpservice.h namerequest.h adapter_old.h \
                                sdpdevice.h sdpattribute.h serviceselectionwidget.h inquiry.h hcisocket.h hcidefault.h \
