Index: driver/XScreenSaver.ad.in
===================================================================
--- driver/XScreenSaver.ad.in.orig
+++ driver/XScreenSaver.ad.in
@@ -404,11 +404,11 @@ XScreenSaver.bourneShell:		/bin/sh
 *passwd.body.label:		This screen is locked.
 *passwd.unlock.label:		OK
 *passwd.login.label:		New Login
 *passwd.user.label:		Username:
 *passwd.thermometer.width:	8
-*passwd.asterisks:              True
+*passwd.asterisks:              False
 *passwd.uname:                  True
 
 *splash.heading.label:		XScreenSaver %s
 *splash.body.label:		Copyright © 1991-2008 by
 *splash.body2.label:		Jamie Zawinski <jwz@jwz.org>
Index: driver/XScreenSaver_ad.h
===================================================================
--- driver/XScreenSaver_ad.h.orig
+++ driver/XScreenSaver_ad.h
@@ -281,11 +281,11 @@
 "*passwd.body.label:		This screen is locked.",
 "*passwd.unlock.label:		OK",
 "*passwd.login.label:		New Login",
 "*passwd.user.label:		Username:",
 "*passwd.thermometer.width:	8",
-"*passwd.asterisks:              True",
+"*passwd.asterisks:              False",
 "*passwd.uname:                  True",
 "*splash.heading.label:		XScreenSaver %s",
 "*splash.body.label:		Copyright © 1991-2008 by",
 "*splash.body2.label:		Jamie Zawinski <jwz@jwz.org>",
 "*splash.demo.label:		Settings",
