diff -Nur kqemu-1.4.0pre1-old/kqemu-linux.c kqemu-1.4.0pre1/kqemu-linux.c
--- kqemu-1.4.0pre1-old/kqemu-linux.c	2009-05-05 11:00:09.000000000 +0300
+++ kqemu-1.4.0pre1/kqemu-linux.c	2009-05-05 11:00:27.000000000 +0300
@@ -66,7 +66,7 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
 module_param(major, int, 0);
 #else
-MODULE_PARM(major,"i");
+module_param(major, int, 0);
 #endif
 
 /* Lock the page at virtual address 'user_addr' and return its
