diff -Nur gnome-mplayer-old/src/thread.c gnome-mplayer/src/thread.c
--- gnome-mplayer-old/src/thread.c	2010-07-19 13:52:14.034180889 +0300
+++ gnome-mplayer/src/thread.c	2010-07-19 13:53:12.362079940 +0300
@@ -439,6 +439,7 @@
             send_command("get_property sub_demux\n", TRUE);
         }
 
+        if (strncmp(vm, "gl2", strlen(vm)) != 0 ) {
         cmd = g_strdup_printf("brightness %i\n", idledata->brightness);
         send_command(cmd, TRUE);
         g_free(cmd);
@@ -458,6 +459,7 @@
         cmd = g_strdup_printf("saturation %i\n", idledata->saturation);
         send_command(cmd, TRUE);
         g_free(cmd);
+        }
 
     }
 
