diff -Nur gnome-mplayer-0.9.8-old/src/support.c gnome-mplayer-0.9.8/src/support.c
--- gnome-mplayer-0.9.8-old/src/support.c	2009-09-18 21:32:46.865495711 +0300
+++ gnome-mplayer-0.9.8/src/support.c	2009-09-18 21:33:12.229744785 +0300
@@ -1009,6 +1009,8 @@
 
     g_key_file_remove_key(config, "gnome-mplayer", "really-quiet", NULL);
     g_key_file_set_string(config, "gnome-mplayer", "msglevel", "all=5");
+    g_key_file_set_string(config, "gnome-mplayer", "nomsgcolor", "1");
+    g_key_file_set_string(config, "gnome-mplayer", "nomsgmodule", "1");
 
     data = g_key_file_to_data(config, NULL, NULL);
     //printf("%i\n%s", strlen(data), data);
