diff -Nur gnome-mplayer-0.9.8-old/src/support.c gnome-mplayer-0.9.8/src/support.c
--- gnome-mplayer-0.9.8-old/src/support.c	2009-09-18 21:59:24.373744369 +0300
+++ gnome-mplayer-0.9.8/src/support.c	2009-09-18 22:03:37.763516773 +0300
@@ -597,6 +597,15 @@
         } else {
             av[ac++] = g_strdup_printf("%s", mplayer_bin);
         }
+
+        if (extraopts != NULL) {
+            char **opts = g_strsplit(extraopts, " ", -1);
+            int i;
+            for (i = 0; opts[i] != NULL; i++)
+                av[ac++] = g_strdup(opts[i]);
+            g_strfreev(opts);
+        }
+
         av[ac++] = g_strdup_printf("-vo");
         av[ac++] = g_strdup_printf("null");
         av[ac++] = g_strdup_printf("-ao");
@@ -756,6 +765,15 @@
         } else {
             av[ac++] = g_strdup_printf("%s", mplayer_bin);
         }
+
+        if (extraopts != NULL) {
+            char **opts = g_strsplit(extraopts, " ", -1);
+            int i;
+            for (i = 0; opts[i] != NULL; i++)
+                av[ac++] = g_strdup(opts[i]);
+            g_strfreev(opts);
+        }
+
         av[ac++] = g_strdup_printf("-vo");
         av[ac++] = g_strdup_printf("null");
         av[ac++] = g_strdup_printf("-ao");
@@ -849,6 +867,15 @@
         } else {
             av[ac++] = g_strdup_printf("%s", mplayer_bin);
         }
+
+        if (extraopts != NULL) {
+            char **opts = g_strsplit(extraopts, " ", -1);
+            int i;
+            for (i = 0; opts[i] != NULL; i++)
+                av[ac++] = g_strdup(opts[i]);
+            g_strfreev(opts);
+        }
+
         av[ac++] = g_strdup_printf("-vo");
         av[ac++] = g_strdup_printf("null");
         av[ac++] = g_strdup_printf("-ao");
@@ -1363,6 +1390,15 @@
     } else {
         av[ac++] = g_strdup_printf("%s", mplayer_bin);
     }
+
+    if (extraopts != NULL) {
+        char **opts = g_strsplit(extraopts, " ", -1);
+        int i;
+        for (i = 0; opts[i] != NULL; i++)
+            av[ac++] = g_strdup(opts[i]);
+        g_strfreev(opts);
+    }
+
     av[ac++] = g_strdup_printf("-vo");
     av[ac++] = g_strdup_printf("null");
     av[ac++] = g_strdup_printf("-ao");
@@ -1584,6 +1620,15 @@
     } else {
         av[ac++] = g_strdup_printf("%s", mplayer_bin);
     }
+
+    if (extraopts != NULL) {
+        char **opts = g_strsplit(extraopts, " ", -1);
+        int i;
+        for (i = 0; opts[i] != NULL; i++)
+            av[ac++] = g_strdup(opts[i]);
+        g_strfreev(opts);
+    }
+
     av[ac++] = g_strdup_printf("-vo");
     av[ac++] = g_strdup_printf("null");
     av[ac++] = g_strdup_printf("-ao");
@@ -3059,6 +3104,15 @@
     } else {
         av[ac++] = g_strdup_printf("%s", mplayer_bin);
     }
+
+    if (extraopts != NULL) {
+        char **opts = g_strsplit(extraopts, " ", -1);
+        int i;
+        for (i = 0; opts[i] != NULL; i++)
+            av[ac++] = g_strdup(opts[i]);
+        g_strfreev(opts);
+    }
+
     av[ac++] = g_strdup_printf("-volume");
     av[ac++] = g_strdup_printf("100");
     av[ac++] = g_strdup_printf("-noidle");
diff -Nur gnome-mplayer-0.9.8-old/src/thread.c gnome-mplayer-0.9.8/src/thread.c
--- gnome-mplayer-0.9.8-old/src/thread.c	2009-09-18 21:59:24.373744369 +0300
+++ gnome-mplayer-0.9.8/src/thread.c	2009-09-18 22:01:17.167744356 +0300
@@ -916,6 +916,14 @@
         argv[arg++] = g_strdup_printf("%s", mplayer_bin);
     }
 
+    if (extraopts != NULL) {
+        char **opts = g_strsplit(extraopts, " ", -1);
+        int i;
+        for (i = 0; opts[i] != NULL; i++)
+            argv[arg++] = g_strdup(opts[i]);
+        g_strfreev(opts);
+    }
+
     // argv[arg++] = g_strdup_printf("-v");
 
     if ((vo != NULL && strlen(vo) > 0) || (ao != NULL && strlen(ao) > 0)) {
