Index: security/nss/cmd/platlibs.mk
===================================================================
RCS file: /cvsroot/mozilla/security/nss/cmd/platlibs.mk,v
retrieving revision 1.66
diff -u -r1.66 platlibs.mk
--- security/nss/cmd/platlibs.mk	4 Feb 2010 18:59:10 -0000	1.66
+++ security/nss/cmd/platlibs.mk	9 Mar 2010 12:06:11 -0000
@@ -80,7 +80,11 @@
 endif
 endif
 
+ifdef NSS_USE_SYSTEM_SQLITE
 SQLITE=-lsqlite3
+else
+SQLITE=-lnsssqlite3
+endif
 
 ifdef NSS_DISABLE_DBM
 DBMLIB = $(NULL)
Index: security/nss/lib/softoken/config.mk
===================================================================
RCS file: /cvsroot/mozilla/security/nss/lib/softoken/config.mk,v
retrieving revision 1.29
diff -u -r1.29 config.mk
--- security/nss/lib/softoken/config.mk	11 Jun 2009 00:55:48 -0000	1.29
+++ security/nss/lib/softoken/config.mk	9 Mar 2010 12:06:12 -0000
@@ -57,12 +57,16 @@
 	-L$(DIST)/lib \
 	-L$(NSSUTIL_LIB_DIR) \
 	-lnssutil3 \
-	-lsqlite3 \
 	-L$(NSPR_LIB_DIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
 	$(NULL)
+ifdef NSS_USE_SYSTEM_SQLITE
+EXTRA_SHARED_LIBS += -lsqlite3
+else
+EXTRA_SHARED_LIBS += -lnsssqlite3
+endif
 else # ! NS_USE_GCC
 
 EXTRA_SHARED_LIBS += \
Index: security/nss/lib/sqlite/manifest.mn
===================================================================
RCS file: /cvsroot/mozilla/security/nss/lib/sqlite/manifest.mn,v
retrieving revision 1.4
diff -u -r1.4 manifest.mn
--- security/nss/lib/sqlite/manifest.mn	9 Jan 2010 05:04:37 -0000	1.4
+++ security/nss/lib/sqlite/manifest.mn	9 Mar 2010 12:06:12 -0000
@@ -38,9 +38,10 @@
 
 MODULE = nss
 
-LIBRARY_NAME = sqlite
+LIBRARY_NAME = nsssqlite
 LIBRARY_VERSION = 3
 MAPFILE = $(OBJDIR)/sqlite.def
+MAPFILE_SOURCE = sqlite.def
 DEFINES += -DSQLITE_THREADSAFE=1
 
 EXPORTS = \
