Index: libsdl1.2-1.2.13/autogen.sh
===================================================================
--- libsdl1.2-1.2.13.orig/autogen.sh	2007-12-31 05:48:39.000000000 +0100
+++ libsdl1.2-1.2.13/autogen.sh	2008-01-02 03:19:59.000000000 +0100
@@ -4,15 +4,9 @@
 echo "This may take a while ..."
 
 # Regenerate configuration files
-cp acinclude.m4 aclocal.m4
-found=false
-for autoconf in autoconf autoconf259 autoconf-2.59
-do if which $autoconf >/dev/null 2>&1; then $autoconf && found=true; break; fi
-done
-if test x$found = xfalse; then
-    echo "Couldn't find autoconf, aborting"
-    exit 1
-fi
+aclocal-1.9
+autoconf
+autoheader
 (cd test; sh autogen.sh)
 
 # Run configure for this platform
Index: libsdl1.2-1.2.13/test/autogen.sh
===================================================================
--- libsdl1.2-1.2.13.orig/test/autogen.sh	2007-12-31 05:48:35.000000000 +0100
+++ libsdl1.2-1.2.13/test/autogen.sh	2008-01-02 03:19:59.000000000 +0100
@@ -1,12 +1,5 @@
 #!/bin/sh
 #
 # Regenerate configuration files
-cp acinclude.m4 aclocal.m4
-found=false
-for autoconf in autoconf autoconf259 autoconf-2.59
-do if which $autoconf >/dev/null 2>&1; then $autoconf && found=true; break; fi
-done
-if test x$found = xfalse; then
-    echo "Couldn't find autoconf, aborting"
-    exit 1
-fi
+aclocal-1.9
+autoconf
