Index: libsdl1.2-1.2.13/src/video/ps2gs/SDL_gsyuv.c
===================================================================
--- libsdl1.2-1.2.13.orig/src/video/ps2gs/SDL_gsyuv.c	2007-12-31 05:48:11.000000000 +0100
+++ libsdl1.2-1.2.13/src/video/ps2gs/SDL_gsyuv.c	2008-01-02 03:20:00.000000000 +0100
@@ -180,7 +180,7 @@
 
 	/* Allocate a DMA area for pixel conversion */
 	bpp = this->screen->format->BytesPerPixel;
-	map_offset = (mapped_len + (PAGE_SIZE - 1)) & ~(PAGE_SIZE - 1);
+	map_offset = (mapped_len + (sysconf(_SC_PAGESIZE) - 1)) & ~(sysconf(_SC_PAGESIZE) - 1);
 	hwdata->dma_len = hwdata->macroblocks * (16 * 16 + 8 * 8 + 8 * 8) +
 	                  width * height * bpp +
 	                  hwdata->macroblocks * (16 * sizeof(long long)) +
