Index: kopete/protocols/irc/ircusercontact.cpp
===================================================================
--- kopete/protocols/irc/ircusercontact.cpp	(revision 699284)
+++ kopete/protocols/irc/ircusercontact.cpp	(working copy)
@@ -115,13 +115,13 @@
 
 				if( !(currentStatus.internalStatus() & IRCProtocol::Away) && newStatus == m_protocol->m_UserStatusAway )
 				{
-					setOnlineStatus( newStatus );
+					//setOnlineStatus( newStatus );
 					//kdDebug(14120) << k_funcinfo << "was NOT away, but is now, channel " << channel->nickName() << endl;
 					adjustInternalOnlineStatusBits(channel, IRCProtocol::Away, AddBits);
 				}
 				else if( (currentStatus.internalStatus() & IRCProtocol::Away) && newStatus == m_protocol->m_UserStatusOnline )
 				{
-					setOnlineStatus( newStatus );
+					//setOnlineStatus( newStatus );
 					//kdDebug(14120) << k_funcinfo << "was away, but not anymore, channel " << channel->nickName() << endl;
 					adjustInternalOnlineStatusBits(channel, IRCProtocol::Away, RemoveBits);
 
