From 5b17c646b1f6a314a0f9ba67f176c00616fc38fc Mon Sep 17 00:00:00 2001
From: Tim Waugh <twaugh@redhat.com>
Date: Mon, 10 Aug 2009 09:26:19 +0100
Subject: [PATCH] Fixed incorrect operator in jpeg and bmp MIME type rules.

---
 conf/mime.types |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/conf/mime.types b/conf/mime.types
index feb4493..bf3249e 100644
--- a/conf/mime.types
+++ b/conf/mime.types
@@ -95,7 +95,7 @@ application/vnd.hp-HPGL		hpgl \
 
 image/gif			gif string(0,GIF87a) string(0,GIF89a)
 image/png			png string(0,<89>PNG)
-image/jpeg			jpeg jpg jpe string(0,<FFD8FF>) &&\
+image/jpeg			jpeg jpg jpe string(0,<FFD8FF>) +\
 				(char(3,0xe0) char(3,0xe1) char(3,0xe2) char(3,0xe3)\
 				 char(3,0xe4) char(3,0xe5) char(3,0xe6) char(3,0xe7)\
 				 char(3,0xe8) char(3,0xe9) char(3,0xea) char(3,0xeb)\
@@ -114,7 +114,7 @@ image/x-sun-raster		ras string(0,<59a66a95>)
 
 #image/fpx			fpx
 image/x-alias			pix short(8,8) short(8,24)
-image/x-bitmap			bmp string(0,BM) && !printable(2,14)
+image/x-bitmap			bmp string(0,BM) + !printable(2,14)
 image/x-icon			ico
 
 ########################################################################
-- 
1.6.2.5

