diff -up cups-1.3.9/filter/pstops.c.str2831 cups-1.3.9/filter/pstops.c
--- cups-1.3.9/filter/pstops.c.str2831	2008-12-09 12:11:55.000000000 +0000
+++ cups-1.3.9/filter/pstops.c	2008-12-09 12:12:19.000000000 +0000
@@ -1547,9 +1547,6 @@ copy_page(cups_file_t  *fp,		/* I - File
       else if (!strncmp(line, "%%Include", 9))
 	continue;
 
-      if (line[0] != '%' && !feature)
-        break;
-
       if (!feature || (doc->number_up == 1 && !doc->fitplot))
 	doc_write(doc, line, linelen);
     }
