diff -ur netpbm-10.35.orig/editor/ppmquantall netpbm-10.35/editor/ppmquantall
--- netpbm-10.35.orig/editor/ppmquantall	2006-09-22 13:47:25.000000000 -0400
+++ netpbm-10.35/editor/ppmquantall	2006-09-22 13:48:07.000000000 -0400
@@ -62,8 +62,8 @@
 # To be robust, we need to use Pnmfile to get that information, or 
 # Put this program in C and use ppm_readppminit().
 
-set widths=()
-set heights=()
+widths=()
+heights=()
 
 for i in ${files[@]}; do
     widths=(${widths[*]} `grep -v '^#' $i | sed '1d; s/ .*//; 2q'`)
