--- netpbm-10.35/converter/pbm/pbmtomacp.c.pbmtomacp	2007-01-29 10:11:08.000000000 +0100
+++ netpbm-10.35/converter/pbm/pbmtomacp.c	2007-02-02 13:55:23.000000000 +0100
@@ -38,7 +38,6 @@
   int argn, rows, cols;
   int left,bottom,right,top;
   int lflg, rflg, tflg, bflg;
-  char name[100];
   const char * const usage = "[-l left] [-r right] [-b bottom] [-t top] [pbmfile]";
 
 
@@ -84,11 +83,9 @@
 
   if ( argn == argc )
   { ifp = stdin;
-    strcpy( name, "noname" );
   }
   else
   { ifp = pm_openr( argv[argn] );
-    strcpy( name, argv[argn] );
     ++argn;
   }
 
