From 53ca60964b3691172f7c95e68b6d754ba55e90ac Mon Sep 17 00:00:00 2001
From: Luca Bigliardi <shammash@artha.org>
Date: Mon, 09  Feb  2009  12:34:29  +0000
Subject: Explicitly link against libm.

---
diff --git a/src/Makefile.am b/src/Makefile.am
index 92cf0d5..dc10b58 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -122,7 +122,7 @@ RGB_la_SOURCES = mixer3/RGB/RGB.c
 
 AM_CPPFLAGS = -I@top_srcdir@/include 
 AM_CFLAGS = -I@top_srcdir@/include 
-AM_LDFLAGS = -module -avoid-version -Wc,-nostartfiles
+AM_LDFLAGS = -module -avoid-version -Wc,-nostartfiles -lm
 
 plugindir = @prefix@/lib/frei0r-1
 
--
cgit v0.7.2-83-g9ec5
