From 058da851b2b536905203db1f170747388996ba78 Mon Sep 17 00:00:00 2001
From: Pierre Ossman <pierre@ossman.eu>
Date: Thu, 4 Jun 2009 07:58:57 +0200
Subject: [PATCH 2/5] cb710: add missing parenthesis

Signed-off-by: Pierre Ossman <pierre@ossman.eu>
---
 drivers/misc/cb710/debug.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/misc/cb710/debug.c b/drivers/misc/cb710/debug.c
index 9da11bc..02358d0 100644
--- a/drivers/misc/cb710/debug.c
+++ b/drivers/misc/cb710/debug.c
@@ -37,7 +37,7 @@ static void cb710_read_regs_##t(void __iomem *iobase,			\
 	unsigned i, j;							\
 									\
 	for (i = 0; i < ARRAY_SIZE(allow); ++i, reg += 16/(t/8)) {	\
-		if (!select & (1 << i))					\
+		if (!(select & (1 << i)))					\
 			continue;					\
 									\
 		for (j = 0; j < 0x10/(t/8); ++j) {			\
-- 
1.6.3

