--- src/amr-nb.c.orig	2007-05-05 12:44:30.000000000 +0300
+++ src/amr-nb.c	2007-12-12 00:02:14.000000000 +0200
@@ -52,4 +52,3 @@
 #define E_IF_exit           Encoder_Interface_exit
 #define E_IF_init()         Encoder_Interface_init(1)
 static unsigned block_size[] = {13,14,16,18,20,21,27,32,6,1,1,1,1,1,1,1};
-#include "amr.h"
--- src/amr-wb.c.orig	2007-05-16 21:49:44.000000000 +0300
+++ src/amr-wb.c	2007-12-12 00:02:13.000000000 +0200
@@ -35,4 +35,3 @@
 #define AMR_NAMES           "amr-wb", "awb"
 #define AMR_PRIV_TOO_BIG    amr_wb_PRIVSIZE_too_big
 #define AMR_RATE            16000
-#include "amr.h"
