diff -urN mpeg4ip-1.5/lib/Makefile.am mpeg4ip-1.5.patched/lib/Makefile.am
--- mpeg4ip-1.5/lib/Makefile.am	2006-03-21 02:08:20 +0300
+++ mpeg4ip-1.5.patched/lib/Makefile.am	2006-05-30 18:37:21 +0400
@@ -8,4 +8,4 @@
 endif
 endif
 
-SUBDIRS = utils audio avi fposrec gnu ismacryp mp4av mp4 mp4v2 msg_queue rtp sdp mpeg2t mpeg2ps srtp $(extradir) 
+SUBDIRS = utils audio avi fposrec gnu ismacryp mp4v2 mp4av mp4 msg_queue rtp sdp mpeg2t mpeg2ps srtp $(extradir) 
diff -urN mpeg4ip-1.5/lib/mp4av/Makefile.am mpeg4ip-1.5.patched/lib/mp4av/Makefile.am
--- mpeg4ip-1.5/lib/mp4av/Makefile.am	2006-05-11 22:17:56 +0400
+++ mpeg4ip-1.5.patched/lib/mp4av/Makefile.am	2006-05-30 18:37:21 +0400
@@ -48,6 +48,12 @@
 	rfcisma.cpp \
 	rfch264.cpp
 
+libmp4av_la_LIBADD = \
+	$(top_builddir)/lib/mp4v2/libmp4v2.la
 
 EXTRA_DIST = libmp4av.dsp
 
+
+$(top_builddir)/lib/mp4v2/libmp4v2.la:
+	make -C `dirname $@` `basename $@`
+
diff -urN mpeg4ip-1.5/lib/mp4v2/util/Makefile.am mpeg4ip-1.5.patched/lib/mp4v2/util/Makefile.am
--- mpeg4ip-1.5/lib/mp4v2/util/Makefile.am	2006-05-11 22:17:57 +0400
+++ mpeg4ip-1.5.patched/lib/mp4v2/util/Makefile.am	2006-05-30 18:37:21 +0400
@@ -41,3 +41,6 @@
 
 EXTRA_DIST = mp4dump60.dsp  \
 	mp4info.dsp mp4tags.dsp
+
+$(top_builddir)/lib/mp4av/libmp4av.la:
+	make -C `dirname $@` `basename $@`
diff -urN mpeg4ip-1.5/lib/mpeg2ps/Makefile.am mpeg4ip-1.5.patched/lib/mpeg2ps/Makefile.am
--- mpeg4ip-1.5/lib/mpeg2ps/Makefile.am	2005-09-29 00:38:27 +0400
+++ mpeg4ip-1.5.patched/lib/mpeg2ps/Makefile.am	2006-05-30 18:37:21 +0400
@@ -5,6 +5,8 @@
 	mpeg2_ps.h \
 	mpeg2ps_util.c
 
+libmpeg2_program_la_LIBADD = \
+	$(top_builddir)/lib/mp4av/libmp4av.la
 
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/lib/mp4av -I$(top_srcdir)/lib/mp4v2 
 AM_CFLAGS = -D_REENTRANT @BILLS_CWARNINGS@
@@ -13,9 +15,9 @@
 bin_PROGRAMS = mpeg_ps_info mpeg_ps_extract
 prog_libadd = \
 	$(top_builddir)/lib/gnu/libmpeg4ip_gnu.la \
+	libmpeg2_program.la \
 	$(top_builddir)/lib/mp4av/libmp4av.la \
-	$(top_builddir)/lib/mp4v2/libmp4v2.la \
-	libmpeg2_program.la 
+	$(top_builddir)/lib/mp4v2/libmp4v2.la
 
 mpeg_ps_info_SOURCES = ps_info.cpp
 mpeg_ps_info_LDADD = $(prog_libadd)
@@ -25,10 +27,7 @@
 
 mpeg2ps_test_SOURCES = test.cpp
 mpeg2ps_test_LDADD = libmpeg2_program.la \
-	$(top_builddir)/lib/gnu/libmpeg4ip_gnu.la \
-	$(top_builddir)/lib/mp4av/libmp4av.la \
-	$(top_builddir)/lib/mp4v2/libmp4v2.la \
-	libmpeg2_program.la 
+		     $(prog_libadd)
 
 EXTRA_DIST=libmpeg2ps.dsp
 
diff -urN mpeg4ip-1.5/lib/msg_queue/Makefile.am mpeg4ip-1.5.patched/lib/msg_queue/Makefile.am
--- mpeg4ip-1.5/lib/msg_queue/Makefile.am	2004-04-08 01:26:48 +0400
+++ mpeg4ip-1.5.patched/lib/msg_queue/Makefile.am	2006-05-30 18:37:21 +0400
@@ -3,6 +3,8 @@
 libmsg_queue_la_SOURCES = \
 	msg_queue.cpp \
 	msg_queue.h
+libmsg_queue_la_LIBADD = \
+	 -lSDL -lpthread
 
 EXTRA_DIST = \
 	libmsg_queue60.dsp
diff -urN mpeg4ip-1.5/lib/SDLAudio/src/Makefile.am mpeg4ip-1.5.patched/lib/SDLAudio/src/Makefile.am
--- mpeg4ip-1.5/lib/SDLAudio/src/Makefile.am	2004-05-13 01:15:24 +0400
+++ mpeg4ip-1.5.patched/lib/SDLAudio/src/Makefile.am	2006-05-30 18:37:21 +0400
@@ -23,7 +23,8 @@
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE)
 libmpeg4ipSDL_la_LIBADD =		\
 	@SDL_EXTRALIBS@		\
-	@SYSTEM_LIBS@
+	@SYSTEM_LIBS@ \
+	-lSDL -lpthread
 libmpeg4ipSDL_la_DEPENDENCIES =	\
 	@SDL_EXTRALIBS@
 
diff -urN mpeg4ip-1.5/player/lib/audio/celp/dec/Makefile.am mpeg4ip-1.5.patched/player/lib/audio/celp/dec/Makefile.am
--- mpeg4ip-1.5/player/lib/audio/celp/dec/Makefile.am	2005-08-12 21:17:17 +0400
+++ mpeg4ip-1.5.patched/player/lib/audio/celp/dec/Makefile.am	2006-05-30 18:37:21 +0400
@@ -47,3 +47,5 @@
 AM_CFLAGS = 
 
 AM_CXXFLAGS = 
+
+libmpeg4ip_celp_la_LIBADD = -lm ../BS/libmpeg4ip_celpbs.la
diff -urN mpeg4ip-1.5/player/lib/audio/faad/Makefile.am mpeg4ip-1.5.patched/player/lib/audio/faad/Makefile.am
--- mpeg4ip-1.5/player/lib/audio/faad/Makefile.am	2005-05-05 00:01:36 +0400
+++ mpeg4ip-1.5.patched/player/lib/audio/faad/Makefile.am	2006-05-30 18:37:21 +0400
@@ -32,6 +32,7 @@
 	tns.h \
 	transfo.h\
 	util.h
+libmpeg4ip_faad_la_LIBADD = -lm
 INCLUDES = -I$(top_srcdir)/include
 AM_CFLAGS = -O2 -Werror -Wall -fexceptions -fno-strict-aliasing
 
diff -urN mpeg4ip-1.5/player/lib/libhttp/Makefile.am mpeg4ip-1.5.patched/player/lib/libhttp/Makefile.am
--- mpeg4ip-1.5/player/lib/libhttp/Makefile.am	2004-02-25 00:58:38 +0300
+++ mpeg4ip-1.5.patched/player/lib/libhttp/Makefile.am	2006-05-30 18:37:21 +0400
@@ -6,7 +6,7 @@
 	http_resp.c \
 	http_util.c
 
-libhttp_la_LIBADD = $(top_builddir)/lib/utils/libutils.la
+libhttp_la_LIBADD = $(top_builddir)/lib/utils/libutils.la -lstdc++
 
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/lib/utils
 
diff -urN mpeg4ip-1.5/player/lib/mp4util/Makefile.am mpeg4ip-1.5.patched/player/lib/mp4util/Makefile.am
--- mpeg4ip-1.5/player/lib/mp4util/Makefile.am	2005-04-20 02:50:01 +0400
+++ mpeg4ip-1.5.patched/player/lib/mp4util/Makefile.am	2006-05-30 18:37:21 +0400
@@ -10,6 +10,7 @@
 
 libh264util_la_SOURCES = \
 	h264_sdp.c
+libh264util_la_LIBADD = $(top_builddir)/lib/mp4v2/libmp4v2.la
 
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/lib/mp4v2
 
diff -urN mpeg4ip-1.5/player/plugin/audio/celp/Makefile.am mpeg4ip-1.5.patched/player/plugin/audio/celp/Makefile.am
--- mpeg4ip-1.5/player/plugin/audio/celp/Makefile.am	2005-08-12 21:17:17 +0400
+++ mpeg4ip-1.5.patched/player/plugin/audio/celp/Makefile.am	2006-05-30 18:37:21 +0400
@@ -7,7 +7,8 @@
 celp_plugin_la_LIBADD = \
 	$(top_builddir)/player/lib/mp4util/libmp4util.la \
 	$(top_builddir)/player/lib/audio/celp/BS/libmpeg4ip_celpbs.la \
-	$(top_builddir)/player/lib/audio/celp/dec/libmpeg4ip_celp.la 
+	$(top_builddir)/player/lib/audio/celp/dec/libmpeg4ip_celp.la \
+	$(top_builddir)/lib/mp4av/libmp4av.la 
 
 INCLUDES = -I$(top_srcdir)/player/src -I$(top_srcdir)/player/lib -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/player/lib/audio/celp -I$(top_srcdir)/player/lib/audio/celp/BS -I$(top_srcdir)/player/lib/audio/celp/dec -I$(top_srcdir)/player/lib/audio/celp/include -I$(top_srcdir)/lib/mp4av -I$(top_srcdir)/lib/sdp -I$(top_srcdir)/lib/mp4v2
 
diff -urN mpeg4ip-1.5/player/plugin/rtp/isma_audio/Makefile.am mpeg4ip-1.5.patched/player/plugin/rtp/isma_audio/Makefile.am
--- mpeg4ip-1.5/player/plugin/rtp/isma_audio/Makefile.am	2005-09-08 21:58:24 +0400
+++ mpeg4ip-1.5.patched/player/plugin/rtp/isma_audio/Makefile.am	2006-05-30 18:37:21 +0400
@@ -15,13 +15,13 @@
 isma_rtp_plugin_la_LIBADD = \
 	$(top_builddir)/lib/mp4av/libmp4av.la \
 	$(top_builddir)/player/lib/mp4util/libmp4util.la \
-	-lm 
+	-lm -lSDL -lpthread
 
 isma_enc_rtp_plugin_la_LIBADD = \
         $(top_builddir)/lib/mp4av/libmp4av.la \
         $(top_builddir)/player/lib/mp4util/libmp4util.la \
         $(top_builddir)/lib/ismacryp/libismacryp.la @SRTPLIB@ \
-        -lm
+        -lm -lSDL -lpthread
 
 INCLUDES = -I$(top_srcdir)/player/src \
 	   -I$(top_srcdir)/player/lib \
diff -urN mpeg4ip-1.5/player/src/codec/mp3/Makefile.am mpeg4ip-1.5.patched/player/src/codec/mp3/Makefile.am
--- mpeg4ip-1.5/player/src/codec/mp3/Makefile.am	2005-08-12 21:17:17 +0400
+++ mpeg4ip-1.5.patched/player/src/codec/mp3/Makefile.am	2006-05-30 18:37:21 +0400
@@ -30,8 +30,8 @@
 endif
 mp3_plugin_la_LIBADD = \
 	-lm $(ID3_LIBS) \
-	$(top_builddir)/lib/fposrec/libmpeg4ip_fposrec.la
-
+	$(top_builddir)/lib/fposrec/libmpeg4ip_fposrec.la \
+	$(top_builddir)/lib/mp4av/libmp4av.la
 
 INCLUDES = -I$(top_srcdir)/player/src -I$(top_srcdir)/player/lib -I$(top_srcdir)/player/lib/audio -I$(top_srcdir)/common/video/mpeg4 -I$(top_srcdir)/include -I$(top_srcdir)/lib -I$(top_srcdir)/lib/mp4v2 -I$(top_srcdir)/lib/sdp -I$(top_srcdir)/lib/utils
 
diff -urN mpeg4ip-1.5/player/src/codec/wav/Makefile.am mpeg4ip-1.5.patched/player/src/codec/wav/Makefile.am
--- mpeg4ip-1.5/player/src/codec/wav/Makefile.am	2005-08-12 21:17:17 +0400
+++ mpeg4ip-1.5.patched/player/src/codec/wav/Makefile.am	2006-05-30 18:37:21 +0400
@@ -5,6 +5,9 @@
 	ourwav.h \
 	wav_file.cpp \
 	wav_file.h
+wav_plugin_la_LIBADD = \
+	$(top_builddir)/lib/mp4av/libmp4av.la \
+	-lSDL -lpthread
 
 libdir = @PLAYER_PLUGIN_DIR@
 
diff -urN mpeg4ip-1.5/player/src/Makefile.am mpeg4ip-1.5.patched/player/src/Makefile.am
--- mpeg4ip-1.5/player/src/Makefile.am	2006-03-21 02:08:27 +0300
+++ mpeg4ip-1.5.patched/player/src/Makefile.am	2006-05-30 18:38:58 +0400
@@ -62,6 +62,10 @@
 	$(SDL_AUDIO_SRC) \
 	video_sdl_sync.cpp \
 	video_sdl_sync.h 
+libmp4syncsdl_la_LIBADD = \
+	$(top_builddir)/lib/SDLAudio/src/libmpeg4ipSDL.la
+
+libmp4sdlvideo_la_LIBADD = -lX11
 
 include_HEADERS = \
 	codec_plugin.h \
diff -urN mpeg4ip-1.5/server/mp4live/gui/Makefile.am mpeg4ip-1.5.patched/server/mp4live/gui/Makefile.am
--- mpeg4ip-1.5/server/mp4live/gui/Makefile.am	2006-03-21 02:08:30 +0300
+++ mpeg4ip-1.5.patched/server/mp4live/gui/Makefile.am	2006-05-30 18:37:21 +0400
@@ -16,7 +16,9 @@
 	support.h \
 	text_dialog.cpp \
 	transmit_dialog.cpp \
-	video_dialog.cpp 
+	video_dialog.cpp
+
+libmp4livegui_la_LIBADD = $(GTK_LIBS)
 
 INCLUDES=@GLIB_CFLAGS@ @GTK_CFLAGS@ \
 	-I$(top_srcdir)/server/mp4live \
