diff -Nur mpeg4ip-1.5.0.1-old/common/video/iso-mpeg4/configure.in mpeg4ip-1.5.0.1/common/video/iso-mpeg4/configure.in
--- mpeg4ip-1.5.0.1-old/common/video/iso-mpeg4/configure.in	2009-05-07 22:46:58.000000000 +0300
+++ mpeg4ip-1.5.0.1/common/video/iso-mpeg4/configure.in	2009-05-07 22:48:23.000000000 +0300
@@ -62,8 +62,8 @@
 [
 save_CFLAG="$CFLAGS"
 save_CXXFLAG="$CXXFLAGS"
-CFLAGS="$CFLAGS -Werror $1"
-CXXFLAGS="$CXXFLAGS -Werror $1"
+CFLAGS="$CFLAGS  $1"
+CXXFLAGS="$CXXFLAGS  $1"
    AC_MSG_CHECKING([whether $3 accepts $1])
    AC_COMPILE_IFELSE(
 [extern void f(void); void f(void){}],
@@ -112,7 +112,7 @@
 	[if test x$enableval = xno; then
 	   AC_MSG_NOTICE([compiler warnings will not be errors])
          else 
-           BASE_ERR="$BASE_ERR -Werror"
+           BASE_ERR="$BASE_ERR "
 	   AC_MSG_NOTICE([enabling compiler warnings as errors])
          fi
 	 ],
diff -Nur mpeg4ip-1.5.0.1-old/common/video/iso-mpeg4/src/Makefile.am mpeg4ip-1.5.0.1/common/video/iso-mpeg4/src/Makefile.am
--- mpeg4ip-1.5.0.1-old/common/video/iso-mpeg4/src/Makefile.am	2009-05-07 22:46:58.000000000 +0300
+++ mpeg4ip-1.5.0.1/common/video/iso-mpeg4/src/Makefile.am	2009-05-07 22:48:13.000000000 +0300
@@ -1,5 +1,5 @@
 
-AM_CXXFLAGS = -D__TRACE_AND_STATS_ -D__DOUBLE_PRECISION_ -D_REENTRANT -DNOCONTROLS -fexceptions -Wall -Werror -D_OBSS_ @BILLS_CPPWARNINGS@
+AM_CXXFLAGS = -D__TRACE_AND_STATS_ -D__DOUBLE_PRECISION_ -D_REENTRANT -DNOCONTROLS -fexceptions -Wall  -D_OBSS_ @BILLS_CPPWARNINGS@
 
 INCLUDES = -I$(srcdir)/../include
 
diff -Nur mpeg4ip-1.5.0.1-old/configure.in mpeg4ip-1.5.0.1/configure.in
--- mpeg4ip-1.5.0.1-old/configure.in	2009-05-07 22:46:58.000000000 +0300
+++ mpeg4ip-1.5.0.1/configure.in	2009-05-07 22:48:23.000000000 +0300
@@ -286,8 +286,8 @@
 [
 save_CFLAG="$CFLAGS"
 save_CXXFLAG="$CXXFLAGS"
-CFLAGS="$CFLAGS -Werror $1"
-CXXFLAGS="$CXXFLAGS -Werror $1"
+CFLAGS="$CFLAGS  $1"
+CXXFLAGS="$CXXFLAGS  $1"
    AC_MSG_CHECKING([whether $3 accepts $1])
    AC_COMPILE_IFELSE(
 [extern void f(void); void f(void){}],
@@ -748,7 +748,7 @@
 	[if test x$enableval = xno; then
 	   AC_MSG_NOTICE([compiler warnings will not be errors])
          else 
-           BASE_ERR="$BASE_ERR -Werror"
+           BASE_ERR="$BASE_ERR "
 	   AC_MSG_NOTICE([enabling compiler warnings as errors])
          fi
 	 ],
diff -Nur mpeg4ip-1.5.0.1-old/lib/gnu/Makefile.am mpeg4ip-1.5.0.1/lib/gnu/Makefile.am
--- mpeg4ip-1.5.0.1-old/lib/gnu/Makefile.am	2009-05-07 22:46:58.000000000 +0300
+++ mpeg4ip-1.5.0.1/lib/gnu/Makefile.am	2009-05-07 22:48:13.000000000 +0300
@@ -12,5 +12,5 @@
 EXTRA_DIST = libmissing60.dsp \
 	need_for_win32.c 
 
-AM_CFLAGS = -D_REENTRANT -Werror -Wall -Wcast-align -Wstrict-prototypes -Wmissing-prototypes
+AM_CFLAGS = -D_REENTRANT  -Wall -Wcast-align -Wstrict-prototypes -Wmissing-prototypes
 
diff -Nur mpeg4ip-1.5.0.1-old/lib/rtp/configure.in mpeg4ip-1.5.0.1/lib/rtp/configure.in
--- mpeg4ip-1.5.0.1-old/lib/rtp/configure.in	2009-05-07 22:46:58.000000000 +0300
+++ mpeg4ip-1.5.0.1/lib/rtp/configure.in	2009-05-07 22:48:23.000000000 +0300
@@ -244,8 +244,8 @@
 [
 save_CFLAG="$CFLAGS"
 save_CXXFLAG="$CXXFLAGS"
-CFLAGS="$CFLAGS -Werror $1"
-CXXFLAGS="$CXXFLAGS -Werror $1"
+CFLAGS="$CFLAGS  $1"
+CXXFLAGS="$CXXFLAGS  $1"
    AC_MSG_CHECKING([whether $3 accepts $1])
    AC_COMPILE_IFELSE(
 [extern void f(void); void f(void){}],
@@ -268,14 +268,14 @@
 	[if test x$enableval = xno; then
 	   AC_MSG_NOTICE([compiler warnings will not be errors])
          else 
-           BASE_ERR="$BASE_ERR -Werror"
+           BASE_ERR="$BASE_ERR "
 	   AC_MSG_NOTICE([enabling compiler warnings as errors])
          fi
 	 ],
 	 [AC_MSG_NOTICE([compiler warnings will not be errs])]) 
 
 BILLS_CWARNINGS="$BASE_ERR"
-CFLAGS="$CFLAGS -Wall -Werror"
+CFLAGS="$CFLAGS -Wall "
 MY_CHECK_CC_OPT_SET([-Wmissing-prototypes], BILLS_CWARNINGS, [c compiler])
 MY_CHECK_CC_OPT_SET([-Wmissing-declarations], BILLS_CWARNINGS, [c compiler])
 MY_CHECK_CC_OPT_SET([-Wbad-function-cast], BILLS_CWARNINGS, [c compiler])
diff -Nur mpeg4ip-1.5.0.1-old/lib/rtp/Makefile.am mpeg4ip-1.5.0.1/lib/rtp/Makefile.am
--- mpeg4ip-1.5.0.1-old/lib/rtp/Makefile.am	2009-05-07 22:46:58.000000000 +0300
+++ mpeg4ip-1.5.0.1/lib/rtp/Makefile.am	2009-05-07 22:48:13.000000000 +0300
@@ -77,7 +77,7 @@
 
 check_PROGRAMS = test_rtp_client test_rtp_server
 
-AM_CFLAGS = -DDEBUG -Wall -Werror
+AM_CFLAGS = -DDEBUG -Wall 
 test_rtp_client_SOURCES = test_rtp_client.c
 test_rtp_client_LDADD = libuclmmbase.la -lsrtp
 test_rtp_server_SOURCES = test_rtp_server.c
