--- src/streamtuner/st-preselections.c.orig	2004-12-13 11:57:52.000000000 +0200
+++ src/streamtuner/st-preselections.c	2006-08-17 11:59:28.000000000 +0300
@@ -390,9 +390,9 @@
 {
   st_preselections_init_handler();
   
-  st_action_register("play-stream", _("Listen to a stream"), "xmms %q");
-  st_action_register("record-stream", _("Record a stream"), "xterm -e streamripper %q");
-  st_action_register("view-web", _("Open a web page"), "epiphany %q");
+  st_action_register("play-stream", _("Listen to a stream"), "kaffeine %q");
+  st_action_register("record-stream", _("Record a stream"), "konsole -e streamripper %q -s -o never");
+  st_action_register("view-web", _("Open a web page"), "firefox %q");
 }
 
 void
