--- src/plugins/xiph/xiph.c.orig	2006-08-28 11:36:32.000000000 +0300
+++ src/plugins/xiph/xiph.c	2006-08-28 16:52:52.000000000 +0300
@@ -956,8 +956,8 @@
 
   init_handler();
 
-  st_action_register("record-stream", _("Record a stream"), "xterm -e streamripper %q");
-  st_action_register("play-stream", _("Listen to a stream"), "xmms %q");
+  st_action_register("record-stream", _("Record a stream"), "xterm -e streamripper %q -s -o never");
+  st_action_register("play-stream", _("Listen to a stream"), "kaffeine %q");
 
   return TRUE;
 }
