diff -up kdelibs-3.5.10/kio/kssl/kopenssl.cc.openssl kdelibs-3.5.10/kio/kssl/kopenssl.cc
--- kdelibs-3.5.10/kio/kssl/kopenssl.cc.openssl	2006-07-22 03:16:39.000000000 -0500
+++ kdelibs-3.5.10/kio/kssl/kopenssl.cc	2009-04-19 16:34:14.000000000 -0500
@@ -329,6 +329,19 @@ KConfig *cfg;
              #ifdef SHLIB_VERSION_NUMBER
              << "libssl.so." SHLIB_VERSION_NUMBER
              #endif
+             << "libssl.so.0.9.8k"
+             << "libssl.so.8"
+             << "libssl.so.0.9.8g"
+             << "libssl.so.7"
+             << "libssl.so.0.9.8b"
+             << "libssl.so.0.9.8a"
+             << "libssl.so.6"
+             << "libssl.so.0.9.7e"
+             << "libssl.so.5"
+             << "libssl.so.0.9.7a"
+             << "libssl.so.4"
+             << "libssl.so.0.9.6b"
+             << "libssl.so.2"
              << "libssl.so"
 	     << "libssl.so.0"
              #endif
@@ -346,6 +359,19 @@ KConfig *cfg;
              #ifdef SHLIB_VERSION_NUMBER
              << "libcrypto.so." SHLIB_VERSION_NUMBER
              #endif
+             << "libcrypto.so.0.9.8k"
+             << "libcrypto.so.8"
+             << "libcrypto.so.0.9.8g"
+             << "libcrypto.so.7"
+             << "libcrypto.so.0.9.8b"
+             << "libcrypto.so.0.9.8a"
+             << "libcrypto.so.6"
+             << "libcrypto.so.0.9.7e"
+             << "libcrypto.so.5"
+             << "libcrypto.so.0.9.7a"
+             << "libcrypto.so.4"
+             << "libcrypto.so.0.9.6b"
+             << "libcrypto.so.2"
              << "libcrypto.so"
 	     << "libcrypto.so.0"
              #endif
