--- logrotate-3.7.9/examples/logrotate-default	2011-02-23 14:56:55.376999935 +0200
+++ logrotate-3.7.9/examples/logrotate-default	2011-02-23 14:57:09.382000196 +0200
@@ -9,10 +9,13 @@
 create
 
 # use date as a suffix of the rotated file
-dateext
+# dateext
 
 # uncomment this if you want your log files compressed
-#compress
+compress
+delaycompress
+notifempty
+missingok
 
 # RPM packages drop log rotation information into this directory
 include /etc/logrotate.d
