Index: ModemManager-0.3_20100429/configure.ac
===================================================================
--- ModemManager-0.3_20100429.orig/configure.ac
+++ ModemManager-0.3_20100429/configure.ac
@@ -84,7 +84,7 @@ AC_ARG_WITH([pppd-plugin-dir], AS_HELP_S
 if test -n "$with_pppd_plugin_dir" ; then
 	PPPD_PLUGIN_DIR="$with_pppd_plugin_dir"
 else
-	PPPD_PLUGIN_DIR="${libdir}/pppd/2.4.4"
+	PPPD_PLUGIN_DIR="${libdir}/pppd/2.4.5"
 fi
 AC_SUBST(PPPD_PLUGIN_DIR)
 
