diff -Nur kdenetwork-3.5.10-old//filesharing/advanced/kcm_sambaconf/usertabimpl.cpp kdenetwork-3.5.10/filesharing/advanced/kcm_sambaconf/usertabimpl.cpp
--- kdenetwork-3.5.10-old//filesharing/advanced/kcm_sambaconf/usertabimpl.cpp	2011-02-15 20:30:42.354000034 +0200
+++ kdenetwork-3.5.10/filesharing/advanced/kcm_sambaconf/usertabimpl.cpp	2011-02-15 20:36:54.756000036 +0200
@@ -264,9 +264,8 @@
     UserSelectDlg *dlg = new UserSelectDlg();
     dlg->init(_specifiedUsers,_share);
 
-    QStringList selectedUsers = dlg->getSelectedUsers();
-
     if (dlg->exec()) {
+        QStringList selectedUsers = dlg->getSelectedUsers();
         for (QStringList::Iterator it = selectedUsers.begin(); it != selectedUsers.end(); ++it)
         {
             addUserToUserTable(*it,dlg->getAccess());
