Index: kdenetwork-3.5.10/kopete/kopete/eventsrc
===================================================================
--- kdenetwork-3.5.10.orig/kopete/kopete/eventsrc
+++ kdenetwork-3.5.10/kopete/kopete/eventsrc
@@ -42,6 +42,7 @@ Comment[sr]=Гласник Kopete
 Comment[sr@Latn]=Glasnik Kopete
 Comment[sv]=Kopete meddelandeklient
 Comment[ta]=உடனடி தூதர்
+Comment[tr]=Kopete Mesajlaşma Uygulaması
 Comment[tg]=Пайёмбари Kopete
 Comment[uk]=Програма обміну повідомленнями Kopete
 Comment[uz]=Kopete xabar almashish vositasi
