--- kopete/kopete/kopetewindow.cpp.orig	2008-02-05 23:27:39.000000000 +0200
+++ kopete/kopete/kopetewindow.cpp	2008-02-05 23:28:02.000000000 +0200
@@ -877,8 +877,8 @@
 		for(Kopete::Account *a = accounts.first(); a; a = accounts.next())
 		{
 			Kopete::Contact *self = a->myself();
-			tt += i18n( "Account tooltip information: <nobr>ICON <b>PROTOCOL:</b> NAME (<i>STATUS</i>)<br/>",
-			            "<nobr><img src=\"kopete-account-icon:%3:%4\"> <b>%1:</b> %2 (<i>%5</i>)<br/>" )
+			tt += i18n( "Account tooltip information: <nobr>ICON <b>PROTOCOL:</b> NAME (<i>STATUS</i>)<nobr/><br>",
+			            "<nobr><img src=\"kopete-account-icon:%3:%4\"> <b>%1:</b> %2 (<i>%5</i>)<nobr/><br>" )
 				.arg( a->protocol()->displayName() ).arg( a->accountLabel(), KURL::encode_string( a->protocol()->pluginId() ),
 				KURL::encode_string( a->accountId() ), self->onlineStatus().description() );
 		}
