Index: openafs-1.5.72/src/cf/osconf.m4
===================================================================
--- openafs-1.5.72.orig/src/cf/osconf.m4
+++ openafs-1.5.72/src/cf/osconf.m4
@@ -3,11 +3,12 @@ AC_DEFUN([OPENAFS_OSCONF], [
 
 dnl defaults, override in case below as needed
 CFLAGS=
-XCFLAGS='${DBG} ${OPTMZ}'
+dnl XCFLAGS='${DBG} ${OPTMZ}'
 RXDEBUG="-DRXDEBUG"
 SHLIB_SUFFIX="so"
-CCOBJ="$CC"
-MT_CC="$CC"
+dnl CC="cc"
+CCOBJ="${CC}"
+MT_CC="${CC}"
 XLIBS="${LIB_AFSDB} ${XBSA_XLIBS} ${LIB_libintl}"
 
 dnl debugging and optimization flag defaults
@@ -47,7 +48,6 @@ case $AFS_SYSNAME in
 		MT_CFLAGS='-D_REENTRANT=1 -pthread -DAFS_PTHREAD_ENV ${XCFLAGS}'
 		MT_LIBS="-pthread -lpthread -lmach -lexc -lc"
 		TXLIBS="-lcurses"
-		XCFLAGS="-D_NO_PROTO -DOSF"
 		SHLIB_LINKER="${CC} -all -shared -expect_unresolved \"*\""
 		;;
 
@@ -61,7 +61,6 @@ case $AFS_SYSNAME in
 		MT_CFLAGS='-D_REENTRANT=1 -pthread -DAFS_PTHREAD_ENV ${XCFLAGS}'
 		MT_LIBS="-pthread -lpthread -lmach -lexc -lc"
 		TXLIBS="-lcurses"
-		XCFLAGS="-D_NO_PROTO -DOSF"
 		SHLIB_LINKER="${CC} -all -shared -expect_unresolved \"*\""
 		;;
 
@@ -76,7 +75,6 @@ case $AFS_SYSNAME in
 		MT_CFLAGS='-D_REENTRANT=1 -pthread -DAFS_PTHREAD_ENV ${XCFLAGS}'
 		MT_LIBS="-pthread -lpthread -lmach -lexc -lc"
 		TXLIBS="-lcurses"
-		XCFLAGS="-D_NO_PROTO -DOSF"
 		SHLIB_LINKER="${CC} -all -shared -expect_unresolved \"*\""
 		;;
 
@@ -90,7 +88,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
@@ -106,7 +103,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
@@ -122,7 +118,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
@@ -138,7 +133,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
@@ -154,7 +148,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
@@ -180,9 +173,6 @@ case $AFS_SYSNAME in
 		SHLIB_SUFFIX="sl"
 		TXLIBS="/usr/lib/libHcurses.a"
 		VFSCK_CFLAGS="-D_FILE64"
-		XCFLAGS0="-ldld -lc -Wp,-H200000 -Wl,-a,archive -DAUTH_DBM_LOG +z -Wl,+k -D_LARGEFILE64_SOURCE"
-		XCFLAGS64="${XCFLAGS0} +DA2.0W"
-		XCFLAGS="${XCFLAGS0} +DA1.0"
 		YACC="/opt/langtools/bin/yacc"
 		SHLIB_LINKER="ld -b"
 		;;
@@ -209,9 +199,6 @@ case $AFS_SYSNAME in
 		SHLIB_SUFFIX="sl"
 		TXLIBS="/usr/lib/libHcurses.a"
 		VFSCK_CFLAGS="-I/usr/old/usr/include -D_FILE64"
-		XCFLAGS0="-ldld -lc -Wp,-H200000 -Wl,-a,archive -DAUTH_DBM_LOG +z -Wl,+k -D_LARGEFILE64_SOURCE"
-		XCFLAGS64="${XCFLAGS0} +DA2.0W"
-		XCFLAGS="${XCFLAGS0} +DA1.0"
 		XLIBELFA="-lelf"
 		#XLIBS="${LIB_AFSDB} -lnsl"
 		YACC="/opt/langtools/bin/yacc"
@@ -240,9 +227,6 @@ case $AFS_SYSNAME in
 		SHLIB_SUFFIX="sl"
 		TXLIBS="/usr/lib/hpux32/libcurses.so"
 		VFSCK_CFLAGS="-I/usr/old/usr/include -D_FILE64"
-		XCFLAGS0="-ldld -lc -Wp,-H200000 -Wl,-a,archive_shared -DAUTH_DBM_LOG +z -Wl,+k -D_LARGEFILE64_SOURCE"
-		XCFLAGS64="${XCFLAGS0} +DD64"
-		XCFLAGS="${XCFLAGS0}"
 		XLIBELFA="-lelf"
 		#XLIBS="${LIB_AFSDB} -lnsl"
 		YACC="/opt/langtools/bin/yacc"
@@ -258,7 +242,6 @@ case $AFS_SYSNAME in
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		SHLIB_LINKER="${MT_CC} -shared"
 		TXLIBS="-lncurses"
-		XCFLAGS="-pipe"
 		YACC="byacc"
 		;;
 
@@ -271,7 +254,6 @@ case $AFS_SYSNAME in
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		SHLIB_LINKER="${MT_CC} -shared"
 		TXLIBS="-lncurses"
-		XCFLAGS="-pipe"
 		YACC="byacc"
 		;;
 
@@ -283,7 +265,6 @@ case $AFS_SYSNAME in
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		SHLIB_LINKER="${MT_CC} -shared"
 		TXLIBS="-lncurses"
-		XCFLAGS="-O2 -pipe -fPIC"
 		YACC="byacc"
 		;;
 
@@ -296,7 +277,6 @@ case $AFS_SYSNAME in
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		SHLIB_LINKER="${MT_CC} -shared"
 		TXLIBS="/usr/lib/libcurses.so"
-		XCFLAGS="-pipe"
 		YACC="yacc"
 		;;
 
@@ -309,7 +289,6 @@ case $AFS_SYSNAME in
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		SHLIB_LINKER="${MT_CC} -shared"
 		TXLIBS="/usr/lib/libcurses.so"
-		XCFLAGS="-pipe"
 		YACC="bison -y"
 		;;
 
@@ -323,7 +302,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE -G0"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
 
@@ -338,7 +316,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
 
@@ -352,7 +329,6 @@ case $AFS_SYSNAME in
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		SHLIB_CFLAGS="-fPIC"
 		TXLIBS="/usr/lib64/libncurses.so"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
 
@@ -366,7 +342,6 @@ case $AFS_SYSNAME in
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		SHLIB_CFLAGS="-fPIC"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE -fPIC"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
@@ -381,7 +356,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
@@ -400,7 +374,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
 
@@ -419,7 +392,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
 
@@ -438,7 +410,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
 
@@ -454,7 +425,6 @@ case $AFS_SYSNAME in
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		SHLIB_LINKER="${MT_CC} -shared"
 		TXLIBS="/usr/lib/libcurses.a"
-		XCFLAGS=
 		YACC="yacc"
 		;;
 
@@ -468,7 +438,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
@@ -480,7 +449,6 @@ case $AFS_SYSNAME in
 		AFSD_LDFLAGS="-F/System/Library/PrivateFrameworks -framework DiskArbitration"
 		LEX="lex -l"
 		REGEX_OBJ="regex.o"
-		XCFLAGS="-traditional-cpp"
 		SHLIB_LINKER="${MT_CC} -dynamiclib"
 		SHLIB_SUFFIX="dylib"
 		;;
@@ -493,7 +461,6 @@ case $AFS_SYSNAME in
 		LEX="lex -l"
 		LWP_OPTMZ="-O2"
 		REGEX_OBJ="regex.o"
-		XCFLAGS="-no-cpp-precomp"
 		SHLIB_LINKER="${MT_CC} -dynamiclib"
 		SHLIB_SUFFIX="dylib"
 		;;
@@ -506,7 +473,6 @@ case $AFS_SYSNAME in
 		LEX="lex -l"
 		LWP_OPTMZ="-O2"
 		REGEX_OBJ="regex.o"
-		XCFLAGS="-no-cpp-precomp"
 		SHLIB_LINKER="${MT_CC} -dynamiclib"
 		SHLIB_SUFFIX="dylib"
 		;;
@@ -519,7 +485,6 @@ case $AFS_SYSNAME in
 		LEX="lex -l"
 		LWP_OPTMZ="-O2"
 		REGEX_OBJ="regex.o"
-		XCFLAGS="-no-cpp-precomp"
 		TXLIBS="-lncurses"
 		SHLIB_LINKER="${MT_CC} -dynamiclib"
 		SHLIB_SUFFIX="dylib"
@@ -536,7 +501,6 @@ case $AFS_SYSNAME in
 		KINCLUDES='-I$(KROOT)/System/Library/Frameworks/Kernel.framework/Headers'
 		LWP_OPTMZ="-O2"
 		REGEX_OBJ="regex.o"
-		XCFLAGS="-no-cpp-precomp"
 		TXLIBS="-lncurses"
 		EXTRA_VLIBOBJS="fstab.o"
 		SHLIB_LINKER="${MT_CC} -dynamiclib"
@@ -615,7 +579,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
@@ -631,7 +594,6 @@ case $AFS_SYSNAME in
 		MT_LIBS="-lpthreads"
 		SHLIB_SUFFIX="o"
 		TXLIBS="-lcurses"
-		XCFLAGS="-K -D_NONSTD_TYPES -D_MBI=void"
 		XLIBS="${LIB_AFSDB} ${LIB_libintl} -ldl"
 		SHLIB_LINKER="${MT_CC} -bM:SRE -berok"
 		AIX32=""
@@ -650,7 +612,6 @@ case $AFS_SYSNAME in
 		MT_LIBS="-lpthreads"
 		SHLIB_SUFFIX="o"
 		TXLIBS="-lcurses"
-		XCFLAGS="-K -D_NONSTD_TYPES -D_MBI=void"
 		XLIBS="${LIB_AFSDB} ${LIB_libintl} -ldl"
 		SHLIB_LINKER="${MT_CC} -bM:SRE -berok"
 		AIX32=""
@@ -668,7 +629,6 @@ case $AFS_SYSNAME in
 		MT_LIBS="-lpthreads"
 		SHLIB_SUFFIX="o"
 		TXLIBS="-lcurses"
-		XCFLAGS="-K -D_NONSTD_TYPES -D_MBI=void"
 		XLIBS="${LIB_AFSDB} ${LIB_libintl} -ldl"
 		SHLIB_LINKER="${MT_CC} -bM:SRE -berok"
 		AIX32=""
@@ -686,7 +646,6 @@ case $AFS_SYSNAME in
 		MT_LIBS="-lpthreads"
 		SHLIB_SUFFIX="o"
 		TXLIBS="-lcurses"
-		XCFLAGS="-K -D_NONSTD_TYPES -D_MBI=void"
 		XLIBS="${LIB_AFSDB} ${LIB_libintl} -ldl"
 		SHLIB_LINKER="${MT_CC} -bM:SRE -berok"
 		AIX32=""
@@ -704,7 +663,6 @@ case $AFS_SYSNAME in
 		MT_LIBS="-lpthreads"
 		SHLIB_SUFFIX="o"
 		TXLIBS="-lcurses"
-		XCFLAGS="-K -D_NONSTD_TYPES -D_MBI=void"
 		XLIBS="${LIB_AFSDB} ${LIB_libintl} -ldl"
 		SHLIB_LINKER="${MT_CC} -bM:SRE -berok"
 		AIX32="#"
@@ -724,7 +682,6 @@ case $AFS_SYSNAME in
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
 		OPTMZ=-O
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
@@ -741,7 +698,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
@@ -760,7 +716,6 @@ case $AFS_SYSNAME in
 		SHLIB_LDFLAGS="-shared -Xlinker -x -Xlinker -Bsymbolic"
 		TXLIBS="-lncurses"
 		OPTMZ=-O
-		XCFLAGS="-D_LARGEFILE64_SOURCE -D__s390x__"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
 		;;
@@ -775,8 +730,6 @@ case $AFS_SYSNAME in
 		MT_CFLAGS='-D_SGI_MP_SOURCE -DAFS_PTHREAD_ENV ${XCFLAGS}'
 		MT_LIBS="-lpthread"
 		TXLIBS="-lcurses"
-		XCFLAGS64="-64 -mips3"
-		XCFLAGS="-o32"
 		XLDFLAGS64="-64"
 		XLDFLAGS="-o32"
 		SHLIB_LINKER="${CC} -shared"
@@ -792,8 +745,6 @@ case $AFS_SYSNAME in
 		MT_CFLAGS='-D_SGI_MP_SOURCE -DAFS_PTHREAD_ENV ${XCFLAGS}'
 		MT_LIBS="-lpthread"
 		TXLIBS="-lcurses"
-		XCFLAGS64="-D_BSD_COMPAT -64 -mips3"
-		XCFLAGS="-D_OLD_TERMIOS -D_BSD_COMPAT -o32"
 		XLDFLAGS64="-64"
 		XLDFLAGS="-o32"
 		SHLIB_LINKER="${CC} -shared"
@@ -809,8 +760,6 @@ case $AFS_SYSNAME in
 		MT_CFLAGS='-D_SGI_MP_SOURCE -DAFS_PTHREAD_ENV ${XCFLAGS}'
 		MT_LIBS="-lpthread"
 		TXLIBS="-lcurses"
-		XCFLAGS64="-D_BSD_COMPAT -64 -mips3"
-		XCFLAGS="-D_OLD_TERMIOS -D_BSD_COMPAT -n32 -woff 1009,1110,1116,1164,1171,1177,1183,1185,1204,1233,1515,1516,1548,1169,1174,1177,1196,1498,1506,1552"
 		XLDFLAGS64="-64"
 		XLDFLAGS="-n32"
 		SHLIB_LINKER="${CC} -shared"
@@ -827,8 +776,6 @@ case $AFS_SYSNAME in
 		MT_CFLAGS='-D_SGI_MP_SOURCE -DAFS_PTHREAD_ENV ${XCFLAGS}'
 		MT_LIBS="-lpthread"
 		TXLIBS="-lcurses"
-		XCFLAGS64="-64 -mips3"
-		XCFLAGS="-n32 -mips3 -woff 1009,1110,1116,1164,1171,1177,1183,1185,1204,1233,1515,1516,1548,1169,1174,1177,1196,1498,1506,1552"
 		XLDFLAGS64="-64 -mips3"
 		XLDFLAGS="-n32 -mips3"
 		SHLIB_LINKER="${CC} -shared"
@@ -844,8 +791,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-fPIC"
 		SHLIB_LDFLAGS="-shared -Xlinker -x"
 		TXLIBS="-lncurses"
-		XCFLAGS="-D_LARGEFILE64_SOURCE"
-		XCFLAGS64="-D_LARGEFILE64_SOURCE -m64"
 		XLDFLAGS64="-m64"
 		YACC="bison -y"
 		SHLIB_LINKER="${MT_CC} -shared"
@@ -858,7 +803,6 @@ case $AFS_SYSNAME in
 		LEX="lex"
 		SHLIB_CFLAGS="-PIC"
 		TXLIBS="-lcurses -ltermcap"
-		XCFLAGS=""
 		XLIBELFA="-lelf"
 		XLIBKVM="-lkvm"
 		XLIBS="${LIB_AFSDB}"
@@ -874,7 +818,6 @@ case $AFS_SYSNAME in
 		MT_LIBS="-lpthread -lsocket"
 		SHLIB_CFLAGS="-KPIC"
 		TXLIBS="-lcurses"
-		XCFLAGS="-dy -Bdynamic"
 		XLIBELFA="-lelf"
 		XLIBKVM="-lkvm"
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
@@ -895,7 +838,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-KPIC"
 		SHLIB_LDFLAGS="-G -Bsymbolic"
 		TXLIBS="-L/usr/ccs/lib -lcurses"
-		XCFLAGS="-dy -Bdynamic"
 		XLIBELFA="-lelf"
 		LD="/usr/ccs/bin/ld"
 		XLIBKVM="-lkvm"
@@ -917,8 +859,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-KPIC"
 		SHLIB_LDFLAGS="-G -Bsymbolic"
 		TXLIBS="-lcurses"
-		XCFLAGS64='${XCFLAGS} -xarch=v9'
-		XCFLAGS="-dy -Bdynamic"
 		XLIBELFA="-lelf"
 		XLIBKVM="-lkvm"
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
@@ -939,8 +879,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-KPIC"
 		SHLIB_LDFLAGS="-G -Bsymbolic"
 		TXLIBS="-lcurses"
-		XCFLAGS64='${XCFLAGS} -xarch=v9'
-		XCFLAGS="-dy -Bdynamic"
 		XLIBELFA="-lelf"
 		XLIBKVM="-lkvm"
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
@@ -961,8 +899,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-KPIC"
 		SHLIB_LDFLAGS="-G -Bsymbolic"
 		TXLIBS="-lcurses"
-		XCFLAGS64='${XCFLAGS} -xarch=v9'
-		XCFLAGS="-dy -Bdynamic"
 		XLIBELFA="-lelf"
 		XLIBKVM="-lkvm"
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
@@ -983,8 +919,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-KPIC"
 		SHLIB_LDFLAGS="-G -Bsymbolic"
 		TXLIBS="-lcurses"
-		XCFLAGS64='${XCFLAGS} -m64'
-		XCFLAGS="-dy -Bdynamic"
 		XLIBELFA="-lelf"
 		XLIBKVM="-lkvm"
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
@@ -1005,8 +939,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-KPIC"
 		SHLIB_LDFLAGS="-G -Bsymbolic"
 		TXLIBS="-lcurses"
-		XCFLAGS64='${XCFLAGS} -xarch=v9'
-		XCFLAGS="-dy -Bdynamic"
 		XLIBELFA="-lelf"
 		XLIBKVM="-lkvm"
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
@@ -1027,8 +959,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-KPIC"
 		SHLIB_LDFLAGS="-G -Bsymbolic"
 		TXLIBS="-lcurses"
-		XCFLAGS64='${XCFLAGS} -xarch=amd64'
-		XCFLAGS="-dy -Bdynamic"
 		XLIBELFA="-lelf"
 		XLIBKVM="-lkvm"
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
@@ -1048,8 +978,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-KPIC"
 		SHLIB_LDFLAGS="-G -Bsymbolic"
 		TXLIBS="-lcurses"
-		XCFLAGS64='${XCFLAGS} -xarch=amd64'
-		XCFLAGS="-dy -Bdynamic"
 		XLIBELFA="-lelf"
 		XLIBKVM="-lkvm"
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
@@ -1069,8 +997,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-KPIC"
 		SHLIB_LDFLAGS="-G -Bsymbolic"
 		TXLIBS="-lcurses"
-		XCFLAGS64='${XCFLAGS} -xarch=amd64'
-		XCFLAGS="-dy -Bdynamic"
 		XLIBELFA="-lelf"
 		XLIBKVM="-lkvm"
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
@@ -1090,8 +1016,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-KPIC"
 		SHLIB_LDFLAGS="-G -Bsymbolic"
 		TXLIBS="-lcurses"
-		XCFLAGS64='${XCFLAGS} -m64'
-		XCFLAGS="-dy -Bdynamic"
 		XLIBELFA="-lelf"
 		XLIBKVM="-lkvm"
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
@@ -1111,8 +1035,6 @@ case $AFS_SYSNAME in
 		SHLIB_CFLAGS="-KPIC"
 		SHLIB_LDFLAGS="-G -Bsymbolic"
 		TXLIBS="-lcurses"
-		XCFLAGS64='${XCFLAGS} -xarch=amd64'
-		XCFLAGS="-dy -Bdynamic"
 		XLIBELFA="-lelf"
 		XLIBKVM="-lkvm"
 		XLIBS="${LIB_AFSDB} -lsocket -lnsl -lintl -ldl"
@@ -1187,11 +1109,7 @@ fi
 CFLAGS_NOERROR=
 
 if test "x$GCC" = "xyes"; then
-  if test "x$enable_warnings" = "xyes"; then
-    XCFLAGS="${XCFLAGS} -Wall -Wstrict-prototypes -Wold-style-definition -Wpointer-arith"
-  fi
   if test "x$enable_checking" != "xno"; then
-    XCFLAGS="${XCFLAGS} -Wall -Wstrict-prototypes -Wold-style-definition -Werror -fdiagnostics-show-option -Wpointer-arith"
     if test "x$enable_checking" != "xall"; then
       CFLAGS_NOERROR="-Wno-error"
       AC_DEFINE(IGNORE_SOME_GCC_WARNINGS, 1, [define to disable some gcc warnings in warnings-as-errors mode])
