--- a/src/util/dirpath.c.old	2006-09-09 14:45:44.000000000 +0200
+++ b/src/util/dirpath.c	2006-09-09 14:46:36.000000000 +0200
@@ -653,7 +653,7 @@
 int
 ConstructLocalBinPath(const char *cpath, char **fullPathBufp)
 {
-    return ConstructLocalPath(cpath, AFSDIR_SERVER_BIN_DIRPATH,
+    return ConstructLocalPath(cpath, AFSDIR_CANONICAL_SERVER_BIN_DIRPATH,
 			      fullPathBufp);
 }
 
@@ -666,6 +666,6 @@
 int
 ConstructLocalLogPath(const char *cpath, char **fullPathBufp)
 {
-    return ConstructLocalPath(cpath, AFSDIR_SERVER_LOGS_DIRPATH,
+    return ConstructLocalPath(cpath, AFSDIR_CANONICAL_SERVER_LOGS_DIRPATH,
 			      fullPathBufp);
 }
