diff -Nur etherape-0.9.7-old/etherape.desktop.in etherape-0.9.7/etherape.desktop.in
--- etherape-0.9.7-old/etherape.desktop.in	2009-09-15 04:35:34.848204251 +0300
+++ etherape-0.9.7/etherape.desktop.in	2009-09-15 04:35:59.886203293 +0300
@@ -2,9 +2,11 @@
 Version=1.0
 Encoding=UTF-8
 Name=EtherApe
+Name[tr]=EtherApe
 Comment=Graphical Network Monitor
-Exec=etherape
+Comment[tr]=Grafiksel Ağ İzleyici
+Exec=xdg-su -c "etherape"
 Terminal=false
 Type=Application
-Icon=etherape.png
-Categories=GNOME;System;Network;
+Icon=etherape
+Categories=System;Monitor;
