diff -up notification-daemon-0.4.0/data/notification-daemon.schemas.in.nodoka notification-daemon-0.4.0/data/notification-daemon.schemas.in
--- notification-daemon-0.4.0/data/notification-daemon.schemas.in.nodoka	2009-04-27 20:55:23.593078616 -0400
+++ notification-daemon-0.4.0/data/notification-daemon.schemas.in	2009-04-27 20:55:38.186830065 -0400
@@ -18,7 +18,7 @@
    <applyto>/apps/notification-daemon/theme</applyto>
    <owner>notification-daemon</owner>
    <type>string</type>
-   <default>standard</default>
+   <default>nodoka</default>
    <locale name="C">
     <short>Current theme</short>
     <long>The theme used when displaying notifications.</long>
