Index: xorg-server-1.7.5.902/configure.ac
===================================================================
--- xorg-server-1.7.5.902.orig/configure.ac
+++ xorg-server-1.7.5.902/configure.ac
@@ -1336,7 +1336,7 @@ PKG_CHECK_MODULES([XSERVERLIBS], [$REQUI
 #
 XSERVER_CFLAGS="${XSERVERCFLAGS_CFLAGS} ${SHA1_CFLAGS}"
 XSERVER_LIBS="$DIX_LIB $CONFIG_LIB $MI_LIB $OS_LIB"
-XSERVER_SYS_LIBS="${XSERVERLIBS_LIBS} ${SYS_LIBS} ${LIBS}"
+XSERVER_SYS_LIBS="${XSERVERLIBS_LIBS} ${SYS_LIBS} ${LIBS} ${SHA1_LIB}"
 AC_SUBST([XSERVER_LIBS])
 AC_SUBST([XSERVER_SYS_LIBS])
 
