Index: kdelibs-3.5.10/admin/cvs.sh
===================================================================
--- kdelibs-3.5.10.orig/admin/cvs.sh
+++ kdelibs-3.5.10/admin/cvs.sh
@@ -68,7 +68,7 @@ case $AUTOMAKE_STRING in
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10*)
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.1[0-5]*)
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
