diff -Nur amarok-1.4.10-old/amarok/configure.in.in amarok-1.4.10/amarok/configure.in.in
--- amarok-1.4.10-old/amarok/configure.in.in	2010-02-16 13:49:51.423666282 +0200
+++ amarok-1.4.10/amarok/configure.in.in	2010-02-16 13:51:43.439666506 +0200
@@ -977,7 +977,7 @@
     ac_cppflags_save=$CPPFLAGS
     CPPFLAGS="$CPPFLAGS -I."
     AC_CHECK_HEADERS(systems.h)
-    AC_CHECK_HEADERS([mp4.h], [have_mp4_h=yes], [],
+    AC_CHECK_HEADERS([mp4v2/mp4v2.h], [have_mp4_h=yes], [],
         [#ifdef HAVE_SYSTEMS_H
          # include <systems.h>
          #endif
diff -Nur amarok-1.4.10-old/amarok/src/metadata/mp4/mp4properties.h amarok-1.4.10/amarok/src/metadata/mp4/mp4properties.h
--- amarok-1.4.10-old/amarok/src/metadata/mp4/mp4properties.h	2010-02-16 13:49:51.347416416 +0200
+++ amarok-1.4.10/amarok/src/metadata/mp4/mp4properties.h	2010-02-16 13:52:32.634666180 +0200
@@ -24,7 +24,7 @@
 
 #include <audioproperties.h>
 #include <tstring.h>
-#include <mp4.h>
+#include <mp4v2/mp4v2.h>
 // mp4.h drags in mp4_config.h that defines these
 // get rid of them so they don't conflict with our config.h
 #undef VERSION
diff -Nur amarok-1.4.10-old/amarok/src/metadata/mp4/mp4tag.h amarok-1.4.10/amarok/src/metadata/mp4/mp4tag.h
--- amarok-1.4.10-old/amarok/src/metadata/mp4/mp4tag.h	2010-02-16 13:49:51.348416396 +0200
+++ amarok-1.4.10/amarok/src/metadata/mp4/mp4tag.h	2010-02-16 13:52:17.458666269 +0200
@@ -23,7 +23,7 @@
 
 #include <tag.h>
 #include "mp4file.h"
-#include <mp4.h>
+#include <mp4v2/mp4v2.h>
 
 namespace TagLib {
 
diff -Nur amarok-1.4.10-old/configure.in amarok-1.4.10/configure.in
--- amarok-1.4.10-old/configure.in	2010-02-16 13:49:51.423666282 +0200
+++ amarok-1.4.10/configure.in	2010-02-16 13:51:16.954666463 +0200
@@ -1057,7 +1057,7 @@
     ac_cppflags_save=$CPPFLAGS
     CPPFLAGS="$CPPFLAGS -I."
     AC_CHECK_HEADERS(systems.h)
-    AC_CHECK_HEADERS([mp4.h], [have_mp4_h=yes], [],
+    AC_CHECK_HEADERS([mp4v2/mp4v2.h], [have_mp4_h=yes], [],
         [#ifdef HAVE_SYSTEMS_H
          # include <systems.h>
          #endif
