diff -rup lloyd-yajl-45a1bdb.orig/CMakeLists.txt lloyd-yajl-45a1bdb.new/CMakeLists.txt
--- lloyd-yajl-45a1bdb.orig/CMakeLists.txt	2009-10-29 23:35:39.000000000 +0000
+++ lloyd-yajl-45a1bdb.new/CMakeLists.txt	2010-01-11 17:14:01.000000000 +0000
@@ -42,7 +42,7 @@ IF (NOT CMAKE_BUILD_TYPE)
   SET(CMAKE_BUILD_TYPE "Release")
 ENDIF (NOT CMAKE_BUILD_TYPE)
 
-SET(CMAKE_C_FLAGS "-Wall")
+SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 IF (WIN32) 
   ADD_DEFINITIONS(-DWIN32)
   SET(linkFlags "/PDB:NONE /INCREMENTAL:NO /OPT:NOREF /OPT:NOICF")
