Index: Firebird-2.1.2.18118-0/configure
===================================================================
--- Firebird-2.1.2.18118-0.orig/configure
+++ Firebird-2.1.2.18118-0/configure
@@ -2285,7 +2285,7 @@ _ACEOF
     LOCK_MANAGER_FLG=Y
     EDITLINE_FLG=Y
     SHRLIB_EXT=so
-    libdir=/usr/lib64
+    libdir=/usr/lib
     CPU_TYPE=amd64
 	RPM64='()(64bit)'
     ;;
Index: Firebird-2.1.2.18118-0/configure.in
===================================================================
--- Firebird-2.1.2.18118-0.orig/configure.in
+++ Firebird-2.1.2.18118-0/configure.in
@@ -137,7 +137,7 @@ dnl CPU_TYPE=ppc64
     LOCK_MANAGER_FLG=Y
     EDITLINE_FLG=Y
     SHRLIB_EXT=so
-    libdir=/usr/lib64
+    libdir=/usr/lib
     CPU_TYPE=amd64
 	RPM64='()(64bit)'
     ;;
