Index: QtCurve-Gtk2-0.69.2/style/icons3
===================================================================
--- QtCurve-Gtk2-0.69.2.orig/style/icons3
+++ QtCurve-Gtk2-0.69.2/style/icons3
@@ -5,7 +5,7 @@ gtk-bold actions/text_bold.png
 gtk-cancel actions/button_cancel.png actions/cancel.png
 gtk-cdrom devices/cdrom_unmount.png
 gtk-clear actions/editclear.png
-gtk-close actions/fileclose.png
+gtk-close actions/button_cancel.png
 gtk-color-picker actions/colorpicker.png
 gtk-connect NONE
 gtk-convert NONE
