---
 drivers/bluetooth/btusb.c |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-2.6.35/drivers/bluetooth/btusb.c
===================================================================
--- linux-2.6.35.orig/drivers/bluetooth/btusb.c
+++ linux-2.6.35/drivers/bluetooth/btusb.c
@@ -1023,6 +1023,8 @@ static int btusb_probe(struct usb_interf
 		return err;
 	}
 
+	usb_enable_autosuspend(data->udev);
+
 	usb_set_intfdata(intf, data);
 
 	return 0;
