Index: kpowersave/src/config/kpowersaverc_default
===================================================================
--- kpowersave.orig/src/config/kpowersaverc_default
+++ kpowersave/src/config/kpowersaverc_default
@@ -3,7 +3,7 @@ lockOnSuspend=true
 lockOnLidClose=true
 ActionOnLidClose=
 ActionOnLidCloseValue=
-ActionOnPowerButton=SHUTDOWN
+ActionOnPowerButton=LOGOUT_DIALOG
 ActionOnPowerButtonValue=
 ActionOnSleepButton=SUSPEND2RAM
 ActionOnS2DiskButton=SUSPEND2DISK
