diff -Nur SuiteSparse/AMD/Makefile SuiteSparse-patched/AMD/Makefile
--- SuiteSparse/AMD/Makefile	2007-05-05 20:23:47.000000000 +0300
+++ SuiteSparse-patched/AMD/Makefile	2008-07-26 13:10:43.000000000 +0300
@@ -10,7 +10,6 @@
 # Do not compile the FORTRAN versions, or MATLAB interface.
 demo:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
 # Compile all C code, including the C-callable routine and the mexFunctions.
 # Do not compile the FORTRAN versions.
diff -Nur SuiteSparse/CAMD/Makefile SuiteSparse-patched/CAMD/Makefile
--- SuiteSparse/CAMD/Makefile	2007-05-05 17:28:33.000000000 +0300
+++ SuiteSparse-patched/CAMD/Makefile	2008-07-26 13:13:20.000000000 +0300
@@ -10,13 +10,11 @@
 # Do not compile the FORTRAN versions, or MATLAB interface.
 demo:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
 # Compile all C code, including the C-callable routine and the mexFunctions.
 # Do not compile the FORTRAN versions.
 all:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	( cd MATLAB ; $(MAKE) )
 
 # compile just the C-callable libraries (not mexFunctions or Demos)
diff -Nur SuiteSparse/CCOLAMD/Makefile SuiteSparse-patched/CCOLAMD/Makefile
--- SuiteSparse/CCOLAMD/Makefile	2007-05-05 16:57:17.000000000 +0300
+++ SuiteSparse-patched/CCOLAMD/Makefile	2008-07-26 13:13:29.000000000 +0300
@@ -10,12 +10,10 @@
 # Do not the MATLAB interface.
 demo:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
 # Compile all C code, including the C-callable routine and the mexFunctions.
 all:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	( cd MATLAB ; $(MAKE) )
 
 # compile just the C-callable libraries (not mexFunctions or Demos)
diff -Nur SuiteSparse/CHOLMOD/Makefile SuiteSparse-patched/CHOLMOD/Makefile
--- SuiteSparse/CHOLMOD/Makefile	2006-11-30 18:40:08.000000000 +0200
+++ SuiteSparse-patched/CHOLMOD/Makefile	2008-07-26 13:11:22.000000000 +0300
@@ -12,7 +12,6 @@
 # Compile the C-callable libraries and the Demo programs.
 all:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 # Compile the C-callable libraries only.
 library:
diff -Nur SuiteSparse/COLAMD/Makefile SuiteSparse-patched/COLAMD/Makefile
--- SuiteSparse/COLAMD/Makefile	2007-05-05 16:40:14.000000000 +0300
+++ SuiteSparse-patched/COLAMD/Makefile	2008-07-26 13:14:08.000000000 +0300
@@ -10,12 +10,10 @@
 # Do not the MATLAB interface.
 demo:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
 # Compile all C code, including the C-callable routine and the mexFunctions.
 all:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	( cd MATLAB ; $(MAKE) )
 
 # compile just the C-callable libraries (not mexFunctions or Demos)
diff -Nur SuiteSparse/CSparse/Makefile SuiteSparse-patched/CSparse/Makefile
--- SuiteSparse/CSparse/Makefile	2007-10-27 21:04:07.000000000 +0300
+++ SuiteSparse-patched/CSparse/Makefile	2008-07-26 13:11:46.000000000 +0300
@@ -2,7 +2,6 @@
 
 C:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 all: C cov
 
diff -Nur SuiteSparse/CXSparse/Makefile SuiteSparse-patched/CXSparse/Makefile
--- SuiteSparse/CXSparse/Makefile	2007-10-27 21:04:07.000000000 +0300
+++ SuiteSparse-patched/CXSparse/Makefile	2008-07-26 13:11:57.000000000 +0300
@@ -2,7 +2,6 @@
 
 C:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 all: C cov
 
diff -Nur SuiteSparse/KLU/Makefile SuiteSparse-patched/KLU/Makefile
--- SuiteSparse/KLU/Makefile	2007-10-28 02:12:35.000000000 +0300
+++ SuiteSparse-patched/KLU/Makefile	2008-07-26 13:12:14.000000000 +0300
@@ -4,7 +4,6 @@
 
 library:
 	( cd Lib ; $(MAKE) )
-	( cd Demo ; $(MAKE) )
 
 clean:
 	( cd Demo ; $(MAKE) clean )
diff -Nur SuiteSparse/LDL/Makefile SuiteSparse-patched/LDL/Makefile
--- SuiteSparse/LDL/Makefile	2007-05-05 20:24:14.000000000 +0300
+++ SuiteSparse-patched/LDL/Makefile	2008-07-26 13:13:35.000000000 +0300
@@ -9,12 +9,10 @@
 # Compile all C code, including the C-callable routine.
 demo:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 
 # Compile all C code, including the C-callable routine and the mexFunctions.
 all:
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	( cd MATLAB ; $(MAKE) )
 
 # compile just the C-callable libraries (not mexFunctions or Demos)
diff -Nur SuiteSparse/UMFPACK/Makefile SuiteSparse-patched/UMFPACK/Makefile
--- SuiteSparse/UMFPACK/Makefile	2007-10-31 00:03:55.000000000 +0200
+++ SuiteSparse-patched/UMFPACK/Makefile	2008-07-26 13:13:00.000000000 +0300
@@ -14,7 +14,6 @@
 	( cd ../AMD ; $(MAKE) library )
 	( cd ../AMD ; $(MAKE) mex )
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	( cd MATLAB ; $(MAKE) )
 	- cat Doc/License
 
@@ -22,7 +21,6 @@
 library:
 	( cd ../AMD ; $(MAKE) library )
 	( cd Lib    ; $(MAKE) )
-	( cd Demo   ; $(MAKE) )
 	- cat Doc/License
 
 # compile the FORTRAN interface and demo program
