Index: network-manager-applet-0.8.2/src/connection-editor/ce-page.c
===================================================================
--- network-manager-applet-0.8.2.orig/src/connection-editor/ce-page.c
+++ network-manager-applet-0.8.2/src/connection-editor/ce-page.c
@@ -407,7 +407,9 @@ ce_page_new_connection (const char *form
 	GSList *connections;
 
 	connection = nm_connection_new ();
-	nm_connection_set_scope (connection, NM_CONNECTION_SCOPE_USER);
+
+	/* We'd rather use the system scope instead of the crappy user scope */
+	nm_connection_set_scope (connection, NM_CONNECTION_SCOPE_SYSTEM);
 
 	s_con = NM_SETTING_CONNECTION (nm_setting_connection_new ());
 	nm_connection_add_setting (connection, NM_SETTING (s_con));
Index: network-manager-applet-0.8.2/src/connection-editor/nm-connection-editor.c
===================================================================
--- network-manager-applet-0.8.2.orig/src/connection-editor/nm-connection-editor.c
+++ network-manager-applet-0.8.2/src/connection-editor/nm-connection-editor.c
@@ -265,7 +265,7 @@ system_checkbutton_toggled_cb (GtkWidget
 	 * unchecked.
 	 */
 	if (   !gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (widget))
-	    && (editor->orig_scope == NM_CONNECTION_SCOPE_USER))
+	    && (editor->orig_scope == NM_CONNECTION_SCOPE_USER || editor->connection_is_new))
 		use_polkit = FALSE;
 
 	ce_polkit_button_set_use_polkit (CE_POLKIT_BUTTON (editor->ok_button), use_polkit);
Index: network-manager-applet-0.8.2/src/connection-editor/nm-connection-editor.h
===================================================================
--- network-manager-applet-0.8.2.orig/src/connection-editor/nm-connection-editor.h
+++ network-manager-applet-0.8.2/src/connection-editor/nm-connection-editor.h
@@ -59,6 +59,7 @@ typedef struct {
 	GtkWidget *cancel_button;
 
 	gboolean busy;
+	gboolean connection_is_new;
 } NMConnectionEditor;
 
 typedef struct {
Index: network-manager-applet-0.8.2/src/connection-editor/nm-connection-list.c
===================================================================
--- network-manager-applet-0.8.2.orig/src/connection-editor/nm-connection-list.c
+++ network-manager-applet-0.8.2/src/connection-editor/nm-connection-list.c
@@ -676,6 +676,9 @@ really_add_connection (NMConnection *con
 		return;
 	}
 
+	/* Mark the connection as unsaved yet */
+	editor->connection_is_new = TRUE;
+
 	g_signal_connect (editor, "done", G_CALLBACK (add_response_cb), info);
 	g_hash_table_insert (info->list->editors, connection, editor);
 
