Index: network-manager-applet-0.8/src/applet.c
===================================================================
--- network-manager-applet-0.8.orig/src/applet.c
+++ network-manager-applet-0.8/src/applet.c
@@ -1636,7 +1636,7 @@ static void nma_menu_show_cb (GtkWidget 
 
 	n_wireless = nma_menu_add_devices (menu, applet);
 
-	nma_menu_add_vpn_submenu (menu, applet);
+	/*nma_menu_add_vpn_submenu (menu, applet);*/
 
 	if (n_wireless > 0 && nm_client_wireless_get_enabled (applet->nm_client)) {
 		/* Add the "Hidden wireless network..." entry */
