diff -ur kdepim-3.5.6.orig/akregator/src/akregator_part.cpp kdepim-3.5.6/akregator/src/akregator_part.cpp
--- kdepim-3.5.6.orig/akregator/src/akregator_part.cpp	2006-07-22 11:14:08.000000000 +0300
+++ kdepim-3.5.6/akregator/src/akregator_part.cpp	2007-02-19 10:41:13.649471382 +0200
@@ -382,6 +382,16 @@
     plan.setAttribute("xmlUrl","http://planetkde.org/rss20.xml");
     mainFolder.appendChild(plan);
 
+    QDomElement planet = doc.createElement( "outline" );
+    planet.setAttribute("text",i18n("Planet Pardus"));
+    planet.setAttribute("xmlUrl","http://planet.pardus.org.tr/rss20.xml");
+    mainFolder.appendChild(planet);
+
+    QDomElement gezegen = doc.createElement( "outline" );
+    gezegen.setAttribute("text",i18n("Pardus Dünyası"));
+    gezegen.setAttribute("xmlUrl","http://gezegen.pardus.org.tr/rss20.xml");
+    mainFolder.appendChild(gezegen);
+
     QDomElement apps = doc.createElement( "outline" );
     apps.setAttribute("text",i18n("KDE Apps"));
     apps.setAttribute("xmlUrl","http://www.kde.org/dot/kde-apps-content.rdf");
