--- libkdepim/Makefile.am
+++ libkdepim/Makefile.am	2005/02/07 16:51:07
@@ -45,7 +45,7 @@
                        $(LIB_KIO) $(LIB_KABC) $(LIB_KPARTS) $(LIB_KIMPROXY) $(LIB_POLL) -lqui
 #               $top_builddir)/libkdepim/resources/libkpimresources.la
 
-include_HEADERS = kdepimmacros.h
+include_HEADERS = kdepimmacros.h kpimprefs.h
 
 AM_CXXFLAGS = -DQT_PLUGIN
 kde_widget_LTLIBRARIES = kdepimwidgets.la

