--- kdepim/kioslaves/sieve/sieve.cpp	2008/10/23 21:44:50	875279
+++ kdepim/kioslaves/sieve/sieve.cpp	2008/10/23 21:47:43	875280
@@ -1290,7 +1290,7 @@
     const int minor = regExp.cap( 2 ).toInt();
     const int patch = regExp.cap( 3 ).toInt();
     const QString vendor = regExp.cap( 4 );
-    if ( major < 2 || (major == 2 && (minor < 3 || (major == 3 && patch < 11))) || (vendor == "-kolab-nocaps") ) {
+    if ( major < 2 || (major == 2 && (minor < 3 || (minor == 3 && patch < 11))) || (vendor == "-kolab-nocaps") ) {
       ksDebug() << k_funcinfo << "Enabling compat mode for Cyrus < 2.3.11 or Cyrus marked as \"kolab-nocaps\"" << endl;
       return true;
     }
