Index: kmail/main.cpp
===================================================================
--- kmail/main.cpp	(revision 595995)
+++ kmail/main.cpp	(working copy)
@@ -105,7 +105,7 @@ int main(int argc, char *argv[])
   kmsetSignalHandler(kmsignalHandler);
 
   kapp->dcopClient()->resume(); // Ok. We are ready for DCOP requests.
-  kmkernel->setStartingUp( false ); // Starting up is finished
+  kmailKernel.setStartingUp( false ); // Starting up is finished
   // Go!
   int ret = kapp->exec();
   // clean up
