--- knode/knnetaccess.cpp
+++ knode/knnetaccess.cpp
@@ -398,57 +398,69 @@
       case KNProtocolClient::TSconnect:
         currMsg = i18n(" Connecting to server...");
         knGlobals.setStatusMsg(currMsg);
-        currentNntpJob->setStatus(currMsg);
+        if ( currentNntpJob )
+          currentNntpJob->setStatus(currMsg);
       break;
       case KNProtocolClient::TSloadGrouplist:
         currMsg = i18n(" Loading group list from disk...");
         knGlobals.setStatusMsg(currMsg);
-        currentNntpJob->setStatus(currMsg);
+        if ( currentNntpJob )
+          currentNntpJob->setStatus(currMsg);
       break;
       case KNProtocolClient::TSwriteGrouplist:
         currMsg = i18n(" Writing group list to disk...");
         knGlobals.setStatusMsg(currMsg);
-        currentNntpJob->setStatus(currMsg);
+        if ( currentNntpJob )
+          currentNntpJob->setStatus(currMsg);
       break;
       case KNProtocolClient::TSdownloadGrouplist:
         currMsg = i18n(" Downloading group list...");
         knGlobals.setStatusMsg(currMsg);
-        currentNntpJob->setStatus(currMsg);
+        if ( currentNntpJob )
+          currentNntpJob->setStatus(currMsg);
       break;
       case KNProtocolClient::TSdownloadNewGroups:
         currMsg = i18n(" Looking for new groups...");
         knGlobals.setStatusMsg(currMsg);
-        currentNntpJob->setStatus(currMsg);
+        if ( currentNntpJob )
+          currentNntpJob->setStatus(currMsg);
       break;
       case KNProtocolClient::TSdownloadDesc:
         currMsg = i18n(" Downloading group descriptions...");
         knGlobals.setStatusMsg(currMsg);
-        currentNntpJob->setStatus(currMsg);
+        if ( currentNntpJob )
+          currentNntpJob->setStatus(currMsg);
       break;
       case KNProtocolClient::TSdownloadNew:
         currMsg = i18n(" Downloading new headers...");
         knGlobals.setStatusMsg(currMsg);
-        currentNntpJob->setStatus(currMsg);
+        if ( currentNntpJob )
+          currentNntpJob->setStatus(currMsg);
       break;
       case KNProtocolClient::TSsortNew:
         currMsg = i18n(" Sorting...");
         knGlobals.setStatusMsg(currMsg);
-        currentNntpJob->setStatus(currMsg);
+        if ( currentNntpJob )
+          currentNntpJob->setStatus(currMsg);
       break;
       case KNProtocolClient::TSdownloadArticle:
         currMsg = i18n(" Downloading article...");
         knGlobals.setStatusMsg(currMsg);
-        currentNntpJob->setStatus(currMsg);
+        if ( currentNntpJob )
+          currentNntpJob->setStatus(currMsg);
       break;
       case KNProtocolClient::TSsendArticle:
         currMsg = i18n(" Sending article...");
         knGlobals.setStatusMsg(currMsg);
-        currentNntpJob->setStatus(currMsg);
+        if ( currentNntpJob )
+          currentNntpJob->setStatus(currMsg);
       break;
       case KNProtocolClient::TSjobStarted:
-        currentNntpJob->setProgress(0);
+        if ( currentNntpJob )
+          currentNntpJob->setProgress(0);
       break;
       case KNProtocolClient::TSprogressUpdate:
+        if ( currentNntpJob )
           currentNntpJob->setProgress(nntpClient->getProgressValue()/10);
       break;
     };
