--- knode/knode_config_privacy.desktop.orig	2008-10-15 15:49:11.000000000 +0300
+++ knode/knode_config_privacy.desktop	2008-10-15 15:49:31.000000000 +0300
@@ -56,7 +56,7 @@
 Name[sv]=Signera och verifiera
 Name[ta]=கையெழுத்து இடுதல்/சரிபார்த்தல்
 Name[tg]=Имзо ва тафтиш
-Name[tr]=İmzalama/Doğruluma
+Name[tr]=İmzalama/Doğrulama
 Name[uk]=Підписування і перевірка
 Name[zh_CN]=签名/校验
 Name[zh_TW]=簽署/檢查
