Index: kdepim-3.5.10/kmail/kmmainwidget.cpp
===================================================================
--- kdepim-3.5.10.orig/kmail/kmmainwidget.cpp
+++ kdepim-3.5.10/kmail/kmmainwidget.cpp
@@ -420,27 +420,27 @@ void KMMainWidget::readConfig(void)
         mFolderTree->addUnreadColumn( i18n("Unread"), 70 );
       else if (totalColumn == 1)
         mFolderTree->addTotalColumn( i18n("Total"), 70 );
-      else if (sizeColumn == 1)
-        mFolderTree->addSizeColumn( i18n("Size"), 70 );
+      //else if (sizeColumn == 1)
+        //mFolderTree->addSizeColumn( i18n("Size"), 70 );
 
       if (unreadColumn == 2)
         mFolderTree->addUnreadColumn( i18n("Unread"), 70 );
       else if (totalColumn == 2)
         mFolderTree->addTotalColumn( i18n("Total"), 70 );
-      else if (sizeColumn == 2)
-        mFolderTree->addSizeColumn( i18n("Size"), 70 );
+      //else if (sizeColumn == 2)
+        //mFolderTree->addSizeColumn( i18n("Size"), 70 );
 
       if (unreadColumn == 3)
         mFolderTree->addUnreadColumn( i18n("Unread"), 70 );
       else if (totalColumn == 3)
         mFolderTree->addTotalColumn( i18n("Total"), 70 );
-      else if (sizeColumn == 3)
-        mFolderTree->addSizeColumn( i18n("Size"), 70 );
+      //else if (sizeColumn == 3)
+        //mFolderTree->addSizeColumn( i18n("Size"), 70 );
 
       mUnreadColumnToggle->setChecked( mFolderTree->isUnreadActive() );
       mUnreadTextToggle->setChecked( !mFolderTree->isUnreadActive() );
       mTotalColumnToggle->setChecked( mFolderTree->isTotalActive() );
-      mSizeColumnToggle->setChecked( mFolderTree->isSizeActive() );
+      //mSizeColumnToggle->setChecked( mFolderTree->isSizeActive() );
 
       mFolderTree->updatePopup();
     }
