Index: kdepim-3.5.10/kmail/mailinglistpropertiesdialog.cpp
===================================================================
--- kdepim-3.5.10.orig/kmail/mailinglistpropertiesdialog.cpp
+++ kdepim-3.5.10/kmail/mailinglistpropertiesdialog.cpp
@@ -84,6 +84,7 @@ MailingListFolderPropertiesDialog::Maili
 		    label, SLOT(setEnabled(bool)) );
   groupLayout->addWidget( label, 4, 0 );
   mMLId = new QLabel( label, "", mlGroup );
+  mMLId->setWordWrap( true );
   groupLayout->addMultiCellWidget( mMLId, 4, 4, 1, 2 );
   mMLId->setEnabled( false );
 
