--- compiler/aslrestype2.c.orig	2005-09-08 12:05:25.000000000 +0200
+++ compiler/aslrestype2.c	2005-09-08 12:08:10.000000000 +0200
@@ -331,7 +331,7 @@
              */
             else if (ResSourceIndex)
             {
-                AslError (ASL_ERROR, ASL_MSG_RESOURCE_SOURCE,
+                AslError (ASL_WARNING, ASL_MSG_RESOURCE_SOURCE,
                     InitializerOp, NULL);
             }
             break;
@@ -545,7 +545,7 @@
              */
             else if (ResSourceIndex)
             {
-                AslError (ASL_ERROR, ASL_MSG_RESOURCE_SOURCE,
+                AslError (ASL_WARNING, ASL_MSG_RESOURCE_SOURCE,
                     InitializerOp, NULL);
             }
             break;
@@ -757,7 +757,7 @@
              */
             else if (ResSourceIndex)
             {
-                AslError (ASL_ERROR, ASL_MSG_RESOURCE_SOURCE,
+                AslError (ASL_WARNING, ASL_MSG_RESOURCE_SOURCE,
                     InitializerOp, NULL);
             }
             break;
@@ -1430,7 +1430,7 @@
              */
             else if (ResSourceIndex)
             {
-                AslError (ASL_ERROR, ASL_MSG_RESOURCE_SOURCE,
+                AslError (ASL_WARNING, ASL_MSG_RESOURCE_SOURCE,
                     InitializerOp, NULL);
             }
             break;
@@ -1639,7 +1639,7 @@
              */
             else if (ResSourceIndex)
             {
-                AslError (ASL_ERROR, ASL_MSG_RESOURCE_SOURCE,
+                AslError (ASL_WARNING, ASL_MSG_RESOURCE_SOURCE,
                     InitializerOp, NULL);
             }
             break;
@@ -1846,7 +1846,7 @@
              */
             else if (ResSourceIndex)
             {
-                AslError (ASL_ERROR, ASL_MSG_RESOURCE_SOURCE,
+                AslError (ASL_WARNING, ASL_MSG_RESOURCE_SOURCE,
                     InitializerOp, NULL);
             }
             break;
@@ -2034,7 +2034,7 @@
              */
             else if (ResSourceIndex)
             {
-                AslError (ASL_ERROR, ASL_MSG_RESOURCE_SOURCE,
+                AslError (ASL_WARNING, ASL_MSG_RESOURCE_SOURCE,
                     InitializerOp, NULL);
             }
             break;
@@ -2234,7 +2234,7 @@
              */
             else if (ResSourceIndex)
             {
-                AslError (ASL_ERROR, ASL_MSG_RESOURCE_SOURCE,
+                AslError (ASL_WARNING, ASL_MSG_RESOURCE_SOURCE,
                     InitializerOp, NULL);
             }
             break;
@@ -2431,7 +2431,7 @@
              */
             else if (ResSourceIndex)
             {
-                AslError (ASL_ERROR, ASL_MSG_RESOURCE_SOURCE,
+                AslError (ASL_WARNING, ASL_MSG_RESOURCE_SOURCE,
                     InitializerOp, NULL);
             }
             break;
@@ -2592,7 +2592,7 @@
              */
             else if (HasResSourceIndex)
             {
-                AslError (ASL_ERROR, ASL_MSG_RESOURCE_SOURCE,
+                AslError (ASL_WARNING, ASL_MSG_RESOURCE_SOURCE,
                     InitializerOp, NULL);
             }
             break;
