diff -urNp libical-0.43.orig/src/test/regression.c libical-0.43/src/test/regression.c
--- libical-0.43.orig/src/test/regression.c	2009-01-13 09:34:39.000000000 +0530
+++ libical-0.43/src/test/regression.c	2009-01-13 09:54:37.000000000 +0530
@@ -34,6 +34,8 @@
 #include <libical/ical.h>
 #include <libical/icalss.h>
 #include <libicalvcal/vobject.h>
+#include <libicalvcal/icalvcal.h>
+#include <libicalvcal/vcc.h>
 
 #include "regression.h"
 
