diff -urNp libical-0.43.orig/src/libicalss/Makefile.in libical-0.43/src/libicalss/Makefile.in
--- libical-0.43.orig/src/libicalss/Makefile.in	2009-01-13 09:34:39.000000000 +0530
+++ libical-0.43/src/libicalss/Makefile.in	2009-01-13 09:52:10.000000000 +0530
@@ -308,7 +308,7 @@ INCLUDES = \
 	$(BDB_INCLUDE)
 
 libicalss_la_LDFLAGS = -version-info 43:0:43
-libicalss_la_LIBADD = $(BDB_LIBFILES) 
+libicalss_la_LIBADD = $(BDB_LIBFILES) ../libical/libical.la
 @DEV_FALSE@libicalss_la_DEVSOURCES = icalsslexer.c icalssyacc.c
 @DEV_TRUE@libicalss_la_DEVSOURCES = icalsslexer.l icalssyacc.y
 libicalss_la_SOURCES = \
