Index: libsdl1.2-1.2.13/src/joystick/SDL_joystick.c
===================================================================
--- libsdl1.2-1.2.13.orig/src/joystick/SDL_joystick.c	2007-12-31 05:47:55.000000000 +0100
+++ libsdl1.2-1.2.13/src/joystick/SDL_joystick.c	2008-01-02 03:20:00.000000000 +0100
@@ -373,7 +373,7 @@
 	/* Remove joystick from list */
 	for ( i=0; SDL_joysticks[i]; ++i ) {
 		if ( joystick == SDL_joysticks[i] ) {
-			SDL_memcpy(&SDL_joysticks[i], &SDL_joysticks[i+1],
+			SDL_memmove(&SDL_joysticks[i], &SDL_joysticks[i+1],
 			       (SDL_numjoysticks-i)*sizeof(joystick));
 			break;
 		}
