diff -Nur SDL-1.2.13-old/src/audio/SDL_audio.c SDL-1.2.13/src/audio/SDL_audio.c
--- SDL-1.2.13-old/src/audio/SDL_audio.c	2008-07-02 16:05:10.000000000 +0300
+++ SDL-1.2.13/src/audio/SDL_audio.c	2008-07-02 16:06:45.000000000 +0300
@@ -39,6 +39,9 @@
 #if SDL_AUDIO_DRIVER_BSD
 	&BSD_AUDIO_bootstrap,
 #endif
+#if SDL_AUDIO_DRIVER_PULSE
+	&PULSE_bootstrap,
+#endif
 #if SDL_AUDIO_DRIVER_OSS
 	&DSP_bootstrap,
 	&DMA_bootstrap,
@@ -46,9 +49,6 @@
 #if SDL_AUDIO_DRIVER_ALSA
 	&ALSA_bootstrap,
 #endif
-#if SDL_AUDIO_DRIVER_PULSE
-	&PULSE_bootstrap,
-#endif
 #if SDL_AUDIO_DRIVER_QNXNTO
 	&QNXNTOAUDIO_bootstrap,
 #endif
