Index: pm-utils-1.3.0/src/Makefile.am
===================================================================
--- pm-utils-1.3.0.orig/src/Makefile.am
+++ pm-utils-1.3.0/src/Makefile.am
@@ -2,23 +2,20 @@ pm_libdir = $(libdir)/pm-utils
 pm_bindir = $(pm_libdir)/bin
 pm_sysconfdir = $(sysconfdir)/pm
 
-pm_bin_PROGRAMS =		\
-	pm-pmu			\
-	pm-reset-swap
+pm_bin_PROGRAMS = \
+	pm-pmu
 
 pm_pmu_SOURCES = pm-pmu.c
 
-pm_reset_swap_SOURCES = pm-reset-swap.c
-
 pm_bin_SCRIPTS = pm-action
 
 bin_SCRIPTS = pm-is-supported
 
-dist_bin_SCRIPTS = on_ac_power import-fdi-quirkdb
+dist_bin_SCRIPTS = on_ac_power
 
 sbin_SCRIPTS = pm-powersave
 
-script_in_files = pm-action.in pm-is-supported.in pm-powersave.in import-fdi-quirkdb.in
+script_in_files = pm-action.in pm-is-supported.in pm-powersave.in
 
 CLEANFILES =  $(script_in_files:.in=)
 
