Index: pardus/pisi/actionsapi/qt4.py
===================================================================
--- pardus.orig/pisi/actionsapi/qt4.py
+++ pardus/pisi/actionsapi/qt4.py
@@ -39,7 +39,7 @@ importdir = "%s/%s/imports" % (libdir, b
 plugindir = "%s/%s/plugins" % (libdir, basename)
 translationdir = "%s/translations" % datadir
 sysconfdir= "/etc"
-qmake = "%s/qmake" % bindir
+qmake = "%s/qmake-qt4" % bindir
 
 class ConfigureError(pisi.actionsapi.Error):
     def __init__(self, value=''):
