Index: pisi-2.2.14/pisi/configfile.py
===================================================================
--- pisi-2.2.14.orig/pisi/configfile.py
+++ pisi-2.2.14/pisi/configfile.py
@@ -45,8 +45,8 @@
 #index_dir = /var/cache/pisi/index
 #packages_dir = /var/cache/pisi/package
 #tmp_dir = /var/pisi
-#kde_dir = /usr/kde/4
-#qt_dir = /usr/qt/4
+#kde_dir = /usr/kde/3.5
+#qt_dir = /usr/qt/3
 
 import os
 import re
@@ -67,7 +67,7 @@ class GeneralDefaults:
     destinationdirectory = "/"
     autoclean = False
     distribution = "Pardus"
-    distribution_release = "2009"
+    distribution_release = "Corporate2"
     architecture = "i686"
     http_proxy = os.getenv("HTTP_PROXY") or None
     https_proxy = os.getenv("HTTPS_PROXY") or None
@@ -107,8 +107,8 @@ class DirectoriesDefaults:
     lock_dir = "/var/lock/subsys"
     index_dir = "/var/lib/pisi/index"
     tmp_dir =  "/var/pisi"
-    kde_dir = "/usr/kde/4"
-    qt_dir = "/usr/qt/4"
+    kde_dir = "/usr/kde/3.5"
+    qt_dir = "/usr/qt/3"
 
 class ConfigurationSection(object):
     """ConfigurationSection class defines a section in the configuration
@@ -206,7 +206,7 @@ class ConfigurationFile(object):
             r'(?P<padding>\s*)'
             r'(?P<value>.*)$'
             )
-        
+
         fp = open(self.filePath, "r+")
         # Default to " = " to match write(), but use the most recent
         # separator found if the file has any options.
@@ -313,4 +313,4 @@ class ConfigurationFile(object):
                 fp.write(sect.getvalue())
 
         fp.close()
-        
+
