--- a/pisi/actionsapi/kerneltools.py	(revision 35933)
+++ b/pisi/actionsapi/kerneltools.py	(revision 35934)
@@ -128,7 +128,10 @@
     autotools.make("ARCH=%s oldconfig" % __getKernelARCH())
 
     # Check configuration with listnewconfig
-    autotools.make("ARCH=%s listnewconfig" % __getKernelARCH())
+    try:
+        autotools.make("ARCH=%s listnewconfig" % __getKernelARCH())
+    except:
+        pass
 
 ###################################
 # Building and installation stuff #
