diff --git a/pisi/config.py b/pisi/config.py
index 31ebc09..feec568 100644
--- a/pisi/config.py
+++ b/pisi/config.py
@@ -49,19 +49,6 @@ class Config(object):
         self.set_options(options)
         self.values = pisi.configfile.ConfigurationFile("/etc/pisi/pisi.conf")
 
-        destdir = self.get_option('destdir')
-        if destdir:
-            if destdir.strip().startswith('/'):
-                self.destdir = destdir
-            else:
-                self.destdir = pisi.util.join_path(os.getcwd(), destdir)
-        else:
-            self.destdir = self.values.general.destinationdirectory
-
-        if not os.path.exists(self.destdir):
-            ctx.ui.warning( _('Destination directory %s does not exist. Creating it.') % self.destdir)
-            os.makedirs(self.destdir)
-
         # get the initial environment variables. this is needed for
         # build process.
         self.environ = copy.deepcopy(os.environ)
@@ -84,7 +71,18 @@ class Config(object):
     # pkg_x_dir: per package directory for storing info type x
 
     def dest_dir(self):
-        return self.destdir
+        destdir = self.get_option('destdir')
+        if destdir:
+            if not destdir.strip().startswith('/'):
+                destdir = pisi.util.join_path(os.getcwd(), destdir)
+        else:
+            destdir = self.values.general.destinationdirectory
+
+        if not os.path.exists(destdir):
+            ctx.ui.warning( _('Destination directory %s does not exist. Creating it.') % destdir)
+            os.makedirs(destdir)
+
+        return destdir
 
     def subdir(self, path):
         subdir = pisi.util.join_path(self.dest_dir(), path)
