Index: actionsapi/perlmodules.py
===================================================================
--- pisi/actionsapi/perlmodules.py	(revision 27623)
+++ pisi/actionsapi/perlmodules.py	(revision 27624)
@@ -62,7 +62,7 @@
         if system('make %s' % parameters):
             raise MakeError, _('Make failed.')
     else:
-        if system('perl Build build'):
+        if system('perl Build %s' % parameters):
             raise MakeError, _('perl build failed.')
 
 def install(parameters = 'install'):
