Index: pisi/operations/delta.py
===================================================================
--- pisi/operations/delta.py	(revision 26980)
+++ pisi/operations/delta.py	(revision 26981)
@@ -153,7 +153,7 @@
     for h in unchanged:
         for _file in files_new[h]:
             path = "/%s" % _file.path
-            if oct(stat.S_IMODE(os.stat(path)[stat.ST_MODE])) != _file.mode:
+            if os.path.exists(path) and oct(stat.S_IMODE(os.stat(path)[stat.ST_MODE])) != _file.mode:
                 permissions.append((path, int(_file.mode, 8)))
 
     return permissions
