Index: paprefs-0.9.9/src/paprefs.cc
===================================================================
--- paprefs-0.9.9.orig/src/paprefs.cc
+++ paprefs-0.9.9/src/paprefs.cc
@@ -237,10 +237,8 @@ void MainWindow::updateSensitive() {
     else
         zeroconfPublishInstallButton->show();
 
-    if (upnpAvailable)
-        upnpInstallButton->hide();
-    else
-        upnpInstallButton->show();
+    /* We don't have rygel in our repositories */
+    upnpInstallButton->hide();
 
     if (rtpRecvAvailable)
         rtpRecvInstallButton->hide();
