--- admin/cvs.sh.orig	2010-05-03 10:54:24.693199933 +0300
+++ admin/cvs.sh	2010-05-03 10:54:44.659201520 +0300
@@ -68,7 +68,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10*)
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10* | automake*1.11*)
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
