Index: alsa-driver/alsa-kernel/pci/hda/hda_beep.c
===================================================================
--- alsa-driver.orig/alsa-kernel/pci/hda/hda_beep.c
+++ alsa-driver/alsa-kernel/pci/hda/hda_beep.c
@@ -122,7 +122,7 @@ int snd_hda_attach_beep_device(struct hd
 	beep->nid = nid;
 	beep->dev = input_dev;
 	beep->codec = codec;
-	beep->enabled = 1;
+	beep->enabled = 0;
 	codec->beep = beep;
 
 	INIT_WORK(&beep->beep_work, &snd_hda_generate_beep);
