diff -ur kdepim-3.5.2.orig/kontact/src/kontactui.rc kdepim-3.5.2/kontact/src/kontactui.rc
--- kdepim-3.5.2.orig/kontact/src/kontactui.rc	2006-04-06 16:48:42.000000000 +0300
+++ kdepim-3.5.2/kontact/src/kontactui.rc	2006-04-06 16:48:22.000000000 +0300
@@ -37,7 +37,6 @@
 <ToolBar position="Top" noMerge="1" name="mainToolBar"><text>Main Toolbar</text>
   <Action name="action_new"/>
   <Merge/>
-  <Action name="help_whats_this"/>
 </ToolBar>
 <ToolBar position="Top" hidden="true" name="navigatorToolBar"><text>Navigator</text>
   <ActionList name="navigator_actionlist" />
