Index: kmail/kmkernel.cpp
===================================================================
--- kmail/kmkernel.cpp	(revision 473061)
+++ kmail/kmkernel.cpp	(working copy)
@@ -1420,7 +1420,7 @@
     KConfigGroupSaver saver(cfg, "Composer");
     if (cfg->readListEntry("pref-charsets").isEmpty())
     {
-      cfg->writeEntry("pref-charsets", "us-ascii,iso-8859-1,locale,utf-8");
+      cfg->writeEntry("pref-charsets", "locale,utf-8");
     }
   }
   readConfig();
