--- kitchensync/src/configguisyncmlhttp.cpp
+++ kitchensync/src/configguisyncmlhttp.cpp
@@ -121,7 +121,7 @@
   mGridLayout->addWidget( label, 5, 0 );
 
   mRecvLimit = new QSpinBox( optionWidget );
-  mRecvLimit->setMinValue( 1 );
+  mRecvLimit->setMinValue( 0 );
   mRecvLimit->setMaxValue( 65536 );
   mGridLayout->addWidget( mRecvLimit, 5, 1 );
 
@@ -130,7 +130,7 @@
   mGridLayout->addWidget( label, 6, 0 );
 
   mMaxObjSize = new QSpinBox( optionWidget );
-  mMaxObjSize->setMinValue( 1 );
+  mMaxObjSize->setMinValue( 0 );
   mMaxObjSize->setMaxValue( 65536 );
   mGridLayout->addWidget( mMaxObjSize, 6, 1 );
 
--- kitchensync/src/configguisyncmlobex.cpp
+++ kitchensync/src/configguisyncmlobex.cpp
@@ -153,7 +153,7 @@
   mGridLayout->addWidget( label, 14, 0 );
 
   mRecvLimit = new QSpinBox( optionsWidget );
-  mRecvLimit->setMinValue( 1 );
+  mRecvLimit->setMinValue( 0 );
   mRecvLimit->setMaxValue( 65536 );
   mGridLayout->addWidget( mRecvLimit, 14, 1 );
 
@@ -162,7 +162,7 @@
   mGridLayout->addWidget( label, 15, 0 );
 
   mMaxObjSize = new QSpinBox( optionsWidget );
-  mMaxObjSize->setMinValue( 1 );
+  mMaxObjSize->setMinValue( 0 );
   mMaxObjSize->setMaxValue( 65536 );
   mGridLayout->addWidget( mMaxObjSize, 15, 1 );
 
