Index: kmail/kmail_part.cpp
===================================================================
--- kmail/kmail_part.cpp	(revision 627754)
+++ kmail/kmail_part.cpp	(working copy)
@@ -146,6 +146,9 @@
 
   KSettings::Dispatcher::self()->registerInstance( KMailFactory::instance(), mKMailKernel,
                                                    SLOT( slotConfigChanged() ) );
+
+  if ( mKMailKernel->firstStart() )
+    AccountWizard::start( mKMailKernel, this );
 }
 
 KMailPart::~KMailPart()
