diff -Nur nmap-5.30BETA1-old/ncat/Makefile.in nmap-5.30BETA1/ncat/Makefile.in
--- nmap-5.30BETA1-old/ncat/Makefile.in	2010-07-07 19:50:14.817507243 +0300
+++ nmap-5.30BETA1/ncat/Makefile.in	2010-07-07 20:03:01.979382693 +0300
@@ -136,7 +136,7 @@
 install: $(TARGET)
 	@echo Installing Ncat;
 	$(SHTOOL) mkdir -f -p -m 755 $(DESTDIR)$(bindir) $(DESTDIR)$(pkgdatadir) $(DESTDIR)$(mandir)/man1
-	$(INSTALL) -c -m 755 -s ncat $(DESTDIR)$(bindir)/ncat
+	$(INSTALL) -c -m 755 ncat $(DESTDIR)$(bindir)/ncat
 	$(INSTALL) -c -m 644 $(DATAFILES) $(DESTDIR)$(pkgdatadir)/
 	$(INSTALL) -c -m 644 docs/$(TARGET).1 $(DESTDIR)$(mandir)/man1/$(TARGET).1
 
