diff -Nur nmap-5.20-old/nse_main.lua nmap-5.20/nse_main.lua
--- nmap-5.20-old/nse_main.lua	2010-01-21 12:22:27.944163474 +0200
+++ nmap-5.20/nse_main.lua	2010-01-21 12:23:32.267169278 +0200
@@ -89,7 +89,7 @@
 do -- Append the nselib directory to the Lua search path
   local t, path = assert(cnse.fetchfile_absolute("nselib/"));
   assert(t == "directory", "could not locate nselib directory!");
-  package.path = package.path..";"..path.."?.lua";
+  package.path = path.."?.lua;"..package.path;
 end
 
 local stdnse = require "stdnse";
