diff -Naur kbarcode/kbarcode-batch.desktop kbarcode.new/kbarcode-batch.desktop
--- kbarcode/kbarcode-batch.desktop	2006-01-14 14:21:59.000000000 +0200
+++ kbarcode.new/kbarcode-batch.desktop	2006-05-19 21:37:12.000000000 +0300
@@ -6,7 +6,9 @@
 DocPath=kbarcode/index.html
 Comment=Barcode and label printing tool (batch printing mode)
 Comment[de]=Werkzeug zum Barcode und Etiketten Druck (Stapelverarbeitung)
+Comment[tr]=Barkod ve etiket yazdırma aracı (toplu yazdırma modu)
 Terminal=false
 Name=KBarcode Batch Printing
 Name[de]=KBarcode Batch Druck
+Name[tr]=KBarcode Toplu Yazdırma
 Categories=Qt;KDE;Office;
diff -Naur kbarcode/kbarcode.desktop kbarcode.new/kbarcode.desktop
--- kbarcode/kbarcode.desktop	2006-01-14 14:24:29.000000000 +0200
+++ kbarcode.new/kbarcode.desktop	2006-05-19 21:41:10.000000000 +0300
@@ -7,7 +7,9 @@
 DocPath=kbarcode/index.html
 Comment=Barcode and label printing tool
 Comment[de]=Werkzeug zum Barcode und Etiketten Druck
+Comment[tr]=Barkod ve etiket yazdırma aracı
 Terminal=0
 Name=KBarcode
 Name[de]=KBarcode
+Name[tr]=KBarcode
 Categories=Qt;KDE;Office;
diff -Naur kbarcode/kbarcode-editor.desktop kbarcode.new/kbarcode-editor.desktop
--- kbarcode/kbarcode-editor.desktop	2006-01-14 14:23:16.000000000 +0200
+++ kbarcode.new/kbarcode-editor.desktop	2006-05-19 21:38:00.000000000 +0300
@@ -6,9 +6,11 @@
 DocPath=kbarcode/index.html
 Comment=Barcode and label printing tool (editor)
 Comment[de]=Werkzeug zum Barcode und Etiketten Druck (Editor)
+Comment[tr]=Barkod ve etiket yazdırma aracı (düzenleyici)
 Terminal=false
 Name=KBarcode Label Editor
 Name[de]=KBarcode Label Editor
+Name[tr]=KBarcode Etiket Düzenleyici
 Categories=Qt;KDE;Office;
 
 
diff -Naur kbarcode/kbarcode-label.desktop kbarcode.new/kbarcode-label.desktop
--- kbarcode/kbarcode-label.desktop	2004-06-30 11:18:36.000000000 +0300
+++ kbarcode.new/kbarcode-label.desktop	2006-05-19 21:39:11.000000000 +0300
@@ -5,5 +5,6 @@
 Icon=kbarcode
 MimeType=application/kbarcode-label
 Name[de]=kbarcode-label.desktop
+Name[tr]=kbarcode-etiket.desktop
 Patterns=*.kbarcode;*.KBARCODE;
 
diff -Naur kbarcode/kbarcode-single.desktop kbarcode.new/kbarcode-single.desktop
--- kbarcode/kbarcode-single.desktop	2006-01-14 14:40:59.000000000 +0200
+++ kbarcode.new/kbarcode-single.desktop	2006-05-19 21:40:09.000000000 +0300
@@ -6,7 +6,9 @@
 DocPath=kbarcode/index.html
 Comment=Barcode generation tool
 Comment[de]=Barcode Erstellung
+Comment[tr]=Barkod oluşturma aracı
 Terminal=0
 Name=KBarcode Barcode Generator
 Name[de]=KBarcode Barcode Generator
+Name[tr]=KBarcode Barkod Üreteci
 Categories=Qt;KDE;Office;
