diff -Nur kradio-snapshot_2006_09_20-old/kradio3/plugins/v4lradio/v4lcfg_interfaces.cpp kradio-snapshot_2006_09_20/kradio3/plugins/v4lradio/v4lcfg_interfaces.cpp
--- kradio-snapshot_2006_09_20-old/kradio3/plugins/v4lradio/v4lcfg_interfaces.cpp	2008-05-15 15:28:20.000000000 +0300
+++ kradio-snapshot_2006_09_20/kradio3/plugins/v4lradio/v4lcfg_interfaces.cpp	2008-05-15 15:30:36.000000000 +0300
@@ -116,7 +116,7 @@
                     setVolumeZeroOnPowerOff(a)
                 )
 
-static QString defaultRDev("/dev/radio");
+static QString defaultRDev("/dev/radio0");
 // static QString defaultMDev("/dev/mixer");
 
 IF_IMPL_QUERY   (   const QString &IV4LCfgClient::queryRadioDevice (),
