diff -dur check-0.9.6.ORIG/src/Makefile.am check-0.9.6/src/Makefile.am
--- check-0.9.6.ORIG/src/Makefile.am	2008-12-29 06:48:45.000000000 -0700
+++ check-0.9.6/src/Makefile.am	2009-04-07 08:51:13.000000000 -0600
@@ -7,7 +7,7 @@
 
 EXTRA_DIST		= check.h.in
 
-AM_CFLAGS		= @GCOV_CFLAGS@ -ansi -pedantic 
+AM_CFLAGS		= @GCOV_CFLAGS@ -pedantic 
 
 
 CFILES =\
diff -dur check-0.9.6.ORIG/src/Makefile.in check-0.9.6/src/Makefile.in
--- check-0.9.6.ORIG/src/Makefile.in	2008-12-29 07:07:01.000000000 -0700
+++ check-0.9.6/src/Makefile.in	2009-04-07 08:51:22.000000000 -0600
@@ -206,7 +206,7 @@
 noinst_LTLIBRARIES = libcheckinternal.la
 include_HEADERS = check.h
 EXTRA_DIST = check.h.in
-AM_CFLAGS = @GCOV_CFLAGS@ -ansi -pedantic 
+AM_CFLAGS = @GCOV_CFLAGS@ -pedantic 
 CFILES = \
 	check.c		\
 	check_error.c	\
