diff -Nur libquicktime-1.1.2-old/plugins/x264/lqt_x264.c libquicktime-1.1.2/plugins/x264/lqt_x264.c
--- libquicktime-1.1.2-old/plugins/x264/lqt_x264.c	2009-10-20 19:58:42.248089435 +0300
+++ libquicktime-1.1.2/plugins/x264/lqt_x264.c	2009-10-20 19:59:27.972105159 +0300
@@ -95,7 +95,7 @@
       .help_string = TRS("Influences how often B-frames are used"),
     },
     {
-      .name =        "x264_b_bframe_pyramid",
+      .name =        "x264_i_bframe_pyramid",
       .real_name =   TRS("B-frame pyramid"),
       .type =        LQT_PARAMETER_INT,
       .val_default = { .val_int = 0 },
diff -Nur libquicktime-1.1.2-old/plugins/x264/x264.c libquicktime-1.1.2/plugins/x264/x264.c
--- libquicktime-1.1.2-old/plugins/x264/x264.c	2009-10-20 19:58:42.248089435 +0300
+++ libquicktime-1.1.2/plugins/x264/x264.c	2009-10-20 19:59:27.981088994 +0300
@@ -67,7 +67,7 @@
   lqt_dump("  i_bframe:                    %d\n", params->i_bframe);          // 0.. X264_BFRAME_MAX
   lqt_dump("  b_bframe_adaptive:           %d\n", params->b_bframe_adaptive);
   lqt_dump("  i_bframe_bias:               %d\n", params->i_bframe_bias);
-  lqt_dump("  b_bframe_pyramid:            %d\n", params->b_bframe_pyramid);
+  lqt_dump("  i_bframe_pyramid:            %d\n", params->i_bframe_pyramid);
   
   lqt_dump("  b_deblocking_filter:         %d\n", params->b_deblocking_filter);
   lqt_dump("  i_deblocking_filter_alphac0: %d\n", params->i_deblocking_filter_alphac0); // -6..6
@@ -782,7 +782,7 @@
   ENUMPARAM("x264_i_bframe_adaptive", codec->params.i_bframe_adaptive, bframe_adaptives);
 #endif
   INTPARAM("x264_i_bframe_bias", codec->params.i_bframe_bias);
-  INTPARAM("x264_b_bframe_pyramid", codec->params.b_bframe_pyramid);
+  INTPARAM("x264_i_bframe_pyramid", codec->params.i_bframe_pyramid);
 
   ENUMPARAM("x264_i_rc_method", codec->params.rc.i_rc_method, rc_methods);
   INTPARAM("x264_i_bitrate", codec->params.rc.i_bitrate);
