diff -ur libgnome-2.28.0.orig/schemas/desktop_gnome_interface.schemas.in.in libgnome-2.28.0/schemas/desktop_gnome_interface.schemas.in.in
--- libgnome-2.28.0.orig/schemas/desktop_gnome_interface.schemas.in.in	2009-08-09 12:48:44.000000000 +0300
+++ libgnome-2.28.0/schemas/desktop_gnome_interface.schemas.in.in	2010-02-06 21:34:28.786917666 +0200
@@ -41,7 +41,7 @@
       <applyto>/desktop/gnome/interface/can_change_accels</applyto>
       <owner>gnome</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
         <short>Can Change Accels</short>
         <long>Whether the user can dynamically type a new accelerator when 
@@ -53,7 +53,7 @@
       <applyto>/desktop/gnome/interface/toolbar_style</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>both-horiz</default>
+      <default>large-toolbar</default>
       <locale name="C">
         <short>Toolbar Style</short>
         <long>Toolbar Style. Valid values are "both", "both-horiz", "icons", and
@@ -65,7 +65,7 @@
       <applyto>/desktop/gnome/interface/menus_have_icons</applyto>
       <owner>gnome</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
         <short>Menus Have Icons</short>
         <long>Whether menus may display an icon next to a menu entry.</long>
