Index: zarafa-6.40.0/src/provider/server/Makefile.in
===================================================================
--- zarafa-6.40.0.orig/src/provider/server/Makefile.in
+++ zarafa-6.40.0/src/provider/server/Makefile.in
@@ -274,7 +274,7 @@ zarafa_server_LDADD = $(UUID_LIBS) $(INT
 	${top_builddir}/common/libcommon_util.la								\
 	${top_builddir}/common/libcommon_mapi.la								\
 	${top_builddir}/common/libcommon_ssl.la									\
-	$(MYSQL_LIBS) $(SSL_LIBS) $(PAM_LIBS) $(KRB5_LIBS) $(TCMALLOC_LIBS) -lrt
+	$(MYSQL_LIBS) $(UUID_LIBS) $(SSL_LIBS) $(PAM_LIBS) $(KRB5_LIBS) $(TCMALLOC_LIBS) -lrt
 
 zarafa_server_SOURCES = ECServer.cpp  \
 						ECSoapServerConnection.h ECSoapServerConnection.cpp \
Index: zarafa-6.40.0/src/provider/server/Makefile.am
===================================================================
--- zarafa-6.40.0.orig/src/provider/server/Makefile.am
+++ zarafa-6.40.0/src/provider/server/Makefile.am
@@ -18,7 +18,7 @@ zarafa_server_LDADD = $(UUID_LIBS) $(INT
 	${top_builddir}/common/libcommon_util.la								\
 	${top_builddir}/common/libcommon_mapi.la								\
 	${top_builddir}/common/libcommon_ssl.la									\
-	$(MYSQL_LIBS) $(SSL_LIBS) $(PAM_LIBS) $(KRB5_LIBS) $(TCMALLOC_LIBS) -lrt
+	$(MYSQL_LIBS) $(UUID_LIBS) $(SSL_LIBS) $(PAM_LIBS) $(KRB5_LIBS) $(TCMALLOC_LIBS) -lrt
 
 zarafa_server_SOURCES = ECServer.cpp  \
 						ECSoapServerConnection.h ECSoapServerConnection.cpp \
