From 125d202e4af6c58383eac073a0b61fd82f4b0b01 Mon Sep 17 00:00:00 2001
From: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date: Fri, 30 Oct 2009 13:28:45 +0000
Subject: [PATCH] [G45] Add MAX helper macro.

---
 i965_drv_video/intel_driver.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/i965_drv_video/intel_driver.h b/i965_drv_video/intel_driver.h
index f768703..84df0aa 100644
--- a/i965_drv_video/intel_driver.h
+++ b/i965_drv_video/intel_driver.h
@@ -43,6 +43,7 @@ struct intel_batchbuffer;
 
 #define ALIGN(i, n)    (((i) + (n) - 1) & ~((n) - 1))
 #define MIN(a, b) ((a) < (b) ? (a) : (b))
+#define MAX(a, b) ((a) > (b) ? (a) : (b))
 
 #define SET_BLOCKED_SIGSET()   do {     \
         sigset_t bl_mask;               \
-- 
1.5.4.3

