Index: libical-0.44/src/libicalss/Makefile.in
===================================================================
--- libical-0.44.orig/src/libicalss/Makefile.in
+++ libical-0.44/src/libicalss/Makefile.in
@@ -307,7 +307,7 @@ INCLUDES = \
 	$(BDB_INCLUDE)
 
 libicalss_la_LDFLAGS = -version-info 44:0:44
-libicalss_la_LIBADD = $(BDB_LIBFILES) 
+libicalss_la_LIBADD = $(BDB_LIBFILES) ../libical/libical.la
 @DEV_FALSE@libicalss_la_DEVSOURCES = icalsslexer.c icalssyacc.c
 @DEV_TRUE@libicalss_la_DEVSOURCES = icalsslexer.l icalssyacc.y
 libicalss_la_SOURCES = \
