diff -Nur ffmpeg-old/cmdutils.c ffmpeg/cmdutils.c
--- ffmpeg-old/cmdutils.c	2009-10-16 20:13:45.101779652 +0300
+++ ffmpeg/cmdutils.c	2009-10-16 20:17:21.236132311 +0300
@@ -333,10 +333,12 @@
 {
     fprintf(stderr, "%s version " FFMPEG_VERSION ", Copyright (c) %d-%d Fabrice Bellard, et al.\n",
             program_name, program_birth_year, this_year);
+    /*
     fprintf(stderr, "  built on %s %s with %s %s\n",
             __DATE__, __TIME__, CC_TYPE, CC_VERSION);
     fprintf(stderr, "  configuration: " FFMPEG_CONFIGURATION "\n");
     print_all_lib_versions(stderr, 1);
+    */
 }
 
 void show_version(void) {
@@ -347,7 +349,7 @@
 void show_license(void)
 {
     printf(
-#if CONFIG_NONFREE
+#if 0
     "This version of %s has nonfree parts compiled in.\n"
     "Therefore it is not legally redistributable.\n",
     program_name
