--- src/cf.data.pre	2007-11-16 00:33:05.000000000 +0100
+++ src/cf.data.pre.oden	2008-01-27 20:40:03.000000000 +0100
@@ -3238,11 +3238,14 @@
 	UID/GID to the user specified below.  The default is to change
 	to UID of @DEFAULT_CACHE_EFFECTIVE_USER@.
 	see also; cache_effective_group
+NOCOMMENT_START
+cache_effective_user @DEFAULT_CACHE_EFFECTIVE_USER@
+NOCOMMENT_END
 DOC_END
 
 NAME: cache_effective_group
 TYPE: string
-DEFAULT: none
+DEFAULT: @DEFAULT_CACHE_EFFECTIVE_USER@
 LOC: Config.effectiveGroup
 DOC_START
 	Squid sets the GID to the effective user's default group ID
@@ -3260,6 +3263,9 @@
 	This option is not recommended by the Squid Team.
 	Our preference is for administrators to configure a secure
 	user account for squid with UID/GID matching system policies.
+NOCOMMENT_START
+cache_effective_group @DEFAULT_CACHE_EFFECTIVE_USER@
+NOCOMMENT_END
 DOC_END
 
 NAME: httpd_suppress_version_string
