--- squid-3.1.0.16-20100224/tools/Makefile.am	2010-02-23 20:39:59.000000000 -0600
+++ squid-3.1.0.16-20100224.dlucio/tools/Makefile.am	2010-02-24 11:50:41.427836247 -0600
@@ -14,6 +14,8 @@
 
 SUBDIRS		= 
 
+DEFAULT_CACHEMGR_CONFIG = $(sysconfdir)/cachemgr.conf
+
 bin_PROGRAMS = \
 	squidclient
 
@@ -23,8 +25,6 @@
 man_MANS = \
 	squidclient.1
 
-DEFAULT_CACHEMGR_CONFIG = $(sysconfdir)/cachemgr.conf
-
 squidclient_SOURCES = squidclient.cc
 cachemgr__CGIEXT__SOURCES = cachemgr.cc
 cachemgr__CGIEXT__CXXFLAGS = -DDEFAULT_CACHEMGR_CONFIG=\"$(DEFAULT_CACHEMGR_CONFIG)\" $(AM_CXXFLAGS)
