diff -Naur frescobaldi-0.7.3.orig/data/frescobaldi.desktop frescobaldi-0.7.3/data/frescobaldi.desktop
--- frescobaldi-0.7.3.orig/data/frescobaldi.desktop	2009-01-10 08:40:29.000000000 +0200
+++ frescobaldi-0.7.3/data/frescobaldi.desktop	2009-01-25 20:28:29.000000000 +0200
@@ -1,5 +1,6 @@
 [Desktop Entry]
 GenericName=LilyPond Music Editor
+GenericName[tr]=LilyPond Müzik Editörü
 GenericName[nl]=Muziek-editor voor LilyPond
 GenericName[es]=Editor de música para LilyPond
 Name=Frescobaldi
@@ -11,5 +12,5 @@
 Type=Application
 Terminal=false
 InitialPreference=8
-Categories=Qt;KDE;Application;Audio;TextEditor;
+Categories=Qt;KDE;Application;AudioVideo;Audio;TextEditor;
 X-Accepts-TextEditUrl=True
