diff -Nur kradio-snapshot_2006_09_20-old/kradio3/plugins/lirc/lircsupport.cpp kradio-snapshot_2006_09_20/kradio3/plugins/lirc/lircsupport.cpp
--- kradio-snapshot_2006_09_20-old/kradio3/plugins/lirc/lircsupport.cpp	2008-05-15 15:40:07.000000000 +0300
+++ kradio-snapshot_2006_09_20/kradio3/plugins/lirc/lircsupport.cpp	2008-05-15 15:40:26.000000000 +0300
@@ -29,6 +29,8 @@
 #include <kaboutdata.h>
 #include <kstandarddirs.h>
 
+#include <cstdlib>
+
 #include "../../src/interfaces/errorlog-interfaces.h"
 #include "../../src/interfaces/radiodevice_interfaces.h"
 #include "../../src/libkradio/stationlist.h"
