Index: hplip-3.9.12/installer/core_install.py
===================================================================
--- hplip-3.9.12.orig/installer/core_install.py
+++ hplip-3.9.12/installer/core_install.py
@@ -600,7 +600,7 @@ class CoreInstall(object):
 
         if not found:
             try:
-                name = file('/etc/issue', 'r').read().lower().strip()
+                name = "Pardus".lower().strip()
             except IOError:
                 # Some O/Ss don't have /etc/issue (Mac)
                 self.distro, self.distro_version = DISTRO_UNKNOWN, '0.0'
