Index: aria2-1.10.0/src/Makefile.am
===================================================================
--- aria2-1.10.0.orig/src/Makefile.am
+++ aria2-1.10.0/src/Makefile.am
@@ -236,9 +236,7 @@ endif # HAVE_LIBEXPAT
 
 endif # ENABLE_XML_RPC
 
-if HAVE_SOME_FALLOCATE
 SRCS += FallocFileAllocationIterator.cc FallocFileAllocationIterator.h
-endif # HAVE_SOME_FALLOCATE
 
 if HAVE_EPOLL
 SRCS += EpollEventPoll.cc EpollEventPoll.h
