diff -p -up samba-3.2.6/source/Makefile.in.cifslinkorder samba-3.2.6/source/Makefile.in
--- samba-3.2.6/source/Makefile.in.cifslinkorder	2008-12-13 20:09:09.000000000 +0100
+++ samba-3.2.6/source/Makefile.in	2008-12-13 20:09:09.000000000 +0100
@@ -1329,7 +1329,7 @@ bin/umount.cifs@EXEEXT@: $(BINARY_PREREQ
 bin/cifs.upcall@EXEEXT@: $(BINARY_PREREQS) $(CIFS_UPCALL_OBJ) $(LIBSMBCLIENT_OBJ1) @BUILD_POPT@ @LIBTALLOC_SHARED@ @LIBTDB_SHARED@ @LIBWBCLIENT_SHARED@
 	@echo Linking $@
 	@$(CC) $(FLAGS) -o $@ $(CIFS_UPCALL_OBJ) $(DYNEXP) $(LDFLAGS) \
-		-lkeyutils $(LIBS) $(LIBSMBCLIENT_OBJ1) $(KRB5LIBS) \
+		$(LIBSMBCLIENT_OBJ1) -lkeyutils $(LIBS) $(KRB5LIBS) \
 		$(LDAP_LIBS) $(POPT_LIBS) $(LIBTALLOC_LIBS) $(WINBIND_LIBS) \
 		$(LIBTDB_LIBS) $(NSCD_LIBS)
 
