--- dbus/dbus-internals.h	2006-10-28 03:41:37.000000000 +0200
+++ dbus/dbus-internals.h	2006-11-12 20:00:01.000000000 +0100
@@ -136,20 +136,20 @@
 #define _dbus_return_if_fail(condition) do {                                       \
    _dbus_assert ((*(const char*)_DBUS_FUNCTION_NAME) != '_');                      \
   if (!(condition)) {                                                              \
-    _dbus_warn_check_failed (_dbus_return_if_fail_warning_format,                  \
-                             _DBUS_FUNCTION_NAME, #condition, __FILE__, __LINE__); \
+    _dbus_warn (_dbus_return_if_fail_warning_format,                               \
+                _DBUS_FUNCTION_NAME, #condition, __FILE__, __LINE__);              \
     return;                                                                        \
   } } while (0)
 
 #define _dbus_return_val_if_fail(condition, val) do {                                   \
    _dbus_assert ((*(const char*)_DBUS_FUNCTION_NAME) != '_');                           \
   if (!(condition)) {                                                                   \
-    _dbus_warn_check_failed (_dbus_return_if_fail_warning_format,                       \
-                             _DBUS_FUNCTION_NAME, #condition, __FILE__, __LINE__);      \
+    _dbus_warn (_dbus_return_if_fail_warning_format,                                    \
+                _DBUS_FUNCTION_NAME, #condition, __FILE__, __LINE__);                   \
     return (val);                                                                       \
   } } while (0)
 
-#endif /* !DBUS_DISABLE_ASSERT */
+#endif /* !DBUS_DISABLE_CHECKS */
 
 #define _DBUS_N_ELEMENTS(array) ((int) (sizeof ((array)) / sizeof ((array)[0])))
 
