--- a/kdeprint/testprint.ps	2005-09-10 11:27:43.000000000 +0300
+++ b/kdeprint/testprint.ps 2010-01-03 22:45:58.000000000 +0200
@@ -3,87 +3,89 @@
 %%Pages: 1
 %%LanguageLevel: 1
 %%DocumentData: Clean7Bit
-%%DocumentSuppliedResources: procset testprint/1.0
+%%DocumentSuppliedResources: procset testprint/1.3
 %%DocumentNeededResources: font Helvetica Helvetica-Bold Times-Roman
-%%Creator: Michael Sweet, Easy Software Products
-%%CreationDate: May 11, 1999
+%%Creator: Michael Sweet, Apple Inc.
+%%CreationDate: D:20090113092400+0800
 %%Title: Test Page
 %%EndComments
 %%BeginProlog
-%%BeginResource procset testprint 1.1 0
+%%BeginResource procset testprint 1.3 0
 %
 %   PostScript test page for the Common UNIX Printing System ("CUPS").
 %
-%   Copyright 1993-2001 Easy Software Products
+%   Copyright 2007-2009 Apple Inc.
+%   Copyright 1993-2007 Easy Software Products
 %
 %   These coded instructions, statements, and computer programs are the
-%   property of Easy Software Products and are protected by Federal
-%   copyright law.  Distribution and use rights are outlined in the file
-%   "LICENSE.txt" which should have been included with this file.  If this
-%   file is missing or damaged please contact Easy Software Products
-%   at:
+%   property of Apple Inc. and are protected by Federal copyright law.
+%   Distribution and use rights are outlined in the file "LICENSE.txt"
+%   which is included with the CUPS source distribution.
 %
-%       Attn: CUPS Licensing Information
-%       Easy Software Products
-%       44141 Airport View Drive, Suite 204
-%       Hollywood, Maryland 20636-3111 USA
-%
-%       Voice: (301) 373-9603
-%       EMail: cups-info@cups.org
-%         WWW: http://www.cups.org
-%
-/OCTANT {			% Draw a color wheel OCTANT...
-				% (name) radius r g b OCTANT -
+/SEXTANT {			% Draw a color wheel sextant...
+				% (name) white radius r g b SEXTANT -
   % Loop through 100 shades...
   0 0.010101 0.98 {
     % Set the color...
-    3 index 1 eq		% R == 1?
-    3 index 1 eq		% G == 1?
-    3 index 1 eq		% B == 1?
-    and and {
-      0 index 4 index mul 	% R * val
-      1 index 4 index mul 	% G * val
-      2 index 4 index mul 	% B * val
+    dup 0.75 le {		% Get "white" value
+      % Start from black
+      dup 0.75 div		% val2 = val / 0.75
+
+      0 index 5 index mul 	% R = R * val2
+      1 index 5 index mul 	% G = G * val2
+      2 index 5 index mul 	% B = B * val2
+
+      4 -1 roll pop		% Discard val2
     } {
-      0 index 4 index mul 	% R * val
-      1 index neg 1 add	add	% + (1 - val)
-      1 index 4 index mul 	% G * val
-      2 index neg 1 add	add	% + (1 - val)
-      2 index 4 index mul 	% B * val
-      3 index neg 1 add	add	% + (1 - val)
+      % Fade to white
+      dup neg 1 add 4 mul	% val2 = (1 - val) * 4
+
+      0 index 5 index mul 	% R = R * val2
+      1 index neg 1 add add	% + (1 - val2)
+      1 index 5 index mul 	% G = G * val2
+      2 index neg 1 add add	% + (1 - val2)
+      2 index 5 index mul 	% B = B * val2
+      3 index neg 1 add add	% + (1 - val2)
+
+      4 -1 roll pop		% Discard val2
     } ifelse
-    setrgbcolor
+    setrgbcolor			% Set the color...
+
+    % Draw the polygon...
+    newpath			% Start a new path...
+    dup 5 index mul		% r1 = radius * val
+    0 0 3 -1 roll 0 60 arc	% Draw the inner arc
 
-    % Draw a polygon...
-    dup 5 index mul dup 0	% x1, y1
-    moveto
-    0.707106781 mul dup lineto	% x2, y2
-
-    0.010101 add 4 index mul dup % x3
-    0.707106781 mul dup lineto	% x3, y3
-    0 lineto			% x4, y4
-    closepath
-    fill
+    dup 0.010101 add 5 index mul% r2 = (radius + 0.010101) * val
+    0 0 3 -1 roll 60 0 arcn	% Draw the outer arc
+
+    closepath			% Close the path
+    fill			% Fill it...
+
+    pop				% Pop value...
   } for
 
   % Draw a line around the polygons...
-  pop pop pop dup
-  0 setgray
-  0 0 moveto
-  dup 0 lineto
-  0.707106781 mul dup lineto
-  closepath
-  stroke
+  pop pop pop dup		% Pop R, G, B, start
+  0 setgray			% Black
+  newpath
+  0 0 moveto			% Center
+  0 0 3 -1 roll 0 60 arc	% Arc around octant
+  closepath			% Back to center
+  stroke			% Stroke it...
 
   % Draw the label...
-  0 exch dup -9 div exch	% text offset = 0, -radius/9
-  dup 0.923879532 mul		% x = radius * cos(22.5)
-  exch 0.382683432 mul		% y = radius * cos(22.5)
-  moveto			% position label
+  dup				% Save radius
+  dup 30 cos mul		% X = radius * cos(30)
+  exch 30 sin mul		% Y = radius * sin(30)
+  moveto			% Position label
+
   gsave
-    22.5 rotate			% rotate label
-    rmoveto			% offset label
-    show			% show label
+    30 rotate			% Rotate label
+    dup 0.05 mul		% Offset to the right
+    exch -0.05 mul		% and down...
+    rmoveto			% Offset label
+    show			% Show label
   grestore
 } bind def
 /CENTER {			% Draw centered text
@@ -125,81 +127,6 @@
   dup 1.2 mul moveto
   (System) show
 } bind def
-/ESPLOGO {			% Draw the ESP logo
-				% height ESPLOGO
-  % Compute the size of the logo...
-  0 0
-  2 index 1.5 mul 3 index
-
-  % Do the "metallic" fill from 10% black to 40% black...
-  1 -0.001 0 {
-    dup			% loopval
-    -0.15 mul		% loopval * -0.15
-    0.9 add		% 0.9 - loopval * 0.15
-    setgray		% set gray shade
-
-    0			% x
-    1 index neg		% loopval
-    1 add		% 1 - loopval
-    3 index		% height
-    mul			% height * (1 - loopval)
-    moveto		% starting point
-
-    dup			% loopval
-    3 index		% width
-    mul			% loopval * width
-    2 index		% height
-    lineto		% Next point
-
-    0			% x
-    2 index		% height
-    lineto		% Next point
-
-    closepath
-    fill
-
-    dup			% loopval
-    0.15 mul		% loopval * 0.15
-    0.6 add		% 0.6 + loopval * 0.15
-    setgray
-
-    dup			% loopval
-    neg 1 add		% 1 - loopval
-    3 index		% width
-    mul			% (1 - loopval) * width
-    0			% y
-    moveto		% Starting point
-
-    2 index		% width
-    exch		% loopval
-    2 index		% height
-    mul			% loopval * height
-    lineto		% Next point
-
-    1 index		% width
-    0			% y
-    lineto		% Next point
-
-    closepath
-    fill
-  } for
-
-  0 setgray rectstroke
-
-  /Helvetica-BoldOblique findfont 1 index 3 div scalefont setfont
-  dup 40 div
-
-  dup 4 mul 1 index 25 mul moveto (E) show
-  dup 10 mul 1 index 15 mul moveto (S) show
-  dup 16 mul 1 index 5 mul moveto (P) show
-
-  /Helvetica-BoldOblique findfont 2 index 5 div scalefont setfont
-  dup 14 mul 1 index 29 mul moveto (asy) show
-  dup 20 mul 1 index 19 mul moveto (oftware) show
-  dup 26 mul 1 index 9 mul moveto (roducts) show
-
-  pop
-} bind def
 %%EndResource
 %%EndProlog
 %%Page: 1 1
@@ -214,7 +141,6 @@
 
   4 setlinewidth			% Draw wide lines
   0 setgray closepath stroke		% Draw a clipping rectangle
-  1 setlinewidth			% Draw normal lines
 
   /pageWidth pageRight pageLeft sub def	% pageWidth = pageRight - pageLeft
   /pageHeight pageTop pageBottom sub def% pageHeight = pageTop - pageBottom
@@ -240,15 +166,115 @@
   /smallFont /Times-Roman findfont	% smallFont = Times-Roman
   pageHeight scalefont def		% size = pageHeight (nominally 11)
 
+  % Draw rulers along the edges...
+  /CENTIMETER 72 2.54 div def
+  /MILLIMETER 72 25.4 div def
+
+  /Times-Roman findfont			% Font for ruler numbers
+  11 scalefont setfont			% 11 points
+
+  gsave					% Left side inches
+    pageLeft 72 mul 0 translate		% Offset left edge
+
+    1 setlinewidth			% Draw normal lines
+    72 72 pageTop 72 mul {		% Height inches
+      dup dup
+      0 exch moveto 24 0 rlineto stroke	% Draw tic mark
+      24 exch pageHeight sub moveto	% Draw number
+      72 div cvi 10 string cvs RIGHT
+    } for
+
+    0.5 setlinewidth			% Draw thin lines
+    18 18 pageTop 72 mul {		% 1/4 inches
+      0 exch moveto 15 0 rlineto stroke	% Draw tic mark
+    } for
+
+    9 9 pageTop 72 mul {		% 1/8 inches
+      0 exch moveto 6 0 rlineto stroke	% Draw tic mark
+    } for
+  grestore
+
+  gsave					% Bottom inches
+    0 pageBottom 72 mul translate	% Offset bottom edge
+
+    1 setlinewidth			% Draw normal lines
+    72 72 pageRight 72 mul {		% Width inches
+      dup dup
+      0 moveto 0 24 rlineto stroke	% Draw tic mark
+      3 add 27 pageHeight sub moveto	% Draw number
+      72 div cvi 10 string cvs show
+    } for
+
+    0.5 setlinewidth			% Draw thin lines
+    18 18 pageRight 72 mul {		% 1/4 inches
+      0 moveto 0 15 rlineto stroke	% Draw tic mark
+    } for
+
+    9 9 pageRight 72 mul {		% 1/8 inches
+      0 moveto 0 6 rlineto stroke	% Draw tic mark
+    } for
+  grestore
+
+  gsave					% Right side centimeters
+    pageRight 72 mul 0 translate	% Offset right edge
+
+    1 setlinewidth			% Draw normal lines
+    CENTIMETER CENTIMETER
+        pageTop 72 mul {		% Height centimeters
+      0 exch moveto -24 0 rlineto stroke% Draw tic mark
+    } for
+    1 1 pageTop 2.54 mul {		% Height labels
+      dup
+      -24 exch CENTIMETER mul
+          pageHeight sub moveto		% Draw number
+      cvi 10 string cvs show
+    } for
+
+    0.5 setlinewidth			% Draw thin lines
+    0 0.5 CENTIMETER mul
+        pageTop 72 mul {		% 1/2 centimeters
+      0 exch moveto -15 0 rlineto stroke% Draw tic mark
+    } for
+    0 MILLIMETER pageTop 72 mul {	% Millimeters
+      0 exch moveto -6 0 rlineto stroke	% Draw tic mark
+    } for
+  grestore
+
+  gsave					% Top centimeters
+    0 pageTop 72 mul translate		% Offset top edge
+
+    1 setlinewidth			% Draw normal lines
+    CENTIMETER CENTIMETER
+        pageRight 72 mul {		% Width centimeters
+      0 moveto 0 -24 rlineto stroke	% Draw tic mark
+    } for
+    1 1 pageRight 2.54 mul {		% Width labels
+      dup
+      CENTIMETER mul 3 add -24 moveto	% Draw number
+      cvi 10 string cvs show
+    } for
+
+    0.5 setlinewidth			% Draw thin lines
+    0 0.5 CENTIMETER mul
+        pageRight 72 mul {		% 1/2 centimeters
+      0 moveto 0 -15 rlineto stroke	% Draw tic mark
+    } for
+    0 MILLIMETER pageRight 72 mul {	% Millimeters
+      0 moveto 0 -6 rlineto stroke	% Draw tic mark
+    } for
+  grestore
+
   % Offset page to account for lower-left margin...
   pageLeft 72 mul
   pageBottom 72 mul
   translate
 
-  % Draw the color wheel...
+  % Set text font and color...
   mediumFont setfont			% Font
   0 setgray				% Color
+  1 setlinewidth			% Draw normal lines
 
+  % Draw the color wheel...
   gsave
     % Position the wheel on the left side...
     pageWidth 18 mul			% x = pageWidth * 1/4 * 72
@@ -259,22 +285,81 @@
     wheelSize
 
     % Draw the colors...
-    dup (C) exch 0 1 1 OCTANT 45 rotate
-    dup (M) exch 1 0 1 OCTANT 45 rotate
-    dup (Y) exch 1 1 0 OCTANT 45 rotate
-    dup (K) exch 0 0 0 OCTANT 45 rotate
-    dup (R) exch 1 0 0 OCTANT 45 rotate
-    dup (G) exch 0 1 0 OCTANT 45 rotate
-    dup (B) exch 0 0 1 OCTANT 45 rotate
-    (W) exch 1 1 1 OCTANT 45 rotate
+    dup (C) 3 -1 roll 0 1 1 SEXTANT 60 rotate
+    dup (M) 3 -1 roll 1 0 1 SEXTANT 60 rotate
+    dup (Y) 3 -1 roll 1 1 0 SEXTANT 60 rotate
+    dup (R) 3 -1 roll 1 0 0 SEXTANT 60 rotate
+    dup (G) 3 -1 roll 0 1 0 SEXTANT 60 rotate
+    dup (B) 3 -1 roll 0 0 1 SEXTANT 60 rotate
+
+    pop
   grestore
 
   % Label the color wheel...
   pageWidth 18 mul			% x = pageWidth * 1/4 * 72
-  pageHeight 44 mul			% y = pageHeight * 19/32 * 72
+  pageHeight 43 mul			% y = pageHeight * 19/32 * 72
   moveto				% Position the text
   (Color Wheel) CENTER			% Show the text centered
 
+  % Draw the gray ramp...
+  gsave
+    % Position the gray ramp in the center...
+    pageWidth 36 mul			% x = pageWidth * 1/2 * 72
+    pageHeight 54 mul			% y = pageHeight * 3/4 * 72
+    wheelSize sub			% - wheelSize
+    translate
+
+    % Loop through 100 shades...
+    0 0.010101 0.98 {
+      % Set the color...
+      dup setgray			% Set the grayscale...
+
+      % Draw the polygon...
+      newpath				% Start a new path...
+
+      wheelSize -0.2 mul		% X = -wheelSize / 5
+      1 index 2 mul wheelSize mul	% Y = val * 2 * wheelSize
+      moveto				% Move there...
+
+      wheelSize 0.4 mul 0 rlineto	% Right side...
+
+      wheelSize 0.2 mul			% X = wheelSize / 5
+      1 index 0.010101 add 2 mul wheelSize mul
+      					% Y = (val + 0.010101) * 2 * wheelSize
+      lineto				% Move there...
+
+      wheelSize -0.4 mul 0 rlineto	% Left side...
+
+      closepath				% Close the path
+      fill				% Fill it...
+
+      pop				% Pop value...
+    } for
+
+    0 setgray				% Black
+
+    newpath				% Start a new path
+    wheelSize -0.2 mul 0 moveto		% Bottom left
+    wheelSize 0.4 mul 0 rlineto		% Bottom right
+    0 wheelSize 2 mul rlineto		% Upper right
+    wheelSize -0.4 mul 0 rlineto	% Upper left
+    closepath				% Close the path
+    stroke				% Stroke it...
+
+    0 wheelSize -0.2 mul moveto		% Center bottom for label
+    (K) CENTER				% Center K at bottom
+
+    0 wheelSize 2.05 mul moveto		% Center top for label
+    (W) CENTER				% Center W at top
+  grestore
+
+  % Label the gray ramp...
+  pageWidth 36 mul			% x = pageWidth * 1/2 * 72
+  pageHeight 43 mul			% y = pageHeight * 19/32 * 72
+  moveto				% Position the text
+  (Gray Ramp) CENTER			% Show the text centered
+
+
   % Draw radial lines...
   gsave
     0 setlinewidth			% 1 pixel lines
@@ -302,7 +387,7 @@
 
   % Label the lines...
   pageWidth 54 mul			% x = pageWidth * 3/4 * 72
-  pageHeight 44 mul			% y = pageHeight * 19/32 * 72
+  pageHeight 43 mul			% y = pageHeight * 19/32 * 72
   moveto				% Position the text
   (1 Degree Radial Lines) CENTER	% Show the text centered
 
@@ -467,56 +552,47 @@
 
   % Serial Number
   pageHeight 2 mul sub			% Move down...
-  2 copy moveto				% Move to x & y
+  moveto				% Move to x & y
   (Serial #: ) RIGHT			% Label
   1 serialnumber NUMBER			% S/N
 
   % Draw the label at the top...
   pageWidth 36 mul			% Center of page
-  pageHeight 68 mul			% Top of page (15/16ths)
-  2 copy moveto				% Position text
+  pageHeight 66 mul			% Top of page (11/12ths)
+  moveto				% Position text
   bigFont setfont			% Font
   (Printer Test Page) CENTER		% Show text centered
 
   % Draw the copyright notice at the bottom...
-  pageWidth 36 mul			% Center of page
-  pageHeight 10 mul			% Bottom of page
-  2 copy moveto				% Position text
-  (Printed Using CUPS v1.1.x) CENTER	% Show text centered
+  pageWidth 17 mul			% Center of page
+  pageHeight 9.5 mul			% Bottom of page
+  moveto				% Position text
+  (Printed with CUPS v1.3.x) show
 
-  pageHeight 2 mul sub			% Move down...
+  pageWidth 17 mul			% Left side of page
+  pageHeight 7.5 mul			% Move down...
   2 copy moveto				% Position text
   smallFont setfont			% Font
-  (Copyright 1993-2001 Easy Software Products, All Rights Reserved.) CENTER
-  pageHeight sub			% Move down...
+  (Copyright 2007-2009 Apple Inc., All Rights Reserved. CUPS and the CUPS logo are) show
+  pageHeight 2 add sub			% Move down...
   2 copy moveto				% Position text
-  (CUPS, and the CUPS logo are the trademark property of) CENTER
-  pageHeight sub			% Move down...
-  2 copy moveto				% Position text
-  (Easy Software Products, 44141 Airport View Drive, Suite 204,) CENTER
-  pageHeight sub			% Move down...
-  2 copy moveto				% Position text
-  (Hollywood, Maryland, 20636-3111, USA.) CENTER
+  (the trademark property of Apple Inc., 1 Infinite Loop, Cupertino, CA 95014, USA.) show
+  pageHeight 2 mul 4 add sub		% Move down...
+  moveto				% Position text
+  (Need help? Contact your operating system vendor or visit "http://www.cups.org/".) show
 
   % Then the CUPS logo....
   gsave
     pageWidth 4 mul
     pageHeight 4 mul
     translate
-    pageWidth 9 mul CUPSLOGO
+    pageWidth 15 mul CUPSLOGO
   grestore
 
-  % And the ESP logo....
-  gsave
-    pageWidth 59 mul
-    pageHeight 4 mul
-    translate
-    pageWidth 6 mul ESPLOGO
-  grestore
 % Show the page...
 grestore
 showpage
 %
-% End of "$Id: testprint.ps 88803 2001-03-26 19:34:11Z waba $".
+% End of "$Id: testprint.ps 8236 2009-01-13 17:27:16Z mike $".
 %
 %%EOF
