Index: kdelibs-3.5.10/kdecore/eventsrc
===================================================================
--- kdelibs-3.5.10.orig/kdecore/eventsrc
+++ kdelibs-3.5.10/kdecore/eventsrc
@@ -1832,7 +1832,7 @@ Comment[vi]=KDE đang thoát.
 Comment[zh_CN]=KDE 正在退出
 Comment[zh_HK]=KDE 正在結束
 Comment[zh_TW]=KDE 正在結束
-default_presentation=1
+default_presentation=0
 default_sound=KDE_Logout_3.ogg
 
 [cancellogout]
