diff -ur kdelibs-3.5.2.orig/kio/misc/fileshareset kdelibs-3.5.2/kio/misc/fileshareset
--- kdelibs-3.5.2.orig/kio/misc/fileshareset	2005-10-10 18:05:43.000000000 +0300
+++ kdelibs-3.5.2/kio/misc/fileshareset	2006-05-02 16:29:08.000000000 +0300
@@ -284,8 +284,7 @@
 	if (system('PATH=/bin:/sbin pidof rpc.mountd >/dev/null') != 0 ||
 	    system('PATH=/bin:/sbin pidof nfsd >/dev/null') != 0) {
 	    # trying to start the server...
-	    system('/etc/init.d/portmap start') if system('/etc/init.d/portmap status') != 0;
-	    system('/etc/init.d/nfs', $_) foreach 'stop', 'start';
+	    system('/bin/service nfs_utils restart');
 	}
 	exit 0;
     }
@@ -406,19 +405,7 @@
     system('/usr/bin/killall -HUP smbd 2>/dev/null');
     if (system('PATH=/bin:/sbin pidof smbd >/dev/null') != 0 ||
 	system('PATH=/bin:/sbin pidof nmbd >/dev/null') != 0) {
-# trying to start the server...
-      if ( -f '/etc/init.d/smb' ) {
-	system('/etc/init.d/smb', $_) foreach 'stop', 'start';
-      }
-      elsif ( -f '/etc/init.d/samba' ) {
-	system('/etc/init.d/samba', $_) foreach 'stop', 'start';
-      }
-	  elsif ( -f '/etc/rc.d/rc.samba' ) {
-	system('/etc/rc.d/rc.samba', $_) foreach 'stop', 'start';
-	  }
-      else {
-	print STDERR "Error: Can't find the samba init script \n";
-      }
+	    system('/bin/service samba restart');
     }
     exit 0;
   }
