Index: kdelibs-3.5.10/kdecore/svgicons/ksvgiconengine.cpp
===================================================================
--- kdelibs-3.5.10.orig/kdecore/svgicons/ksvgiconengine.cpp
+++ kdelibs-3.5.10/kdecore/svgicons/ksvgiconengine.cpp
@@ -538,6 +538,9 @@ KSVGIconEngine::~KSVGIconEngine()
 
 bool KSVGIconEngine::load(int width, int height, const QString &path)
 {
+	if (path.isEmpty())
+		return false;
+
 	QDomDocument svgDocument("svg");
 	QFile file(path);
 
