--- kdelibs-3.5.10-old/kioslave/ftp/ftp.cc	2011-02-14 10:45:44.677000003 +0200
+++ kdelibs-3.5.10/kioslave/ftp/ftp.cc	2011-02-14 11:26:15.563999700 +0200
@@ -1216,9 +1216,9 @@
   if( !ftpOpenConnection(loginImplicit) )
         return;
 
-  QString path = remoteEncoding()->encode(url);
+  QCString path = remoteEncoding()->encode(url);
   QCString buf = "mkd ";
-  buf += remoteEncoding()->encode(path);
+  buf += path;
 
   if( !ftpSendCmd( buf ) || (m_iRespType != 2) )
   {
