Index: kdecore/kglobalsettings.cpp
===================================================================
--- kdecore/kglobalsettings.cpp	(revision 574096)
+++ kdecore/kglobalsettings.cpp	(working copy)
@@ -94,7 +94,7 @@
     int tearoff;
     bool effectsenabled;
     KConfigGroup g( KGlobal::config(), "KDE" );
-    effectsenabled = g.readBoolEntry( "EffectsEnabled", false);
+    effectsenabled = g.readBoolEntry( "EffectsEnabled", true);
     tearoff = g.readNumEntry("InsertTearOffHandle", KDE_DEFAULT_INSERTTEAROFFHANDLES);
     return effectsenabled ? (TearOffHandle) tearoff : Disable;
 }
@@ -511,7 +511,7 @@
       s_autostartPath->append('/');
 
     // Document Path
-    *s_documentPath = g.readPathEntry( "Documents", 
+    *s_documentPath = g.readPathEntry( "Documents",
 #ifdef Q_WS_WIN
         getWin32ShellFoldersPath("Personal")
 #else
@@ -703,8 +703,7 @@
 bool KGlobalSettings::showIconsOnPushButtons()
 {
     KConfigGroup g( KGlobal::config(), "KDE" );
-    return g.readBoolEntry("ShowIconsOnPushButtons",
-        KDE_DEFAULT_ICON_ON_PUSHBUTTON);
+    return g.readBoolEntry("ShowIconsOnPushButtons", true);
 }
 
 bool KGlobalSettings::showFilePreview(const KURL &url)
