diff -ur kdelibs-3.4.92.orig/kioslave/http/kcookiejar/kcookiejar.cpp kdelibs-3.4.92/kioslave/http/kcookiejar/kcookiejar.cpp
--- kdelibs-3.4.92.orig/kioslave/http/kcookiejar/kcookiejar.cpp	2005-09-10 11:26:41.000000000 +0300
+++ kdelibs-3.4.92/kioslave/http/kcookiejar/kcookiejar.cpp	2005-10-20 03:58:56.000000000 +0300
@@ -1518,7 +1518,7 @@
     m_rejectCrossDomainCookies = _config->readBoolEntry( "RejectCrossDomainCookies", true );
     m_autoAcceptSessionCookies = _config->readBoolEntry( "AcceptSessionCookies", true );
     m_ignoreCookieExpirationDate = _config->readBoolEntry( "IgnoreExpirationDate", false );
-    QString value = _config->readEntry("CookieGlobalAdvice", L1("Ask"));
+    QString value = _config->readEntry("CookieGlobalAdvice", L1("Accept"));
     m_globalAdvice = strToAdvice(value);
 
     // Reset current domain settings first.
