diff -ur kdelibs-3.4.92.orig/kdecore/kglobalsettings.h kdelibs-3.4.92/kdecore/kglobalsettings.h
--- kdelibs-3.4.92.orig/kdecore/kglobalsettings.h	2005-10-10 18:06:03.000000000 +0300
+++ kdelibs-3.4.92/kdecore/kglobalsettings.h	2005-10-20 03:53:02.000000000 +0300
@@ -23,7 +23,7 @@
 #include <qfont.h>
 #include "kdelibs_export.h"
 
-#define KDE_DEFAULT_SINGLECLICK true
+#define KDE_DEFAULT_SINGLECLICK false
 #define KDE_DEFAULT_INSERTTEAROFFHANDLES 0
 #define KDE_DEFAULT_AUTOSELECTDELAY -1
 #define KDE_DEFAULT_CHANGECURSOR true
