Index: kdecore/kglobalsettings.cpp
===================================================================
--- kdecore/kglobalsettings.cpp	(revision 668676)
+++ kdecore/kglobalsettings.cpp	(working copy)
@@ -322,7 +322,7 @@
         return *_generalFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _generalFont = new QFont("Sans Serif", 10);
+    _generalFont = new QFont("DejaVu Sans", 10);
     _generalFont->setPointSize(10);
     _generalFont->setStyleHint(QFont::SansSerif);
 
@@ -338,7 +338,7 @@
         return *_fixedFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _fixedFont = new QFont("Monospace", 10);
+    _fixedFont = new QFont("DejaVu Sans Mono", 10);
     _fixedFont->setPointSize(10);
     _fixedFont->setStyleHint(QFont::TypeWriter);
 
@@ -354,7 +354,7 @@
         return *_toolBarFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _toolBarFont = new QFont("Sans Serif", 10);
+    _toolBarFont = new QFont("DejaVu Sans", 10);
     _toolBarFont->setPointSize(10);
     _toolBarFont->setStyleHint(QFont::SansSerif);
 
@@ -370,7 +370,7 @@
         return *_menuFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _menuFont = new QFont("Sans Serif", 10);
+    _menuFont = new QFont("DejaVu Sans", 10);
     _menuFont->setPointSize(10);
     _menuFont->setStyleHint(QFont::SansSerif);
 
@@ -386,12 +386,12 @@
         return *_windowTitleFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _windowTitleFont = new QFont("Sans Serif", 9, QFont::Bold);
+    _windowTitleFont = new QFont("DejaVu Sans", 9, QFont::Bold);
     _windowTitleFont->setPointSize(10);
     _windowTitleFont->setStyleHint(QFont::SansSerif);
 
-    KConfigGroup g( KGlobal::config(), "WM" );
-    *_windowTitleFont = g.readFontEntry("activeFont", _windowTitleFont); // inconsistency
+    //KConfigGroup g( KGlobal::config(), "WM" );
+    //*_windowTitleFont = g.readFontEntry("activeFont", _windowTitleFont); // inconsistency
 
     return *_windowTitleFont;
 }
@@ -402,7 +402,7 @@
         return *_taskbarFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _taskbarFont = new QFont("Sans Serif", 10);
+    _taskbarFont = new QFont("DejaVu Sans", 10);
     _taskbarFont->setPointSize(10);
     _taskbarFont->setStyleHint(QFont::SansSerif);
 
