diff -ur kdelibs-3.4.92.orig/khtml/khtmldefaults.h kdelibs-3.4.92/khtml/khtmldefaults.h
--- kdelibs-3.4.92.orig/khtml/khtmldefaults.h	2005-10-10 18:06:13.000000000 +0300
+++ kdelibs-3.4.92/khtml/khtmldefaults.h	2005-10-20 03:50:41.000000000 +0300
@@ -24,10 +24,10 @@
 #define HTML_DEFAULT_VLNK_COLOR Qt::magenta
 #define HTML_DEFAULT_BASE_COLOR Qt::white
 
-#define HTML_DEFAULT_VIEW_FONT "Sans Serif"
-#define HTML_DEFAULT_VIEW_FIXED_FONT "Monospace"
-#define HTML_DEFAULT_VIEW_SERIF_FONT "Serif"
-#define HTML_DEFAULT_VIEW_SANSSERIF_FONT "Sans Serif"
-#define HTML_DEFAULT_VIEW_CURSIVE_FONT "Sans Serif"
-#define HTML_DEFAULT_VIEW_FANTASY_FONT "Sans Serif"
-#define HTML_DEFAULT_MIN_FONT_SIZE 7 // everything smaller is usually unreadable.
+#define HTML_DEFAULT_VIEW_FONT "DejaVu Sans"
+#define HTML_DEFAULT_VIEW_FIXED_FONT "DejaVu Sans Mono"
+#define HTML_DEFAULT_VIEW_SERIF_FONT "DejaVu Sans"
+#define HTML_DEFAULT_VIEW_SANSSERIF_FONT "DejaVu Sans"
+#define HTML_DEFAULT_VIEW_CURSIVE_FONT "DejaVu Sans"
+#define HTML_DEFAULT_VIEW_FANTASY_FONT "DejaVu Sans"
+#define HTML_DEFAULT_MIN_FONT_SIZE 8 // everything smaller is usually unreadable.
