diff -ur kdelibs-3.5.5.orig/kdeprint/management/networkscanner.cpp kdelibs-3.5.5/kdeprint/management/networkscanner.cpp
--- kdelibs-3.5.5.orig/kdeprint/management/networkscanner.cpp	2005-10-10 18:06:32.000000000 +0300
+++ kdelibs-3.5.5/kdeprint/management/networkscanner.cpp	2006-11-23 10:55:41.000000000 +0200
@@ -344,6 +344,8 @@
 	mm->setFixedWidth(fontMetrics().width(mm->text())+10);
 
 	mask_ = new QLineEdit(dummy);
+	// make sure address area is not overly small
+	mask_->setMinimumWidth(fontMetrics().width("192.168.123"));
 	mask_->setAlignment(Qt::AlignRight);
 	port_ = new QComboBox(true,dummy);
         if ( port_->lineEdit() )
