diff -ur kdelibs-3.5.0.orig/kdeui/kaboutdialog.cpp kdelibs-3.5.0/kdeui/kaboutdialog.cpp
--- kdelibs-3.5.0.orig/kdeui/kaboutdialog.cpp	2005-12-13 16:26:54.000000000 +0200
+++ kdelibs-3.5.0/kdeui/kaboutdialog.cpp	2005-12-13 16:27:41.000000000 +0200
@@ -20,6 +20,8 @@
  *
  */
 
+#include "config.h"
+
 #include <qclipboard.h>
 #include <qimage.h>
 #include <qlabel.h>
@@ -699,8 +701,9 @@
   else
     kdDebug(291) << "no kapp" << endl;
 
-  const QString msg1 = i18n("%1 %2 (Using KDE %3)").arg(appName).arg(version).
-    arg(QString::fromLatin1(KDE_VERSION_STRING));
+  const QString msg1 = i18n("%1 %2 (Using KDE %3 %4)").arg(appName).arg(version).
+    arg(QString::fromLatin1(KDE_VERSION_STRING)).
+    arg(QString::fromLatin1(KDE_DISTRIBUTION_TEXT));
   const QString msg2 = !year.isEmpty() ? i18n("%1 %2, %3").arg('©').arg(year).
     arg(author) : QString::fromLatin1("");
 
