Index: kdeui/kbugreport.cpp
===================================================================
--- kdeui/kbugreport.cpp	(revision 489954)
+++ kdeui/kbugreport.cpp	(working copy)
@@ -274,13 +274,7 @@
 
 void KBugReport::updateURL()
 {
-    KURL url ( "http://bugs.kde.org/wizard.cgi" );
-    url.addQueryItem( "os", d->os );
-    url.addQueryItem( "compiler", KDE_COMPILER_VERSION );
-    url.addQueryItem( "kdeVersion", d->kde_version );
-    url.addQueryItem( "appVersion", m_strVersion );
-    url.addQueryItem( "package", d->appcombo->currentText() );
-    url.addQueryItem( "kbugreport", "1" );
+    KURL url ( "http://bugs.uludag.org.tr/enter_bug.cgi" );
     d->url = url;
 }
 
