diff -up hplip-3.9.8/ip/ipmain.c.no-asm hplip-3.9.8/ip/ipmain.c
--- hplip-3.9.8/ip/ipmain.c.no-asm	2009-08-19 11:35:39.770825475 +0100
+++ hplip-3.9.8/ip/ipmain.c	2009-08-19 11:36:25.767828346 +0100
@@ -265,9 +265,6 @@ static IP_XFORM_TBL * const xformJumpTab
 void fatalBreakPoint (void)
 {
    /* do nothing */
-#if defined _DEBUG
-    __asm int 3;
-#endif
    PRINT0 (_T("\nhit fatalBreakPoint!\n"));
 }
 
diff -up hplip-3.9.8/ip/xfax.c.no-asm hplip-3.9.8/ip/xfax.c
--- hplip-3.9.8/ip/xfax.c.no-asm	2009-08-19 11:35:45.484827783 +0100
+++ hplip-3.9.8/ip/xfax.c	2009-08-19 11:36:39.815701961 +0100
@@ -1680,19 +1680,6 @@ static BOOL bits_flush_to_eol (
                  & ((1u<<n_bits) - 1u);                                      \
 }
 
-#if 0
-
-#define BITS_LOAD(g, num_bits, par_result) {                                 \
-    int n_bits = (int)(num_bits);                                            \
-                                                                             \
-    par_result = g->gb_cache;                                                \
-                                                                             \
-    asm ("extract %1,%2,%0"                                                  \
-        : "=d" (par_result)                                                  \
-        : "dI" (g->gb_cache_cnt - n_bits), "dI" (n_bits), "0" (par_result)); \
-}
-
-#endif
 
 
 
