Index: jack-audio-connection-kit-0.116.2/configure.ac
===================================================================
--- jack-audio-connection-kit-0.116.2.orig/configure.ac
+++ jack-audio-connection-kit-0.116.2/configure.ac
@@ -132,7 +132,7 @@ case "${host_os}" in
     linux*)
     case "${host_cpu}" in
 	x86_64|mips64|ppc64|sparc64|s390x)
-        libnn=lib64
+        libnn=lib
 	;;
     esac
     ;;
