Index: libsmbios-2.2.6/Makefile.in
===================================================================
--- libsmbios-2.2.6.orig/Makefile.in
+++ libsmbios-2.2.6/Makefile.in
@@ -2597,13 +2597,6 @@ install-cliPYTHON: $(cli_PYTHON)
 	    $(cliPYTHON_INSTALL) "$$b$$p" "$(DESTDIR)$(clidir)/$$f"; \
 	  else :; fi; \
 	done; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(clidir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(clidir)" $$dlist; \
-	  fi; \
-	else :; fi
 
 uninstall-cliPYTHON:
 	@$(NORMAL_UNINSTALL)
@@ -2625,13 +2618,6 @@ install-dist_yumpluginPYTHON: $(dist_yum
 	    $(dist_yumpluginPYTHON_INSTALL) "$$b$$p" "$(DESTDIR)$(yumplugindir)/$$f"; \
 	  else :; fi; \
 	done; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(yumplugindir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(yumplugindir)" $$dlist; \
-	  fi; \
-	else :; fi
 
 uninstall-dist_yumpluginPYTHON:
 	@$(NORMAL_UNINSTALL)
@@ -2653,13 +2639,6 @@ install-pkgpythonPYTHON: $(pkgpython_PYT
 	    $(pkgpythonPYTHON_INSTALL) "$$b$$p" "$(DESTDIR)$(pkgpythondir)/$$f"; \
 	  else :; fi; \
 	done; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pkgpythondir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pkgpythondir)" $$dlist; \
-	  fi; \
-	else :; fi
 
 uninstall-pkgpythonPYTHON:
 	@$(NORMAL_UNINSTALL)
