--- gstreamer/mediaobject.cpp   2009/02/27 18:00:46 932979
+++ gstreamer/mediaobject.cpp   2009/02/27 18:01:12 932980
@@ -141,9 +141,9 @@
     if (m_resumeState)
         return;
 
-    if (m_state == Phonon::PlayingState || m_state == Phonon::PausedState) {
+    if (m_pendingState == Phonon::PlayingState || m_pendingState == Phonon::PausedState) {
         m_resumeState = true;
-        m_oldState = m_state;
+        m_oldState = m_pendingState;
         m_oldPos = getPipelinePos();
     }
 }

