Index: gstreamer/mediaobject.cpp
===================================================================
--- gstreamer/mediaobject.cpp   (revision 1040729)
+++ gstreamer/mediaobject.cpp   (working copy)
@@ -484,7 +484,7 @@
     gst_object_ref (GST_OBJECT (m_pipeline));
     gst_object_sink (GST_OBJECT (m_pipeline));

-    m_decodebin = gst_element_factory_make ("decodebin", NULL);
+    m_decodebin = gst_element_factory_make ("decodebin2", NULL);
     g_signal_connect (m_decodebin, "new-decoded-pad", G_CALLBACK (&cb_newpad), this);
     g_signal_connect (m_decodebin, "unknown-type", G_CALLBACK (&cb_unknown_type), this);
     g_signal_connect (m_decodebin, "no-more-pads", G_CALLBACK (&cb_no_more_pads), this);
Index: gstreamer/gsthelper.cpp
===================================================================
--- gstreamer/gsthelper.cpp     (revision 1040694)
+++ gstreamer/gsthelper.cpp     (working copy)
@@ -121,7 +121,7 @@
 {
     GstElement *playbin = 0;
     //init playbin and add to our pipeline
-    playbin = gst_element_factory_make("playbin", NULL);
+    playbin = gst_element_factory_make("playbin2", NULL);

     //Create an identity element to redirect sound
     GstElement *audioSinkBin =  gst_bin_new (NULL);

