--- xvidcap.desktop	2007-04-24 18:15:42.000000000 +0300
+++ xvidcap.desktop	2007-04-25 22:58:38.000000000 +0300
@@ -1,10 +1,12 @@
 [Desktop Entry]
 Type=Application
 Encoding=UTF-8
-Name=XVidCap Screen Capture
+Name=XVidCap
 GenericName=X11 Screencam
+GenericName[tr]=Görüntü Yakalama Aracı
 Comment=X11 Screencam
 Comment[de]=X11 Videoaufnahme
+Comment[tr]=Görüntü Yakalama Aracı
 Icon=xvidcap.png
 TryExec=xvidcap
 Exec=xvidcap
