diff -Nuar src/kredentials.desktop src/kredentials.desktop
--- src/kredentials.desktop	2011-06-20 10:46:08.847000071 +0300
+++ src/kredentials.desktop	2011-06-20 10:51:48.843000074 +0300
@@ -1,8 +1,8 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=kredentials
-Name[xx]=xxkredentialsxx
 Exec=kredentials
 Icon=kredentials
 Type=Application
 Comment=Panel Applet to renew authentication credentials
+Comment[tr]=Kerberos ticketlarını yönetmek için KDE uygulaması
