diff -Naur raptor-1.4.19-orig/configure.ac raptor-1.4.19/configure.ac
--- raptor-1.4.19-orig/configure.ac	2009-07-28 12:00:15.928703425 +0300
+++ raptor-1.4.19/configure.ac	2009-07-28 12:00:26.040453431 +0300
@@ -1266,8 +1266,6 @@
 raptor.spec
 raptor.rdf
 data/Makefile
-docs/Makefile
-docs/version.xml
 examples/Makefile
 src/Makefile
 tests/Makefile
@@ -1285,9 +1283,6 @@
 AC_CONFIG_FILES([src/raptor-config], [chmod +x src/raptor-config])
 AC_CONFIG_FILES([raptor-src-config], [chmod +x raptor-src-config])
 
-dnl Check for gtk-doc and docbook
-GTK_DOC_CHECK([1.3])
-
 AC_OUTPUT
 
 AC_MSG_RESULT([
diff -Naur raptor-1.4.19-orig/Makefile.am raptor-1.4.19/Makefile.am
--- raptor-1.4.19-orig/Makefile.am	2009-07-28 12:00:15.956703251 +0300
+++ raptor-1.4.19/Makefile.am	2009-07-28 12:00:36.393416814 +0300
@@ -24,7 +24,7 @@
 
 noinst_SCRIPTS = raptor-src-config
 
-SUBDIRS= src utils docs data tests win32 examples librdfa
+SUBDIRS= src utils data tests win32 examples librdfa
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = raptor.pc
