Index: qt-x11-opensource-src-4.5.3/src/gui/itemviews/qlistwidget.cpp
===================================================================
--- qt-x11-opensource-src-4.5.3.orig/src/gui/itemviews/qlistwidget.cpp
+++ qt-x11-opensource-src-4.5.3/src/gui/itemviews/qlistwidget.cpp
@@ -687,7 +687,7 @@ QVariant QListWidgetItem::data(int role)
 */
 bool QListWidgetItem::operator<(const QListWidgetItem &other) const
 {
-    return text() < other.text();
+    return QString::localeAwareCompare(text(), other.text()) < 0;
 }
 
 #ifndef QT_NO_DATASTREAM
