Index: etc/shadow
===================================================================
--- etc/shadow	(revision 19668)
+++ etc/shadow	(revision 19821)
@@ -13,5 +13,7 @@
 man:*:9797:0:::::
 games:*:9797:0:::::
 dbus:*:9797:0:::::
+hal:*:9797:0:::::
+polkit:*:9797:0:::::
 guest:*:9797:0:::::
 nobody:*:9797:0:::::
Index: etc/group
===================================================================
--- etc/group	(revision 19668)
+++ etc/group	(revision 19821)
@@ -21,5 +21,7 @@
 tape::26:root
 video::27:root
 dbus:x:101:dbus
+hal:x:102:hal
+polkit:x:103:polkit
 nogroup::65533:
 nobody::65534:
Index: etc/passwd
===================================================================
--- etc/passwd	(revision 19668)
+++ etc/passwd	(revision 19821)
@@ -15,4 +15,6 @@
 sshd:x:22:22:sshd:/dev/null:/bin/false
 at:x:25:25:at:/var/spool/cron/atjobs:/bin/false
 dbus:x:101:101:D-Bus:/dev/null:/bin/false
+hal:x:102:102:Hal:/dev/null:/bin/false
+polkit:x:103:103:PolicyKit:/dev/null:/bin/false
 nobody:x:65534:65534:nobody:/:/bin/false
