Index: omnibook-20090720/laptop.h
===================================================================
--- omnibook-20090720.orig/laptop.h
+++ omnibook-20090720/laptop.h
@@ -664,6 +664,15 @@ static struct  dmi_system_id omnibook_id
 	},
 	{
 		.callback = dmi_matched,
+		.ident = "Toshiba Satellite U400",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Satellite Pro U400"),
+		},
+		.driver_data = (void*) TSM70
+	},
+	{
+		.callback = dmi_matched,
 		.ident = "Toshiba Satellite P100",
 		.matches = {
 			DMI_MATCH(DMI_SYS_VENDOR, "TOSHIBA"),
Index: omnibook-20090720/init.c
===================================================================
--- omnibook-20090720.orig/init.c
+++ omnibook-20090720/init.c
@@ -536,4 +536,6 @@ MODULE_ALIAS("dmi:*:svnTOSHIBA:pnSatelli
 MODULE_ALIAS("dmi:*:svnTOSHIBA:pnSatelliteA210:pvrPSAEGE-01M00KTE:*");
 /* Toshiba Satellite A300 */
 MODULE_ALIAS("dmi:*:svnTOSHIBA:pnSatelliteA300:pvrPSAJ0E-01300QTE:*");
+/* Toshiba Satellite Pro U400 */
+MODULE_ALIAS("dmi:*:svnTOSHIBA:pnSatelliteProU400:pvrPSU45E-00700GTE:*");
 /* End of file */
