Index: cupsddk-1.2.3/Makedefs.in
===================================================================
--- cupsddk-1.2.3.orig/Makedefs.in
+++ cupsddk-1.2.3/Makedefs.in
@@ -49,7 +49,7 @@ RM		=	@RM@ -f
 # Installation programs...
 #
 
-INSTALL_BIN	=	$(LIBTOOL) $(INSTALL) -m 755 -s
+INSTALL_BIN	=	$(LIBTOOL) $(INSTALL) -m 755
 INSTALL_DATA	=	$(INSTALL) -m 644
 INSTALL_DIR	=	$(INSTALL) -d
 INSTALL_MAN	=	$(INSTALL) -m 644
