diff -Nur libvorbis-1.2.3-old/lib/codebook.c libvorbis-1.2.3/lib/codebook.c
--- libvorbis-1.2.3-old/lib/codebook.c	2009-11-10 22:55:23.514524806 +0200
+++ libvorbis-1.2.3/lib/codebook.c	2009-11-10 22:57:04.605284571 +0200
@@ -198,6 +198,7 @@
       for(i=0;i<s->entries;){
         long num=oggpack_read(opb,_ilog(s->entries-i));
         if(num==-1)goto _eofout;
+        if(length>32)goto _errout;
         for(j=0;j<num && i<s->entries;j++,i++)
           s->lengthlist[i]=length;
         length++;
