--- kdeedu-3.5.10.orig/keduca/resources/keducabuilder.desktop	2011-06-15 14:42:02.655000024 +0300
+++ kdeedu-3.5.10/keduca/resources/keducabuilder.desktop	2011-06-16 08:38:50.333000019 +0300
@@ -121,5 +121,5 @@
 Path=
 Terminal=false
 Type=Application
-X-DCOP-ServiceType=Multi
 Categories=Qt;KDE;Education;
+InitialPreference=10
--- kdeedu-3.5.10.orig/keduca/resources/keduca.desktop	2011-06-15 14:42:02.655000024 +0300
+++ kdeedu-3.5.10/keduca/resources/keduca.desktop	2011-06-15 14:43:03.271000029 +0300
@@ -133,3 +133,4 @@
 Type=Application
 X-DCOP-ServiceType=Multi
 Categories=Qt;KDE;Education;
+InitialPreference=20
