Index: libechonest-1.1.0/CMakeLists.txt
===================================================================
--- libechonest-1.1.0.orig/CMakeLists.txt
+++ libechonest-1.1.0/CMakeLists.txt
@@ -13,6 +13,7 @@ find_package(QJSON REQUIRED)
 
 include( ${QT_USE_FILE} )
 
+LINK_DIRECTORIES( "/usr/qt/4/lib" )
 set( ECHONEST_LIB_MAJOR_VERSION "1" )
 set( ECHONEST_LIB_MINOR_VERSION "1" )
 set( ECHONEST_LIB_PATCH_VERSION "0" )
