diff -Nur gfxtheme-pardus-0.11.4-old/src/common.inc gfxtheme-pardus-0.11.4/src/common.inc
--- gfxtheme-pardus-0.11.4-old/src/common.inc	2011-01-25 16:50:00.424171711 +0200
+++ gfxtheme-pardus-0.11.4/src/common.inc	2011-01-25 16:50:52.765112231 +0200
@@ -1181,7 +1181,7 @@
     /lang.displayed true def
   } {
     pop
-    "en_US" dup setlang pop setkeymap
+    "tr_TR" dup setlang pop setkeymap
   } ifelse
 
   font.large setfont
