diff -up foomatic-db-20100204/db/source/PPD/Kyocera/de/Kyocera_Mita_FS-1020D_de.ppd.device-id-kyocera-mita-fs-1020d foomatic-db-20100204/db/source/PPD/Kyocera/de/Kyocera_Mita_FS-1020D_de.ppd
--- foomatic-db-20100204/db/source/PPD/Kyocera/de/Kyocera_Mita_FS-1020D_de.ppd.device-id-kyocera-mita-fs-1020d	2010-03-19 10:18:40.667621007 +0000
+++ foomatic-db-20100204/db/source/PPD/Kyocera/de/Kyocera_Mita_FS-1020D_de.ppd	2010-03-19 12:48:25.333620595 +0000
@@ -88,7 +88,7 @@
 *Protocols: PJL TBCP
 
 *1284Modes Parallel: Compat Nibble ECP
-*1284DeviceID: "MFG:Kyocera Mita;MODEL:Kyocera Mita FS-1020D;COMMAND SET: POSTSCRIPT,PJL,PCL"
+*1284DeviceID: "MFG:Kyocera;MDL:FS-1020D;CMD:PCLXL,PostScript Emulation,PCL5E,PJL;"
 
 *% JCL Information 
 *JCLBegin: "<1B>%-12345X@PJL JOB<0A>"
diff -up foomatic-db-20100204/db/source/PPD/Kyocera/en/Kyocera_Mita_FS-1020D_en.ppd.device-id-kyocera-mita-fs-1020d foomatic-db-20100204/db/source/PPD/Kyocera/en/Kyocera_Mita_FS-1020D_en.ppd
--- foomatic-db-20100204/db/source/PPD/Kyocera/en/Kyocera_Mita_FS-1020D_en.ppd.device-id-kyocera-mita-fs-1020d	2010-03-19 10:18:41.231620596 +0000
+++ foomatic-db-20100204/db/source/PPD/Kyocera/en/Kyocera_Mita_FS-1020D_en.ppd	2010-03-19 12:48:25.362620955 +0000
@@ -88,7 +88,7 @@
 *Protocols: PJL TBCP
 
 *1284Modes Parallel: Compat Nibble ECP
-*1284DeviceID: "MFG:Kyocera Mita;MODEL:Kyocera Mita FS-1020D;COMMAND SET: POSTSCRIPT,PJL,PCL"
+*1284DeviceID: "MFG:Kyocera;MDL:FS-1020D;CMD:PCLXL,PostScript Emulation,PCL5E,PJL;"
 
 *% JCL Information 
 *JCLBegin: "<1B>%-12345X@PJL JOB<0A>"
diff -up foomatic-db-20100204/db/source/PPD/Kyocera/es/Kyocera_Mita_FS-1020D_es.ppd.device-id-kyocera-mita-fs-1020d foomatic-db-20100204/db/source/PPD/Kyocera/es/Kyocera_Mita_FS-1020D_es.ppd
--- foomatic-db-20100204/db/source/PPD/Kyocera/es/Kyocera_Mita_FS-1020D_es.ppd.device-id-kyocera-mita-fs-1020d	2010-03-19 10:18:42.142622002 +0000
+++ foomatic-db-20100204/db/source/PPD/Kyocera/es/Kyocera_Mita_FS-1020D_es.ppd	2010-03-19 12:48:25.394621206 +0000
@@ -88,7 +88,7 @@
 *Protocols: PJL TBCP
 
 *1284Modes Parallel: Compat Nibble ECP
-*1284DeviceID: "MFG:Kyocera Mita;MODEL:Kyocera Mita FS-1020D;COMMAND SET: POSTSCRIPT,PJL,PCL"
+*1284DeviceID: "MFG:Kyocera;MDL:FS-1020D;CMD:PCLXL,PostScript Emulation,PCL5E,PJL;"
 
 *% JCL Information 
 *JCLBegin: "<1B>%-12345X@PJL JOB<0A>"
diff -up foomatic-db-20100204/db/source/PPD/Kyocera/fr/Kyocera_Mita_FS-1020D_fr.ppd.device-id-kyocera-mita-fs-1020d foomatic-db-20100204/db/source/PPD/Kyocera/fr/Kyocera_Mita_FS-1020D_fr.ppd
--- foomatic-db-20100204/db/source/PPD/Kyocera/fr/Kyocera_Mita_FS-1020D_fr.ppd.device-id-kyocera-mita-fs-1020d	2010-03-19 10:18:39.883623183 +0000
+++ foomatic-db-20100204/db/source/PPD/Kyocera/fr/Kyocera_Mita_FS-1020D_fr.ppd	2010-03-19 12:48:25.263620271 +0000
@@ -88,7 +88,7 @@
 *Protocols: PJL TBCP
 
 *1284Modes Parallel: Compat Nibble ECP
-*1284DeviceID: "MFG:Kyocera Mita;MODEL:Kyocera Mita FS-1020D;COMMAND SET: POSTSCRIPT,PJL,PCL"
+*1284DeviceID: "MFG:Kyocera;MDL:FS-1020D;CMD:PCLXL,PostScript Emulation,PCL5E,PJL;"
 
 *% JCL Information 
 *JCLBegin: "<1B>%-12345X@PJL JOB<0A>"
diff -up foomatic-db-20100204/db/source/PPD/Kyocera/it/Kyocera_Mita_FS-1020D_it.ppd.device-id-kyocera-mita-fs-1020d foomatic-db-20100204/db/source/PPD/Kyocera/it/Kyocera_Mita_FS-1020D_it.ppd
--- foomatic-db-20100204/db/source/PPD/Kyocera/it/Kyocera_Mita_FS-1020D_it.ppd.device-id-kyocera-mita-fs-1020d	2010-03-19 10:18:39.911620599 +0000
+++ foomatic-db-20100204/db/source/PPD/Kyocera/it/Kyocera_Mita_FS-1020D_it.ppd	2010-03-19 12:48:25.310621922 +0000
@@ -88,7 +88,7 @@
 *Protocols: PJL TBCP
 
 *1284Modes Parallel: Compat Nibble ECP
-*1284DeviceID: "MFG:Kyocera Mita;MODEL:Kyocera Mita FS-1020D;COMMAND SET: POSTSCRIPT,PJL,PCL"
+*1284DeviceID: "MFG:Kyocera;MDL:FS-1020D;CMD:PCLXL,PostScript Emulation,PCL5E,PJL;"
 
 *% JCL Information 
 *JCLBegin: "<1B>%-12345X@PJL JOB<0A>"
diff -up foomatic-db-20100204/db/source/PPD/Kyocera/pt/Kyocera_Mita_FS-1020D_pt.ppd.device-id-kyocera-mita-fs-1020d foomatic-db-20100204/db/source/PPD/Kyocera/pt/Kyocera_Mita_FS-1020D_pt.ppd
--- foomatic-db-20100204/db/source/PPD/Kyocera/pt/Kyocera_Mita_FS-1020D_pt.ppd.device-id-kyocera-mita-fs-1020d	2010-03-19 10:18:39.393747204 +0000
+++ foomatic-db-20100204/db/source/PPD/Kyocera/pt/Kyocera_Mita_FS-1020D_pt.ppd	2010-03-19 12:48:25.227621183 +0000
@@ -88,7 +88,7 @@
 *Protocols: PJL TBCP
 
 *1284Modes Parallel: Compat Nibble ECP
-*1284DeviceID: "MFG:Kyocera Mita;MODEL:Kyocera Mita FS-1020D;COMMAND SET: POSTSCRIPT,PJL,PCL"
+*1284DeviceID: "MFG:Kyocera;MDL:FS-1020D;CMD:PCLXL,PostScript Emulation,PCL5E,PJL;"
 
 *% JCL Information 
 *JCLBegin: "<1B>%-12345X@PJL JOB<0A>"
diff -up foomatic-db-20100204/db/source/printer/Kyocera-FS-1020D.xml.device-id-kyocera-mita-fs-1020d foomatic-db-20100204/db/source/printer/Kyocera-FS-1020D.xml
--- foomatic-db-20100204/db/source/printer/Kyocera-FS-1020D.xml.device-id-kyocera-mita-fs-1020d	2010-03-19 10:19:14.771753487 +0000
+++ foomatic-db-20100204/db/source/printer/Kyocera-FS-1020D.xml	2010-03-19 11:47:20.308749300 +0000
@@ -19,10 +19,10 @@
   </lang>
   <autodetect>
     <general>
-      <ieee1284>MFG:Kyocera Mita;Model:Kyocera Mita FS-1020D;COMMAND SET: POSTSCRIPT,PJL,PCL</ieee1284>
-      <commandset>POSTSCRIPT,PJL,PCL</commandset>
-      <manufacturer>Kyocera Mita</manufacturer>
-      <model>Kyocera Mita FS-1020D</model>
+      <ieee1284>MFG:Kyocera;MDL:FS-1020D;</ieee1284>
+      <commandset>PCLXL,PostScript Emulation,PCL5E,PJL</commandset>
+      <manufacturer>Kyocera</manufacturer>
+      <model>FS-1020D</model>
     </general>
   </autodetect>
   <functionality>A</functionality>
