Index: pkinit-nss-0.7.8-1/configure.ac
===================================================================
--- pkinit-nss-0.7.8-1.orig/configure.ac
+++ pkinit-nss-0.7.8-1/configure.ac
@@ -99,7 +99,7 @@ else
 	AC_MSG_RESULT([Requested build for $krb5_version.])
 fi
 case "$krb5_version" in
-	1.7)
+	1.7*)
 		AC_MSG_RESULT([Building for Kerberos 1.7.])
 		if test x$ac_cv_header_krb5_preauth_plugin_h = xno ; then
 			BACKPORT_CPPFLAGS='-I$(top_srcdir)/backport-1.7'
