--- arts-1.3.1/flow/audioioalsa.cc.orig	2004-11-19 19:32:07.977923623 +0100
+++ arts-1.3.1/flow/audioioalsa.cc	2004-11-19 19:42:47.977558161 +0100
@@ -358,12 +358,9 @@
 
 		case autoDetect:
 				/*
-				 * that the ALSA driver could be compiled doesn't say anything
-				 * about whether it will work (the user might be using an OSS
-				 * kernel driver) so we'll use a value less than the OSS one
-				 * here, because OSS will most certainly work (ALSA's OSS emu)
+				 * alsa as default
 				 */
-				return 5;
+				return 15;
 			break;
 
 		default:
