--- arts-1.1.4/mcop/dispatcher.cc.orig	2003-09-29 22:34:19.000000000 +0200
+++ arts-1.1.4/mcop/dispatcher.cc	2003-09-29 22:36:50.000000000 +0200
@@ -214,8 +214,7 @@
 	orig_sigpipe = signal(SIGPIPE,SIG_IGN);
 	if(orig_sigpipe != SIG_DFL)
 	{
-		cerr << "mcop warning: user defined signal handler found for"
-		        " SIG_PIPE, overriding" << endl;
+		arts_debug("mcop warning: user defined signal handler found for SIG_PIPE, overriding");
 	}
 	
 	StartupManager::startup();
