Index: mcop/debug.cc
===================================================================
RCS file: /home/kde/arts/mcop/debug.cc,v
retrieving revision 1.14
diff -u -3 -p -r1.14 debug.cc
--- mcop/debug.cc	13 Oct 2003 19:59:41 -0000	1.14
+++ mcop/debug.cc	24 Feb 2004 17:07:40 -0000
@@ -92,7 +92,6 @@ static void output_message(Debug::Level 
 		  buff = arts_strdup_printf("%s -w 'Sound server warning message:\n\n%s' &", messageAppName, quoted_msg);
 		  break;
 	  case Debug::lInfo:
-		  buff = arts_strdup_printf("%s -i 'Sound server informational message:\n\n%s' &", messageAppName, quoted_msg);
 		  break;
 	  default:
 		  break; // avoid compile warning
