--- vlc-0.9.6/libs/loader/module.c.org	2007-01-23 11:55:14.000000000 +0200
+++ vlc-0.9.6/libs/loader/module.c	    2007-01-23 11:55:40.000000000 +0200
@@ -377,7 +377,7 @@
 HMODULE WINAPI LoadLibraryExA(LPCSTR libname, HANDLE hfile, DWORD flags)
 {
 	WINE_MODREF *wm = 0;
-	char* listpath[] = { "", "", "/usr/lib/win32", "/usr/local/lib/win32", "/usr/lib/codecs", "/usr/local/lib/codecs", 0 };
+	char* listpath[] = { "", "", "/usr/lib/essential", "/usr/local/lib/win32", "/usr/lib/codecs", "/usr/local/lib/codecs", 0 };
 	extern char* def_path;
 	char path[512];
 	char checked[2000];
