diff -Naur vlc-1.0.0.orig/modules/audio_output/pulse.c vlc-1.0.0/modules/audio_output/pulse.c
--- vlc-1.0.0.orig/modules/audio_output/pulse.c	2009-06-15 16:30:21.000000000 +0300
+++ vlc-1.0.0/modules/audio_output/pulse.c	2009-07-09 15:39:11.883405289 +0300
@@ -97,7 +97,7 @@
 vlc_module_begin ()
     set_shortname( "Pulse Audio" )
     set_description( N_("Pulseaudio audio output") )
-    set_capability( "audio output", 40 )
+    set_capability( "audio output", 200 )
     set_category( CAT_AUDIO )
     set_subcategory( SUBCAT_AUDIO_AOUT )
     add_shortcut( "pulseaudio" )
