--- src/sigtool.cpp.orig	2009-02-24 10:34:31.000000000 +0200
+++ src/sigtool.cpp	2009-02-24 10:34:52.000000000 +0200
@@ -98,7 +98,7 @@
 	notes_layout->addMultiCellWidget(notes_hlp, 1,1, 0,1);
 	QHBoxLayout *notes_box = new QHBoxLayout(notes_hlp, KDialog::spacingHint() );
 	
-	notes_label = new QLabel(i18n(text), notes_hlp);
+	notes_label = new QLabel(text, notes_hlp);
 	notes_label->setAlignment( int( QLabel::WordBreak | QLabel::AlignVCenter ) );
 	//notes_label->setFixedSize(notes_label->sizeHint());
 	notes_box->addWidget(notes_label);
