--- Imlib/misc.c	2007-08-16 15:20:53.000000000 +0200
+++ Imlib/misc.c	2007-08-16 15:21:32.000000000 +0200
@@ -423,6 +423,8 @@
 	      if (xvir[i].depth > max)
 		max = xvir[i].depth;
 	    }
+	  if (max > 24)
+	    max = 24;
 	  if (max > 8)
 	    {
 	      id->x.depth = max;
@@ -455,6 +455,8 @@
 	      if (xvir[i].depth > max)
 		max = xvir[i].depth;
 	    }
+	  if (max > 24)
+	    max = 24;
 	  if (max > 8)
 	    {
 	      id->x.depth = max;
@@ -1004,6 +1006,8 @@
 	      if (xvir[i].depth > max)
 		max = xvir[i].depth;
 	    }
+	  if (max > 24)
+	    max = 24;
 	  if (max > 8)
 	    {
 	      id->x.depth = max;
@@ -1026,6 +1028,8 @@
 	      if (xvir[i].depth > max)
 		max = xvir[i].depth;
 	    }
+	  if (max > 24)
+	    max = 24;
 	  if (max > 8)
 	    {
 	      id->x.depth = max;
