--- cmake/Ts.cmake.orig	2009-07-08 23:47:01.000000000 +0300
+++ cmake/Ts.cmake	2009-07-08 23:48:16.000000000 +0300
@@ -64,7 +64,7 @@
                 
             SET(qm_files ${qm_files} ${_outXml} ${_out})
 
-			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/${BIN_DIR}/i18n")
+			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/avidemux/i18n")
         ENDFOREACH(ts_input ${ts_files})
 
         SET(${_sources} ${${_sources}} ${qm_files})
@@ -95,7 +95,7 @@
                 
             SET(qm_files ${qm_files} ${_out})
 
-			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/${BIN_DIR}/i18n")
+			INSTALL(FILES ${_out} DESTINATION "${CMAKE_INSTALL_PREFIX}/share/avidemux/i18n")
         ENDFOREACH(ts_input ${ts_files})
 
         SET(${_sources} ${${_sources}} ${qm_files})
--- avidemux/ADM_userInterfaces/ADM_QT4/ui_support.cpp.orig	2009-07-08 23:45:45.000000000 +0300
+++ avidemux/ADM_userInterfaces/ADM_QT4/ui_support.cpp	2009-07-08 23:46:25.000000000 +0300
@@ -73,7 +73,7 @@
 #ifdef __APPLE__
 	QString appdir = QCoreApplication::applicationDirPath() + "/../Resources/locale/";
 #else
-	QString appdir = QCoreApplication::applicationDirPath() + "/i18n/";
+	QString appdir = "/usr/share/avidemux/i18n/";
 #endif
 
 	loadTranslation(&qtTranslator, appdir + "qt_" + QLocale::system().name());
@@ -164,4 +164,4 @@
 	}
 
 	return "Qt";
-}
\ No newline at end of file
+}
