diff -up openchange-0.9-COCHRANE/libmapi/IMAPISession.c.talloc-doublefree openchange-0.9-COCHRANE/libmapi/IMAPISession.c
--- openchange-0.9-COCHRANE/libmapi/IMAPISession.c.talloc-doublefree	2010-12-02 08:57:01.325338027 +0100
+++ openchange-0.9-COCHRANE/libmapi/IMAPISession.c	2010-12-02 08:57:24.449338002 +0100
@@ -94,7 +94,6 @@ static enum MAPISTATUS FindGoodServer(st
 	MAPIFreeBuffer(SPropTagArray);
 	MAPIFreeBuffer(MId_array);
 	MAPIFreeBuffer(server_dn);
-	MAPIFreeBuffer(SRowSet);
 	OPENCHANGE_RETVAL_IF(retval, retval, mem_ctx);
 
 	/* Step 5. Extract host from ncacn_ip_tcp binding string */
