diff -up libmapi-0.8.2-ROMULUS/Doxyfile.in.xml-doc libmapi-0.8.2-ROMULUS/Doxyfile.in
--- libmapi-0.8.2-ROMULUS/Doxyfile.in.xml-doc	2008-04-02 13:16:12.000000000 +0200
+++ libmapi-0.8.2-ROMULUS/Doxyfile.in	2010-01-06 19:12:49.000000000 +0100
@@ -231,7 +231,7 @@ SUBGROUPING            = YES
 # Private class members and static file members will be hidden unless 
 # the EXTRACT_PRIVATE and EXTRACT_STATIC tags are set to YES
 
-EXTRACT_ALL            = NO
+EXTRACT_ALL            = YES
 
 # If the EXTRACT_PRIVATE tag is set to YES all private members of a class 
 # will be included in the documentation.
@@ -925,13 +925,13 @@ MAN_LINKS              = YES
 # generate an XML file that captures the structure of 
 # the code including all documentation.
 
-GENERATE_XML           = NO
+GENERATE_XML           = YES
 
 # The XML_OUTPUT tag is used to specify where the XML pages will be put. 
 # If a relative path is entered the value of OUTPUT_DIRECTORY will be 
 # put in front of it. If left blank `xml' will be used as the default path.
 
-XML_OUTPUT             = xml
+XML_OUTPUT             = xml/overview
 
 # The XML_SCHEMA tag can be used to specify an XML schema, 
 # which can be used by a validating XML parser to check the 
diff -up libmapi-0.8.2-ROMULUS/libmapiadmin/Doxyfile.in.xml-doc libmapi-0.8.2-ROMULUS/libmapiadmin/Doxyfile.in
--- libmapi-0.8.2-ROMULUS/libmapiadmin/Doxyfile.in.xml-doc	2008-09-13 07:11:06.000000000 +0200
+++ libmapi-0.8.2-ROMULUS/libmapiadmin/Doxyfile.in	2010-01-06 19:13:04.000000000 +0100
@@ -231,7 +231,7 @@ SUBGROUPING            = YES
 # Private class members and static file members will be hidden unless 
 # the EXTRACT_PRIVATE and EXTRACT_STATIC tags are set to YES
 
-EXTRACT_ALL            = NO
+EXTRACT_ALL            = YES
 
 # If the EXTRACT_PRIVATE tag is set to YES all private members of a class 
 # will be included in the documentation.
@@ -923,13 +923,13 @@ MAN_LINKS              = YES
 # generate an XML file that captures the structure of 
 # the code including all documentation.
 
-GENERATE_XML           = NO
+GENERATE_XML           = YES
 
 # The XML_OUTPUT tag is used to specify where the XML pages will be put. 
 # If a relative path is entered the value of OUTPUT_DIRECTORY will be 
 # put in front of it. If left blank `xml' will be used as the default path.
 
-XML_OUTPUT             = xml
+XML_OUTPUT             = xml/libmapiadmin
 
 # The XML_SCHEMA tag can be used to specify an XML schema, 
 # which can be used by a validating XML parser to check the 
diff -up libmapi-0.8.2-ROMULUS/libmapi/Doxyfile.in.xml-doc libmapi-0.8.2-ROMULUS/libmapi/Doxyfile.in
--- libmapi-0.8.2-ROMULUS/libmapi/Doxyfile.in.xml-doc	2008-09-10 13:23:14.000000000 +0200
+++ libmapi-0.8.2-ROMULUS/libmapi/Doxyfile.in	2010-01-06 19:12:42.000000000 +0100
@@ -231,7 +231,7 @@ SUBGROUPING            = YES
 # Private class members and static file members will be hidden unless 
 # the EXTRACT_PRIVATE and EXTRACT_STATIC tags are set to YES
 
-EXTRACT_ALL            = NO
+EXTRACT_ALL            = YES
 
 # If the EXTRACT_PRIVATE tag is set to YES all private members of a class 
 # will be included in the documentation.
@@ -929,13 +929,13 @@ MAN_LINKS              = YES
 # generate an XML file that captures the structure of 
 # the code including all documentation.
 
-GENERATE_XML           = NO
+GENERATE_XML           = YES
 
 # The XML_OUTPUT tag is used to specify where the XML pages will be put. 
 # If a relative path is entered the value of OUTPUT_DIRECTORY will be 
 # put in front of it. If left blank `xml' will be used as the default path.
 
-XML_OUTPUT             = xml
+XML_OUTPUT             = xml/libmapi
 
 # The XML_SCHEMA tag can be used to specify an XML schema, 
 # which can be used by a validating XML parser to check the 
diff -up libmapi-0.8.2-ROMULUS/libmapi++/Doxyfile.in.xml-doc libmapi-0.8.2-ROMULUS/libmapi++/Doxyfile.in
--- libmapi-0.8.2-ROMULUS/libmapi++/Doxyfile.in.xml-doc	2008-12-14 03:27:25.000000000 +0100
+++ libmapi-0.8.2-ROMULUS/libmapi++/Doxyfile.in	2010-01-06 19:12:57.000000000 +0100
@@ -231,7 +231,7 @@ SUBGROUPING            = YES
 # Private class members and static file members will be hidden unless 
 # the EXTRACT_PRIVATE and EXTRACT_STATIC tags are set to YES
 
-EXTRACT_ALL            = NO
+EXTRACT_ALL            = YES
 
 # If the EXTRACT_PRIVATE tag is set to YES all private members of a class 
 # will be included in the documentation.
@@ -908,13 +908,13 @@ MAN_LINKS              = YES
 # generate an XML file that captures the structure of 
 # the code including all documentation.
 
-GENERATE_XML           = NO
+GENERATE_XML           = YES
 
 # The XML_OUTPUT tag is used to specify where the XML pages will be put. 
 # If a relative path is entered the value of OUTPUT_DIRECTORY will be 
 # put in front of it. If left blank `xml' will be used as the default path.
 
-XML_OUTPUT             = xml
+XML_OUTPUT             = xml/libmapi++
 
 # The XML_SCHEMA tag can be used to specify an XML schema, 
 # which can be used by a validating XML parser to check the 
diff -up libmapi-0.8.2-ROMULUS/libocpf/Doxyfile.in.xml-doc libmapi-0.8.2-ROMULUS/libocpf/Doxyfile.in
--- libmapi-0.8.2-ROMULUS/libocpf/Doxyfile.in.xml-doc	2008-09-10 13:23:14.000000000 +0200
+++ libmapi-0.8.2-ROMULUS/libocpf/Doxyfile.in	2010-01-06 19:13:12.000000000 +0100
@@ -231,7 +231,7 @@ SUBGROUPING            = YES
 # Private class members and static file members will be hidden unless 
 # the EXTRACT_PRIVATE and EXTRACT_STATIC tags are set to YES
 
-EXTRACT_ALL            = NO
+EXTRACT_ALL            = YES
 
 # If the EXTRACT_PRIVATE tag is set to YES all private members of a class 
 # will be included in the documentation.
@@ -927,13 +927,13 @@ MAN_LINKS              = YES
 # generate an XML file that captures the structure of 
 # the code including all documentation.
 
-GENERATE_XML           = NO
+GENERATE_XML           = YES
 
 # The XML_OUTPUT tag is used to specify where the XML pages will be put. 
 # If a relative path is entered the value of OUTPUT_DIRECTORY will be 
 # put in front of it. If left blank `xml' will be used as the default path.
 
-XML_OUTPUT             = xml
+XML_OUTPUT             = xml/libocpf
 
 # The XML_SCHEMA tag can be used to specify an XML schema, 
 # which can be used by a validating XML parser to check the 
diff -up libmapi-0.8.2-ROMULUS/Makefile.xml-doc libmapi-0.8.2-ROMULUS/Makefile
--- libmapi-0.8.2-ROMULUS/Makefile.xml-doc	2009-03-02 11:27:03.000000000 +0100
+++ libmapi-0.8.2-ROMULUS/Makefile	2010-01-06 18:18:45.000000000 +0100
@@ -1422,6 +1422,7 @@ doxygen:	
 		echo "Doxify API documentation: HTML and man pages";		\
 		mkdir -p apidocs/html;						\
 		mkdir -p apidocs/man;						\
+		mkdir -p apidocs/xml;						\
 		$(DOXYGEN) Doxyfile;						\
 		$(DOXYGEN) libmapi/Doxyfile;					\
 		$(DOXYGEN) libmapiadmin/Doxyfile;				\
diff -up libmapi-0.8.2-ROMULUS/mapiproxy/Doxyfile.in.xml-doc libmapi-0.8.2-ROMULUS/mapiproxy/Doxyfile.in
--- libmapi-0.8.2-ROMULUS/mapiproxy/Doxyfile.in.xml-doc	2009-01-05 11:42:44.000000000 +0100
+++ libmapi-0.8.2-ROMULUS/mapiproxy/Doxyfile.in	2010-01-06 19:13:21.000000000 +0100
@@ -286,7 +286,7 @@ SYMBOL_CACHE_SIZE      = 0
 # Private class members and static file members will be hidden unless 
 # the EXTRACT_PRIVATE and EXTRACT_STATIC tags are set to YES
 
-EXTRACT_ALL            = NO
+EXTRACT_ALL            = YES
 
 # If the EXTRACT_PRIVATE tag is set to YES all private members of a class 
 # will be included in the documentation.
@@ -1113,13 +1113,13 @@ MAN_LINKS              = YES
 # generate an XML file that captures the structure of 
 # the code including all documentation.
 
-GENERATE_XML           = NO
+GENERATE_XML           = YES
 
 # The XML_OUTPUT tag is used to specify where the XML pages will be put. 
 # If a relative path is entered the value of OUTPUT_DIRECTORY will be 
 # put in front of it. If left blank `xml' will be used as the default path.
 
-XML_OUTPUT             = xml
+XML_OUTPUT             = xml/mapiproxy
 
 # The XML_SCHEMA tag can be used to specify an XML schema, 
 # which can be used by a validating XML parser to check the 
diff -up libmapi-0.8.2-ROMULUS/utils/mapitest/Doxyfile.in.xml-doc libmapi-0.8.2-ROMULUS/utils/mapitest/Doxyfile.in
--- libmapi-0.8.2-ROMULUS/utils/mapitest/Doxyfile.in.xml-doc	2008-12-14 03:27:25.000000000 +0100
+++ libmapi-0.8.2-ROMULUS/utils/mapitest/Doxyfile.in	2010-01-06 19:13:28.000000000 +0100
@@ -231,7 +231,7 @@ SUBGROUPING            = YES
 # Private class members and static file members will be hidden unless 
 # the EXTRACT_PRIVATE and EXTRACT_STATIC tags are set to YES
 
-EXTRACT_ALL            = NO
+EXTRACT_ALL            = YES
 
 # If the EXTRACT_PRIVATE tag is set to YES all private members of a class 
 # will be included in the documentation.
@@ -921,13 +921,13 @@ MAN_LINKS              = YES
 # generate an XML file that captures the structure of 
 # the code including all documentation.
 
-GENERATE_XML           = NO
+GENERATE_XML           = YES
 
 # The XML_OUTPUT tag is used to specify where the XML pages will be put. 
 # If a relative path is entered the value of OUTPUT_DIRECTORY will be 
 # put in front of it. If left blank `xml' will be used as the default path.
 
-XML_OUTPUT             = xml
+XML_OUTPUT             = xml/mapitest
 
 # The XML_SCHEMA tag can be used to specify an XML schema, 
 # which can be used by a validating XML parser to check the 
