diff -Nuar kdelibs-3.5.10.orig/mimetypes/application/x-pisi.desktop kdelibs-3.5.10/mimetypes/application/x-pisi.desktop
--- kdelibs-3.5.10.orig/mimetypes/application/x-pisi.desktop	1970-01-01 02:00:00.000000000 +0200
+++ kdelibs-3.5.10/mimetypes/application/x-pisi.desktop	2011-06-27 10:12:49.274000046 +0300
@@ -0,0 +1,20 @@
+[Desktop Entry]
+Comment=Pisi File
+Comment[da]=Pisi-program
+Comment[de]=Pisi-Programm
+Comment[es]=Programa en Pisi
+Comment[fr]=Programme Pisi
+Comment[hu]=Pisi-program
+Comment[it]=Programma Pisi
+Comment[nl]=Pisi-programma
+Comment[pl]=Program w Pisi
+Comment[pt_BR]=Programa Pisi
+Comment[ru]=Программа на Pisi
+Comment[sr]=Pisi програм
+Comment[sv]=Pisi-program
+Comment[tr]=Pisi Programı
+Comment[uk]=Програма PISI
+Icon=package-manager
+Type=MimeType
+MimeType=application/x-pisi
+Patterns=*.pisi
--- kdelibs-3.5.10.orig/mimetypes/application/Makefile.am	2011-06-27 09:31:02.615000002 +0300
+++ kdelibs-3.5.10/mimetypes/application/Makefile.am	2011-06-27 13:55:21.811000407 +0300
@@ -67,7 +67,7 @@
 	x-java-jnlp-file.desktop xml-dtd.desktop fits.desktop \
 	x-rar-compressed.desktop \
 	x-sqlite2.desktop x-sqlite3.desktop chm.desktop \
-	x-mplayer2.desktop
+	x-mplayer2.desktop x-pisi.desktop
 
 mimetypeapplicationdatadir = $(kde_mimedir)/application
 
