Index: kdelibs-3.5.10/interfaces/ktexteditor/kcm_ktexteditor.desktop
===================================================================
--- kdelibs-3.5.10.orig/interfaces/ktexteditor/kcm_ktexteditor.desktop
+++ kdelibs-3.5.10/interfaces/ktexteditor/kcm_ktexteditor.desktop
@@ -152,6 +152,7 @@ Comment[ta]=உரை மற்றும
 Comment[te]=కార్యక్రమాలకు వాచకం సరిచేయు సేవ ఒక వాచకం విక్షణ మరియు సరిచెయుటకు సహాయపడును. కెడిఈ కార్యక్రమాలు వాచకం సరిచెయు సౌకర్యాలు వున్నట్టైతె దిని సేవ ను ఉపయొగించొకొవలెను
 Comment[tg]=Хидмати таҳриргари матн барномаҳоро бо намоишгар ва таҳриргари матн таъмин мекунад. Барномаҳои KDE дорои таҳриргари матн бояд ин хидмат истифода барад.
 Comment[th]=บริการตัวแก้ไขข้อความ จะจัดเตรียมตัวดูข้อความและตัวแก้ไขข้อความให้กับแอพพลิเคชัน แอพพลิเคชันของ KDE ที่เตรียมความสามารถในการแก้ไขข้อความมาให้ ควรที่จะใช้บริการนี้
+Comment[tr]=Metin düzenleyici servisi uygulamaların metin görüntülemesini ve düzenlemesini sağlar. Metin düzenleyen KDE uygulamalarının bu servisi kullanmaları gerekir.
 Comment[tt]=Mäten üzgärtü xezmäte törle yazılımnarnı mäten kürsätkeçe/üzgärtkeçe belän tä'min itä. Mäten üzgärtüçe KDE-yazılımnar bu xezmätne qullanırğa tieş.
 Comment[uk]=Служба "Текстовий редактор" постачає програми з переглядачем та редактором тексту. Програми KDE, які надають можливість редагування тексту, повинні користуватись цією службою.
 Comment[uz]=Matn tarhrirchi xizmati dasturlarni matn koʻruvchi va tahrirchi bilan taʼminlaydi. Matn tahrirlash imkoniyatini yaratuvchi KDE dasturlari shu xizmatni ishlatishi kerak.
