diff -Nur akode-2.0.2-old/akode/plugins/ffmpeg_decoder/ffmpeg_decoder.cpp akode-2.0.2/akode/plugins/ffmpeg_decoder/ffmpeg_decoder.cpp
--- akode-2.0.2-old/akode/plugins/ffmpeg_decoder/ffmpeg_decoder.cpp	2008-04-11 09:53:13.000000000 +0300
+++ akode-2.0.2/akode/plugins/ffmpeg_decoder/ffmpeg_decoder.cpp	2008-04-11 09:55:08.000000000 +0300
@@ -25,10 +25,12 @@
 #include "audioframe.h"
 #include "decoder.h"
 
+extern "C" {
 #include <assert.h>
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
-#include <ffmpeg/avio.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
+#include <libavformat/avio.h>
+}
 
 #include "ffmpeg_decoder.h"
 #include <iostream>
