--- cups-1.4.6/cups/language.c.orig	2011-02-22 16:09:23.390000054 +0200
+++ cups-1.4.6/cups/language.c	2011-02-22 16:11:34.248000054 +0200
@@ -44,6 +44,7 @@
 #include "debug.h"
 #include <stdlib.h>
 #include <errno.h>
+#include <locale.h>
 #ifdef HAVE_LANGINFO_H
 #  include <langinfo.h>
 #endif /* HAVE_LANGINFO_H */
@@ -599,8 +600,9 @@
   else
   {
    /*
-    * Copy the parts of the locale string over safely...
+    * Copy the parts of the locale string over safely and ignore locale dependant character conversion...
     */
+    setlocale(LC_CTYPE, "C");
 
     for (ptr = langname; *language; language ++)
       if (*language == '_' || *language == '-' || *language == '.')
