diff -Nur util-linux-ng-2.14.2-old/misc-utils/Makefile.am util-linux-ng-2.14.2/misc-utils/Makefile.am
--- util-linux-ng-2.14.2-old/misc-utils/Makefile.am	2009-02-17 01:38:50.000000000 +0200
+++ util-linux-ng-2.14.2/misc-utils/Makefile.am	2009-02-17 01:39:11.000000000 +0200
@@ -4,7 +4,7 @@
 
 bin_PROGRAMS =
 
-usrbinexec_PROGRAMS = cal ddate hostid logger look mcookie \
+usrbinexec_PROGRAMS = cal ddate logger look mcookie \
 	namei script whereis scriptreplay
 EXTRA_DIST += README.cal README.ddate README.namei README.namei2
 
@@ -14,7 +14,7 @@
 
 CLEANFILES = chkdupexe
 
-dist_man_MANS = cal.1 chkdupexe.1 ddate.1 hostid.1 logger.1 look.1 mcookie.1 \
+dist_man_MANS = cal.1 chkdupexe.1 ddate.1 logger.1 look.1 mcookie.1 \
 	namei.1 script.1 whereis.1 scriptreplay.1
 
 if HAVE_TINFO
