Index: luatex-beta-0.40.6/source/texk/web2c/luatexdir/image/pdftoepdf.cc
===================================================================
--- luatex-beta-0.40.6.orig/source/texk/web2c/luatexdir/image/pdftoepdf.cc
+++ luatex-beta-0.40.6/source/texk/web2c/luatexdir/image/pdftoepdf.cc
@@ -666,7 +666,7 @@ static void writeEncodings()
     }
     for (r = encodingList; r != NULL; r = n) {
         n = r->next;
-        delete r->font;
+	r->font->decRefCnt();
         delete r;
     }
 }
