Index: luatex-beta-0.40.6/source/texk/web2c/configure.ac
===================================================================
--- luatex-beta-0.40.6.orig/source/texk/web2c/configure.ac
+++ luatex-beta-0.40.6/source/texk/web2c/configure.ac
@@ -251,6 +251,7 @@ KPSE_FREETYPE2_FLAGS
 KPSE_TECKIT_FLAGS
 KPSE_ICU_FLAGS
 KPSE_OBSDCOMPAT_FLAGS
+PKG_CHECK_MODULES(ZZIP, zziplib )
 
 dnl Write output here, instead of putting a zillion -D's on the command line.
 AC_CONFIG_HEADERS([c-auto.h:c-auto.in],
Index: luatex-beta-0.40.6/source/texk/web2c/luatexdir/am/luasocket.am
===================================================================
--- luatex-beta-0.40.6.orig/source/texk/web2c/luatexdir/am/luasocket.am
+++ luatex-beta-0.40.6/source/texk/web2c/luatexdir/am/luasocket.am
@@ -7,9 +7,8 @@
 ##
 EXTRA_LIBRARIES += libluasocket.a
 
-libluasocket_a_DEPENDENCIES = $(ZLIB_DEPEND) libzzip.a 
+libluasocket_a_DEPENDENCIES = $(ZLIB_DEPEND)
 
-$(libluasocket_a_OBJECTS): libzzip.a 
 
 libluasocket_a_CPPFLAGS = \
 	-DLUASOCKET_DEBUG -I$(top_srcdir)/luatexdir/luasocket -I$(srcdir)/luatexdir/lua51
Index: luatex-beta-0.40.6/source/texk/web2c/luatexdir/am/luatex.am
===================================================================
--- luatex-beta-0.40.6.orig/source/texk/web2c/luatexdir/am/luatex.am
+++ luatex-beta-0.40.6/source/texk/web2c/luatexdir/am/luatex.am
@@ -15,20 +15,20 @@ nodist_EXTRA_luatex_SOURCES = dummy.cxx
 
 luatex_tangle = WEBINPUTS=$(srcdir)/luatexdir $(buildenv) $(LUATANGLE)
 
-luatex_CPPFLAGS = $(ZLIB_INCLUDES) $(LIBPNG_INCLUDES) $(XPDF_INCLUDES)
+luatex_CPPFLAGS = $(ZLIB_INCLUDES) $(LIBPNG_INCLUDES) $(XPDF_INCLUDES) $(ZZIP_CFLAGS)
 luatex_CPPFLAGS += $(OBSDCOMPAT_INCLUDES) -I$(srcdir)/libmd5
 luatex_CPPFLAGS += -Iluatexdir -I$(srcdir)/luatexdir -I$(srcdir)/luatexdir/lua51 -I$(srcdir)/mplibdir
 luatex_CPPFLAGS += -Dextra_version_info=`date +-%Y%m%d%H`
 
-luatex_ldadd = libluatex.a libff.a libluamisc.a libzzip.a libluasocket.a liblua51.a
-luatex_ldadd += $(LIBPNG_LIBS) $(ZLIB_LIBS) $(XPDF_LIBS)
+luatex_ldadd = libluatex.a libff.a libluamisc.a libluasocket.a liblua51.a
+luatex_ldadd += $(LIBPNG_LIBS) $(ZLIB_LIBS) $(XPDF_LIBS) $(ZZIP_LIBS)
 luatex_ldadd += $(OBSDCOMPAT_LIBS) libmd5.a libmplib.a 
 
 luatex_LDADD = $(luatex_ldadd) $(LDADD) $(lua_socketlibs)
 
 luatex_DEPENDENCIES = $(proglib) libluatex.a
 luatex_DEPENDENCIES += $(LIBPNG_DEPEND) $(ZLIB_DEPEND) $(XPDF_DEPEND)
-luatex_DEPENDENCIES += $(OBSDCOMPAT_DEPEND) libmd5.a libmplib.a
+luatex_DEPENDENCIES += $(OBSDCOMPAT_DEPEND) libmd5.a libmplib.a liblua51.a
 
 $(luatex_OBJECTS): libluatex.a libmplib.a
 
Index: luatex-beta-0.40.6/source/texk/web2c/luatexdir/am/luamisc.am
===================================================================
--- luatex-beta-0.40.6.orig/source/texk/web2c/luatexdir/am/luamisc.am
+++ luatex-beta-0.40.6/source/texk/web2c/luatexdir/am/luamisc.am
@@ -14,8 +14,7 @@ $(libluamisc_a_OBJECTS): libluasocket.a
 
 libluamisc_a_CPPFLAGS = \
 	$(ZLIB_INCLUDES) \
-	-I$(top_srcdir)/luatexdir/lua51 \
-	-I$(top_srcdir)/luatexdir/zziplib
+	-I$(top_srcdir)/luatexdir/lua51
 
 libluamisc_a_SOURCES = \
 	luatexdir/luafilesystem/src/lfs.c \
