Index: luatex-beta-0.40.6/source/texk/kpathsea/texmf.cnf
===================================================================
--- luatex-beta-0.40.6.orig/source/texk/kpathsea/texmf.cnf
+++ luatex-beta-0.40.6/source/texk/kpathsea/texmf.cnf
@@ -493,7 +493,7 @@ try_std_extension_first = t
 shell_escape = p
 
 % No spaces in this command list.
-shell_escape_commands = \
+shell_escape_commands = \\
 bibtex,convert,dvips,epstopdf,epspdf,etex,fc-match,gnuplot,\
 kpsewhich,latex,luatex,lualatex,makeindex,mpost,\
 pdfcrop,pdflatex,pdfluatex,ps2pdf,ps4pdf,pstopdf,pygmentize,\
