Index: luatex-0.40.5/source/texk/web2c/luatexdir/image/pdftoepdf.cc
===================================================================
--- luatex-0.40.5.orig/source/texk/web2c/luatexdir/image/pdftoepdf.cc	2009-10-17 16:32:12.000000000 +0400
+++ luatex-0.40.5/source/texk/web2c/luatexdir/image/pdftoepdf.cc	2009-10-17 16:32:12.000000000 +0400
@@ -720,7 +720,7 @@
     // this works only for PDF 1.x -- but since any versions of PDF newer
     // than 1.x will not be backwards compatible to PDF 1.x, pdfTeX will
     // then have to changed drastically anyway.
-    pdf_version_found = pdf_doc->doc->getPDFVersion();
+    pdf_version_found = pdf_doc->doc->getPDFMajorVersion() + 0.1 * pdf_doc->doc->getPDFMinorVersion();
     pdf_version_wanted = 1 + (minor_pdf_version_wanted * 0.1);
     if (pdf_version_found > pdf_version_wanted + 0.01) {
         char msg[] =
