diff -Nur yakuake-2.8.1-orig/yakuake/src/yakuake.desktop yakuake-2.8.1/yakuake/src/yakuake.desktop
--- yakuake-2.8.1-orig/yakuake/src/yakuake.desktop	2011-01-29 17:45:42.293000036 +0200
+++ yakuake-2.8.1/yakuake/src/yakuake.desktop	2011-01-29 17:51:27.819999892 +0200
@@ -6,6 +6,8 @@
 Icon=yakuake
 Type=Application
 Categories=Qt;KDE;System;TerminalEmulator;
+GenericName=Quake-style Terminal Emulator
+GenericName[tr]=Quake-stili Terminal Emülatörü
 Comment=A Quake-style terminal emulator based on KDE Konsole technology.
 Comment[ar]=مضاهي نافذة أوامر من نوع Quake معتمد على تقنية KDE Konsole.
 Comment[bg]=Терминален емулатор в стил Quake, базиран на конзолната технология на KDE.
