diff -dur apollon-1.0.2.1.orig/apollon/libapollon/giftsocket.cpp apollon-1.0.2.1/apollon/libapollon/giftsocket.cpp
--- apollon-1.0.2.1.orig/apollon/libapollon/giftsocket.cpp	2004-01-28 21:48:33.000000000 +0200
+++ apollon-1.0.2.1/apollon/libapollon/giftsocket.cpp	2007-04-15 20:06:27.000000000 +0300
@@ -92,7 +92,8 @@
 {
 	QCString localdata = m_codec->fromUnicode(data);
 
-	return QSocket::writeBlock(localdata, len);
+	//return QSocket::writeBlock(localdata, len);
+	return QSocket::writeBlock(localdata, localdata.length());
 }
 
 }


