diff -Nur libxslt-1.1.24-old/doc/Makefile.in libxslt-1.1.24/doc/Makefile.in
--- libxslt-1.1.24-old/doc/Makefile.in	2009-02-03 02:38:16.000000000 +0200
+++ libxslt-1.1.24/doc/Makefile.in	2009-02-03 02:38:19.000000000 +0200
@@ -210,7 +210,7 @@
 top_srcdir = @top_srcdir@
 
 # The name of the module.
-DOC_MODULE = libxslt-$(VERSION)
+DOC_MODULE = libxslt
 
 # The top-level SGML file.
 DOC_MAIN_XML_FILE = libxslt.xml
