diff -up system-config-printer-1.1.15/statereason.py.localize-statereason system-config-printer-1.1.15/statereason.py
--- system-config-printer-1.1.15/statereason.py.localize-statereason	2009-11-30 12:43:08.000000000 +0000
+++ system-config-printer-1.1.15/statereason.py	2009-12-03 14:57:17.285051607 +0000
@@ -138,7 +138,7 @@ class StateReason:
                     reason = localized_reason[:-2]
                 else:
                     reason = self.get_reason()
-            except cups.IPPError:
+            except (cups.IPPError, RuntimeError):
                 reason = self.get_reason()
 
             text = _("Printer '%s': '%s'.") % (self.get_printer (), reason)
