diff -Nur mplayer-20100114-old/configure mplayer-20100114/configure
--- mplayer-20100114-old/configure	2010-01-14 16:49:44.352804883 +0200
+++ mplayer-20100114/configure	2010-01-14 16:50:15.236804916 +0200
@@ -5123,7 +5123,7 @@
 if test "$_vaapi" = yes ; then
   def_vaapi='#define CONFIG_VAAPI 1'
   libs_mencoder="$libs_mencoder -lva"
-  libs_mplayer="$libs_mplayer -lva-x11"
+  libs_mplayer="$libs_mplayer -lva -lva-x11"
   _vomodules="vaapi $_vomodules"
 else
   def_vaapi='#define CONFIG_VAAPI 0'
