diff -Nur mplayer-20100114-old/libavcodec/mpeg4videodec.c mplayer-20100114/libavcodec/mpeg4videodec.c
--- mplayer-20100114-old/libavcodec/mpeg4videodec.c	2010-01-14 16:11:36.427930406 +0200
+++ mplayer-20100114/libavcodec/mpeg4videodec.c	2010-01-14 16:12:43.773930125 +0200
@@ -2119,7 +2119,7 @@
     for(;;) {
         if(get_bits_count(gb) >= gb->size_in_bits){
             if(gb->size_in_bits==8 && (s->divx_version || s->xvid_build)){
-                av_log(s->avctx, AV_LOG_ERROR, "frame skip %d\n", gb->size_in_bits);
+                //av_log(s->avctx, AV_LOG_ERROR, "frame skip %d\n", gb->size_in_bits);
                 return FRAME_SKIPPED; //divx bug
             }else
                 return -1; //end of stream
