Index: etc/codecs.conf
===================================================================
--- etc/codecs.conf	(revision 30265)
+++ etc/codecs.conf	(revision 30264)
@@ -3967,6 +3967,20 @@
   driver acm
   dll "msadp32.acm"
 
+audiocodec mp3
+  info "mp3lib MPEG layer-2, layer-3"
+  status working
+  comment "Optimized to MMX/SSE/3Dnow!"
+  format 0x50        ; layer-1 && layer-2
+  format 0x55        ; layer-3
+  format 0x5500736d  ; "ms\0\x55" older mp3 fcc (MOV files)
+  format 0x5000736d  ; "ms\0\x50" older mp2 fcc (MOV files)
+  format 0x55005354  ;  broken file
+  fourcc ".mp3"      ; CBR/VBR MP3 (MOV files)
+  fourcc "MP3 "      ; used in .nsv files
+  fourcc "LAME"      ; used in mythtv .nuv files
+  driver mp3lib
+
 audiocodec ffpcmdaud
   info "D-Cinema audio (FFmpeg)"
   status untested
@@ -4051,26 +4065,11 @@
   comment "integer only"
   status working
   format 0x50
-  format 0x5000736d  ; "ms\0\x50" older mp2 fcc (MOV files)
   fourcc ".mp2"      ; MP2 (MOV files)
   fourcc ".mp1"      ; MP1 (MOV files)
   driver ffmpeg
   dll "mp2"
 
-audiocodec mp3
-  info "mp3lib MPEG layer-2, layer-3"
-  status working
-  comment "Optimized to MMX/SSE/3Dnow!"
-  format 0x50        ; layer-1 && layer-2
-  format 0x55        ; layer-3
-  format 0x5500736d  ; "ms\0\x55" older mp3 fcc (MOV files)
-  format 0x5000736d  ; "ms\0\x50" older mp2 fcc (MOV files)
-  format 0x55005354  ;  broken file
-  fourcc ".mp3"      ; CBR/VBR MP3 (MOV files)
-  fourcc "MP3 "      ; used in .nsv files
-  fourcc "LAME"      ; used in mythtv .nuv files
-  driver mp3lib
-
 audiocodec mad
   info "libMAD MPEG layer 1-2-3"
   status working
