diff -Nur mplayer-20090910-old/libmpdemux/demux_mkv.c mplayer-20090910/libmpdemux/demux_mkv.c
--- mplayer-20090910-old/libmpdemux/demux_mkv.c	2009-09-10 22:36:47.830942639 +0300
+++ mplayer-20090910/libmpdemux/demux_mkv.c	2009-09-10 22:38:21.226943913 +0300
@@ -45,6 +45,7 @@
 #include "libass/ass_mp.h"
 
 #include "libavutil/common.h"
+#include "libavformat/avio.h"
 
 #ifdef CONFIG_QTX_CODECS
 #include "loader/qtx/qtxsdk/components.h"
@@ -1755,6 +1756,8 @@
         track->a_formattag = mmioFOURCC('W', 'V', 'P', 'K');
       else if (!strcmp(track->codec_id, MKV_A_TRUEHD))
         track->a_formattag = mmioFOURCC('T', 'R', 'H', 'D');
+      else if (!strcmp(track->codec_id, MKV_A_TTA))
+        track->a_formattag = mmioFOURCC('T', 'T', 'A', '1');
       else if (!strcmp(track->codec_id, MKV_A_FLAC))
         {
           if (track->private_data == NULL || track->private_size == 0)
@@ -1998,6 +2001,20 @@
   else if (track->a_formattag == mmioFOURCC('W', 'V', 'P', 'K') ||
            track->a_formattag == mmioFOURCC('T', 'R', 'H', 'D'))
     {  /* do nothing, still works */  }
+  else if (track->a_formattag == mmioFOURCC('T', 'T', 'A', '1'))
+    {
+        sh_a->codecdata_len = 30;
+        sh_a->codecdata = malloc(30);
+        ByteIOContext b;
+        init_put_byte(&b, sh_a->codecdata, sh_a->codecdata_len, 1,
+                      NULL, NULL, NULL, NULL);
+        put_buffer(&b, (uint8_t *) "TTA1", 4);
+        put_le16(&b, 1);
+        put_le16(&b, track->a_channels);
+        put_le16(&b, track->a_bps);
+        put_le32(&b, track->a_sfreq);
+        put_le32(&b, mkv_d->duration * track->a_sfreq);
+    }
   else if (!track->ms_compat || (track->private_size < sizeof(WAVEFORMATEX)))
     {
       free_sh_audio(demuxer, track->tnum);
diff -Nur mplayer-20090910-old/libmpdemux/matroska.h mplayer-20090910/libmpdemux/matroska.h
--- mplayer-20090910-old/libmpdemux/matroska.h	2009-09-10 22:36:47.830942639 +0300
+++ mplayer-20090910/libmpdemux/matroska.h	2009-09-10 22:36:58.572947036 +0300
@@ -52,6 +52,7 @@
 #define MKV_A_FLAC       "A_FLAC"
 #define MKV_A_WAVPACK    "A_WAVPACK4"
 #define MKV_A_TRUEHD     "A_TRUEHD"
+#define MKV_A_TTA        "A_TTA1"
 
 #define MKV_V_MSCOMP     "V_MS/VFW/FOURCC"
 #define MKV_V_REALV10    "V_REAL/RV10"
