--- bash_completion.orig/bash_completion	2011-06-20 12:01:33.880000024 +0300
+++ bash_completion/bash_completion	2011-06-20 12:02:46.723000026 +0300
@@ -108,6 +108,12 @@
 complete -f -X '!*.@(exe|EXE|com|COM|scr|SCR|exe.so)' wine
 complete -f -X '!*.@(zip|ZIP|z|Z|gz|GZ|tgz|TGZ)' bzme
 complete -f -X '!*.@(?([xX]|[sS])[hH][tT][mM]?([lL]))' netscape mozilla lynx opera galeon curl dillo elinks amaya
+complete -f -X '!*.@(sxw|stw|sxg|sgl|doc|dot|rtf|txt|htm|html|odt|ott|odm)' lowriter
+complete -f -X '!*.@(sxi|sti|pps|ppt|pot|odp|otp)' loimpress
+complete -f -X '!*.@(sxc|stc|xls|xlw|xlt|csv|ods|ots)' localc
+complete -f -X '!*.@(sxd|std|sda|sdd|odg|otg)' lodraw
+complete -f -X '!*.@(sxm|smf|mml|odf)' lomath
+complete -f -X '!*.odb' lobase
 complete -f -X '!*.@(sxw|stw|sxg|sgl|doc|dot|rtf|txt|htm|html|odt|ott|odm)' oowriter
 complete -f -X '!*.@(sxi|sti|pps|ppt|pot|odp|otp)' ooimpress
 complete -f -X '!*.@(sxc|stc|xls|xlw|xlt|csv|ods|ots)' oocalc
