--- a/src/screens/ScrPanel.py	(revision 36076)
+++ b/src/screens/ScrPanel.py	(revision 36090)
@@ -114,7 +114,16 @@
         Kicker = qtxml.QDomElement
         Kicker = dom.elementsByTagName("kicker").item(0).toElement()
 
-        kickerConf.writeEntry("LegacyKMenu",not self.checkKickoff.isChecked())
+        shouldRestart = False
+
+        if kickerConf.readEntry("LegacyKMenu") == "1" \
+                and self.checkKickoff.isChecked():
+            shouldRestart = True
+        elif kickerConf.readEntry("LegacyKMenu") == "0" \
+                and not self.checkKickoff.isChecked():
+            shouldRestart = True
+
+        kickerConf.writeEntry("LegacyKMenu", not self.checkKickoff.isChecked())
         kickerConf.writeEntry("Transparent", self.getProperty(Kicker, "Transparent", "value"))
         kickerConf.writeEntry("SizePercentage", self.getProperty(Kicker, "SizePercentage", "value"))
         kickerConf.writeEntry("CustomSize", self.getProperty(Kicker, "CustomSize", "value"))
@@ -122,8 +131,12 @@
         kickerConf.writeEntry("Alignment", self.getProperty(Kicker, "Alignment", "value"))
         kickerConf.sync()
 
-        # restart kicker
-        client.send("kicker", "kicker", "restart()", "")
+        if shouldRestart:
+            # Restart kicker as switching to kickoff needs that
+            client.send("kicker", "kicker", "restart()", "")
+        else:
+            # For other things, configure() is faster and safer
+            client.send("kicker", "kicker", "configure()", "")
 
         """
         # we don't have a style-changing anymore.
