Index: src/kaptan.py
===================================================================
--- src/kaptan.py	(revision 32356)
+++ src/kaptan.py	(working copy)
@@ -29,7 +29,7 @@
 import screens.ScrWallpaper as ScrWallpaper
 import screens.ScrPackage as ScrPackage
 import screens.ScrMouse as ScrMouse
-import screens.ScrNetwork as ScrNetwork
+#import screens.ScrNetwork as ScrNetwork
 import screens.ScrPanel as ScrPanel
 import screens.ScrKeyboard as ScrKeyboard
 import screens.ScrGoodbye as ScrGoodbye
@@ -71,7 +71,7 @@
                  ScrPanel,
                  ScrMultiple,
                  ScrWallpaper,
-                 ScrNetwork,
+                 #ScrNetwork,
                  ScrGoodbye]
 else:
     #set avaiable screens
@@ -80,7 +80,7 @@
                      ScrPanel,
                      ScrMultiple,
                      ScrWallpaper,
-                     ScrNetwork,
+                     #ScrNetwork,
                      ScrPackage,
                      ScrGoodbye]
 
