diff -Nur iproute2-2.6.29-1-old/tc/Makefile iproute2-2.6.29-1/tc/Makefile
--- iproute2-2.6.29-1-old/tc/Makefile	2009-05-10 00:45:18.000000000 +0300
+++ iproute2-2.6.29-1/tc/Makefile	2009-05-10 00:46:24.000000000 +0300
@@ -100,7 +100,7 @@
 	rm -f emp_ematch.yacc.output
 
 q_atm.so: q_atm.c
-	$(CC) $(CFLAGS) -shared -fpic -o q_atm.so q_atm.c -latm
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared -fpic -o q_atm.so q_atm.c -latm
 
 %.yacc.c: %.y
 	$(YACC) $(YACCFLAGS) -o $@ $<
