diff -Naur phonon-4.3.1-orig/gstreamer/gstreamer.desktop phonon-4.3.1/gstreamer/gstreamer.desktop
--- phonon-4.3.1-orig/gstreamer/gstreamer.desktop	2009-05-14 17:09:59.635628898 +0300
+++ phonon-4.3.1/gstreamer/gstreamer.desktop	2009-05-14 17:10:42.184654810 +0300
@@ -7,7 +7,7 @@
 X-KDE-PhononBackendInfo-Version=0.1
 X-KDE-PhononBackendInfo-Website=http://gstreamer.freedesktop.org/
 Icon=phonon-gstreamer
-InitialPreference=10
+InitialPreference=15
 
 Name=GStreamer
 Name[pa]=ਜੀਸਟੀਰਮਰ
diff -Naur phonon-4.3.1-orig/xine/xine.desktop phonon-4.3.1/xine/xine.desktop
--- phonon-4.3.1-orig/xine/xine.desktop	2009-05-14 17:09:59.638628939 +0300
+++ phonon-4.3.1/xine/xine.desktop	2009-05-14 17:10:46.768629941 +0300
@@ -7,7 +7,7 @@
 X-KDE-PhononBackendInfo-Version=0.2.60
 X-KDE-PhononBackendInfo-Website=http://www.xinehq.de/
 Icon=phonon-xine
-InitialPreference=15
+InitialPreference=10
 
 Name=Xine
 Name[x-test]=xxXinexx
