--- a/gstreamer/mediaobject.cpp	(révision 966234)
+++ b/gstreamer/mediaobject.cpp	(copie de travail)
@@ -902,10 +902,27 @@
             setError(tr("Could not open media source."));
         break;
 
-    case MediaSource::Disc: // CD tracks can be specified by setting the url in the following way uri=cdda:4
+    case MediaSource::Disc:
         {
-            QUrl cdurl(QLatin1String("cdda://"));
-            if (createPipefromURL(cdurl))
+	    QString mediaUrl;
+	    switch (source.discType()) {
+	    case Phonon::NoDisc:
+                qWarning() << "I should never get to see a MediaSource that is a disc but doesn't specify which one";
+                return;
+            case Phonon::Cd:  // CD tracks can be specified by setting the url in the following way uri=cdda:4
+                mediaUrl = QLatin1String("cdda://");
+                break;
+            case Phonon::Dvd:
+                mediaUrl = QLatin1String("dvd://");
+                break;
+            case Phonon::Vcd:
+                mediaUrl = QLatin1String("vcd://");
+                break;
+            default:
+                qWarning() <<  "media " << source.discType() << " not implemented";
+                return;
+            }
+            if (!mediaUrl.isEmpty() && createPipefromURL(QUrl(mediaUrl)))
                 m_loading = true;
             else
                 setError(tr("Could not open media source."));
