--- includes/dhcpd.h.orig	2008-08-29 20:48:57.000000000 +0300
+++ includes/dhcpd.h	2009-05-15 11:53:53.000000000 +0300
@@ -1279,49 +1279,49 @@
 #else /* !DEBUG */
 
 #ifndef _PATH_DHCPD_CONF
-#define _PATH_DHCPD_CONF	"/etc/dhcpd.conf"
+#define _PATH_DHCPD_CONF	"/etc/dhcp/dhcpd.conf"
 #endif /* DEBUG */
 
 #ifndef _PATH_DHCPD_DB
-#define _PATH_DHCPD_DB		LOCALSTATEDIR"/db/dhcpd.leases"
+#define _PATH_DHCPD_DB  "/var/lib/dhcp/dhcpd.leases"
 #endif
 
 #ifndef _PATH_DHCPD6_DB
-#define _PATH_DHCPD6_DB		LOCALSTATEDIR"/db/dhcpd6.leases"
+#define _PATH_DHCPD6_DB  "/var/lib/dhcp/dhcpd6.leases"
 #endif
 
 #ifndef _PATH_DHCPD_PID
-#define _PATH_DHCPD_PID		LOCALSTATEDIR"/run/dhcpd.pid"
+#define _PATH_DHCPD_PID  "/var/run/dhcpd.pid"
 #endif
 
 #ifndef _PATH_DHCPD6_PID
-#define _PATH_DHCPD6_PID	LOCALSTATEDIR"/run/dhcpd6.pid"
+#define _PATH_DHCPD6_PID  "/var/run/dhcpd6.pid"
 #endif
 
 #endif /* DEBUG */
 
 #ifndef _PATH_DHCLIENT_CONF
-#define _PATH_DHCLIENT_CONF	"/etc/dhclient.conf"
+#define _PATH_DHCLIENT_CONF  "/etc/dhcp/dhclient.conf"
 #endif
 
 #ifndef _PATH_DHCLIENT_SCRIPT
-#define _PATH_DHCLIENT_SCRIPT	"/sbin/dhclient-script"
+#define _PATH_DHCLIENT_SCRIPT  "/usr/sbin/dhclient-script"
 #endif
 
 #ifndef _PATH_DHCLIENT_PID
-#define _PATH_DHCLIENT_PID	LOCALSTATEDIR"/run/dhclient.pid"
+#define _PATH_DHCLIENT_PID  "/var/run/dhclient.pid"
 #endif
 
 #ifndef _PATH_DHCLIENT6_PID
-#define _PATH_DHCLIENT6_PID	LOCALSTATEDIR"/run/dhclient6.pid"
+#define _PATH_DHCLIENT6_PID  "/var/run/dhclient6.pid"
 #endif
 
 #ifndef _PATH_DHCLIENT_DB
-#define _PATH_DHCLIENT_DB	LOCALSTATEDIR"/db/dhclient.leases"
+#define _PATH_DHCLIENT_DB  "/var/lib/dhcp/dhclient.leases"
 #endif
 
 #ifndef _PATH_DHCLIENT6_DB
-#define _PATH_DHCLIENT6_DB	LOCALSTATEDIR"/db/dhclient6.leases"
+#define _PATH_DHCLIENT6_DB  "/var/lib/dhcp/dhclient6.leases"
 #endif
 
 #ifndef _PATH_RESOLV_CONF
@@ -1329,11 +1329,11 @@
 #endif
 
 #ifndef _PATH_DHCRELAY_PID
-#define _PATH_DHCRELAY_PID	LOCALSTATEDIR"/run/dhcrelay.pid"
+#define _PATH_DHCRELAY_PID  "/var/run/dhcrelay.pid"
 #endif
 
 #ifndef _PATH_DHCRELAY6_PID
-#define _PATH_DHCRELAY6_PID	LOCALSTATEDIR"/run/dhcrelay6.pid"
+#define _PATH_DHCRELAY6_PID  "/var/run/dhcrelay6.pid"
 #endif
 
 #ifndef DHCPD_LOG_FACILITY
