From c82f50c4fdfb989793851c74bf05a88fe78a91a2 Mon Sep 17 00:00:00 2001
From: Johan Hedberg <johan.hedberg@nokia.com>
Date: Sun, 19 Dec 2010 13:35:13 +0200
Subject: [PATCH] Get rid of unnecessary manager_get_default_adapter function

---
 src/manager.c |    5 -----
 src/manager.h |    1 -
 2 files changed, 0 insertions(+), 6 deletions(-)

Index: bluez-4.82/src/manager.c
===================================================================
--- bluez-4.82.orig/src/manager.c
+++ bluez-4.82/src/manager.c
@@ -454,11 +454,6 @@ int manager_stop_adapter(int id)
 	return adapter_stop(adapter);
 }
 
-int manager_get_default_adapter()
-{
-	return default_adapter_id;
-}
-
 void manager_set_default_adapter(int id)
 {
 	struct btd_adapter *adapter;
Index: bluez-4.82/src/manager.h
===================================================================
--- bluez-4.82.orig/src/manager.h
+++ bluez-4.82/src/manager.h
@@ -41,7 +41,6 @@ int manager_unregister_adapter(int id);
 int manager_start_adapter(int id);
 int manager_stop_adapter(int id);
 void manager_add_adapter(const char *path);
-int manager_get_default_adapter();
 void manager_set_default_adapter(int id);
 void manager_update_svc(struct btd_adapter *adapter, uint8_t svc);
 void btd_manager_set_offline(gboolean offline);
