--- src/kdedistutils.py.orig	2008-02-19 15:32:27.000000000 +0200
+++ src/kdedistutils.py	2008-02-19 15:32:59.000000000 +0200
@@ -2111,7 +2111,12 @@
             else:
                 accu += c
     if accu!='':
-        result.append(accu)
+        if type==1:
+            result.append(int(accu))
+        else:
+            result.append(accu)
+
+
     return result
 
 ###########################################################################
