diff -ur pykdeextensions-0.4.0/src/kdedistutils.py pykdeextensions/src/kdedistutils.py
--- pykdeextensions-0.4.0/src/kdedistutils.py	2005-09-19 19:15:13.000000000 +0000
+++ pykdeextensions/src/kdedistutils.py	2005-11-01 13:33:07.000000000 +0000
@@ -708,7 +708,7 @@
             python_version = '%i.%i' % (sys.version_info[0],sys.version_info[1])
             
             # Build the 'stub' code.
-            cppcode = self.cpptemplate % {"moduledir": self.data_dir,
+            cppcode = self.cpptemplate % {"moduledir": ask_kde_config("--install data --expandvars").strip()+"/"+libraryname,
                                             "modulename": modulename,
                                             "factoryfunction": factoryfunction,
                                             "python_version": python_version}

