diff -Naur pykdeextensions-0.4.0-orig/src/kdedesigner.py pykdeextensions-0.4.0/src/kdedesigner.py
--- pykdeextensions-0.4.0-orig/src/kdedesigner.py	2009-12-15 10:44:38.765251336 +0200
+++ pykdeextensions-0.4.0/src/kdedesigner.py	2009-12-15 11:42:50.711250846 +0200
@@ -18,7 +18,7 @@
 import qtuicompiler
 
 python_import = __builtin__.__import__
-def load(*args):
+def load(*args, **kwargs):
     qtuicompiler.DynamicImport(args,True)
-    return apply(python_import,args)
+    return apply(python_import, args, kwargs)
 __builtin__.__import__ = load
