Index: pykdeextensions-0.4.0/src/kdedistutils.py
===================================================================
--- pykdeextensions-0.4.0.orig/src/kdedistutils.py
+++ pykdeextensions-0.4.0/src/kdedistutils.py
@@ -2201,7 +2201,4 @@ def compile_qtdesigner(ui_files,
 
 ###########################################################################
 def get_libdir_name():
-    if os.uname()[4] in ['x86_64','mips64','ppc64','sparc64','s390x']:
-        return 'lib64'
-    else:
-        return 'lib'
+    return 'lib'
