--- pykdeextensions-0.4.0/src/qtuicompiler.py	2005-05-22 02:56:46.000000000 +0300
+++ py-custom/src/qtuicompiler.py	2005-10-05 21:59:33.000000000 +0300
@@ -35,7 +35,7 @@
         pyuic_exe = find_executable('pyuic')
 
 ############################################################################
-def CompileUI(ui_file_name, py_file_name=None, kde=False):
+def CompileUI(ui_file_name, py_file_name=None, kde=True):
     pyuic_exe = find_executable('pyuic',pyqt_configuration.default_bin_dir)
     if pyuic_exe is None:
         # Search on the $Path.
