Index: kdelibs-3.5.10/kdeprint/management/kmwizard.cpp
===================================================================
--- kdelibs-3.5.10.orig/kdeprint/management/kmwizard.cpp
+++ kdelibs-3.5.10/kdeprint/management/kmwizard.cpp
@@ -53,7 +53,7 @@
 KMWizard::KMWizard(QWidget *parent, const char *name)
 : QDialog(parent,name,true)
 {
-	m_start = KMWizard::Start;
+	m_start = KMWizard::Backend;
 	m_end = KMWizard::End;
 	m_inclusive = true;
 	m_printer = new KMPrinter();
@@ -108,7 +108,7 @@ KMWizard::KMWizard(QWidget *parent, cons
 	btn_->addWidget(m_cancel);
 
 	// create standard pages
-	addPage(new KMWInfoPage(this));
+	//addPage(new KMWInfoPage(this));
 	m_backend = new KMWBackend(this);
 	addPage(m_backend);
 	addPage(new KMWPassword(this));
