diff -ur k3b-1.0.4.orig/libk3b/projects/datacd/k3bisooptions.cpp k3b-1.0.4/libk3b/projects/datacd/k3bisooptions.cpp
--- k3b-1.0.4.orig/libk3b/projects/datacd/k3bisooptions.cpp	2007-11-02 11:55:40.000000000 +0200
+++ k3b-1.0.4/libk3b/projects/datacd/k3bisooptions.cpp	2008-01-04 15:56:28.000000000 +0200
@@ -52,7 +52,7 @@
 
   m_doNotCacheInodes = true;
 
-  m_isoLevel = 2;
+  m_isoLevel = 3;
 
   m_discardSymlinks = false;
   m_discardBrokenSymlinks = false;
