Index: plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
===================================================================
--- plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp.orig
+++ plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
@@ -194,7 +194,11 @@ QString K3bFFMpegFile::typeComment() con
     return i18n("Windows Media v1");
   case CODEC_ID_WMAV2:
     return i18n("Windows Media v2");
+#if LIBAVCODEC_VERSION_MAJOR < 52
   case CODEC_ID_MP3LAME:
+#else
+  case CODEC_ID_MP3:
+#endif
     return i18n("MPEG 1 Layer III");
   case CODEC_ID_AAC:
     return i18n("Advanced Audio Coding (AAC)");
@@ -289,8 +293,10 @@ int K3bFFMpegFile::fillOutputBuffer()
 
 #ifdef FFMPEG_BUILD_PRE_4629
     int len = avcodec_decode_audio( &d->formatContext->streams[0]->codec,
-#else
+#elif LIBAVCODEC_VERSION_MAJOR < 52
     int len = avcodec_decode_audio( d->formatContext->streams[0]->codec,
+#else
+    int len = avcodec_decode_audio2( d->formatContext->streams[0]->codec,
 #endif
 				    (short*)d->outputBuffer, &d->outputBufferSize,
 				    d->packetData, d->packetSize );
Index: libk3bdevice/configure.in.in
===================================================================
--- libk3bdevice/configure.in.in.orig
+++ libk3bdevice/configure.in.in
@@ -128,7 +128,7 @@ if test "x$ac_cv_use_hal" = "xyes" ; the
   dbus_lib=NOTFOUND
   dbus=NOTFOUND
 
-  search_incs="$kde_includes /usr/include /usr/include/dbus-1.0 /usr/local/include /usr/local/include/dbus-1.0"
+  search_incs="$kde_includes $kde_includes/dbus-1.0 /usr/include /usr/include/dbus-1.0 /usr/local/include /usr/local/include/dbus-1.0"
   AC_FIND_FILE(dbus/dbus.h, $search_incs, dbus_incdir)
 
   search_incs_arch_deps="$kde_includes /usr/lib64/dbus-1.0/include /usr/lib/dbus-1.0/include /usr/local/lib/dbus-1.0/include"
@@ -165,7 +165,7 @@ if test "x$ac_cv_use_hal" = "xyes" ; the
   dbusqt_lib=NOTFOUND
   dbusqt=NOTFOUND
 
-  search_incs="$kde_includes /usr/include /usr/include/dbus-1.0 /usr/local/include /usr/local/include/dbus-1.0"
+  search_incs="$kde_includes $kde_includes/dbus-1.0 /usr/include /usr/include/dbus-1.0 /usr/local/include /usr/local/include/dbus-1.0"
   AC_FIND_FILE(dbus/connection.h, $search_incs, dbusqt_incdir)
 
   if test -r $dbusqt_incdir/dbus/connection.h ; then
Index: src/k3b.desktop
===================================================================
--- src/k3b.desktop.orig
+++ src/k3b.desktop
@@ -117,7 +117,7 @@ Name=K3b
 Name[ar]= K3b
 Name[bn]=কে-থ্রি-বি
 Name[hi]=के3बी
-MimeType=application/x-k3b
+MimeType=application/x-k3b;
 X-KDE-StartupNotify=true
 X-KDE-NativeMimeType=application/x-k3b;application/x-iso
 Categories=KDE;Application;AudioVideo;DiscBurning;
Index: src/Makefile.am
===================================================================
--- src/Makefile.am.orig
+++ src/Makefile.am
@@ -32,10 +32,7 @@ k3b_LDADD = ./option/liboption.la ./rip/
 
 SUBDIRS = option misc rip projects fastscale pics icons konqi mimetypes sounds
 
-xdg_apps_DATA = k3b.desktop
-
-silent_DATA = k3b-iso.desktop k3b-cue.desktop
-silentdir = $(kde_appsdir)/.hidden/
+xdg_apps_DATA = k3b.desktop k3b-iso.desktop k3b-cue.desktop
 
 ###################
 # this 10 paths are KDE specific. Use them:
Index: libk3b/jobs/k3bvideodvdtitletranscodingjob.cpp
===================================================================
--- libk3b/jobs/k3bvideodvdtitletranscodingjob.cpp.orig
+++ libk3b/jobs/k3bvideodvdtitletranscodingjob.cpp
@@ -27,6 +27,7 @@
 #include <qfile.h>
 #include <qfileinfo.h>
 
+inline const int qMax(const int a, const int b) { if (a < b) return b; return a; }
 
 class K3bVideoDVDTitleTranscodingJob::Private
 {
@@ -206,10 +207,11 @@ void K3bVideoDVDTitleTranscodingJob::sta
     *d->process << "--nice" << "19";
 
   // we only need 100 steps, but to make sure we use 150
+  int progressRate = qMax( 1, ( int )m_dvd[m_titleNumber-1].playbackTime().totalFrames()/150 );
   if ( d->usedTranscodeBin->version.simplify() >= K3bVersion( 1, 1, 0 ) )
-      *d->process << "--progress_meter" << "2" << "--progress_rate" << QString::number(m_dvd[m_titleNumber-1].playbackTime().totalFrames()/150);
+      *d->process << "--progress_meter" << "2" << "--progress_rate" << QString::number(progressRate);
   else
-      *d->process << "--print_status" << QString::number(m_dvd[m_titleNumber-1].playbackTime().totalFrames()/150);
+      *d->process << "--print_status" << QString::number(progressRate);
 
   // the input
   *d->process << "-i" << m_dvd.device()->blockDeviceName();
