diff -ur k3b-1.0.1.orig/src/k3bsystemproblemdialog.cpp k3b-1.0.1/src/k3bsystemproblemdialog.cpp
--- k3b-1.0.1.orig/src/k3bsystemproblemdialog.cpp	2007-04-18 18:43:19.000000000 +0300
+++ k3b-1.0.1/src/k3bsystemproblemdialog.cpp	2007-04-19 15:31:34.479771557 +0300
@@ -203,7 +203,7 @@
 					   false ) );
       }
     
-#ifdef Q_OS_LINUX
+#if 0
 
       //
       // Since kernel 2.6.8 older cdrecord versions are not able to use the SCSI subsystem when running suid root anymore
@@ -247,7 +247,7 @@
 					 false ) );
     }
     else {
-#ifdef Q_OS_LINUX
+#if 0
 #ifdef CDRECORD_SUID_ROOT_CHECK
       if( !k3bcore->externalBinManager()->binObject( "cdrdao" )->hasFeature( "suidroot" ) && getuid() != 0 )
 	problems.append( K3bSystemProblem( K3bSystemProblem::CRITICAL,
