Index: hal-0.5.14/fdi/policy/10osvendor/10-x11-input.fdi
===================================================================
--- hal-0.5.14.orig/fdi/policy/10osvendor/10-x11-input.fdi
+++ hal-0.5.14/fdi/policy/10osvendor/10-x11-input.fdi
@@ -13,6 +13,17 @@
        <merge key="input.x11_driver" type="string">evdev</merge>
     </match>
 
+    <match key="input.product" contains_outof="QUANTA OpticalTouchScreen;QUANTA Optical Touch Screen">
+       <merge key="input.x11_driver" type="string">evdev</merge>
+       <merge key="input.x11_options.ReportingMode" type="string">Raw</merge>
+       <merge key="input.x11_options.PacketCount" type="string">13</merge>
+       <merge key="input.x11_options.OpcodePressure" type="string">852034</merge>
+       <merge key="input.x11_options.OpcodeX" type="string">65584</merge>
+       <merge key="input.x11_options.OpcodeY" type="string">65585</merge>
+       <merge key="input.x11_options.CalibrationModel" type="string">1</merge>
+    </match>
+
+
     <!-- FIXME: Support tablets too. -->
     <match key="info.capabilities" contains="input.mouse">
       <merge key="input.x11_driver" type="string">mouse</merge>
