diff -ru kdegraphics-3.4.2.orig/kpdf/xpdf/splash/SplashFTFont.cc kdegraphics-3.4.2/kpdf/xpdf/splash/SplashFTFont.cc
--- kdegraphics-3.4.2.orig/kpdf/xpdf/splash/SplashFTFont.cc	2005-07-20 12:00:36.000000000 +0200
+++ kdegraphics-3.4.2/kpdf/xpdf/splash/SplashFTFont.cc	2005-08-31 20:04:30.000000000 +0200
@@ -156,7 +156,8 @@
   // if we have the FT2 bytecode interpreter, autohinting won't be used
 #ifdef TT_CONFIG_OPTION_BYTECODE_INTERPRETER
   if (FT_Load_Glyph(ff->face, gid,
-		    aa ? FT_LOAD_NO_BITMAP : FT_LOAD_DEFAULT)) {
+		    aa ? FT_LOAD_NO_HINTING | FT_LOAD_NO_BITMAP
+		       : FT_LOAD_DEFAULT)) {
     return gFalse;
   }
 #else
kdegraphics-3.4.2/kpdf/xpdf/splashだけに発見: SplashFTFont.cc.~1~
