--- a/ksvg/plugin/backends/agg/Makefile.am
+++ b/ksvg/plugin/backends/agg/Makefile.am	2004/12/07 12:27:36
@@ -1,8 +1,8 @@
 kde_module_LTLIBRARIES = libksvgrendereragg.la
 
 # Damn agg2 has no configuration querying possibility!
-AGG_CFLAGS = -I/cvs/agg2/include/
-AGG_LIBS = /cvs/agg2/src/libagg.a
+AGG_CFLAGS = -I/usr/include/agg2/
+AGG_LIBS = -lagg
 
 libksvgrendereragg_la_SOURCES = BezierPathAgg.cpp GlyphTracerAgg.cpp AggCanvas.cpp AggCanvasItems.cpp AggCanvasFactory.cpp
 libksvgrendereragg_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
--- a/ksvg/plugin/backends/Makefile.am
+++ b/ksvg/plugin/backends/Makefile.am	2004/12/07 12:26:13
@@ -1 +1 @@
-SUBDIRS = libart #agg
+SUBDIRS = libart agg
