diff -ur kdegraphics-3.5.5.orig/kolourpaint/kpmainwindow_view.cpp kdegraphics-3.5.5/kolourpaint/kpmainwindow_view.cpp
--- kdegraphics-3.5.5.orig/kolourpaint/kpmainwindow_view.cpp	2005-09-10 11:19:07.000000000 +0300
+++ kdegraphics-3.5.5/kolourpaint/kpmainwindow_view.cpp	2006-11-18 01:48:21.000000000 +0200
@@ -213,7 +213,7 @@
 // private
 QString kpMainWindow::zoomLevelToString (int zoomLevel)
 {
-    return i18n ("%1%").arg (zoomLevel);
+    return QString("%1%").arg(zoomLevel);
 }
 
 // private
