diff -ur basket-1.0.orig/src/basket_part.rc basket-1.0/src/basket_part.rc
--- basket-1.0.orig/src/basket_part.rc	2007-03-09 14:35:20.000000000 +0200
+++ basket-1.0/src/basket_part.rc	2007-03-09 14:39:45.000000000 +0200
@@ -105,7 +105,7 @@
    <Action name="help_whats_this" />
    <Action name="help_show_tip" />
    <Separator/>
-   <Action name="help_welcome_baskets" />
+<!--   <Action name="help_welcome_baskets" />-->
 <!--   <Action name="help_report_bug" />-->
    <Action name="likeback_send_a_comment" />
    <Separator/>
diff -ur basket-1.0.orig/src/basketui.rc basket-1.0/src/basketui.rc
--- basket-1.0.orig/src/basketui.rc	2007-03-09 14:35:20.000000000 +0200
+++ basket-1.0/src/basketui.rc	2007-03-09 14:39:55.000000000 +0200
@@ -108,7 +108,7 @@
    <Action name="help_whats_this" />
    <Action name="help_show_tip" />
    <Separator/>
-   <Action name="help_welcome_baskets" />
+<!--   <Action name="help_welcome_baskets" />-->
 <!--   <Action name="help_report_bug" />-->
    <Action name="likeback_send_a_comment" />
    <Separator/>
diff -ur basket-1.0.orig/src/bnpview.cpp basket-1.0/src/bnpview.cpp
--- basket-1.0.orig/src/bnpview.cpp	2007-03-09 14:35:20.000000000 +0200
+++ basket-1.0/src/bnpview.cpp	2007-03-09 14:41:31.000000000 +0200
@@ -191,6 +191,7 @@
 
 void BNPView::addWelcomeBaskets()
 {
+#if 0
 	// Possible paths where to find the welcome basket archive, trying the translated one, and falling back to the English one:
 	QStringList possiblePaths;
 	if (QString(KGlobal::locale()->encoding()) == QString("UTF-8")) { // Welcome baskets are encoded in UTF-8. If the system is not, then use the English version:
@@ -212,6 +213,7 @@
 	// Extract:
 	if (!path.isEmpty())
 		Archive::open(path);
+#endif
 }
 
 void BNPView::onFirstShow()
