Index: gst-plugins-bad-0.10.9/ext/x264/gstx264enc.c
===================================================================
--- gst-plugins-bad-0.10.9.orig/ext/x264/gstx264enc.c
+++ gst-plugins-bad-0.10.9/ext/x264/gstx264enc.c
@@ -390,9 +390,6 @@ gst_x264_enc_log_callback (gpointer priv
       gst_level = GST_LEVEL_LOG;
       break;
   }
-
-  gst_debug_log_valist (x264_enc_debug, gst_level, "", "", 0, object, format,
-      args);
 }
 
 /* initialize the new element
