diff -Nur gst-plugins-ugly-0.10.12-old/ext/x264/gstx264enc.c gst-plugins-ugly-0.10.12/ext/x264/gstx264enc.c
--- gst-plugins-ugly-0.10.12-old/ext/x264/gstx264enc.c	2009-10-20 20:05:29.967215196 +0300
+++ gst-plugins-ugly-0.10.12/ext/x264/gstx264enc.c	2009-10-20 20:05:54.774865249 +0300
@@ -573,7 +573,7 @@
   encoder->x264param.analyse.i_noise_reduction = encoder->noise_reduction;
   encoder->x264param.i_frame_reference = encoder->ref;
   encoder->x264param.i_bframe = encoder->bframes;
-  encoder->x264param.b_bframe_pyramid = encoder->b_pyramid;
+  encoder->x264param.i_bframe_pyramid = encoder->b_pyramid;
 #if X264_BUILD < 63
   encoder->x264param.b_bframe_adaptive = encoder->b_adapt;
 #else
