Index: clementine-0.6_2629/CMakeLists.txt
===================================================================
--- clementine-0.6_2629.orig/CMakeLists.txt
+++ clementine-0.6_2629/CMakeLists.txt
@@ -9,6 +9,7 @@ include(cmake/Rpm.cmake)
 include(cmake/SipBindings.cmake)
 
 find_package(Qt4 4.5.0 REQUIRED QtCore QtGui QtOpenGL QtSql QtNetwork QtXml)
+LINK_DIRECTORIES( "/usr/qt/4/lib" )
 if(UNIX AND NOT APPLE)
   option(ENABLE_DBUS "Enable D-Bus, MPRIS and native notifications. Required for DeviceKit and Wii remote support" ON)
   if(ENABLE_DBUS)
