Index: chromium-13.0.782.107/webkit/plugins/npapi/plugin_list_posix.cc
===================================================================
--- chromium-13.0.782.107.orig/webkit/plugins/npapi/plugin_list_posix.cc
+++ chromium-13.0.782.107/webkit/plugins/npapi/plugin_list_posix.cc
@@ -91,6 +91,7 @@ bool IsUndesirablePlugin(const WebPlugin
 bool IsBlacklistedPlugin(const FilePath& path) {
   const char* kBlackListedPlugins[] = {
     "nppdf.so",           // Adobe PDF
+    "gecko-mediaplayer",  // Gecko Media Player, still has some regression
   };
   std::string filename = path.BaseName().value();
   for (size_t i = 0; i < arraysize(kBlackListedPlugins); i++) {
