? kioslave/smb/kio_smb_la_all_cpp.cpp
Index: kioslave/smb/kio_smb_auth.cpp
===================================================================
RCS file: /home/kde/kdebase/kioslave/smb/kio_smb_auth.cpp,v
retrieving revision 1.29
diff -u -3 -p -r1.29 kio_smb_auth.cpp
--- kioslave/smb/kio_smb_auth.cpp	22 Nov 2004 14:01:23 -0000	1.29
+++ kioslave/smb/kio_smb_auth.cpp	4 Mar 2005 16:25:37 -0000
@@ -144,6 +144,7 @@ bool SMBSlave::checkPassword(SMBUrl &url
     if ( openPassDlg(info) ) {
         kdDebug(KIO_SMB) << "openPassDlg returned " << info.username << endl;
         url.setUser(info.username);
+        url.setPass(info.password);
         return true;
     }
     kdDebug(KIO_SMB) << "no value from openPassDlg\n";
