diff -ur kdebase-3.4.92.orig/kcontrol/fonts/fonts.cpp kdebase-3.4.92/kcontrol/fonts/fonts.cpp
--- kdebase-3.4.92.orig/kcontrol/fonts/fonts.cpp	2005-10-10 18:03:43.000000000 +0300
+++ kdebase-3.4.92/kcontrol/fonts/fonts.cpp	2005-10-24 01:20:53.000000000 +0300
@@ -508,11 +508,11 @@
 
   // Keep in sync with kdelibs/kdecore/kglobalsettings.cpp
 
-  QFont f0("Sans Serif", 10);
-  QFont f1("Monospace", 10);
-  QFont f2("Sans Serif", 10);
-  QFont f3("Sans Serif", 9, QFont::Bold);
-  QFont f4("Sans Serif", 10);
+  QFont f0("DejaVu Sans", 10);
+  QFont f1("DejaVu Sans Mono", 10);
+  QFont f2("DejaVu Sans", 10);
+  QFont f3("DejaVu Sans", 9, QFont::Bold);
+  QFont f4("DejaVu Sans", 10);
 
   f0.setPointSize(10);
   f1.setPointSize(10);
