diff -ur kdebase-3.4.92.orig/kwin/workspace.cpp kdebase-3.4.92/kwin/workspace.cpp
--- kdebase-3.4.92.orig/kwin/workspace.cpp	2005-10-10 18:04:22.000000000 +0300
+++ kdebase-3.4.92/kwin/workspace.cpp	2005-10-24 01:59:47.000000000 +0300
@@ -906,7 +906,7 @@
         groupname.sprintf("Desktops-screen-%d", screen_number);
     KConfigGroupSaver saver(c,groupname);
 
-    int n = c->readNumEntry("Number", 4);
+    int n = c->readNumEntry("Number", 2);
     number_of_desktops = n;
     delete workarea;
     workarea = new QRect[ n + 1 ];
