diff -ur kdebase-3.5.3.orig/kicker/kicker/core/containerarea.cpp kdebase-3.5.3/kicker/kicker/core/containerarea.cpp
--- kdebase-3.5.3.orig/kicker/kicker/core/containerarea.cpp	2006-07-09 01:16:23.000000000 +0300
+++ kdebase-3.5.3/kicker/kicker/core/containerarea.cpp	2006-07-09 01:18:28.000000000 +0300
@@ -641,10 +641,6 @@
                                           bool isImmutable,
                                           int insertionIndex)
 {
-    if (!canAddContainers())
-    {
-        return 0;
-    }
 
     AppletContainer* a = PluginManager::the()->createAppletContainer(
         info.desktopFile(),
