diff -ur kdebase-3.5.0.orig/ksplashml/wndmain.cpp kdebase-3.5.0/ksplashml/wndmain.cpp
--- kdebase-3.5.0.orig/ksplashml/wndmain.cpp	2005-11-17 15:43:55.000000000 +0200
+++ kdebase-3.5.0/ksplashml/wndmain.cpp	2005-11-17 15:42:01.000000000 +0200
@@ -95,7 +95,7 @@
   if (mTheme.isEmpty())
   {
     config->setGroup( "KSplash" );
-    mTheme = config->readEntry( "Theme", "Default" );
+    mTheme = config->readEntry( "Theme", "Pardus" );
   }
   loadTheme( mTheme ); // Guaranteed to return a valid theme.
 }
