diff -ur kdebase-3.4.92.orig/libkonq/konq_settings.cc kdebase-3.4.92/libkonq/konq_settings.cc
--- kdebase-3.4.92.orig/libkonq/konq_settings.cc	2005-10-10 18:04:21.000000000 +0300
+++ kdebase-3.4.92/libkonq/konq_settings.cc	2005-10-24 02:16:27.000000000 +0300
@@ -90,7 +90,7 @@
   
   d->m_iconTextWidth = config->readNumEntry( "TextWidth", DEFAULT_TEXTWIDTH );
   if ( d->m_iconTextWidth == DEFAULT_TEXTWIDTH )
-    d->m_iconTextWidth = QFontMetrics(m_standardFont).width("0000000000");
+    d->m_iconTextWidth = QFontMetrics(m_standardFont).width("0000000000000");
 
   m_iconTextHeight = config->readNumEntry( "TextHeight", 0 );
   if ( m_iconTextHeight == 0 ) {
