diff -ur kdebase-3.4.92.orig/kcontrol/kio/kcookiespolicies.cpp kdebase-3.4.92/kcontrol/kio/kcookiespolicies.cpp
--- kdebase-3.4.92.orig/kcontrol/kio/kcookiespolicies.cpp	2005-10-10 18:03:48.000000000 +0300
+++ kdebase-3.4.92/kcontrol/kio/kcookiespolicies.cpp	2005-10-24 01:32:46.000000000 +0300
@@ -291,7 +291,7 @@
   cookiesEnabled( enableCookies );
 
   KCookieAdvice::Value advice = KCookieAdvice::strToAdvice (cfg.readEntry(
-                                               "CookieGlobalAdvice", "Ask"));
+                                               "CookieGlobalAdvice", "Accept"));
   switch (advice)
   {
     case KCookieAdvice::Accept:
