diff -ur kdebase-3.5.7.orig/konqueror/konq_frame.cc kdebase-3.5.7/konqueror/konq_frame.cc
--- kdebase-3.5.7.orig/konqueror/konq_frame.cc	2006-05-22 21:12:59.000000000 +0300
+++ kdebase-3.5.7/konqueror/konq_frame.cc	2007-06-17 13:40:19.000000000 +0300
@@ -34,6 +34,7 @@
 #include <kprogress.h>
 #include <klocale.h>
 #include <ksqueezedtextlabel.h>
+#include <networkstatusindicator.h>
 
 #include "konq_events.h"
 #include "konq_frame.h"
@@ -97,6 +98,10 @@
     m_progressBar->hide();
     addWidget( m_progressBar, 0, true /*permanent->right align*/ );
 
+    StatusBarNetworkStatusIndicator * indicator = new StatusBarNetworkStatusIndicator( this, "networkstatusindicator" );
+    addWidget( indicator, 0, false );
+    indicator->init();
+
     fontChange(QFont());
     installEventFilter( this );
 }
diff -ur kdebase-3.5.7.orig/konqueror/Makefile.am kdebase-3.5.7/konqueror/Makefile.am
--- kdebase-3.5.7.orig/konqueror/Makefile.am	2005-10-10 18:04:20.000000000 +0300
+++ kdebase-3.5.7/konqueror/Makefile.am	2007-06-17 13:40:31.000000000 +0300
@@ -49,7 +49,7 @@
 	konq_view.h konq_viewmgr.h konq_extensionmanager.h version.h
 
 konqueror_la_LDFLAGS = -module $(KDE_PLUGIN) $(all_libraries)
-konqueror_la_LIBADD = ../libkonq/libkonq.la libkonqueror_intern.la $(LIBMALLOC) $(LIB_KUTILS)
+konqueror_la_LIBADD = ../libkonq/libkonq.la libkonqueror_intern.la $(LIBMALLOC) $(LIB_KUTILS) -lconnectionmanager
 
 # Hmm, this experiment of a static konq failed, don't trust it...
 # (but feel free to fix it :)
