diff -ur kdebase-3.5.2.orig/kcontrol/kicker/extensionInfo.cpp kdebase-3.5.2/kcontrol/kicker/extensionInfo.cpp
--- kdebase-3.5.2.orig/kcontrol/kicker/extensionInfo.cpp	2006-05-01 14:29:36.000000000 +0300
+++ kdebase-3.5.2/kcontrol/kicker/extensionInfo.cpp	2006-05-01 14:30:34.000000000 +0300
@@ -47,9 +47,9 @@
         _useStdSizes = true;
         _customSizeMin = 24;
         _customSizeMax = 256;
-        _customSize = 56;
+        _customSize = 32;
         _showLeftHB     = false;
-        _showRightHB    = true;
+        _showRightHB    = false;
 	for (int i=0;i<4;i++) _allowedPosition[i]=true;
     }
     else
@@ -201,7 +201,7 @@
     _xineramaScreen = QApplication::desktop()->primaryScreen();
     _size           = 2;
     _showLeftHB     = false;
-    _showRightHB    = true;
+    _showRightHB    = false;
     _hideButtonSize = 14;
     _autohidePanel  = false;
     _backgroundHide = false;
diff -ur kdebase-3.5.2.orig/kcontrol/kicker/positiontab_impl.cpp kdebase-3.5.2/kcontrol/kicker/positiontab_impl.cpp
--- kdebase-3.5.2.orig/kcontrol/kicker/positiontab_impl.cpp	2006-05-01 14:29:36.000000000 +0300
+++ kdebase-3.5.2/kcontrol/kicker/positiontab_impl.cpp	2006-05-01 14:30:34.000000000 +0300
@@ -204,7 +204,7 @@
         m_panelAlign = AlignLeft;
     }
 
-    m_panelSize->setCurrentItem(KPanelExtension::SizeNormal);
+    m_panelSize->setCurrentItem(KPanelExtension::SizeCustom);
 
     // update the magic drawing
     lengthenPanel(-1);
diff -ur kdebase-3.5.2.orig/kicker/applets/clock/clockapplet.kcfg kdebase-3.5.2/kicker/applets/clock/clockapplet.kcfg
--- kdebase-3.5.2.orig/kicker/applets/clock/clockapplet.kcfg	2006-05-01 14:29:36.000000000 +0300
+++ kdebase-3.5.2/kicker/applets/clock/clockapplet.kcfg	2006-05-01 14:30:34.000000000 +0300
@@ -14,7 +14,7 @@
         <choice name="Analog"/>
         <choice name="Fuzzy"/>
       </choices>
-      <default>Digital</default>
+      <default>Plain</default>
     </entry>
   </group>
   <group name="Date">
@@ -42,7 +42,7 @@
     </entry>
     <entry name="PlainShowDate" type="Bool" key="Show_Date">
       <label>Show date.</label>
-      <default>true</default>
+      <default>false</default>
     </entry>
     <entry name="PlainShowDayOfWeek" type="Bool" key="Show_DayOfWeek">
       <label>Show day of week.</label>
@@ -56,7 +56,7 @@
       <label>Font for the clock.</label>
       <code>
 defFont=KGlobalSettings::generalFont();
-defFont.setPointSize(8);
+defFont.setPointSize(14);
 defFont.setBold(true);
       </code>
       <default code="true">defFont</default>
diff -ur kdebase-3.5.2.orig/kicker/kicker/buttons/kbutton.cpp kdebase-3.5.2/kicker/kicker/buttons/kbutton.cpp
--- kdebase-3.5.2.orig/kicker/kicker/buttons/kbutton.cpp	2006-05-01 14:29:36.000000000 +0300
+++ kdebase-3.5.2/kicker/kicker/buttons/kbutton.cpp	2006-05-01 14:31:05.000000000 +0300
@@ -49,9 +49,15 @@
 
     if (KickerSettings::showKMenuText())
     {
-        setButtonText(KickerSettings::kMenuText());
-        setFont(KickerSettings::buttonFont());
-        setTextColor(KickerSettings::buttonTextColor());
+        setDrawArrow(false);
+
+        setButtonText("PARDUS");
+	QFont f = KickerSettings::buttonFont();
+	f.setBold(true);
+	f.setFamily("pardus");
+	setFont(f);
+	setTextColor(KickerSettings::buttonTextColor());
+	setFontPercent(0.55); // a little bigger font for KMenu
     }
 }
 
@@ -60,6 +66,11 @@
     MenuManager::the()->unregisterKButton(this);
 }
 
+int KButton::widthForHeight(int height) const
+{
+  return PanelButton::widthForHeight(height) + 10;
+}
+
 void KButton::properties()
 {
     KApplication::startServiceByDesktopName("kmenuedit", QStringList(),
Yalnızca kdebase-3.5.2/kicker/kicker/buttons'da: kbutton.cpp~
diff -ur kdebase-3.5.2.orig/kicker/kicker/buttons/kbutton.h kdebase-3.5.2/kicker/kicker/buttons/kbutton.h
--- kdebase-3.5.2.orig/kicker/kicker/buttons/kbutton.h	2006-05-01 14:29:36.000000000 +0300
+++ kdebase-3.5.2/kicker/kicker/buttons/kbutton.h	2006-05-01 14:30:34.000000000 +0300
@@ -41,6 +41,8 @@
 
     virtual void properties();
 
+    int widthForHeight(int height) const;
+
 protected:
     virtual QString tileName() { return "KMenu"; }
     virtual void initPopup();
diff -ur kdebase-3.5.2.orig/kicker/kicker/core/default-apps kdebase-3.5.2/kicker/kicker/core/default-apps
--- kdebase-3.5.2.orig/kicker/kicker/core/default-apps	2006-05-01 14:29:36.000000000 +0300
+++ kdebase-3.5.2/kicker/kicker/core/default-apps	2006-05-01 14:30:34.000000000 +0300
@@ -1,2 +1,3 @@
 Home.desktop
-Internet/konqbrowser.desktop
+Internet/Kontact.desktop
+Internet/mozillafirefox.desktop
diff -ur kdebase-3.5.2.orig/kicker/kicker/core/extensionSettings.kcfg kdebase-3.5.2/kicker/kicker/core/extensionSettings.kcfg
--- kdebase-3.5.2.orig/kicker/kicker/core/extensionSettings.kcfg	2006-05-01 14:29:36.000000000 +0300
+++ kdebase-3.5.2/kicker/kicker/core/extensionSettings.kcfg	2006-05-01 14:30:34.000000000 +0300
@@ -101,12 +101,12 @@
 
 <entry name="Size" type="Int">
       <label>Size</label>
-      <default code="true">KPanelExtension::SizeNormal</default>
+      <default code="true">KPanelExtension::SizeCustom</default>
    </entry>
 
 <entry name="CustomSize" type="Int" >
       <label>Custom size</label>
-      <default>58</default>
+      <default>32</default>
       <min>16</min>
    </entry>
 
diff -ur kdebase-3.5.2.orig/kicker/libkicker/kickerSettings.kcfg kdebase-3.5.2/kicker/libkicker/kickerSettings.kcfg
--- kdebase-3.5.2.orig/kicker/libkicker/kickerSettings.kcfg	2006-05-01 14:29:36.000000000 +0300
+++ kdebase-3.5.2/kicker/libkicker/kickerSettings.kcfg	2006-05-01 14:30:34.000000000 +0300
@@ -12,7 +12,7 @@
 
 <entry name="Locked" type="Bool" >
       <label>When this option is enabled, the panels may not be moved and items cannot be removed or added</label>
-      <default>false</default>
+      <default>true</default>
    </entry>
 
 <entry name="ConserveSpace" type="Bool" >
@@ -128,7 +128,7 @@
 
 <entry name="MenuEntryHeight" type="Int" >
       <label>Height of menu entries in pixels</label>
-      <default>0</default>
+      <default>22</default>
    </entry>
 
 <entry name="ShowHiddenFiles" type="Bool" >
@@ -154,7 +154,6 @@
 
 <entry name="MenuExtensions" key="Extensions" type="StringList" >
       <label>Optional Menus</label>
-      <default>prefmenu.desktop,systemmenu.desktop</default>
    </entry>
 
 <entry name="RecentAppsStat" type="StringList" >
@@ -163,13 +162,13 @@
 
 <entry name="NumVisibleEntries" type="UInt" >
       <label>Number of visible entries</label>
-      <default>5</default>
+      <default>3</default>
       <max>100</max>
    </entry>
 
 <entry name="RecentVsOften" type="Bool" >
       <label>Show most recently used applications rather than most frequently used</label>
-      <default>false</default>
+      <default>true</default>
    </entry>
 
 </group>
@@ -267,12 +266,12 @@
 
 <entry name="ShowKMenuText" key="ShowText" type="Bool" >
       <label>Show text on the K Menu button</label>
-      <default>false</default>
+      <default>true</default>
    </entry>
 
 <entry name="KMenuText" key="Text" type="String" >
       <label>Text to be shown on K Menu Button</label>
-      <default code="true">i18n("Applications")</default>
+      <default code="true">i18n("Pardus")</default>
    </entry>
 
 </group>
Index: kcontrol/kicker/hidingtab_impl.cpp
===================================================================
--- a/kcontrol/kicker/hidingtab_impl.cpp	(revision 583463)
+++ b/kcontrol/kicker/hidingtab_impl.cpp	(working copy)
@@ -200,7 +200,7 @@
    m_delaySpinBox->setValue( 3 );
    m_autoHideSwitch->setChecked( false );
    m_lHB->setChecked(  false );
-   m_rHB->setChecked(  true );
+   m_rHB->setChecked(  false );
    m_animateHiding->setChecked( true );
    m_hideSlider->setValue( 10 );
    m_delaySpinBox->setValue( 3 );
Index: kcontrol/kicker/menutab_impl.cpp
===================================================================
--- a/kcontrol/kicker/menutab_impl.cpp	(revision 583463)
+++ b/kcontrol/kicker/menutab_impl.cpp	(working copy)
@@ -123,7 +123,7 @@
         }
     }
 
-    m_showFrequent->setChecked(true);
+    m_showFrequent->setChecked(false);
 
     if ( useDefaults )
        emit changed();
