Index: ksplashml/kcmksplash/installer.cpp
===================================================================
--- ksplashml/kcmksplash/installer.cpp	(revision 583469)
+++ ksplashml/kcmksplash/installer.cpp	(working copy)
@@ -261,7 +261,7 @@
   KConfig cnf("ksplashrc");
   cnf.setReadDefaults( useDefaults );
   cnf.setGroup("KSplash");
-  QString curTheme = cnf.readEntry("Theme","Default");
+  QString curTheme = cnf.readEntry("Theme","Pardus");
   mThemesList->setCurrentItem(findTheme(curTheme));
   emit changed( useDefaults );
 }
