--- kdebase-3.5.0.orig/kdm/kfrontend/genkdmconf.c	2005-12-06 13:55:33.000000000 +0200
+++ kdebase-3.5.0/kdm/kfrontend/genkdmconf.c	2005-12-07 17:52:32.000000000 +0200
@@ -651,6 +651,9 @@
 "[ -f /etc/xprofile ] && . /etc/xprofile\n"
 "[ -f $HOME/.xprofile ] && . $HOME/.xprofile\n"
 "\n"
+"[ -f /etc/X11/Xmodmap ] && xmodmap /etc/X11/Xmodmap\n"
+"[ -f $HOME/.Xmodmap ] && xmodmap $HOME/.Xmodmap\n"
+"\n"
 "case $session in\n"
 "  \"\")\n"
 "    exec xmessage -center -buttons OK:0 -default OK \"Sorry, $DESKTOP_SESSION is no valid session.\"\n"
