Index: kdebase-3.5.10/kcontrol/fonts/kxftconfig.h
===================================================================
--- kdebase-3.5.10.orig/kcontrol/fonts/kxftconfig.h
+++ kdebase-3.5.10/kcontrol/fonts/kxftconfig.h
@@ -108,7 +108,7 @@ class KXftConfig
 #else
         SubPixel(Type t=None, char *s=NULL, char *e=NULL) : Item(s, e), type(t) {}
 #endif
-        void reset() { Item::reset(); type=None; }
+        void reset() { Item::reset(); type=Rgb; }
 
         Type type;
     };
@@ -142,7 +142,7 @@ class KXftConfig
         Hint(Style s, QDomNode &n) : Item(n), style(s) {}
         Hint(Style s=NotSet)       : style(s)          {}
 
-        void reset() { Item::reset(); style=NotSet; }
+        void reset() { Item::reset(); style=Full; }
 
         Style style;
     };
