Index: kdebase-3.5.10/kioslave/media/medianotifier/medianotifier.cpp
===================================================================
--- kdebase-3.5.10.orig/kioslave/media/medianotifier/medianotifier.cpp
+++ kdebase-3.5.10/kioslave/media/medianotifier/medianotifier.cpp
@@ -49,12 +49,14 @@ MediaNotifier::MediaNotifier(const QCStr
 	connectDCOPSignal( "kded", "mediamanager", "mediumChanged(QString, bool)",
 	                   "onMediumChange(QString, bool)", true );
 
-    m_freeTimer = new QTimer( this );
-    connect( m_freeTimer, SIGNAL( timeout() ), SLOT( checkFreeDiskSpace() ) );
+	m_freeDialog = 0;
+	if (!QFile::exists("/var/run/pardus/livemedia")) {
+		m_freeTimer = new QTimer( this );
+		connect( m_freeTimer, SIGNAL( timeout() ), SLOT( checkFreeDiskSpace() ) );
 
-    m_freeTimer->start(0, true);
-    m_freeTimer->start( 1000*60*3 /* 3 minutes */ );
-    m_freeDialog = 0;
+		m_freeTimer->start(0, true);
+		m_freeTimer->start( 1000*60*3 /* 3 minutes */ );
+	}
 }
 
 MediaNotifier::~MediaNotifier()
