diff -ur kdebase-3.5.5.orig/kioslave/media/mediamanager/halbackend.cpp kdebase-3.5.5/kioslave/media/mediamanager/halbackend.cpp
--- kdebase-3.5.5.orig/kioslave/media/mediamanager/halbackend.cpp	2006-11-08 17:09:06.000000000 +0200
+++ kdebase-3.5.5/kioslave/media/mediamanager/halbackend.cpp	2006-11-08 17:09:39.000000000 +0200
@@ -837,7 +837,7 @@
 
     if (valids.contains("shortname"))
     {
-        QString svalue = config.readEntry("shortname", "lower").lower();
+        QString svalue = config.readEntry("shortname", "mixed").lower();
         if (svalue == "winnt")
             result << "shortname=winnt";
         else if (svalue == "win95")
