--- kcontrol/kfontinst/viewpart/FontViewPart.cpp.orig	2008-11-11 14:05:38.000000000 +0200
+++ kcontrol/kfontinst/viewpart/FontViewPart.cpp	2008-11-11 14:08:42.000000000 +0200
@@ -198,7 +198,7 @@
     itsInstallButton->setShown(st && itsShowInstallButton);
     itsToolsFrame->setShown(itsInstallButton->isShown()||itsFaceSelector->isShown());
     itsChangeTextAction->setEnabled(st);
-    itsPrintAction->setEnabled(st && KFI_KIO_FONTS_PROTOCOL==m_url.protocol());
+    itsPrintAction->setEnabled(st || KFI_KIO_FONTS_PROTOCOL==m_url.protocol());
 }
 
 void CFontViewPart::install()
