Index: kdebase-3.5.10/ksmserver/shutdowndlg.cpp
===================================================================
--- kdebase-3.5.10.orig/ksmserver/shutdowndlg.cpp
+++ kdebase-3.5.10/ksmserver/shutdowndlg.cpp
@@ -168,16 +168,19 @@ KSMShutdownDlg::KSMShutdownDlg( QWidget*
     vbox = new QVBoxLayout( frame, 2 * KDialog::marginHint(),
                             2 * KDialog::spacingHint() );
 
+    /*
     QLabel* label = new QLabel( i18n("End Session for \"%1\"").arg(KUser().loginName()), frame );
     QFont fnt = label->font();
     fnt.setBold( true );
     fnt.setPointSize( fnt.pointSize() * 3 / 2 );
     label->setFont( fnt );
     vbox->addWidget( label, 0, AlignHCenter );
+    */
 
-    QHBoxLayout* hbox = new QHBoxLayout( vbox, 2 * KDialog::spacingHint() );
+    //QHBoxLayout* hbox = new QHBoxLayout( vbox, 2 * KDialog::spacingHint() );
 
     // konqy
+    /*
     QFrame* lfrm = new QFrame( frame );
     lfrm->setFrameStyle( QFrame::Panel | QFrame::Sunken );
     hbox->addWidget( lfrm, AlignCenter );
@@ -186,9 +189,10 @@ KSMShutdownDlg::KSMShutdownDlg( QWidget*
     icon->setPixmap( UserIcon( "shutdownkonq" ) );
     lfrm->setFixedSize( icon->sizeHint());
     icon->setFixedSize( icon->sizeHint());
+    */
 
     // right column (buttons)
-    QVBoxLayout* buttonlay = new QVBoxLayout( hbox, 2 * KDialog::spacingHint() );
+    QVBoxLayout* buttonlay = new QVBoxLayout( vbox, 2 * KDialog::spacingHint() );
     buttonlay->setAlignment( Qt::AlignHCenter );
 
     buttonlay->addStretch( 1 );
Index: kdebase-3.5.10/kdm/kfrontend/kdmshutdown.cpp
===================================================================
--- kdebase-3.5.10.orig/kdm/kfrontend/kdmshutdown.cpp
+++ kdebase-3.5.10/kdm/kfrontend/kdmshutdown.cpp
@@ -478,10 +478,12 @@ KDMSlimShutdown::KDMSlimShutdown( QWidge
 	QFrame *lfrm = new QFrame( this );
 	lfrm->setFrameStyle( QFrame::Panel | QFrame::Sunken );
 	hbox->addWidget( lfrm, AlignCenter );
+    /*
 	QLabel *icon = new QLabel( lfrm );
 	icon->setPixmap( QPixmap( locate( "data", "kdm/pics/shutdown.jpg" ) ) );
 	QVBoxLayout *iconlay = new QVBoxLayout( lfrm );
 	iconlay->addWidget( icon );
+    */
 
 	QVBoxLayout *buttonlay = new QVBoxLayout( hbox, KDsh );
 
