Index: kcontrol/screensaver/scrnsave.cpp
===================================================================
--- kcontrol/screensaver/scrnsave.cpp	(revision 500759)
+++ kcontrol/screensaver/scrnsave.cpp	(working copy)
@@ -100,13 +100,15 @@
 
     // Add KDE specific screensaver path
     QString relPath="System/ScreenSavers/";
+
     KServiceGroup::Ptr servGroup = KServiceGroup::baseGroup( "screensavers" );
+#if 0
     if (servGroup)
     {
       relPath=servGroup->relPath();
       kdDebug() << "relPath=" << relPath << endl;
     }
-
+#endif
     KGlobal::dirs()->addResourceType("scrsav",
                                      KGlobal::dirs()->kde_default("apps") +
                                      relPath);
