--- startkde	2008-01-13 16:53:31.000000000 +0200
+++ startkde	2008-01-13 16:54:38.000000000 +0200
@@ -319,6 +319,11 @@
 
 echo 'startkde: Starting up...'  1>&2
 
+# Launch D-Bus session bus
+if test -z "$DBUS_SESSION_BUS_ADDRESS" ; then
+    eval `dbus-launch --sh-syntax --exit-with-session`
+fi
+
 # run KPersonalizer before the session, if this is the first login
 if test "$kpersonalizerrc_general_firstlogin" = "true"; then
     # start only dcopserver, don't start whole kdeinit (takes too long)
