diff -ur kdebase-3.5.5.orig/kdesktop/init/System.desktop kdebase-3.5.5/kdesktop/init/System.desktop
--- kdebase-3.5.5.orig/kdesktop/init/System.desktop	2006-11-25 16:00:38.000000000 +0200
+++ kdebase-3.5.5/kdesktop/init/System.desktop	2006-10-01 20:32:02.000000000 +0300
@@ -129,6 +129,6 @@
 GenericName[vi]=Đường dẫn Hệ thống
 GenericName[zh_CN]=系统定位
 GenericName[zh_TW]=系統位置
-URL=system:/
+URL=sysinfo:/
 Icon=system
 Type=Link
