Index: kdebase-3.5.10/kioslave/home/home.protocol
===================================================================
--- kdebase-3.5.10.orig/kioslave/home/home.protocol
+++ kdebase-3.5.10/kioslave/home/home.protocol
@@ -10,7 +10,7 @@ makedir=true
 deleting=true
 linking=true
 moving=true
-Icon=folder_home
+Icon=folder_home2
 maxInstances=4
 #TODO DocPath=kioslave/file.html
 Class=:local
Index: kdebase-3.5.10/kioslave/system/entries/home.desktop
===================================================================
--- kdebase-3.5.10.orig/kioslave/system/entries/home.desktop
+++ kdebase-3.5.10/kioslave/system/entries/home.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Link
 Path=$HOME
-Icon=folder_home
+Icon=folder_home2
 Name=Home Folder
 Name[af]=Tuis Gids
 Name[ar]=منزلي
@@ -67,7 +67,7 @@ Name[ta]=வீட்டு அடைவ
 Name[te]=ఇంటి ఫొల్డర్
 Name[tg]=Феҳристи хонагӣ
 Name[th]=โฟลเดอร์ส่วนตัว
-Name[tr]=Başlangıç Dizini
+Name[tr]=Ev Dizini
 Name[tt]=Ana Törgäk
 Name[uk]=Домашня тека
 Name[uz]=Uy jildi
Index: kdebase-3.5.10/kioslave/system/entries/users.desktop
===================================================================
--- kdebase-3.5.10.orig/kioslave/system/entries/users.desktop
+++ kdebase-3.5.10/kioslave/system/entries/users.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Link
 URL=home:/
-Icon=folder_home2
+Icon=folder_home
 Name=Users Folders
 Name[af]=Gebruiker Gidse
 Name[ar]=مجلّدات المستخدمين
Index: kdebase-3.5.10/konqueror/Home.desktop
===================================================================
--- kdebase-3.5.10.orig/konqueror/Home.desktop
+++ kdebase-3.5.10/konqueror/Home.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Exec=kfmclient openProfile filemanagement
-Icon=folder_home
+Icon=folder_home2
 Terminal=false
 
 Name=Home
@@ -72,7 +72,7 @@ Name[ta]=தொடக்கம்
 Name[te]=ఇల్లు
 Name[tg]=Компютери Ман
 Name[th]=พื้นที่ส่วนตัว
-Name[tr]=Başlangıç
+Name[tr]=Ev Dizini
 Name[tt]=Anabit
 Name[uk]=Домівка
 Name[uz]=Uy
Index: kdebase-3.5.10/konqueror/sidebar/trees/init/home.desktop
===================================================================
--- kdebase-3.5.10.orig/konqueror/sidebar/trees/init/home.desktop
+++ kdebase-3.5.10/konqueror/sidebar/trees/init/home.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Link
 URL=file:$HOME
-Icon=folder_home
+Icon=folder_home2
 Name=Home Folder
 Name[af]=Tuis Gids
 Name[ar]=منزلي
@@ -67,7 +67,7 @@ Name[ta]=வீட்டு அடைவ
 Name[te]=ఇంటి ఫొల్డర్
 Name[tg]=Феҳристи хонагӣ
 Name[th]=โฟลเดอร์ส่วนตัว
-Name[tr]=Başlangıç Dizini
+Name[tr]=Ev Dizini
 Name[tt]=Ana Törgäk
 Name[uk]=Домашня тека
 Name[uz]=Uy jildi
Index: kdebase-3.5.10/kdesktop/init/Home.desktop
===================================================================
--- kdebase-3.5.10.orig/kdesktop/init/Home.desktop
+++ kdebase-3.5.10/kdesktop/init/Home.desktop
@@ -67,7 +67,7 @@ Name[ta]=தொடக்கம்
 Name[te]=ఇల్లు
 Name[tg]=Компютери Ман
 Name[th]=พื้นที่ส่วนตัว
-Name[tr]=Başlangıç
+Name[tr]=Ev Dizini
 Name[tt]=Anabit
 Name[uk]=Домівка
 Name[uz]=Uy
Index: kdebase-3.5.10/khotkeys/data/konqueror_gestures_kde321.khotkeys
===================================================================
--- kdebase-3.5.10.orig/khotkeys/data/konqueror_gestures_kde321.khotkeys
+++ kdebase-3.5.10/khotkeys/data/konqueror_gestures_kde321.khotkeys
@@ -1417,7 +1417,7 @@ Name[ta]=தொடக்கம்
 Name[te]=ఇల్లు
 Name[tg]=Компютери Ман
 Name[th]=พื้นที่ส่วนตัว
-Name[tr]=Başlangıç
+Name[tr]=Ev Dizini
 Name[tt]=Anabit
 Name[uk]=Домівка
 Name[uz]=Uy
