Index: kdebase-3.5.10/startkde
===================================================================
--- kdebase-3.5.10.orig/startkde
+++ kdebase-3.5.10/startkde
@@ -14,6 +14,15 @@ if kcheckrunning >/dev/null 2>&1; then
 	exit 1
 fi
 
+# We need at least 6 MB in HOME to startup
+homefreespace=`df -P ${HOME}/|tail -n 1|awk '{print $4}'`
+if [ "$homefreespace" -lt "10000" ]; then
+    echo "There is not enough space in $HOME"
+    xmessage -buttons OK -geometry 500x100 -center -timeout 300 "There is not enough free space in your home directory ($HOME).
+Please clean up the disk partition containing your home directory." > /dev/null 2>/dev/null
+    exit 1
+fi
+
 export PATH="/usr/kde/3.5/bin:${PATH}"
 
 cd ${HOME}
