--- Makefile.am	2007-03-03 09:12:37.000000000 +0200
+++ Makefile.am.new	2007-04-02 14:48:43.000000000 +0300
@@ -90,14 +90,3 @@
 
 endif
 
-install-data-local:
-	@$(NORMAL_INSTALL)
-	@if test ! -f $(DESTDIR)$(NSS_LDAP_PATH_CONF); then \
-		$(mkinstalldirs) $(DESTDIR)$(dir $(NSS_LDAP_PATH_CONF)); \
-		$(INSTALL_DATA) -o $(INST_UID) -g $(INST_GID) $(srcdir)/ldap.conf $(DESTDIR)$(NSS_LDAP_PATH_CONF); \
-	fi
-	$(INSTALL_DATA) -o $(INST_UID) -g $(INST_GID) $(srcdir)/nsswitch.ldap $(DESTDIR)$(sysconfdir)/nsswitch.ldap; 
-
-uninstall-local:
-	@$(NORMAL_UNINSTALL)
-
