diff -Naur nss_ldap-264-orig/Makefile.am nss_ldap-264/Makefile.am
--- nss_ldap-264-orig/Makefile.am	2009-10-13 15:01:30.806613155 +0300
+++ nss_ldap-264/Makefile.am	2009-10-13 15:02:32.733864650 +0300
@@ -65,7 +65,7 @@
 
 # AIX install instructions per doc/README.AIX
 
-install-exec-local: nss_ldap.so NSS_LDAP
+install-exec-local: NSS_LDAP
 	$(mkinstalldirs) $(DESTDIR)$(libdir)/netsvc/dynload
 	$(INSTALL_PROGRAM) -o $(INST_UID) -g $(INST_GID) nss_ldap.so $(DESTDIR)$(libdir)/netsvc/dynload/nss_ldap.so
 	$(mkinstalldirs) $(DESTDIR)$(libdir)/security
@@ -75,7 +75,7 @@
 
 # Linux, Solaris, other platform install instructions
 
-install-exec-local: nss_ldap.so
+install-exec-local: 
 	@$(NORMAL_INSTALL)
 if GLIBC
 	-rm -f $(DESTDIR)$(libdir)/$(NSS_LDAP_LIBC_VERSIONED)
