Index: v4l2capture-1.2/v4l2capture.c
===================================================================
--- v4l2capture-1.2.orig/v4l2capture.c
+++ v4l2capture-1.2/v4l2capture.c
@@ -142,13 +142,29 @@ static PyObject *Video_device_get_info(V
 {
   ASSERT_OPEN;
   struct v4l2_capability caps;
+  PyObject *list;
+
+  list = PyList_New(0);
 
   if(my_ioctl(self->fd, VIDIOC_QUERYCAP, &caps))
     {
       return NULL;
     }
 
-  return Py_BuildValue("sss", caps.driver, caps.card, caps.bus_info);
+  if ((caps.capabilities & V4L2_CAP_VIDEO_CAPTURE) > 0)
+    PyList_Append(list, PyString_FromString("capture"));
+  if ((caps.capabilities & V4L2_CAP_VIDEO_OUTPUT) > 0)
+    PyList_Append(list, PyString_FromString("video_output"));
+  if ((caps.capabilities & V4L2_CAP_VIDEO_OVERLAY) > 0)
+    PyList_Append(list, PyString_FromString("video_overlay"));
+  if ((caps.capabilities & V4L2_CAP_AUDIO) > 0)
+    PyList_Append(list, PyString_FromString("audio"));
+  if ((caps.capabilities & V4L2_CAP_TUNER) > 0)
+    PyList_Append(list, PyString_FromString("tuner"));
+  if ((caps.capabilities & V4L2_CAP_RADIO) > 0)
+    PyList_Append(list, PyString_FromString("radio"));
+
+  return Py_BuildValue("sssO", caps.driver, caps.card, caps.bus_info, list);
 }
 
 static PyObject *Video_device_set_format(Video_device *self, PyObject *args)
Index: v4l2capture-1.2/list_devices.py
===================================================================
--- v4l2capture-1.2.orig/list_devices.py
+++ v4l2capture-1.2/list_devices.py
@@ -19,7 +19,7 @@ for file_name in file_names:
     print path
     try:
         video = v4l2capture.Video_device(path)
-        print "    driver:    %s\n    card:      %s\n    bus info:  %s" \
+        print "    driver:    %s\n    card:      %s\n    bus info:  %s    capabilities:  %s\n" \
             % video.get_info()
         video.close()
     except IOError, e:
