https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=218379


Index: gdb-6.8.50.20081128/gdb/minsyms.c
===================================================================
--- gdb-6.8.50.20081128.orig/gdb/minsyms.c	2008-10-01 18:56:52.000000000 +0200
+++ gdb-6.8.50.20081128/gdb/minsyms.c	2008-12-02 23:24:27.000000000 +0100
@@ -544,6 +544,11 @@ lookup_minimal_symbol_by_pc_section_1 (C
 			 don't fill the bfd_section member, so don't
 			 throw away symbols on those platforms.  */
 		      && SYMBOL_OBJ_SECTION (&msymbol[hi]) != NULL
+		      /* Don't ignore symbols for solib trampolines.
+			 Limit its sideeffects - only for non-0 sized trampolines.
+			 Red Hat Bug 200533 with its regression Bug 218379.  */
+		      && (MSYMBOL_TYPE (&msymbol[hi]) != mst_solib_trampoline
+		          || MSYMBOL_SIZE (&msymbol[hi]))
 		      && (!matching_obj_sections
 			  (SYMBOL_OBJ_SECTION (&msymbol[hi]), section)))
 		    {
