diff -Naur cvs-1.11.23.org/lib/getline.c cvs-1.11.23/lib/getline.c
--- cvs-1.11.23.org/lib/getline.c	2009-12-24 15:48:33.000000000 +0000
+++ cvs-1.11.23/lib/getline.c	2009-12-24 15:50:26.000000000 +0000
@@ -155,7 +155,7 @@
 }
 
 int
-getline (lineptr, n, stream)
+my_getline (lineptr, n, stream)
      char **lineptr;
      size_t *n;
      FILE *stream;
diff -Naur cvs-1.11.23.org/lib/getline.h cvs-1.11.23/lib/getline.h
--- cvs-1.11.23.org/lib/getline.h	2009-12-24 15:48:33.000000000 +0000
+++ cvs-1.11.23/lib/getline.h	2009-12-24 15:50:13.000000000 +0000
@@ -12,7 +12,7 @@
 #define GETLINE_NO_LIMIT -1
 
 int
-  getline __PROTO ((char **_lineptr, size_t *_n, FILE *_stream));
+  my_getline __PROTO ((char **_lineptr, size_t *_n, FILE *_stream));
 int
   getline_safe __PROTO ((char **_lineptr, size_t *_n, FILE *_stream,
                          int limit));
