diff -Nur wxGTK-2.8.7-old/configure.in wxGTK-2.8.7/configure.in
--- wxGTK-2.8.7-old/configure.in	2008-04-26 02:01:48.000000000 +0300
+++ wxGTK-2.8.7/configure.in	2008-04-26 02:03:18.000000000 +0300
@@ -5259,15 +5259,7 @@
 dnl Checks specific to contrib libs (MMedia)
 dnl ------------------------------------------
 
-dnl Check for ESD:
 EXTRALIBS_ESD=
-AC_CHECK_LIB(esd, esd_close, [
-    AC_CHECK_HEADERS([esd.h], [
-    EXTRALIBS_ESD="-lesd"
-  ],
-  [],
-  [ ])
-])
 AC_SUBST(EXTRALIBS_ESD)
 
 dnl check for known CD-ROM interface
