diff -Nur gettext-0.17-old/gettext-tools/gnulib-m4/openmp.m4 gettext-0.17/gettext-tools/gnulib-m4/openmp.m4
--- gettext-0.17-old/gettext-tools/gnulib-m4/openmp.m4	2010-10-06 10:24:42.772211877 +0300
+++ gettext-0.17/gettext-tools/gnulib-m4/openmp.m4	2010-10-06 10:25:58.745150447 +0300
@@ -6,6 +6,8 @@
 
 dnl This file can be removed once we assume autoconf >= 2.62.
 
+m4_version_prereq([2.62],, [
+
 # _AC_LANG_OPENMP
 # ---------------
 # Expands to some language dependent source code for testing the presence of
@@ -90,3 +92,6 @@
   fi
   AC_SUBST([OPENMP_]_AC_LANG_PREFIX[FLAGS])
 ])
+
+])
+
