diff -Nur sysinfo-1.5.orig/admin/cvs.sh sysinfo-1.5/admin/cvs.sh
--- sysinfo-1.5.orig/admin/cvs.sh	2007-02-27 19:21:51.000000000 +0000
+++ sysinfo-1.5/admin/cvs.sh	2009-11-15 17:01:43.801083918 +0000
@@ -68,7 +68,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10*)
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10* | automake*1.11*)
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
