Index: sysinfo/sysinfo.cpp
===================================================================
--- sysinfo/sysinfo.cpp (revision 18426)
+++ sysinfo/sysinfo.cpp (working copy)
@@ -268,13 +268,13 @@
     {
         const int mem_unit = info.mem_unit;
         unsigned long int usage,percent,peer;
-        usage = info.totalram - info.freeram * mem_unit;
+        usage = ( info.totalram - info.freeram ) * mem_unit;
         peer = info.totalram / 100;
         peer == 0 ? percent = 0 : percent = usage / peer;

         m_info[MEM_TOTALRAM] = formattedUnit( info.totalram * mem_unit ,0);
         m_info[MEM_FREERAM] = formattedUnit( info.freeram * mem_unit );
-        m_info[MEM_USAGE] = formattedUnit( (info.totalram - info.freeram) * mem_unit);
+        m_info[MEM_USAGE] = formattedUnit( usage );
         m_info[MEM_TOTALSWAP] = formattedUnit( info.totalswap * mem_unit );
         m_info[MEM_FREESWAP] = formattedUnit( info.freeswap * mem_unit );

