diff -Nur gnome-mplayer-0.9.8-old/src/main.c gnome-mplayer-0.9.8/src/main.c
--- gnome-mplayer-0.9.8-old/src/main.c	2009-09-18 19:43:56.380514107 +0300
+++ gnome-mplayer-0.9.8/src/main.c	2009-09-18 19:44:04.239494698 +0300
@@ -844,6 +844,8 @@
         mplayer_bin = NULL;
     }
     extraopts = gm_pref_store_get_string(gm_store, EXTRAOPTS);
+    if (extraopts == NULL)
+        extraopts = "-nomsgcolor -nomsgmodule";
     use_pulse_flat_volume = gm_pref_store_get_boolean(gm_store, USE_PULSE_FLAT_VOLUME);
 
     remember_loc = gm_pref_store_get_boolean(gm_store, REMEMBER_LOC);
