diff -Nur gnome-mplayer-0.9.8-old/src/gui.c gnome-mplayer-0.9.8/src/gui.c
--- gnome-mplayer-0.9.8-old/src/gui.c	2009-09-18 21:10:19.828742894 +0300
+++ gnome-mplayer-0.9.8/src/gui.c	2009-09-18 21:11:00.655493254 +0300
@@ -1259,7 +1259,7 @@
     IdleData *idle = (IdleData *) data;
     gchar *buf = NULL;
 
-    if (GTK_IS_WIDGET(vol_slider) && idle->volume >= 0 && idle->volume <= 100) {
+    if (GTK_IS_WIDGET(vol_slider) && idle->volume > 0) {
         //printf("setting slider to %f\n", idle->volume);
 #ifdef GTK2_12_ENABLED
         if (rpcontrols != NULL && g_strcasecmp(rpcontrols, "volumeslider") == 0) {
@@ -1268,8 +1268,6 @@
             g_strlcpy(idledata->vol_tooltip, buf, 128);
             gtk_tooltips_set_tip(volume_tip, vol_slider, idle->vol_tooltip, NULL);
             g_free(buf);
-        } else {
-            gtk_scale_button_set_value(GTK_SCALE_BUTTON(vol_slider), idle->volume);
         }
 
 #else
