diff -Nuar src/kredentials.cpp src/kredentials.cpp
--- src/kredentials.cpp	2011-06-17 13:33:02.012000159 +0300
+++ src/kredentials.cpp	2011-06-17 13:43:21.244000148 +0300
@@ -214,7 +214,7 @@
     }
     else if(tktRenewableExpirationTime < now)
     {
-	KMessageBox::information(0, "Your tickets have outlived their renewable lifetime and can't be renewed.", 0, 0);
+	KMessageBox::information(0, i18n("Your tickets have outlived their renewable lifetime and can't be renewed."), 0, 0);
 	LOG << "tktRenewableExpirationTime has passed: ";
 	LOG << "tktRenewableExpirationTime = " << 
 	    tktRenewableExpirationTime << ", now = " << now << endl;
@@ -232,7 +232,7 @@
     else
     {
 	if(doNotify){
-	    KPassivePopup::message("Kerberos tickets have been renewed", 0);
+	    KPassivePopup::message(i18n("Kerberos tickets have been renewed"), 0);
 	}
     }
     // restart the timer here, regardless of whether we currently
@@ -242,7 +242,7 @@
     startTimer(1000);
     if(authenticated > 0){
 	if( !runAklog() ){
-	    KMessageBox::sorry(0, "Unable to run aklog", 0, 0);
+	    KMessageBox::sorry(0, i18n("Unable to run aklog"), 0, 0);
 	}
 		
 	LOG << "WarnTime: " << renewWarningTime << " " << 
@@ -257,9 +257,9 @@
 		LOG << "RESET: Renew=" << renewWarningFlag << endl;
 
 		QString msgString = 
-		    QString("Kerberos tickets will permanently expire on ")
+		    QString(i18n("Kerberos tickets will permanently expire on "))
 		    +  QString(ctime(&tktRenewableExpirationTime)) +
-		    QString(" You may want to renew them now.");
+		    QString(i18n(" You may want to renew them now."));
 		KMessageBox::information(0, msgString, 0, 0);
 	    }
 
@@ -298,7 +298,7 @@
     if(!authenticated)
     {
 	KMessageBox::information(0, 
-				 "You do not have any valid tickets.", 
+				 i18n("You do not have any valid tickets."), 
 				 "Kerberos", 0, 0);
     }
     else
@@ -307,22 +307,22 @@
 	if(pme){
 	    const krb5::principal& me=*pme;
 	    if(me.getDataLength()){
-		msgString = QString("Your tickets as ")
+		msgString = QString(i18n("Your tickets as "))
 		    +QString(me.getName())+QString(" ");
 	    }else{
-		msgString = QString("Your tickets ");
+		msgString = QString(i18n("Your tickets "));
 	    }
 	}
-	msgString+=QString("are\n Valid until ") + 
+	msgString+=QString(i18n("are\n Valid until ")) + 
 	    QString(ctime(&tktExpirationTime));
 	if(tktRenewableExpirationTime > time(0))
 	{
-	    msgString += QString("\nRenewable until ") + 
+	    msgString += QString(i18n("\nRenewable until ")) + 
 		QString(ctime(&tktRenewableExpirationTime));
 	}
 	else
 	{
-	    msgString += QString("\nTickets are not renewable");
+	    msgString += QString(i18n("\nTickets are not renewable"));
 	}
 	KMessageBox::information(0, msgString, "Kerberos", 0, 0);
     }
