diff -up cups-1.3.8/scheduler/auth.c.str2101 cups-1.3.8/scheduler/auth.c
--- cups-1.3.8/scheduler/auth.c.str2101	2008-08-11 15:12:12.000000000 +0100
+++ cups-1.3.8/scheduler/auth.c	2008-08-11 15:12:37.000000000 +0100
@@ -2098,7 +2098,7 @@ cupsdIsAuthorized(cupsd_client_t *con,	/
 	  return (HTTP_OK);
       }
 
-      return (HTTP_UNAUTHORIZED);
+      return (HTTP_FORBIDDEN);
     }
 #endif /* HAVE_AUTHORIZATION_H */
 
@@ -2122,7 +2122,7 @@ cupsdIsAuthorized(cupsd_client_t *con,	/
         return (HTTP_OK);
     }
 
-    return (HTTP_UNAUTHORIZED);
+    return (HTTP_FORBIDDEN);
   }
 
  /*
@@ -2159,7 +2159,7 @@ cupsdIsAuthorized(cupsd_client_t *con,	/
   cupsdLogMessage(CUPSD_LOG_DEBUG,
                   "cupsdIsAuthorized: User not in group(s)!");
 
-  return (HTTP_UNAUTHORIZED);
+  return (HTTP_FORBIDDEN);
 }
 
 
