diff -Nur nut-2.2.1-pre1-old/m4/nut_check_libgd.m4 nut-2.2.1-pre1/m4/nut_check_libgd.m4
--- nut-2.2.1-pre1-old/m4/nut_check_libgd.m4	2007-10-19 10:30:31.000000000 +0300
+++ nut-2.2.1-pre1/m4/nut_check_libgd.m4	2007-10-19 13:22:31.000000000 +0300
@@ -19,7 +19,7 @@
    dnl arguments
 
    CFLAGS=""
-   LDFLAGS="-L/usr/X11R6/lib -lgd -lpng -lz -ljpeg -lfreetype -lm -lXpm -lX11"
+   LDFLAGS="-lgd -lpng -lz -ljpeg -lfreetype -lm"
 
    GD_VERSION=`gdlib-config --version 2>/dev/null`
    if (test "$?" != "0")
