Index: pycairo-1.4.12/cairo/Makefile.in
===================================================================
--- pycairo-1.4.12.orig/cairo/Makefile.in
+++ pycairo-1.4.12/cairo/Makefile.in
@@ -390,13 +390,6 @@ install-pycairoexecPYTHON: $(pycairoexec
 	    $(pycairoexecPYTHON_INSTALL) "$$b$$p" "$(DESTDIR)$(pycairoexecdir)/$$f"; \
 	  else :; fi; \
 	done; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pycairoexecdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pycairoexecdir)" $$dlist; \
-	  fi; \
-	else :; fi
 
 uninstall-pycairoexecPYTHON:
 	@$(NORMAL_UNINSTALL)
