From 4be7acd98f49b3b1bb12c9fa5dd52b7f9ffd90fb Mon Sep 17 00:00:00 2001
From: Dario Freddi <drf@kde.org>
Date: Mon, 28 Sep 2009 15:58:12 +0200
Subject: [PATCH 18/19] Fix QNativeImage constructor

Signed-off-by: Dario Freddi <drf@kde.org>
---
 src/gui/image/qnativeimage.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/gui/image/qnativeimage.cpp b/src/gui/image/qnativeimage.cpp
index a0c86db..c1bc794 100644
--- a/src/gui/image/qnativeimage.cpp
+++ b/src/gui/image/qnativeimage.cpp
@@ -144,7 +144,7 @@ QImage::Format QNativeImage::systemFormat()
 #elif defined(Q_WS_X11) && !defined(QT_NO_MITSHM)
 
 QNativeImage::QNativeImage(int width, int height, QImage::Format format,bool /* isTextBuffer */, QWidget *widget)
-    : xshmimg(0), xshmpm(0)
+    : xshmimg(0)
 {
     if (!X11->use_mitshm) {
         image = QImage(width, height, format);
-- 
1.6.5.2

