diff -Nur kdetv-0.8.9-old/configure.in kdetv-0.8.9/configure.in
--- kdetv-0.8.9-old/configure.in	2007-04-27 18:43:46.573556185 +0300
+++ kdetv-0.8.9/configure.in	2007-04-27 18:44:00.277472874 +0300
@@ -477,7 +477,6 @@
 AC_CONFIG_FILES([ kdetv/plugins/filter/chromakill/Makefile ])
 AC_CONFIG_FILES([ kdetv/plugins/filter/colourinversion/Makefile ])
 AC_CONFIG_FILES([ kdetv/plugins/filter/greedy/Makefile ])
-AC_CONFIG_FILES([ kdetv/plugins/filter/greedy2frame/Makefile ])
 AC_CONFIG_FILES([ kdetv/plugins/filter/greedyh/Makefile ])
 AC_CONFIG_FILES([ kdetv/plugins/filter/linearblend/Makefile ])
 AC_CONFIG_FILES([ kdetv/plugins/filter/mirror/Makefile ])
diff -Nur kdetv-0.8.9-old/kdetv/plugins/filter/Makefile.am kdetv-0.8.9/kdetv/plugins/filter/Makefile.am
--- kdetv-0.8.9-old/kdetv/plugins/filter/Makefile.am	2007-04-27 18:43:46.581554969 +0300
+++ kdetv-0.8.9/kdetv/plugins/filter/Makefile.am	2007-04-27 18:44:47.838242537 +0300
@@ -1,5 +1,5 @@
 if ARCH_X86
-FILTER_ARCHSPECIFIC_PLUGINS = greedy greedyh greedy2frame tomsmocomp chromakill mirror sharpness
+FILTER_ARCHSPECIFIC_PLUGINS = greedy greedyh tomsmocomp chromakill mirror sharpness
 else
 FILTER_ARCHSPECIFIC_PLUGINS =
 endif
diff -Nur kdetv-0.8.9-old/kdetv/plugins/filter/Makefile.in kdetv-0.8.9/kdetv/plugins/filter/Makefile.in
--- kdetv-0.8.9-old/kdetv/plugins/filter/Makefile.in	2007-04-27 18:43:46.581554969 +0300
+++ kdetv-0.8.9/kdetv/plugins/filter/Makefile.in	2007-04-27 18:44:47.840242234 +0300
@@ -65,7 +65,7 @@
 ETAGS = etags
 CTAGS = ctags
 DIST_SUBDIRS = linearblend bilinear overscan colourinversion none \
-	greedy greedyh greedy2frame tomsmocomp chromakill mirror \
+	greedy greedyh tomsmocomp chromakill mirror \
 	sharpness
 #>- DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 #>+ 1
@@ -340,7 +340,7 @@
 xdg_directorydir = @xdg_directorydir@
 xdg_menudir = @xdg_menudir@
 @ARCH_X86_FALSE@FILTER_ARCHSPECIFIC_PLUGINS = 
-@ARCH_X86_TRUE@FILTER_ARCHSPECIFIC_PLUGINS = greedy greedyh greedy2frame tomsmocomp chromakill mirror sharpness
+@ARCH_X86_TRUE@FILTER_ARCHSPECIFIC_PLUGINS = greedy greedyh tomsmocomp chromakill mirror sharpness
 SUBDIRS = linearblend bilinear overscan colourinversion none $(FILTER_ARCHSPECIFIC_PLUGINS)
 #>- all: all-recursive
 #>+ 1
