--- plug-ins/pygimp/Makefile.in.orig	2008-09-05 09:04:06.000000000 +0300
+++ plug-ins/pygimp/Makefile.in	2008-09-05 09:04:34.000000000 +0300
@@ -710,13 +710,6 @@
 	    $(pygimpPYTHON_INSTALL) "$$b$$p" "$(DESTDIR)$(pygimpdir)/$$f"; \
 	  else :; fi; \
 	done; \
-	if test -n "$$dlist"; then \
-	  if test -z "$(DESTDIR)"; then \
-	    PYTHON=$(PYTHON) $(py_compile) --basedir "$(pygimpdir)" $$dlist; \
-	  else \
-	    PYTHON=$(PYTHON) $(py_compile) --destdir "$(DESTDIR)" --basedir "$(pygimpdir)" $$dlist; \
-	  fi; \
-	else :; fi
 
 uninstall-pygimpPYTHON:
 	@$(NORMAL_UNINSTALL)
