--- 981-move-freezer-h-include-for-userspace.patch-old/include/linux/fs.h   2009-07-02 09:27:13.000000000 +1000
+++ 981-move-freezer-h-include-for-userspace.patch-new/include/linux/fs.h   2009-07-02 09:26:24.000000000 +1000
@@ -8,7 +8,6 @@
 
 #include <linux/limits.h>
 #include <linux/ioctl.h>
-#include <linux/freezer.h>
 
 /*
  * It's silly to have NR_OPEN bigger than NR_FILE, but you can change
@@ -378,6 +377,7 @@ struct inodes_stat_t {
 #include <linux/capability.h>
 #include <linux/semaphore.h>
 #include <linux/fiemap.h>
+#include <linux/freezer.h>
 
 #include <asm/atomic.h>
 #include <asm/byteorder.h>

