--- python/Makefile.am~	2008-02-21 22:53:06.000000000 +0100
+++ python/Makefile.am	2009-01-26 11:03:15.000000000 +0100
@@ -31,7 +31,7 @@
 
 libxml2mod_la_SOURCES = libxml.c types.c
 nodist_libxml2mod_la_SOURCES = libxml2-py.c
-libxml2mod_la_LIBADD = $(mylibs) @CYGWIN_EXTRA_PYTHON_LIBADD@
+libxml2mod_la_LIBADD = $(mylibs) @CYGWIN_EXTRA_PYTHON_LIBADD@ `python-config --libs`
 
 libxml.c: libxml2-py.h # to generate before to compile
 
--- python/Makefile.in~	2009-01-18 22:39:05.000000000 +0100
+++ python/Makefile.in	2009-01-26 11:03:42.000000000 +0100
@@ -340,7 +340,7 @@
 @WITH_PYTHON_TRUE@python_LTLIBRARIES = libxml2mod.la
 @WITH_PYTHON_TRUE@libxml2mod_la_SOURCES = libxml.c types.c
 @WITH_PYTHON_TRUE@nodist_libxml2mod_la_SOURCES = libxml2-py.c
-@WITH_PYTHON_TRUE@libxml2mod_la_LIBADD = $(mylibs) @CYGWIN_EXTRA_PYTHON_LIBADD@
+@WITH_PYTHON_TRUE@libxml2mod_la_LIBADD = $(mylibs) @CYGWIN_EXTRA_PYTHON_LIBADD@ `python-config --libs`
 @WITH_PYTHON_TRUE@python_DATA = \
 @WITH_PYTHON_TRUE@	libxml2.py
 
