--- cups-1.2.3/cups/http.c.wbuffer	2006-09-13 16:41:57.000000000 +0100
+++ cups-1.2.3/cups/http.c	2006-09-13 16:42:23.000000000 +0100
@@ -1886,7 +1886,7 @@
 
   if (length > 0)
   {
-    if (http->wused && (length + http->wused) > sizeof(http->wbuffer))
+    if (http->wused && (length + http->wused) >= sizeof(http->wbuffer))
     {
       DEBUG_printf(("    flushing buffer (wused=%d, length=%d)\n",
                     http->wused, length));
