Index: tuxonice-userui-1.0/Makefile
===================================================================
--- tuxonice-userui-1.0.orig/Makefile
+++ tuxonice-userui-1.0/Makefile
@@ -38,10 +38,10 @@ usplash:
 	$(MAKE) -C $@ all
 
 tuxoniceui_text: $(CORE_OBJECTS) userui_text.o
-	$(CC) $(LDFLAGS) -static $^ -o $@
+	$(CC) $(LDFLAGS) $^ -o $@
 
 tuxoniceui_fbsplash: fbsplash $(CORE_OBJECTS)
-	$(CC) $(LDFLAGS) -static $(CORE_OBJECTS) fbsplash/userui_fbsplash.o -o $@ $(FBSPLASH_LIBS)
+	$(CC) $(LDFLAGS) $(CORE_OBJECTS) fbsplash/userui_fbsplash.o -o $@ $(FBSPLASH_LIBS)
 
 tuxoniceui_usplash: usplash $(CORE_OBJECTS) usplash/userui_usplash.o
 	$(CC) $(filter-out -static,$(LDFLAGS)) $(CORE_OBJECTS) usplash/userui_usplash.o -o $@ $(USPLASH_LIBS)
