diff -Nur slmodem-2.9.11-20080126.orig/drivers/Makefile slmodem-2.9.11-20080126/drivers/Makefile
--- slmodem-2.9.11-20080126.orig/drivers/Makefile	2005-04-05 18:17:50.000000000 +0300
+++ slmodem-2.9.11-20080126/drivers/Makefile	2008-06-13 18:53:00.000000000 +0300
@@ -20,7 +20,7 @@
 
 EXTRA_CFLAGS = -I$(obj) -I$(obj)/../modem
 
-obj-m := slamr.o slusb.o
+obj-m := slamr.o
 
 slamr-objs:= amrmo_init.o sysdep_amr.o amrlibs.o
 slusb-objs:= st7554.o
