--- src/cal3d_converter.cpp.orig	2008-04-15 15:16:15.000000000 +0300
+++ src/cal3d_converter.cpp	2008-04-15 15:16:31.000000000 +0300
@@ -9,6 +9,7 @@
 // your option) any later version.                                            //
 //****************************************************************************//
 
+#include <string.h>
 #include "cal3d/cal3d.h"
 
 #define SKELETON 0
--- src/cal3d/loader.cpp.orig	2008-04-15 15:16:44.000000000 +0300
+++ src/cal3d/loader.cpp	2008-04-15 15:17:06.000000000 +0300
@@ -16,6 +16,7 @@
 // Includes                                                                   //
 //****************************************************************************//
 
+#include <memory>
 #include "cal3d/loader.h"
 #include "cal3d/error.h"
 #include "cal3d/matrix.h"
