diff -Nur avidemux_2.5.1-old/plugins/ADM_videoEncoder/ADM_vidEnc_x264/encoder.cpp avidemux_2.5.1/plugins/ADM_videoEncoder/ADM_vidEnc_x264/encoder.cpp
--- avidemux_2.5.1-old/plugins/ADM_videoEncoder/ADM_vidEnc_x264/encoder.cpp	2009-10-20 20:27:57.420964334 +0300
+++ avidemux_2.5.1/plugins/ADM_videoEncoder/ADM_vidEnc_x264/encoder.cpp	2009-10-20 20:29:49.188964467 +0300
@@ -601,7 +601,7 @@
 	printf("[x264] i_bframe = %d\n", x264Param->i_bframe);
 	printf("[x264] i_bframe_bias = %d\n", x264Param->i_bframe_bias);
 	printf("[x264] i_frame_reference = %d\n", x264Param->i_frame_reference);
-	printf("[x264] b_bframe_pyramid = %d\n", x264Param->b_bframe_pyramid);
+	printf("[x264] i_bframe_pyramid = %d\n", x264Param->i_bframe_pyramid);
 #if X264_BUILD >= 63
 	printf("[x264] i_bframe_adaptive = %d\n", x264Param->i_bframe_adaptive);
 #else
diff -Nur avidemux_2.5.1-old/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp avidemux_2.5.1/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp
--- avidemux_2.5.1-old/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp	2009-10-20 20:27:57.420964334 +0300
+++ avidemux_2.5.1/plugins/ADM_videoEncoder/ADM_vidEnc_x264/x264Options.cpp	2009-10-20 20:30:12.676004885 +0300
@@ -332,12 +332,12 @@
 
 bool x264Options::getBFrameReferences(void)
 {
-	return _param.b_bframe_pyramid;
+	return _param.i_bframe_pyramid;
 }
 
 void x264Options::setBFrameReferences(bool bFrameReferences)
 {
-	_param.b_bframe_pyramid = bFrameReferences;
+	_param.i_bframe_pyramid = bFrameReferences;
 }
 
 bool x264Options::getLoopFilter(void)
