Index: unixODBC-2.2.12/configure.in
===================================================================
--- unixODBC-2.2.12.orig/configure.in
+++ unixODBC-2.2.12/configure.in
@@ -344,13 +344,6 @@ if test "x$thread" = "xtrue"; then
         AC_DEFINE(_REENTRANT)
         gotthread="yes";
         THREADLIB="-lpthread"
-        if test "x$ac_cv_prog_gcc"="xyes"; then
-          dnl Check if the compiler will build with -pthread as a option
-          AC_CHECK_COMP_OPT(pthread)
-        else
-          dnl Check if the compiler will build with -mt as a option
-          AC_CHECK_COMP_OPT(mt)
-        fi
       ])
     fi
 
