--- DRVConfig/MiniSQL/Makefile.am
+++ DRVConfig/MiniSQL/Makefile.am
@@ -8,3 +8,4 @@
 
 libodbcminiS_la_SOURCES = odbcminiS.c
 	
+libdir = @libdir@/unixODBC
--- DRVConfig/MySQL/Makefile.am
+++ DRVConfig/MySQL/Makefile.am
@@ -8,3 +8,4 @@
 
 libodbcmyS_la_SOURCES = odbcmyS.c
 	
+libdir = @libdir@/unixODBC
--- DRVConfig/Oracle/Makefile.am
+++ DRVConfig/Oracle/Makefile.am
@@ -8,3 +8,4 @@
 
 liboraodbcS_la_SOURCES = oraodbcS.c
 	
+libdir = @libdir@/unixODBC
--- DRVConfig/PostgreSQL/Makefile.am
+++ DRVConfig/PostgreSQL/Makefile.am
@@ -8,3 +8,4 @@
 
 libodbcpsqlS_la_SOURCES = odbcpsqlS.c
 	
+libdir = @libdir@/unixODBC
--- DRVConfig/drvcfg1/Makefile.am
+++ DRVConfig/drvcfg1/Makefile.am
@@ -8,3 +8,4 @@
 
 libodbcdrvcfg1S_la_SOURCES = drvcfg1.c
 	
+libdir = @libdir@/unixODBC
--- DRVConfig/drvcfg2/Makefile.am
+++ DRVConfig/drvcfg2/Makefile.am
@@ -8,3 +8,4 @@
 
 libodbcdrvcfg2S_la_SOURCES = drvcfg2.c
 	
+libdir = @libdir@/unixODBC
--- DRVConfig/esoob/Makefile.am
+++ DRVConfig/esoob/Makefile.am
@@ -8,3 +8,4 @@
 
 libesoobS_la_SOURCES = esoobS.c
 	
+libdir = @libdir@/unixODBC
--- DRVConfig/nn/Makefile.am
+++ DRVConfig/nn/Makefile.am
@@ -8,3 +8,4 @@
 
 libodbcnnS_la_SOURCES = drvcfg.c
 	
+libdir = @libdir@/unixODBC
--- DRVConfig/oplodbc/Makefile.am
+++ DRVConfig/oplodbc/Makefile.am
@@ -8,3 +8,4 @@
 
 liboplodbcS_la_SOURCES = oplodbc.c
 	
+libdir = @libdir@/unixODBC
--- DRVConfig/sapdb/Makefile.am
+++ DRVConfig/sapdb/Makefile.am
@@ -9,3 +9,5 @@
 libsapdbS_la_SOURCES = sapdb.c
 
 EXTRA_DIST = README 
+
+libdir = @libdir@/unixODBC
--- DRVConfig/tds/Makefile.am
+++ DRVConfig/tds/Makefile.am
@@ -8,3 +8,4 @@
 
 libtdsS_la_SOURCES = tdsS.c
 	
+libdir = @libdir@/unixODBC
--- DRVConfig/txt/Makefile.am
+++ DRVConfig/txt/Makefile.am
@@ -8,3 +8,4 @@
 
 libodbctxtS_la_SOURCES = drvcfg.c
 	
+libdir = @libdir@/unixODBC
--- Drivers/Postgre7.1/Makefile.am
+++ Drivers/Postgre7.1/Makefile.am
@@ -69,3 +69,5 @@
 	statement.c \
 	tuple.c \
 	tuplelist.c
+
+libdir = @libdir@/unixODBC
--- Drivers/PostgreSQL/Makefile.am
+++ Drivers/PostgreSQL/Makefile.am
@@ -64,3 +64,5 @@
 	statement.c \
 	tuple.c \
 	tuplelist.c
+
+libdir = @libdir@/unixODBC
--- Drivers/nn/Makefile.am
+++ Drivers/nn/Makefile.am
@@ -76,3 +76,4 @@
 	nndate.c \
 	nntp.c
 
+libdir = @libdir@/unixODBC
--- Drivers/txt/Makefile.am
+++ Drivers/txt/Makefile.am
@@ -112,3 +112,4 @@
 	driver.h \
 	driverextras.h
 
+libdir = @libdir@/unixODBC
--- DRVConfig/Mimer/Makefile.am
+++ DRVConfig/Mimer/Makefile.am
@@ -8,3 +8,4 @@
 
 libmimerS_la_SOURCES = mimerS.c
 	
+libdir = @libdir@/unixODBC
--- Drivers/template/Makefile.am
+++ Drivers/template/Makefile.am
@@ -101,3 +101,5 @@
 	_NativeTypeDesc.c \
 	_NativeTypeLength.c \
 	_NativeTypePrecision.c
+
+libdir = @libdir@/unixODBC
