--- Drivers/Postgre7.1/socket.c
+++ Drivers/Postgre7.1/socket.c
@@ -100,7 +100,7 @@
 {
 struct hostent *host;
 struct sockaddr_in sadr;
-unsigned long iaddr;
+struct in_addr iaddr;
 
 	if (self->socket != -1) {
 		self->errornumber = SOCKET_ALREADY_CONNECTED;
@@ -113,8 +113,8 @@
 	/*	If it is a valid IP address, use it.
 		Otherwise use hostname lookup. 
 	*/
-	iaddr = inet_addr(hostname);
-	if (iaddr == INADDR_NONE) {
+	iaddr.s_addr = inet_addr(hostname);
+	if (iaddr.s_addr == INADDR_NONE) {
 		host = gethostbyname(hostname);
 		if (host == NULL) {
 			self->errornumber = SOCKET_HOST_NOT_FOUND;
--- Drivers/PostgreSQL/socket.c
+++ Drivers/PostgreSQL/socket.c
@@ -102,7 +102,7 @@
 {
 struct hostent *host;
 struct sockaddr_in sadr;
-unsigned long iaddr;
+struct in_addr iaddr;
 
 	if (self->socket != -1) {
 		self->errornumber = SOCKET_ALREADY_CONNECTED;
@@ -115,8 +115,8 @@
 	/*	If it is a valid IP address, use it.
 		Otherwise use hostname lookup. 
 	*/
-	iaddr = inet_addr(hostname);
-	if (iaddr == INADDR_NONE) {
+	iaddr.s_addr = inet_addr(hostname);
+	if (iaddr.s_addr == INADDR_NONE) {
 		host = gethostbyname(hostname);
 		if (host == NULL) {
 			self->errornumber = SOCKET_HOST_NOT_FOUND;
