--- sqp/sqpStoreDataType.c
+++ sqp/sqpStoreDataType.c
@@ -6,9 +6,20 @@
  *
  ****************************************/
 
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
 #include "sqp.h"
 #include <sqlext.h>
 
+#ifndef WIN32
+#ifndef HAVE_STRICMP
+#define stricmp(s1,s2) 		strcasecmp(s1,s2)
+#define strnicmp(s1,s2,n)	strncasecmp(s1,s2,n)
+#endif
+#endif
+
 void sqpStoreDataType( char *pszType, int nPrecision, int nScale )
 {
     HSQPDATATYPE    hDataType;
--- sqp/Makefile.am
+++ sqp/Makefile.am
@@ -46,6 +46,8 @@
 
 endif
 
+AM_YFLAGS = -d
+
 EXTRA_DIST = \
 	yac.h
 
