Index: unixODBC-2.2.12/DataManager/Makefile.am
===================================================================
--- unixODBC-2.2.12.orig/DataManager/Makefile.am
+++ unixODBC-2.2.12/DataManager/Makefile.am
@@ -9,10 +9,11 @@ DataManager_LDADD = \
 	$(QT_LIBS) \
 	../ini/libinilc.la \
 	../odbcinst/libodbcinst.la \
+	../ini/libinilc.la \
 	../DriverManager/libodbc.la 
 
 DataManager_DEPENDANCIES = \
-    ../odbcinst/libodbcinst.la \
+	../odbcinst/libodbcinst.la \
 	../DriverManager/libodbc.la 
 
 DataManager_SOURCES = \
Index: unixODBC-2.2.12/DataManagerII/Makefile.am
===================================================================
--- unixODBC-2.2.12.orig/DataManagerII/Makefile.am
+++ unixODBC-2.2.12/DataManagerII/Makefile.am
@@ -9,10 +9,11 @@ DataManagerII_LDADD = \
 	$(QT_LIBS) \
 	../ini/libinilc.la \
 	../odbcinst/libodbcinst.la \
+	../ini/libinilc.la \
 	../DriverManager/libodbc.la 
 
 DataManagerII_DEPENDANCIES = \
-    ../odbcinst/libodbcinst.la \
+	../odbcinst/libodbcinst.la \
 	../DriverManager/libodbc.la 
 
 DataManagerII_SOURCES = \
