Index: libmpdemux/demux_mkv.c
===================================================================
--- libmpdemux/demux_mkv.c	(revisione 28646)
+++ libmpdemux/demux_mkv.c	(copia locale)
@@ -2989,6 +2989,9 @@
           mkv_index_t *index = NULL;
           int seek_id = (demuxer->video->id < 0) ? demuxer->audio->id : demuxer->video->id;  
 
+          if (target_timecode == 0)
+              index = mkv_d->indexes;
+
           /* let's find the entry in the indexes with the smallest */
           /* difference to the wanted timecode. */
           for (i=0; i < mkv_d->num_indexes; i++)
