diff -Nur mplayer-20080119-old/libmpdemux/demux_mkv.c mplayer-20080119/libmpdemux/demux_mkv.c
--- mplayer-20080119-old/libmpdemux/demux_mkv.c	2008-01-19 20:37:44.007105800 +0200
+++ mplayer-20080119/libmpdemux/demux_mkv.c	2008-01-19 20:40:28.409112896 +0200
@@ -1909,6 +1909,8 @@
         track->a_formattag = 0x0055;
       else if (!strncmp(track->codec_id, MKV_A_AC3, strlen(MKV_A_AC3)))
         track->a_formattag = 0x2000;
+      else if (!strncmp(track->codec_id, MKV_A_EAC3, strlen(MKV_A_EAC3)))
+        track->a_formattag = 0x2002;
       else if (!strcmp(track->codec_id, MKV_A_DTS))
         track->a_formattag = 0x2001;
       else if (!strcmp(track->codec_id, MKV_A_PCM) ||
@@ -1993,6 +1995,7 @@
       sh_a->wf->nBlockAlign = 1152;
     }
   else if ((track->a_formattag == 0x2000) || /* AC3 */
+           (track->a_formattag == 0x2002) || /* EAC3 */
            (track->a_formattag == 0x2001)) /* DTS */
     {
       free(sh_a->wf);
diff -Nur mplayer-20080119-old/libmpdemux/matroska.h mplayer-20080119/libmpdemux/matroska.h
--- mplayer-20080119-old/libmpdemux/matroska.h	2008-01-19 20:37:44.007105800 +0200
+++ mplayer-20080119/libmpdemux/matroska.h	2008-01-19 20:39:26.791480200 +0200
@@ -18,6 +18,7 @@
 #define MKV_A_AAC_4LTP   "A_AAC/MPEG4/LTP"
 #define MKV_A_AAC        "A_AAC"
 #define MKV_A_AC3        "A_AC3"
+#define MKV_A_EAC3       "A_EAC3"
 #define MKV_A_DTS        "A_DTS"
 #define MKV_A_MP2        "A_MPEG/L2"
 #define MKV_A_MP3        "A_MPEG/L3"
