diff -Nur mplayer-20061209-old/libavcodec/h263.c mplayer-20061209/libavcodec/h263.c
--- mplayer-20061209-old/libavcodec/h263.c	2006-12-12 19:56:11.000000000 +0200
+++ mplayer-20061209/libavcodec/h263.c	2006-12-12 19:59:52.000000000 +0200
@@ -6076,7 +6076,7 @@
     for(;;) {
         if(get_bits_count(gb) >= gb->size_in_bits){
             if(gb->size_in_bits==8 && (s->divx_version || s->xvid_build)){
-                av_log(s->avctx, AV_LOG_ERROR, "frame skip %d\n", gb->size_in_bits);
+                //av_log(s->avctx, AV_LOG_ERROR, "frame skip %d\n", gb->size_in_bits);
                 return FRAME_SKIPPED; //divx bug
             }else
                 return -1; //end of stream
