diff -Nur mplayer-20080119-old/libmpcodecs/dec_audio.c mplayer-20080119/libmpcodecs/dec_audio.c
--- mplayer-20080119-old/libmpcodecs/dec_audio.c	2008-01-19 19:36:46.287163752 +0200
+++ mplayer-20080119/libmpcodecs/dec_audio.c	2008-01-19 19:39:30.965128896 +0200
@@ -2,6 +2,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <assert.h>
+#include <locale.h>
 
 #include "config.h"
 #include "mp_msg.h"
@@ -131,6 +132,9 @@
 static int init_audio(sh_audio_t *sh_audio, char *codecname, char *afm,
 		      int status, stringset_t *selected)
 {
+    char *locale_data = setlocale(LC_CTYPE, "");
+    setlocale(LC_CTYPE, "C");
+
     unsigned int orig_fourcc = sh_audio->wf ? sh_audio->wf->wFormatTag : 0;
     int force = 0;
     if (codecname && codecname[0] == '+') {
@@ -219,8 +223,10 @@
 	    continue;		// try next...
 	}
 	// Yeah! We got it!
+    setlocale(LC_CTYPE, locale_data);
 	return 1;
     }
+    setlocale(LC_CTYPE, locale_data);
     return 0;
 }
 
diff -Nur mplayer-20080119-old/libmpcodecs/dec_video.c mplayer-20080119/libmpcodecs/dec_video.c
--- mplayer-20080119-old/libmpcodecs/dec_video.c	2008-01-19 19:36:46.284164208 +0200
+++ mplayer-20080119/libmpcodecs/dec_video.c	2008-01-19 19:38:38.946037000 +0200
@@ -7,6 +7,7 @@
 #endif
 #include <stdlib.h>
 #include <unistd.h>
+#include <locale.h>
 
 #include "mp_msg.h"
 #include "help_mp.h"
@@ -177,6 +178,9 @@
 
 static int init_video(sh_video_t *sh_video,char* codecname,char* vfm,int status,
                stringset_t *selected){
+    char *locale_data = setlocale(LC_CTYPE, "");
+    setlocale(LC_CTYPE, "C");
+
     int force = 0;
     unsigned int orig_fourcc=sh_video->bih?sh_video->bih->biCompression:0;
     sh_video->codec=NULL;
@@ -273,8 +277,10 @@
 	}
 	// Yeah! We got it!
 	sh_video->initialized=1;
+    setlocale(LC_CTYPE, locale_data);
 	return 1;
     }
+    setlocale(LC_CTYPE, locale_data);
     return 0;
 }
 
