Index: command.c
===================================================================
--- command.c	(revision 31961)
+++ command.c	(working copy)
@@ -1543,7 +1543,7 @@
 #endif
 
     if (source == SUB_SOURCE_VOBSUB) {
-        vobsub_id = source_pos;
+        vobsub_id = vobsub_get_id_by_index(vo_vobsub, source_pos);
     } else if (source == SUB_SOURCE_SUBS) {
         mpctx->set_of_sub_pos = source_pos;
 #ifdef CONFIG_ASS
