diff --git a/systemv/hal_lpadmin b/systemv/hal_lpadmin
index 16bfc08..d5c6441 100755
--- a/systemv/hal_lpadmin
+++ b/systemv/hal_lpadmin
@@ -723,9 +723,8 @@ class HalPrinter:
 
             if faxuri:
                 faxname = self.faxname
-                if p == None:
-                    cupsppds = self.cups_connection.getPPDs ()
-                    p = cupshelpers.ppds.PPDs (cupsppds)
+                cupsppds = self.cups_connection.getPPDs ()
+                p = cupshelpers.ppds.PPDs (cupsppds)
                 (status, faxppd) = p.getPPDNameFromDeviceID ("HP", "Fax",
                                                         "HP Fax", [], faxuri)
                 info = "Fax queue for %s %s" % (self.make, self.model)
