--- src/plugins/live365/live365.c.orig	2006-08-28 11:36:32.000000000 +0300
+++ src/plugins/live365/live365.c	2006-08-28 16:51:01.000000000 +0300
@@ -1410,8 +1410,8 @@
   init_handler();
 
   st_action_register("record-stream", _("Record a stream"), "xterm -e streamripper %q");
-  st_action_register("view-web", _("Open a web page"), "epiphany %q");
-  st_action_register("play-stream", _("Listen to a stream"), "xmms %q");
+  st_action_register("view-web", _("Open a web page"), "firefox %q");
+  st_action_register("play-stream", _("Listen to a stream"), "kaffeine %q");
 
   return TRUE;
 }
