--- src/plugins/local/local.c.orig	2006-08-28 11:36:32.000000000 +0300
+++ src/plugins/local/local.c	2006-08-28 16:51:26.000000000 +0300
@@ -1057,8 +1057,8 @@
 
   init_handler();
 
-  st_action_register("play-m3u", _("Listen to a .m3u file"), "xmms %q");
-  st_action_register("view-web", _("Open a web page"), "epiphany %q");
+  st_action_register("play-m3u", _("Listen to a .m3u file"), "kaffeine %q");
+  st_action_register("view-web", _("Open a web page"), "firefox %q");
 
   return TRUE;
 }
