--- src/plugins/shoutcasts/houtcast.c.orig	2006-08-28 11:36:32.000000000 +0300
+++ src/plugins/shoutcast/shoutcast.c	2006-08-28 16:52:20.000000000 +0300
@@ -959,9 +959,9 @@
 
   init_handler();
   
-  st_action_register("play-m3u", _("Listen to a .m3u file"), "xmms %q");
-  st_action_register("record-stream", _("Record a stream"), "xterm -e streamripper %q");
-  st_action_register("view-web", _("Open a web page"), "epiphany %q");
+  st_action_register("play-m3u", _("Listen to a .m3u file"), "kaffeine %q");
+  st_action_register("record-stream", _("Record a stream"), "xterm -e streamripper %q -s -o never");
+  st_action_register("view-web", _("Open a web page"), "firefox %q");
 
   return TRUE;
 }
