--- src/streamtuner/st-action.c.orig	2004-12-18 11:56:42.000000000 +0200
+++ src/streamtuner/st-action.c	2006-08-17 12:30:47.000000000 +0300
@@ -123,7 +123,7 @@
 {
   actions = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, NULL);
 
-  st_action_register("view-web", _("Open a web page"), "epiphany %q");
+  st_action_register("view-web", _("Open a web page"), "firefox %q");
   st_action_register("view-ghelp", _("Open a GNOME help document"), "yelp %q");
 }
 
