--- kaudiocreator/kaudiocreator.desktop.orig	2009-06-20 12:55:17.000000000 +0300
+++ kaudiocreator/kaudiocreator.desktop	2009-06-20 12:55:42.000000000 +0300
@@ -63,7 +63,7 @@
 GenericName[ro]=Extractor de CD
 GenericName[ru]=Копирование CD
 GenericName[sv]=Cd-lagring
-GenericName[tr]=CD 
+GenericName[tr]=CD Kodlayıcı
 GenericName[uk]=Видирання КД
 GenericName[x-test]=xxCD Ripperxx
 Exec=kaudiocreator
