--- hastymail2.conf.example	2010-09-30 06:25:21.000000000 +0300
+++ hastymail2.conf	2011-05-17 11:58:30.118000238 +0300
@@ -432,7 +432,7 @@
   # uk_UA       Ukranian
   # zh_CN       Chinese
 
-site_default_lang = en_US
+site_default_lang = tr_TR
 
   # site_default_timezone
   # ---------------------
@@ -440,7 +440,7 @@
   # options. This only works with PHP5. If you are running PHP4 it should
   # be set to false
 
-site_default_timezone = false
+site_default_timezone = Europe/Istanbul
 
   # site_page_title
   # ---------------
@@ -492,14 +492,14 @@
   # RC4 encryption that uses the following key to encrypt user passwords
   # It has been reported that keys over 93 characters do NOT work properly
 
-site_key = asdfasdfasdfasdfasdf
+site_key = 0685c70b530b401838e5d3fc6994e424ec893aaa
 
   # site_logo
   # ---------
   # This is the logo that appears after the page icon and before the folder
   # list in the upper left hand corner of every page
 
-site_logo = <span>Hm<span class="super">2</span></span>
+site_logo = <span>HastyMail<span class="super">2</span></span>
 
 
   # special folders
@@ -757,7 +757,7 @@
 
   # timezone (PHP5 only) 
 
-default_timezone = America/Chicago
+default_timezone = Europe/Istanbul
 
   # first page after login.
 
@@ -770,7 +770,7 @@
 
   # Interface language
 
-default_lang = en_US
+default_lang = tr_TR
 
   # Show the folder list/tree 
 
