From 24f5017b232a525b7226e49730616c0b26a0de36 Mon Sep 17 00:00:00 2001
From: Colin Guthrie <development@colin.guthr.ie>
Date: Tue, 5 Aug 2008 20:43:41 +0100
Subject: [PATCH 503/503] Change the default resample method to speex-fixed-0 (Mandriva bug #36084)

---
 src/daemon/daemon.conf.in |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

Index: pulseaudio-0.9.21/src/daemon/daemon.conf.in
===================================================================
--- pulseaudio-0.9.21.orig/src/daemon/daemon.conf.in
+++ pulseaudio-0.9.21/src/daemon/daemon.conf.in
@@ -50,6 +50,13 @@
 ; log-time = no
 ; log-backtrace = 0
 
+# for (supposed) better quality
+; resample-method = speex-float-3
+# for better speed (and lower quality
+; resample-method = trivial
+# good compromise
+resample-method = speex-float-0
+
 ; resample-method = speex-float-3
 ; enable-remixing = yes
 ; enable-lfe-remixing = no
