--- /var/tmp/pisi/cdrdao-1.2.2-3/work/cdrdao-1.2.2/configure.ac	2006-09-20 13:46:02.000000000 +0300
+++ configure.ac	2006-09-22 15:39:43.000000000 +0300
@@ -254,7 +254,7 @@
       AC_CHECK_FILE(/usr/include/schily.h, [sgsysinc=yes])
       AC_CHECK_FILE(/usr/local/include/schily.h, [sgsysinc=yes])
       AC_CHECK_FILE(/usr/include/schily/schily.h,
-         [sgsysinc=yes; scsilib_incl="-I/usr/include/schily"])
+         [sgsysinc=yes; scsilib_incl="-I/usr/include"])
       AC_CHECK_FILE(/usr/local/include/schily/schily.h,
          [sgsysinc=yes; scsilib_incl="-I/usr/local/include/schily"])
     else
@@ -315,17 +315,6 @@
     esac
 fi
 
-dnl If we're using custom libsch/schily libs, make sure it'll compile
-if test "$en_custom_libscg" = yes; then
-  AC_LIBSCG(0.5,[have_scg05=yes],[AC_MSG_ERROR([No cdrtools libraries found. You need to install cdrtools version >= 1.10 (a.k.a. cdrecord-devel). To build the included fallback scsi library, please use --with-scglib=pkg (see INSTALL file)])])
-  AC_LIBSCG(0.7,[have_scg07=yes], [])
-
-  dnl Special case when we're compiling with cdrtools 1.10
-  if test "$have_scg05" = yes -a ! "$have_scg07" = yes; then
-    AC_DEFINE(USE_OLDSCGLIB,1,"")
-  fi
-fi
-
 dnl Selection of buffer code
 case "$host" in
   *)
--- dao/ScsiIf-lib.cc	2006-09-19 12:07:11.000000000 +0300
+++ dao/ScsiIf-lib.cc	2006-09-22 15:41:51.000000000 +0300
@@ -29,7 +29,7 @@
 
 #include "xconfig.h"
 
-#include "standard.h"
+#include "schily/standard.h"
 #include "scg/scgcmd.h"
 #include "scg/scsitransp.h"
 #include "scg/scsireg.h"
