Index: src/repodialog.py
===================================================================
--- src/repodialog.py	(revision 36117)
+++ src/repodialog.py	(working copy)
@@ -22,5 +22,5 @@
         QtGui.QDialog.__init__(self, parent)
         self.setupUi(self)
         arch = pisi.ctx.config.values.general.architecture
-        self.repoAddress.addItem("http://packages.pardus.org.tr/pardus/corporate2/stable/%s/pisi-index.xml.xz" % arch)
+        self.repoAddress.addItem("http://packages.pardus.org.tr/pardus/kurumsal2/stable/%s/pisi-index.xml.xz" % arch)
 
