Index: kftpgrabber-0.8.1/kftpgrabber/src/misc/kftpgrabber.kcfg
===================================================================
--- kftpgrabber-0.8.1.orig/kftpgrabber/src/misc/kftpgrabber.kcfg
+++ kftpgrabber-0.8.1/kftpgrabber/src/misc/kftpgrabber.kcfg
@@ -95,7 +95,7 @@
     </entry>
     
     <entry name="defEncoding" type="String">
-      <default>iso 8859-1</default>
+      <default>utf8</default>
       <label>The default site encoding.</label>
     </entry>
     
