From 758aac63b97d6f5a3a6a5d74d7d0af12cd46cd3d Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Mon, 8 Jun 2009 18:22:19 +0200
Subject: [PATCH 125/127] sample: fix build on BE archs

---
 src/pulsecore/sample-util.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/pulsecore/sample-util.c b/src/pulsecore/sample-util.c
index dda3883..5b8ccf5 100644
--- a/src/pulsecore/sample-util.c
+++ b/src/pulsecore/sample-util.c
@@ -1182,7 +1182,7 @@ pa_memchunk* pa_silence_memchunk_get(pa_silence_cache *cache, pa_mempool *pool,
             case PA_SAMPLE_S24LE:
             case PA_SAMPLE_S24BE:
             case PA_SAMPLE_S24_32LE:
-            case PA_SAMPLE_S24_32RE:
+            case PA_SAMPLE_S24_32BE:
             case PA_SAMPLE_FLOAT32LE:
             case PA_SAMPLE_FLOAT32BE:
                 cache->blocks[PA_SAMPLE_S16LE] = b = silence_memblock_new(pool, 0);
-- 
1.6.3

