--- pulseaudio-0.9.10.orig/src/daemon/org.pulseaudio.policy.in	2008-03-28 01:38:09.000000000 +0200
+++ pulseaudio-0.9.10/src/daemon/org.pulseaudio.policy.in	2008-04-20 14:05:16.000000000 +0300
@@ -45,7 +45,7 @@
     <defaults>
       <allow_any>no</allow_any>
       <allow_inactive>no</allow_inactive>
-      <allow_active>no</allow_active>
+      <allow_active>yes</allow_active>
     </defaults>
   </action>
 
