--- xinetd-2.3.14/xinetd/reconfig.c.orig	2010-06-01 12:58:18.000000000 +0200
+++ xinetd-2.3.14/xinetd/reconfig.c	2010-06-01 12:58:23.000000000 +0200
@@ -138,8 +138,8 @@ void hard_reconfig( void )
           *      b. Terminate running servers and cancel retry attempts, in case
           *         of reconfiguration
           */
-         svc_deactivate( osp ) ;
          terminate_servers( osp ) ;
+         svc_deactivate( osp ) ;
          cancel_service_retries( osp ) ;
 
          /*
