diff -up btrfs-progs-0.19/Makefile.orig btrfs-progs-0.19/Makefile
--- btrfs-progs-0.19/Makefile.orig	2009-06-19 08:05:49.151573429 -0400
+++ btrfs-progs-0.19/Makefile	2009-06-19 08:08:35.192509803 -0400
@@ -16,7 +16,7 @@ prefix ?= /usr/local
 bindir = $(prefix)/bin
 LIBS=-luuid
 
-progs = btrfsctl mkfs.btrfs btrfs-debug-tree btrfs-show btrfs-vol btrfsck
+progs = btrfsctl mkfs.btrfs btrfs-debug-tree btrfs-show btrfs-vol btrfsck btrfs-image btrfstune
 
 # make C=1 to enable sparse
 ifdef C
