Index: wxPython/src/gtk/_core_wrap.cpp
===================================================================
--- wxPython.orig/src/gtk/_core_wrap.cpp
+++ wxPython/src/gtk/_core_wrap.cpp
@@ -28673,6 +28673,7 @@ fail:
 }
 
 
+#if 0
 SWIGINTERN PyObject *_wrap_ShowEvent_IsShown(PyObject *SWIGUNUSEDPARM(self), PyObject *args) {
   PyObject *resultobj = 0;
   wxShowEvent *arg1 = (wxShowEvent *) 0 ;
@@ -28701,6 +28702,7 @@ SWIGINTERN PyObject *_wrap_ShowEvent_IsS
 fail:
   return NULL;
 }
+#endif
 
 
 SWIGINTERN PyObject *ShowEvent_swigregister(PyObject *SWIGUNUSEDPARM(self), PyObject *args) {
@@ -61296,7 +61298,6 @@ static PyMethodDef SwigMethods[] = {
 	 { (char *)"new_ShowEvent", (PyCFunction) _wrap_new_ShowEvent, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"ShowEvent_SetShow", (PyCFunction) _wrap_ShowEvent_SetShow, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"ShowEvent_GetShow", (PyCFunction)_wrap_ShowEvent_GetShow, METH_O, NULL},
-	 { (char *)"ShowEvent_IsShown", (PyCFunction)_wrap_ShowEvent_IsShown, METH_O, NULL},
 	 { (char *)"ShowEvent_swigregister", ShowEvent_swigregister, METH_VARARGS, NULL},
 	 { (char *)"ShowEvent_swiginit", ShowEvent_swiginit, METH_VARARGS, NULL},
 	 { (char *)"new_IconizeEvent", (PyCFunction) _wrap_new_IconizeEvent, METH_VARARGS | METH_KEYWORDS, NULL},
Index: wxPython/src/gtk/_gdi_wrap.cpp
===================================================================
--- wxPython.orig/src/gtk/_gdi_wrap.cpp
+++ wxPython/src/gtk/_gdi_wrap.cpp
@@ -20579,6 +20579,7 @@ fail:
 }
 
 
+#if 0
 SWIGINTERN PyObject *_wrap_DC_SetDeviceClippingRegion(PyObject *SWIGUNUSEDPARM(self), PyObject *args, PyObject *kwargs) {
   PyObject *resultobj = 0;
   wxDC *arg1 = (wxDC *) 0 ;
@@ -20618,6 +20619,7 @@ SWIGINTERN PyObject *_wrap_DC_SetDeviceC
 fail:
   return NULL;
 }
+#endif
 
 
 SWIGINTERN PyObject *_wrap_DC_DrawLines(PyObject *SWIGUNUSEDPARM(self), PyObject *args, PyObject *kwargs) {
@@ -30753,6 +30755,7 @@ fail:
 }
 
 
+#if 0
 SWIGINTERN PyObject *_wrap_GraphicsRenderer_GetCairoRenderer(PyObject *SWIGUNUSEDPARM(self), PyObject *args) {
   PyObject *resultobj = 0;
   wxGraphicsRenderer *result = 0 ;
@@ -30767,6 +30770,7 @@ SWIGINTERN PyObject *_wrap_GraphicsRende
 fail:
   return NULL;
 }
+#endif
 
 
 SWIGINTERN PyObject *_wrap_GraphicsRenderer_CreateContext__SWIG_0(PyObject *SWIGUNUSEDPARM(self), int nobjs, PyObject **swig_obj) {
@@ -40069,7 +40073,6 @@ static PyMethodDef SwigMethods[] = {
 	 { (char *)"DC_SetClippingRegionPointSize", (PyCFunction) _wrap_DC_SetClippingRegionPointSize, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"DC_SetClippingRegionAsRegion", (PyCFunction) _wrap_DC_SetClippingRegionAsRegion, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"DC_SetClippingRect", (PyCFunction) _wrap_DC_SetClippingRect, METH_VARARGS | METH_KEYWORDS, NULL},
-	 { (char *)"DC_SetDeviceClippingRegion", (PyCFunction) _wrap_DC_SetDeviceClippingRegion, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"DC_DrawLines", (PyCFunction) _wrap_DC_DrawLines, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"DC_DrawPolygon", (PyCFunction) _wrap_DC_DrawPolygon, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"DC_DrawLabel", (PyCFunction) _wrap_DC_DrawLabel, METH_VARARGS | METH_KEYWORDS, NULL},
@@ -40341,7 +40344,6 @@ static PyMethodDef SwigMethods[] = {
 	 { (char *)"GraphicsContext_swigregister", GraphicsContext_swigregister, METH_VARARGS, NULL},
 	 { (char *)"delete_GraphicsRenderer", (PyCFunction)_wrap_delete_GraphicsRenderer, METH_O, NULL},
 	 { (char *)"GraphicsRenderer_GetDefaultRenderer", (PyCFunction)_wrap_GraphicsRenderer_GetDefaultRenderer, METH_NOARGS, NULL},
-	 { (char *)"GraphicsRenderer_GetCairoRenderer", (PyCFunction)_wrap_GraphicsRenderer_GetCairoRenderer, METH_NOARGS, NULL},
 	 { (char *)"GraphicsRenderer_CreateContext", _wrap_GraphicsRenderer_CreateContext, METH_VARARGS, NULL},
 	 { (char *)"GraphicsRenderer_CreateMeasuringContext", (PyCFunction)_wrap_GraphicsRenderer_CreateMeasuringContext, METH_O, NULL},
 	 { (char *)"GraphicsRenderer_CreateContextFromNativeContext", (PyCFunction) _wrap_GraphicsRenderer_CreateContextFromNativeContext, METH_VARARGS | METH_KEYWORDS, NULL},
Index: wxPython/src/gtk/xrc_wrap.cpp
===================================================================
--- wxPython.orig/src/gtk/xrc_wrap.cpp
+++ wxPython/src/gtk/xrc_wrap.cpp
@@ -6361,6 +6361,7 @@ fail:
 }
 
 
+#if 0
 SWIGINTERN PyObject *_wrap_XmlNode_GetAttribute(PyObject *SWIGUNUSEDPARM(self), PyObject *args, PyObject *kwargs) {
   PyObject *resultobj = 0;
   wxXmlNode *arg1 = (wxXmlNode *) 0 ;
@@ -6516,6 +6517,7 @@ SWIGINTERN PyObject *_wrap_XmlNode_GetAt
 fail:
   return NULL;
 }
+#endif
 
 
 SWIGINTERN PyObject *XmlNode_swigregister(PyObject *SWIGUNUSEDPARM(self), PyObject *args) {
@@ -9050,9 +9052,6 @@ static PyMethodDef SwigMethods[] = {
 	 { (char *)"XmlNode_SetNext", (PyCFunction) _wrap_XmlNode_SetNext, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"XmlNode_SetChildren", (PyCFunction) _wrap_XmlNode_SetChildren, METH_VARARGS | METH_KEYWORDS, NULL},
 	 { (char *)"XmlNode_SetProperties", (PyCFunction) _wrap_XmlNode_SetProperties, METH_VARARGS | METH_KEYWORDS, NULL},
-	 { (char *)"XmlNode_GetAttribute", (PyCFunction) _wrap_XmlNode_GetAttribute, METH_VARARGS | METH_KEYWORDS, NULL},
-	 { (char *)"XmlNode_AddAttribute", (PyCFunction) _wrap_XmlNode_AddAttribute, METH_VARARGS | METH_KEYWORDS, NULL},
-	 { (char *)"XmlNode_GetAttributes", (PyCFunction)_wrap_XmlNode_GetAttributes, METH_O, NULL},
 	 { (char *)"XmlNode_swigregister", XmlNode_swigregister, METH_VARARGS, NULL},
 	 { (char *)"XmlNode_swiginit", XmlNode_swiginit, METH_VARARGS, NULL},
 	 { (char *)"new_XmlDocument", (PyCFunction) _wrap_new_XmlDocument, METH_VARARGS | METH_KEYWORDS, NULL},
Index: wxPython/src/_dc.i
===================================================================
--- wxPython.orig/src/_dc.i
+++ wxPython/src/_dc.i
@@ -379,7 +379,6 @@ screen is damaged.", "
     %Rename(SetClippingRegionAsRegion, void, SetClippingRegion(const wxRegion& region));
     %Rename(SetClippingRect, void, SetClippingRegion(const wxRect& rect));
 
-    void SetDeviceClippingRegion(const wxRegion& region);
     
     DocDeclAStr(
         void , DrawLines(int points, wxPoint* points_array,
Index: wxPython/src/_event.i
===================================================================
--- wxPython.orig/src/_event.i
+++ wxPython/src/_event.i
@@ -1873,7 +1873,6 @@ public:
 
     void SetShow(bool show);
     bool GetShow() const;
-    bool IsShown() const;
     
     %property(Show, GetShow, SetShow, doc="See `GetShow` and `SetShow`");
 };
Index: wxPython/src/_graphics.i
===================================================================
--- wxPython.orig/src/_graphics.i
+++ wxPython/src/_graphics.i
@@ -288,11 +288,6 @@ public :
                         "wx.GraphicsRenderer is not available on this platform.");
         return NULL;
     }
-    static wxGraphicsRenderer* GetCairoRenderer() {
-        PyErr_SetString(PyExc_NotImplementedError,
-                        "wx.GraphicsRenderer is not available on this platform.");
-        return NULL;
-    }   
 
     virtual wxGraphicsContext * CreateContext( const wxWindowDC& ) { return NULL; }
     virtual wxGraphicsContext * CreateContextFromNativeContext( void *  ) { return NULL; }
@@ -1029,7 +1024,6 @@ public :
 
     // %newobject GetDefaultRenderer;  ???
     static wxGraphicsRenderer* GetDefaultRenderer();
-    static wxGraphicsRenderer* GetCairoRenderer();
 
     %nokwargs CreateContext;
     %newobject CreateContext;
Index: wxPython/src/_xml.i
===================================================================
--- wxPython.orig/src/_xml.i
+++ wxPython/src/_xml.i
@@ -134,11 +134,6 @@ public:
 
     void SetProperties(wxXmlProperty *prop);
 
-    wxString GetAttribute(const wxString& attrName,
-                          const wxString& defaultVal) const;
-    //bool GetAttribute(const wxString& attrName, wxString *value) const;
-    void AddAttribute(const wxString& attrName, const wxString& value);
-    wxXmlProperty* GetAttributes() const;
 
     
     %property(Children, GetChildren, SetChildren, doc="See `GetChildren` and `SetChildren`");
