diff -ur kaffeine-0.8.4.orig/kaffeine/src/kaffeine.desktop kaffeine-0.8.4/kaffeine/src/kaffeine.desktop
--- kaffeine-0.8.4.orig/kaffeine/src/kaffeine.desktop	2007-04-13 17:54:43.000000000 +0300
+++ kaffeine-0.8.4/kaffeine/src/kaffeine.desktop	2007-04-14 18:51:10.882500316 +0300
@@ -65,4 +65,5 @@
 Name[xx]=xxKaffeinexx
 Terminal=false
 Type=Application
+InitialPreference=15
 X-KDE-HasTempFileOption=true
