--- gtk/help_dlg.c
+++ gtk/help_dlg.c
@@ -51,7 +51,7 @@
 #endif
 
 
-#define HELP_DIR	"help"
+#define HELP_DIR	"/usr/share/wireshark/help"
 
 
 #define NOTEBOOK_KEY    "notebook_key"
@@ -164,10 +164,8 @@
   char *relative_path, *absolute_path;
   help_page_t *page;
 
-  relative_path = g_strconcat(HELP_DIR, G_DIR_SEPARATOR_S, filename, NULL);
-  absolute_path = get_datafile_path(relative_path);
+  absolute_path = g_strconcat(HELP_DIR, G_DIR_SEPARATOR_S, filename, NULL);
   text_page = text_page_new(absolute_path);
-  g_free(relative_path);
   gtk_widget_show(text_page);
 
   page = g_malloc(sizeof (help_page_t));
@@ -223,12 +221,11 @@
     return;
   }
 
-  help_toc_file_path = get_datafile_path(HELP_DIR G_DIR_SEPARATOR_S "toc");
+  help_toc_file_path = HELP_DIR G_DIR_SEPARATOR_S "toc";
   help_toc_file = eth_fopen(help_toc_file_path, "r");
   if (help_toc_file == NULL) {
     simple_dialog(ESD_TYPE_ERROR, ESD_BTN_OK, "Could not open file \"%s\": %s",
                   help_toc_file_path, strerror(errno));
-    g_free(help_toc_file_path);
     return;
   }
 
