Index: cups-1.3.11/conf/mime.convs.in
===================================================================
--- cups-1.3.11.orig/conf/mime.convs.in
+++ cups-1.3.11/conf/mime.convs.in
@@ -39,7 +39,7 @@
 #
 
 application/pdf		application/vnd.cups-postscript	66	pdftops
-application/postscript	application/vnd.cups-postscript	66	pstops
+application/postscript	application/vnd.cups-postscript	100	pstops
 application/vnd.hp-HPGL	application/postscript	66	hpgltops
 application/x-cshell	application/postscript	33	texttops
 application/x-csource	application/postscript	33	texttops
